{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "source": [
    "# SON: Zhu-Tokita-Takenouchi-Kim (ZTTK) Syndrome (OMIM #617140)\n",
    "\n",
    "[Zhu-Tokita-Takenouchi-Kim Syndrome (OMIM #617140)](https://omim.org/entry/617140) is a rare developmental disorder caused by pathogenic variants in the *SON* gene. This syndrome, also known as **ZTTK Syndrome**, is characterized by a range of symptoms including intellectual disability, developmental delays, and craniofacial abnormalities. The disease is associated mainly with protein-truncating variants of the *SON* gene, and it is inherited in an autosomal dominant manner."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2025-01-16\n",
      "Using gpsea version 0.9.4\n"
     ]
    }
   ],
   "source": [
    "import hpotk\n",
    "import gpsea\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo()\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using gpsea version {gpsea.__version__}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "source": [
    "### SON\n",
    "We user the [Matched Annotation from NCBI and EMBL-EBI (MANE)](https://www.ncbi.nlm.nih.gov/refseq/MANE/) transcript and the corresponding protein identifier for SON."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "outputs": [],
   "source": [
    "from gpsea.preprocessing import configure_default_protein_metadata_service\n",
    "gene_symbol = 'SON'\n",
    "mane_tx_id = 'NM_138927.4'\n",
    "mane_protein_id = 'NP_620305.2'\n",
    "pms = configure_default_protein_metadata_service()\n",
    "protein_meta = pms.annotate(mane_protein_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 100%|██████████| 52/52 [00:02<00:00, 21.17 individuals/s] \n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store('0.1.24') as ps:\n",
    "    phenopackets = tuple(ps.iter_cohort_phenopackets(gene_symbol))\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort, validation = load_phenopackets(  \n",
    "    phenopackets=phenopackets,\n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "validation.summarize() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "\n",
       "<h1>GPSEA cohort analysis</h1>\n",
       "<div id=\"cohort-summary\">\n",
       "  <p>\n",
       "    Successfully loaded 52 individuals.\n",
       "\n",
       "    \n",
       "    26 were recorded as male,\n",
       "    26 as female,\n",
       "    and 0 as unknown sex.\n",
       "\n",
       "    \n",
       "    \n",
       "    No information about individuals' vital status was reported.\n",
       "    \n",
       "\n",
       "    \n",
       "    0 individuals had disease onset information\n",
       "    and 0 had information about the age of last encounter.\n",
       "  </p>\n",
       "\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "</div>\n",
       "\n",
       "\n",
       "<div class=\"cards\">\n",
       "  \n",
       "  <div id=\"hpo\" class=\"card\">\n",
       "    <h2>HPO terms</h2>\n",
       "    \n",
       "    \n",
       "    <table>\n",
       "      The cohort included 47 distinct HPO terms.\n",
       "      <caption>\n",
       "        \n",
       "        Top 10 most common HPO terms.\n",
       "        \n",
       "        \n",
       "      </caption>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>HPO Term</th>\n",
       "        </tr>\n",
       "\n",
       "        \n",
       "        <tr>\n",
       "          <td>31</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0001252\">Hypotonia</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>30</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0011968\">Feeding difficulties</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>27</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000750\">Delayed speech and language development</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>26</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0004322\">Short stature</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>24</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0002119\">Ventriculomegaly</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>24</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000494\">Downslanted palpebral fissures</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>24</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0001250\">Seizure</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>22</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000369\">Low-set ears</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>22</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0010864\">Intellectual disability, severe</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>21</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000486\">Strabismus</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div id=\"measurements\" class=\"card\">\n",
       "    <h2>Measurements</h2>\n",
       "    \n",
       "    \n",
       "    No data regarding measurement assays were provided.\n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div id=\"diseases\" class=\"card\">\n",
       "    <h2>Diseases</h2>\n",
       "    \n",
       "    \n",
       "    The cohort members were diagnosed with 1 disease.\n",
       "    <table>\n",
       "      \n",
       "\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>Disease</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>52</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/disease/OMIM:617140\">ZTTK SYNDROME</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  \n",
       "  <div class=\"card\">\n",
       "    <h2>Variants</h2>\n",
       "    <p>\n",
       "      A total of 35 unique variants\n",
       "      were identified in the cohort. Variants were annotated with respect to NM_138927.4.\n",
       "    </p>\n",
       "    <table>\n",
       "      <caption>\n",
       "        <h3>Top 10 variants</h3>\n",
       "      </caption>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>Variant key</th>\n",
       "          <th>HGVS</th>\n",
       "          <th>Variant Class</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>13</td>\n",
       "          <td>21_33554981_33554985_CAGTT_C</td>\n",
       "          <td>c.5753_5756del (p.Val1918GlufsTer87)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>21_33553078_33553083_TGTATG_T</td>\n",
       "          <td>c.3852_3856del (p.Met1284IlefsTer2)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>21_33552565_33552565_C_T</td>\n",
       "          <td>c.3334C>T (p.Arg1112Ter)</td>\n",
       "          <td>STOP_GAINED</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>21_33551111_33551113_CAG_C</td>\n",
       "          <td>c.1881_1882del (p.Val629AlafsTer56)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>21_33555240_33555241_TG_T</td>\n",
       "          <td>c.6010del (p.Val2004TrpfsTer2)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>SO:1000029_HGNC:11183_SON</td>\n",
       "          <td>1000029_HGNC (None)</td>\n",
       "          <td>TRANSCRIPT_ABLATION</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>1</td>\n",
       "          <td>21_33553905_33553906_TG_T</td>\n",
       "          <td>c.4678del (p.Glu1560LysfsTer63)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>1</td>\n",
       "          <td>21_33553870_33553871_CA_C</td>\n",
       "          <td>c.4640del (p.His1547LeufsTer76)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>1</td>\n",
       "          <td>21_33553586_33553588_TCA_T</td>\n",
       "          <td>c.4358_4359del (p.Thr1453SerfsTer11)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>1</td>\n",
       "          <td>21_33550674_33550675_GC_G</td>\n",
       "          <td>c.1444del (p.Leu482CysfsTer4)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div class=\"card\">\n",
       "    \n",
       "    <h2>Variant effects</h2>\n",
       "    <p>\n",
       "      The effects were predicted for NM_138927.4\n",
       "    </p>\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th>Variant effect</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "          <td>40 (77%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>STOP_GAINED</td>\n",
       "          <td>5 (10%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "          <td>3 (6%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>INFRAME_DELETION</td>\n",
       "          <td>2 (4%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>TRANSCRIPT_ABLATION</td>\n",
       "          <td>2 (4%)</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x16bd7a3c0>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import CohortViewer\n",
    "viewer = CohortViewer(hpo)\n",
    "viewer.process(cohort=cohort, transcript_id=mane_tx_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.model import ProteinMetadata\n",
    "import pandas as pd\n",
    "df = pd.DataFrame([{\"region\":\"1-1\", \"category\": \"Repeat\", \"start\": 1006, \"end\": 1011},\n",
    "{\"region\":\"1-2\", \"category\": \"Repeat\", \"start\": 1014, \"end\": 1019},\n",
    "{\"region\":\"1-3\", \"category\": \"Repeat\", \"start\": 1021, \"end\": 1026},\n",
    "{\"category\": \"Repeat\", \"start\":1030,\"end\":1035,\"region\":\"1-4\"},\n",
    "{\"category\": \"Repeat\", \"start\":1038,\"end\":1043,\"region\":\"1-5\"},\n",
    "{\"category\": \"Repeat\", \"start\":1046,\"end\":1051,\"region\":\"1-6\"},\n",
    "{\"category\": \"Repeat\", \"start\":1055,\"end\":1060,\"region\":\"1-7\"},\n",
    "{\"category\": \"Repeat\", \"start\":1063,\"end\":1068,\"region\":\"1-8\"},\n",
    "{\"category\": \"Repeat\", \"start\":1071,\"end\":1076,\"region\":\"1-9\"},\n",
    "{\"category\": \"Repeat\", \"start\":1080,\"end\":1085,\"region\":\"1-10\"},\n",
    "{\"category\": \"Repeat\", \"start\":1089,\"end\":1094,\"region\":\"1-11\"},\n",
    "{\"category\": \"Repeat\", \"start\":1100,\"end\":1105,\"region\":\"1-12\"},\n",
    "{\"category\": \"Repeat\", \"start\":1111,\"end\":1116,\"region\":\"1-13\"},\n",
    "{\"category\": \"Repeat\", \"start\":1121,\"end\":1126,\"region\":\"1-14\"},\n",
    "{\"category\": \"Repeat\", \"start\":1925,\"end\":1931,\"region\":\"2-1\"},\n",
    "{\"category\": \"Repeat\", \"start\":1934,\"end\":1952,\"region\":\"3-1\"},\n",
    "{\"category\": \"Repeat\", \"start\":1953,\"end\":1959,\"region\":\"2-2\"},\n",
    "{\"category\": \"Repeat\", \"start\":1960,\"end\":1966,\"region\":\"2-3\"},\n",
    "{\"category\": \"Repeat\", \"start\":1967,\"end\":1973,\"region\":\"2-4\"},\n",
    "{\"category\": \"Repeat\", \"start\":1974,\"end\":1980,\"region\":\"2-5\"},\n",
    "{\"category\": \"Repeat\", \"start\":1981,\"end\":1987,\"region\":\"2-6\"},\n",
    "{\"category\": \"Repeat\", \"start\":1988,\"end\":1994,\"region\":\"2-7\"},\n",
    "{\"category\": \"Repeat\", \"start\":1995,\"end\":2013,\"region\":\"3-2\"},\n",
    "{\"category\":\"Domain\",\"start\":2305,\"end\":2351,\"region\":\"G-patch\"},\n",
    "{\"category\":\"Domain\",\"start\":2371,\"end\":2426,\"region\":\"DRBM\"},\n",
    "{\"category\":\"Region\",\"start\":24,\"end\":56,\"region\":\"Disordered\"},\n",
    "{\"category\":\"Region\",\"start\":77,\"end\":155,\"region\":\"Disordered\"},\n",
    "{\"category\":\"Region\",\"start\":305,\"end\":328,\"region\":\"Disordered\"},\n",
    "{\"category\":\"Region\",\"start\":406,\"end\":442,\"region\":\"Disordered\"},\n",
    "{\"category\":\"Region\",\"start\":726,\"end\":895,\"region\":\"17 X 10 AA tandem repeats of L-A-[ST]-[NSG]-[TS]-MDSQM\"},\n",
    "{\"category\":\"Region\",\"start\":912,\"end\":988,\"region\":\"11 X 7 AA tandem repeats of [DR]-P-Y-R-[LI][AG][QHP]\"},\n",
    "{\"category\":\"Region\",\"start\":1006,\"end\":1126,\"region\":\"14 X 6 AA repeats of [ED]-R-S-M-M-S\"},\n",
    "{\"category\":\"Region\",\"start\":1144,\"end\":1236,\"region\":\"Disordered\",\n",
    "\"evidences\":[{\"evidenceCode\":\"ECO:0000256\",\"source\":\"SAM\"},\n",
    "{\"category\":\"Region\",\"start\":1147,\"end\":1179,\"region\":\"3 X 11 AA tandem repats of P-P-L-P-P-E-E-P-P-[TME]-[MTG]\"},\n",
    "{\"category\":\"Region\",\"start\":1359,\"end\":1390,\"region\":\"4 X 8 AA tandem repeats of V-L-E-SS-[AVT]-VT\"},\n",
    "{\"category\":\"Region\",\"start\":1645,\"end\":1722,\"region\":\"Disordered\"},\n",
    "{\"category\":\"Region\",\"start\":1754,\"end\":2054,\"region\":\"Disordered\"},\n",
    "{\"category\":\"Region\",\"start\":1925,\"end\":1994,\"region\":\"7 X 7 AA repeats of P-S-R-R-S-R-[TS]\"},\n",
    "{\"category\":\"Region\",\"start\":1934,\"end\":2013,\"region\":\"2 X 19 AA repeats of P-S-R-R-R-R-S-R-S-V-V-R-R-R-S-F-S-I-S\"},\n",
    "{\"category\":\"Region\",\"start\":2013,\"end\":2039,\"region\":\"3 X tandem repeats of [ST]-P-[VLI]-R-[RL]-[RK]-[RF]-S-R\"},\n",
    "{\"category\":\"Region\",\"start\":2200,\"end\":2220,\"region\":\"Disordered\"},\n",
    "{\"category\":\"Compositional bias\",\"start\":24,\"end\":45,\"region\":\"Polar residues\"},\n",
    "{\"category\":\"Compositional bias\",\"start\":77,\"end\":91,\"region\":\"Basic and acidic residues\"},\n",
    "{\"category\":\"Compositional bias\",\"start\":107,\"end\":127,\"region\":\"Basic residues\"},\n",
    "{\"category\":\"Compositional bias\",\"start\":128,\"end\":155,\"region\":\"Basic and acidic residues\"},\n",
    "{\"category\":\"Compositional bias\",\"start\":1147,\"end\":1180,\"region\":\"Pro residues\"}]},\n",
    "{\"category\":\"Compositional bias\",\"start\":1186,\"end\":1236,\"region\":\"Polar residues\"},\n",
    "{\"category\":\"Compositional bias\",\"start\":1780,\"end\":1827,\"region\":\"Basic and acidic residues\"},\n",
    "{\"category\":\"Compositional bias\",\"start\":1841,\"end\":1914,\"region\":\"Basic residues\"},\n",
    "{\"category\":\"Compositional bias\",\"start\":1921,\"end\":1945,\"region\":\"Basic residues\"},\n",
    "{\"category\":\"Compositional bias\",\"start\":1958,\"end\":2006,\"region\":\"Basic residues\"},\n",
    "{\"category\":\"Compositional bias\",\"start\":2019,\"end\":2043,\"region\":\"Basic residues\"}\n",
    "])\n",
    "protein_meta = ProteinMetadata.from_feature_frame(protein_id=mane_protein_id, label=\"SON\", features=df, protein_length=2426)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "zero-size array to reduction operation maximum which has no identity",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[16], line 12\u001b[0m\n\u001b[1;32m     10\u001b[0m son_fig, ax \u001b[38;5;241m=\u001b[39m plt\u001b[38;5;241m.\u001b[39msubplots(figsize\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m15\u001b[39m, \u001b[38;5;241m8\u001b[39m))\n\u001b[1;32m     11\u001b[0m visualizer \u001b[38;5;241m=\u001b[39m ProteinVisualizer()\n\u001b[0;32m---> 12\u001b[0m \u001b[43mvisualizer\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdraw_protein\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     13\u001b[0m \u001b[43m     \u001b[49m\u001b[43mcohort\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     14\u001b[0m \u001b[43m    \u001b[49m\u001b[43mprotein_meta\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     15\u001b[0m \u001b[43m   \u001b[49m\n\u001b[1;32m     16\u001b[0m \u001b[43m    \u001b[49m\u001b[43max\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43max\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     17\u001b[0m \u001b[43m)\u001b[49m\n\u001b[1;32m     18\u001b[0m son_fig\u001b[38;5;241m.\u001b[39mtight_layout()\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.13/site-packages/gpsea/view/_protein_visualizer.py:89\u001b[0m, in \u001b[0;36mProteinVisualizer.draw_protein\u001b[0;34m(self, cohort, protein_metadata, ax, **kwargs)\u001b[0m\n\u001b[1;32m     78\u001b[0m feature_handler \u001b[38;5;241m=\u001b[39m DrawableProteinFeatureHandler(\n\u001b[1;32m     79\u001b[0m     starts,\n\u001b[1;32m     80\u001b[0m     ends,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     83\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_available_colors,\n\u001b[1;32m     84\u001b[0m )\n\u001b[1;32m     85\u001b[0m max_overlapping_features \u001b[38;5;241m=\u001b[39m sweep_line(\n\u001b[1;32m     86\u001b[0m     [(f\u001b[38;5;241m.\u001b[39mmin_pos_abs, f\u001b[38;5;241m.\u001b[39mmax_pos_abs) \u001b[38;5;28;01mfor\u001b[39;00m f \u001b[38;5;129;01min\u001b[39;00m feature_handler\u001b[38;5;241m.\u001b[39mfeatures]  \u001b[38;5;66;03m# define intervals\u001b[39;00m\n\u001b[1;32m     87\u001b[0m )\n\u001b[0;32m---> 89\u001b[0m variant_handler \u001b[38;5;241m=\u001b[39m \u001b[43mDrawableProteinVariantHandler\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     90\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcohort\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcohort\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     91\u001b[0m \u001b[43m    \u001b[49m\u001b[43mprotein_meta\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mprotein_metadata\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     92\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     94\u001b[0m x_ticks \u001b[38;5;241m=\u001b[39m generate_ticks(apprx_n_ticks\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m6\u001b[39m, \u001b[38;5;28mmin\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m, \u001b[38;5;28mmax\u001b[39m\u001b[38;5;241m=\u001b[39mprotein_metadata\u001b[38;5;241m.\u001b[39mprotein_length)\n\u001b[1;32m     95\u001b[0m y_ticks \u001b[38;5;241m=\u001b[39m generate_ticks(apprx_n_ticks\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m5\u001b[39m, \u001b[38;5;28mmin\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0\u001b[39m, \u001b[38;5;28mmax\u001b[39m\u001b[38;5;241m=\u001b[39mvariant_handler\u001b[38;5;241m.\u001b[39mmax_marker_count)\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.13/site-packages/gpsea/view/_protein_visualizer.py:417\u001b[0m, in \u001b[0;36mDrawableProteinVariantHandler.__init__\u001b[0;34m(self, cohort, protein_meta, aggregation_method)\u001b[0m\n\u001b[1;32m    373\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mUnsupported aggregation method \u001b[39m\u001b[38;5;132;01m{\u001b[39;00maggregation_method\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m    375\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mvariant_effect2color \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m    376\u001b[0m     VariantEffect\u001b[38;5;241m.\u001b[39mTRANSCRIPT_ABLATION: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m#ff0000\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    377\u001b[0m     VariantEffect\u001b[38;5;241m.\u001b[39mSPLICE_ACCEPTOR_VARIANT: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m#00ff00\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    415\u001b[0m     VariantEffect\u001b[38;5;241m.\u001b[39mSEQUENCE_VARIANT: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m#33ff00\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    416\u001b[0m }\n\u001b[0;32m--> 417\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmax_marker_count \u001b[38;5;241m=\u001b[39m \u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_pvis\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmarker_counts\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    418\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39maggregation_method \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstandard\u001b[39m\u001b[38;5;124m'\u001b[39m:\n\u001b[1;32m    419\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mvariants \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_generate_variant_markers()\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.13/site-packages/numpy/_core/fromnumeric.py:3164\u001b[0m, in \u001b[0;36mmax\u001b[0;34m(a, axis, out, keepdims, initial, where)\u001b[0m\n\u001b[1;32m   3052\u001b[0m \u001b[38;5;129m@array_function_dispatch\u001b[39m(_max_dispatcher)\n\u001b[1;32m   3053\u001b[0m \u001b[38;5;129m@set_module\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mnumpy\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m   3054\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21mmax\u001b[39m(a, axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, out\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, keepdims\u001b[38;5;241m=\u001b[39mnp\u001b[38;5;241m.\u001b[39m_NoValue, initial\u001b[38;5;241m=\u001b[39mnp\u001b[38;5;241m.\u001b[39m_NoValue,\n\u001b[1;32m   3055\u001b[0m          where\u001b[38;5;241m=\u001b[39mnp\u001b[38;5;241m.\u001b[39m_NoValue):\n\u001b[1;32m   3056\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m   3057\u001b[0m \u001b[38;5;124;03m    Return the maximum of an array or maximum along an axis.\u001b[39;00m\n\u001b[1;32m   3058\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   3162\u001b[0m \u001b[38;5;124;03m    5\u001b[39;00m\n\u001b[1;32m   3163\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m-> 3164\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_wrapreduction\u001b[49m\u001b[43m(\u001b[49m\u001b[43ma\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmaximum\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mmax\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3165\u001b[0m \u001b[43m                          \u001b[49m\u001b[43mkeepdims\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkeepdims\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43minitial\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minitial\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mwhere\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mwhere\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.13/site-packages/numpy/_core/fromnumeric.py:86\u001b[0m, in \u001b[0;36m_wrapreduction\u001b[0;34m(obj, ufunc, method, axis, dtype, out, **kwargs)\u001b[0m\n\u001b[1;32m     83\u001b[0m         \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m     84\u001b[0m             \u001b[38;5;28;01mreturn\u001b[39;00m reduction(axis\u001b[38;5;241m=\u001b[39maxis, out\u001b[38;5;241m=\u001b[39mout, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mpasskwargs)\n\u001b[0;32m---> 86\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mufunc\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mreduce\u001b[49m\u001b[43m(\u001b[49m\u001b[43mobj\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mout\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mpasskwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[0;31mValueError\u001b[0m: zero-size array to reduction operation maximum which has no identity"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAABMkAAAKZCAYAAACiDnxZAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjAsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvlHJYcgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAJoRJREFUeJzt3XtsV/X9+PFXASmaCepQUMThvKFRQUERL1tMUBKNzj/MEI0Q5iVOJUrnBnihXqY4dYYlVIm3uH8YTKPGCIEpapyTjAiaaCYaRS0hcpuBMlRQ+Hxzzu/XjmpxFEtbeD0eySdyTs9pz8fkTcuz7/M+VZVKpRIAAAAAkFiXjr4AAAAAAOhoIhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOm1OpK99tprccEFF8QhhxwSVVVV8dxzz/3Pc1599dU4+eSTo7q6Oo488sh48sknd/Z6AQAAAKDjI9nGjRtj0KBBUVdXt0PHf/zxx3H++efH2WefHW+//XbceOONceWVV8b8+fN35noBAAAAoM1VVSqVyk6fXFUVzz77bFx00UXbPWbixIkxZ86cePfdd5v2XXLJJbFu3bqYN2/ezn5pAAAAAGgz3WIXW7hwYYwYMaLZvpEjR5YzyrZn06ZN5avR1q1b4/PPP48f//jHZZgDAAAAIKdKpRIbNmwolwLr0qXL7hPJVq5cGX369Gm2r9huaGiIL7/8Mvbee+/vnDN16tS44447dvWlAQAAALCbWr58eRx66KG7TyTbGZMnT46ampqm7fXr18dhhx1WvvmePXt26LUBAAAA0HGKiVf9+/ePfffdt00/7y6PZH379o1Vq1Y121dsF7GrpVlkheIpmMXr24pzRDIAAAAAqtp4Sa62u3FzO4YPHx4LFixotu/FF18s9wMAAABAZ9DqSPaf//wn3n777fJV+Pjjj8s/19fXN90qOWbMmKbjr7nmmli2bFn87ne/i6VLl8ZDDz0Uf/3rX2PChAlt+T4AAAAAoP0i2ZtvvhknnXRS+SoUa4cVf54yZUq5/dlnnzUFs8Lhhx8ec+bMKWePDRo0KP74xz/GY489Vj7hEgAAAAA6g6pK8dzM3WBBtl69epUL+FuTDAAAACCvhl3UiXb5mmQAAAAA0NmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6OxXJ6urqYsCAAdGjR48YNmxYLFq06HuPnzZtWhxzzDGx9957R//+/WPChAnx1Vdf7ew1AwAAAEDHRrLZs2dHTU1N1NbWxpIlS2LQoEExcuTIWL16dYvHz5w5MyZNmlQe/95778Xjjz9efo6bb765La4fAAAAANo/kj344INx1VVXxbhx4+K4446LGTNmxD777BNPPPFEi8e/8cYbccYZZ8Sll15azj4799xzY/To0f9z9hkAAAAAdMpItnnz5li8eHGMGDHiv5+gS5dye+HChS2ec/rpp5fnNEaxZcuWxdy5c+O8887b7tfZtGlTNDQ0NHsBAAAAwK7SrTUHr127NrZs2RJ9+vRptr/YXrp0aYvnFDPIivPOPPPMqFQq8c0338Q111zzvbdbTp06Ne64447WXBoAAAAAdN6nW7766qtxzz33xEMPPVSuYfbMM8/EnDlz4q677truOZMnT47169c3vZYvX76rLxMAAACAxFo1k6x3797RtWvXWLVqVbP9xXbfvn1bPOe2226Lyy+/PK688spy+4QTToiNGzfG1VdfHbfcckt5u+a3VVdXly8AAAAA6HQzybp37x5DhgyJBQsWNO3bunVruT18+PAWz/niiy++E8KK0FYobr8EAAAAgN1qJlmhpqYmxo4dG0OHDo1TTz01pk2bVs4MK552WRgzZkz069evXFescMEFF5RPxDzppJNi2LBh8eGHH5azy4r9jbEMAAAAAHarSDZq1KhYs2ZNTJkyJVauXBmDBw+OefPmNS3mX19f32zm2K233hpVVVXlf1esWBEHHnhgGcjuvvvutn0nAAAAALCTqiq7wT2PDQ0N0atXr3IR/549e3b05QAAAACwh3WiXf50SwAAAADo7EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAIL2dimR1dXUxYMCA6NGjRwwbNiwWLVr0vcevW7currvuujj44IOjuro6jj766Jg7d+7OXjMAAAAAtKlurT1h9uzZUVNTEzNmzCgD2bRp02LkyJHx/vvvx0EHHfSd4zdv3hznnHNO+bGnn346+vXrF59++mnst99+bfUeAAAAAOAHqapUKpXWnFCEsVNOOSWmT59ebm/dujX69+8f48ePj0mTJn3n+CKm3X///bF06dLYa6+9duoiGxoaolevXrF+/fro2bPnTn0OAAAAAHZ/DbuoE7XqdstiVtjixYtjxIgR//0EXbqU2wsXLmzxnOeffz6GDx9e3m7Zp0+fOP744+Oee+6JLVu2/PCrBwAAAID2vt1y7dq1ZdwqYte2iu1iplhLli1bFi+//HJcdtll5TpkH374YVx77bXx9ddfR21tbYvnbNq0qXxtWwgBAAAAYLd9umVxO2axHtkjjzwSQ4YMiVGjRsUtt9xS3oa5PVOnTi2nzTW+its5AQAAAKBTRLLevXtH165dY9WqVc32F9t9+/Zt8ZziiZbF0yyL8xode+yxsXLlyvL2zZZMnjy5vK+08bV8+fLWXCYAAAAA7LpI1r1793I22IIFC5rNFCu2i3XHWnLGGWeUt1gWxzX64IMPynhWfL6WVFdXlwuvbfsCAAAAgE5zu2VNTU08+uij8ec//znee++9+PWvfx0bN26McePGlR8fM2ZMOROsUfHxzz//PG644YYyjs2ZM6dcuL9YyB8AAAAAdruF+wvFmmJr1qyJKVOmlLdMDh48OObNm9e0mH99fX35xMtGxXpi8+fPjwkTJsSJJ54Y/fr1K4PZxIkT2/adAAAAAMBOqqpUKpXo5IqnWxYL+Bfrk7n1EgAAACCvhl3UiXb50y0BAAAAoLMTyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0tupSFZXVxcDBgyIHj16xLBhw2LRokU7dN6sWbOiqqoqLrroop35sgAAAADQOSLZ7Nmzo6amJmpra2PJkiUxaNCgGDlyZKxevfp7z/vkk0/ipptuirPOOuuHXC8AAAAAdHwke/DBB+Oqq66KcePGxXHHHRczZsyIffbZJ5544ontnrNly5a47LLL4o477oif/vSnP/SaAQAAAKDjItnmzZtj8eLFMWLEiP9+gi5dyu2FCxdu97w777wzDjrooLjiiit26Ots2rQpGhoamr0AAAAAoFNEsrVr15azwvr06dNsf7G9cuXKFs95/fXX4/HHH49HH310h7/O1KlTo1evXk2v/v37t+YyAQAAAKDzPN1yw4YNcfnll5eBrHfv3jt83uTJk2P9+vVNr+XLl+/KywQAAAAguW6tObgIXV27do1Vq1Y1219s9+3b9zvHf/TRR+WC/RdccEHTvq1bt/6/L9ytW7z//vtxxBFHfOe86urq8gUAAAAAnW4mWffu3WPIkCGxYMGCZtGr2B4+fPh3jh84cGC888478fbbbze9Lrzwwjj77LPLP7uNEgAAAIDdbiZZoaamJsaOHRtDhw6NU089NaZNmxYbN24sn3ZZGDNmTPTr169cV6xHjx5x/PHHNzt/v/32K//77f0AAAAAsNtEslGjRsWaNWtiypQp5WL9gwcPjnnz5jUt5l9fX18+8RIAAAAAdhdVlUqlEp1cQ0ND+ZTLYhH/nj17dvTlAAAAALCHdSJTvgAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0tupSFZXVxcDBgyIHj16xLBhw2LRokXbPfbRRx+Ns846K/bff//yNWLEiO89HgAAAAA6fSSbPXt21NTURG1tbSxZsiQGDRoUI0eOjNWrV7d4/KuvvhqjR4+OV155JRYuXBj9+/ePc889N1asWNEW1w8AAAAAP1hVpVKptOaEYubYKaecEtOnTy+3t27dWoav8ePHx6RJk/7n+Vu2bClnlBXnjxkzZoe+ZkNDQ/Tq1SvWr18fPXv2bM3lAgAAALAHadhFnahVM8k2b94cixcvLm+ZbPoEXbqU28UssR3xxRdfxNdffx0HHHDAdo/ZtGlT+Ya3fQEAAADArtKqSLZ27dpyJlifPn2a7S+2V65cuUOfY+LEiXHIIYc0C23fNnXq1LIINr6KmWoAAAAAsEc83fLee++NWbNmxbPPPlsu+r89kydPLqfMNb6WL1/enpcJAAAAQDLdWnNw7969o2vXrrFq1apm+4vtvn37fu+5DzzwQBnJXnrppTjxxBO/99jq6uryBQAAAACdbiZZ9+7dY8iQIbFgwYKmfcXC/cX28OHDt3vefffdF3fddVfMmzcvhg4d+sOuGAAAAAA6ciZZoaamJsaOHVvGrlNPPTWmTZsWGzdujHHjxpUfL55Y2a9fv3JdscIf/vCHmDJlSsycOTMGDBjQtHbZj370o/IFAAAAALtdJBs1alSsWbOmDF9F8Bo8eHA5Q6xxMf/6+vryiZeNHn744fKpmBdffHGzz1NbWxu33357W7wHAAAAAPhBqiqVSiU6uYaGhvIpl8Ui/j179uzoywEAAABgD+tE7fp0SwAAAADojEQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAIL2dimR1dXUxYMCA6NGjRwwbNiwWLVr0vcc/9dRTMXDgwPL4E044IebOnbuz1wsAAAAAHR/JZs+eHTU1NVFbWxtLliyJQYMGxciRI2P16tUtHv/GG2/E6NGj44orroi33norLrroovL17rvvtsX1AwAAAMAPVlWpVCqtOaGYOXbKKafE9OnTy+2tW7dG//79Y/z48TFp0qTvHD9q1KjYuHFjvPDCC037TjvttBg8eHDMmDFjh75mQ0ND9OrVK9avXx89e/ZszeUCAAAAsAdp2EWdqFtrDt68eXMsXrw4Jk+e3LSvS5cuMWLEiFi4cGGL5xT7i5ln2ypmnj333HPb/TqbNm0qX42KN934PwEAAACAvBr+fx9q5byvto1ka9eujS1btkSfPn2a7S+2ly5d2uI5K1eubPH4Yv/2TJ06Ne64447v7C9mrAEAAADAv//973JGWYdEsvZSzFTbdvbZunXr4ic/+UnU19e36ZsH2qbgFwF7+fLlboeGTsgYhc7L+ITOzRiFzqu44/Cwww6LAw44oE0/b6siWe/evaNr166xatWqZvuL7b59+7Z4TrG/NccXqqury9e3FYHMX07QORVj0/iEzssYhc7L+ITOzRiFzqtYAqxNP19rDu7evXsMGTIkFixY0LSvWLi/2B4+fHiL5xT7tz2+8OKLL273eAAAAABob62+3bK4DXLs2LExdOjQOPXUU2PatGnl0yvHjRtXfnzMmDHRr1+/cl2xwg033BA///nP449//GOcf/75MWvWrHjzzTfjkUceaft3AwAAAADtEclGjRoVa9asiSlTppSL7w8ePDjmzZvXtDh/sW7YttPdTj/99Jg5c2bceuutcfPNN8dRRx1VPtny+OOP3+GvWdx6WVtb2+ItmEDHMj6hczNGofMyPqFzM0Yh3/isqrT18zIBAAAAYDfTtiucAQAAAMBuSCQDAAAAID2RDAAAAID0RDIAAAAA0us0kayuri4GDBgQPXr0iGHDhsWiRYu+9/innnoqBg4cWB5/wgknxNy5c9vtWiGb1ozPRx99NM4666zYf//9y9eIESP+53gG2vd7aKNZs2ZFVVVVXHTRRbv8GiGr1o7PdevWxXXXXRcHH3xw+cSuo48+2s+50InG6LRp0+KYY46JvffeO/r37x8TJkyIr776qt2uF7J47bXX4oILLohDDjmk/Hn1ueee+5/nvPrqq3HyySeX3z+PPPLIePLJJ3fPSDZ79uyoqakpH9+5ZMmSGDRoUIwcOTJWr17d4vFvvPFGjB49Oq644op46623yh/ui9e7777b7tcOe7rWjs/iL6ZifL7yyiuxcOHC8oeHc889N1asWNHu1w4ZtHaMNvrkk0/ipptuKqM20DnG5+bNm+Occ84px+fTTz8d77//fvnLp379+rX7tUMGrR2jM2fOjEmTJpXHv/fee/H444+Xn+Pmm29u92uHPd3GjRvLMVmE7B3x8ccfx/nnnx9nn312vP3223HjjTfGlVdeGfPnz2/V162qVCqV6GBFsT/llFNi+vTp5fbWrVvLf1iPHz++/Evo20aNGlX+D3vhhRea9p122mkxePDgmDFjRrteO+zpWjs+v23Lli3ljLLi/DFjxrTDFUMuOzNGi3H5s5/9LH71q1/F3//+93Lmyo78dg7YteOz+Dn2/vvvj6VLl8Zee+3VAVcMubR2jF5//fVlHFuwYEHTvt/85jfxz3/+M15//fV2vXbIpKqqKp599tnvvfth4sSJMWfOnGaTpy655JLy59x58+btPjPJit+YLV68uLwlq1GXLl3K7WIWSkuK/dseXyiK//aOB9pvfH7bF198EV9//XUccMABu/BKIaedHaN33nlnHHTQQeWMbKDzjM/nn38+hg8fXt5u2adPnzj++OPjnnvuKcM20PFj9PTTTy/Pabwlc9myZeXt0Oedd167XTcQu7QTdYsOtnbt2vIbf/GDwLaK7eK3aC1ZuXJli8cX+4GOHZ8tFf3iPvJv/4UFdMwYLX7TXdweUkxDBzrX+Cz+wf3yyy/HZZddVv7D+8MPP4xrr722/GVTcXsX0LFj9NJLLy3PO/PMM6O4Ieubb76Ja665xu2W0AlsrxM1NDTEl19+Wa4juFvMJAP2XPfee2+5MHgxNbZYDBXoWBs2bIjLL7+8XOOod+/eHX05wLcUt3oVszwfeeSRGDJkSLnEyC233GI5EegkirV3i9mdDz30ULmG2TPPPFPe3nXXXXd19KUBbaTDZ5IVP6R37do1Vq1a1Wx/sd23b98Wzyn2t+Z4oP3GZ6MHHnigjGQvvfRSnHjiibv4SiGn1o7Rjz76qFwQvHhS0Lb/KC9069atXCT8iCOOaIcrhz3fznwPLZ5oWaxFVpzX6Nhjjy1/O17cGta9e/ddft2Qxc6M0dtuu638ZVOxGHjhhBNOKNfKvvrqq8ugXdyuCXSM7XWinj177vAsskKHj+Lim33xm7JtFz8sfmAvtos1GVpS7N/2+MKLL7643eOB9hufhfvuu6/8jVqxQOLQoUPb6Wohn9aO0YEDB8Y777xT3mrZ+LrwwgubngJULFYMdNz30DPOOKO8xbIxXhc++OCDMp4JZNDxY7RYa/fbIawxaneC5+FBasPbqhNVOoFZs2ZVqqurK08++WTlX//6V+Xqq6+u7LfffpWVK1eWH7/88ssrkyZNajr+H//4R6Vbt26VBx54oPLee+9VamtrK3vttVflnXfe6cB3AXum1o7Pe++9t9K9e/fK008/Xfnss8+aXhs2bOjAdwF7rtaO0W8bO3Zs5Re/+EU7XjHk0drxWV9fX9l3330r119/feX999+vvPDCC5WDDjqo8vvf/74D3wXsuVo7Rot/dxZj9C9/+Utl2bJllb/97W+VI444ovLLX/6yA98F7Jk2bNhQeeutt8pXka4efPDB8s+ffvpp+fFibBZjtFExJvfZZ5/Kb3/727IT1dXVVbp27VqZN29eq75uh99uWSjWW1izZk1MmTKlnE4+ePDgcgZK46Jr9fX1zYp98VSRmTNnxq233louknjUUUeVj64vngAEdOz4fPjhh8tbQi6++OJmn6dYcPj2229v9+uHPV1rxyjQecdnMZtz/vz5MWHChHKpgn79+sUNN9xQPgQH6PgxWvz7s6qqqvzvihUr4sADDyyXMLj77rs78F3AnunNN98s73ZoVFNTU/537Nix8eSTT8Znn31WjtFGhx9+eLlGYPE99E9/+lMceuih8dhjj5VPuGyNqqKUteH7AAAAAIDdjl8tAwAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAARHb/B3bDz3bg6PK3AAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 1500x800 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from gpsea.model.genome import GRCh38\n",
    "from gpsea.model import ProteinMetadata\n",
    "from gpsea.preprocessing import configure_default_protein_metadata_service, VVMultiCoordinateService\n",
    "from gpsea.view import ProteinVisualizer\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "txc_service = VVMultiCoordinateService(genome_build=GRCh38)\n",
    "pms = configure_default_protein_metadata_service()\n",
    "protein_meta = pms.ppr\n",
    "tx_coordinates = txc_service.fetch(mane_tx_id)\n",
    "son_fig, ax = plt.subplots(figsize=(15, 8))\n",
    "visualizer = ProteinVisualizer()\n",
    "visualizer.draw_protein(\n",
    "     cohort,\n",
    "    protein_meta,\n",
    "    ax=ax,\n",
    ")\n",
    "son_fig.tight_layout()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Protein SON: NP_620305.2</h1>\n",
       "<div class=\"card\">\n",
       "  <p>\n",
       "    Protein SON has 52 annotated protein featureses.\n",
       "    0 were found to be located within these features,\n",
       "    and 0 were located elsewhere.\n",
       "    The total length of the protein is 2426 amino acid residues.\n",
       "  </p>\n",
       "  <table>\n",
       "    <tbody>\n",
       "        <tr>\n",
       "          <th>Count</th>\n",
       "          <th>Name</th>\n",
       "          <th>Type</th>\n",
       "          <th>Coordinates</th>\n",
       "          <th>Variants</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Disordered</td>\n",
       "          <td>Region</td>\n",
       "          <td>25 - 56</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Polar residues</td>\n",
       "          <td>Compositional bias</td>\n",
       "          <td>25 - 45</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Disordered</td>\n",
       "          <td>Region</td>\n",
       "          <td>78 - 155</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Basic and acidic residues</td>\n",
       "          <td>Compositional bias</td>\n",
       "          <td>78 - 91</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Basic residues</td>\n",
       "          <td>Compositional bias</td>\n",
       "          <td>108 - 127</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Basic and acidic residues</td>\n",
       "          <td>Compositional bias</td>\n",
       "          <td>129 - 155</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Disordered</td>\n",
       "          <td>Region</td>\n",
       "          <td>306 - 328</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Disordered</td>\n",
       "          <td>Region</td>\n",
       "          <td>407 - 442</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>17 X 10 AA tandem repeats of L-A-[ST]-[NSG]-[TS]-MDSQM</td>\n",
       "          <td>Region</td>\n",
       "          <td>727 - 895</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>11 X 7 AA tandem repeats of [DR]-P-Y-R-[LI][AG][QHP]</td>\n",
       "          <td>Region</td>\n",
       "          <td>913 - 988</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-1</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1007 - 1011</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>14 X 6 AA repeats of [ED]-R-S-M-M-S</td>\n",
       "          <td>Region</td>\n",
       "          <td>1007 - 1126</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-2</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1015 - 1019</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-3</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1022 - 1026</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-4</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1031 - 1035</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-5</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1039 - 1043</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-6</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1047 - 1051</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-7</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1056 - 1060</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-8</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1064 - 1068</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-9</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1072 - 1076</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-10</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1081 - 1085</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-11</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1090 - 1094</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-12</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1101 - 1105</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-13</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1112 - 1116</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>1-14</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1122 - 1126</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Disordered</td>\n",
       "          <td>Region</td>\n",
       "          <td>1145 - 1236</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>3 X 11 AA tandem repats of P-P-L-P-P-E-E-P-P-[TME]-[MTG]</td>\n",
       "          <td>Region</td>\n",
       "          <td>1148 - 1179</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Pro residues</td>\n",
       "          <td>Compositional bias</td>\n",
       "          <td>1148 - 1180</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Polar residues</td>\n",
       "          <td>Compositional bias</td>\n",
       "          <td>1187 - 1236</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>4 X 8 AA tandem repeats of V-L-E-SS-[AVT]-VT</td>\n",
       "          <td>Region</td>\n",
       "          <td>1360 - 1390</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Disordered</td>\n",
       "          <td>Region</td>\n",
       "          <td>1646 - 1722</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Disordered</td>\n",
       "          <td>Region</td>\n",
       "          <td>1755 - 2054</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Basic and acidic residues</td>\n",
       "          <td>Compositional bias</td>\n",
       "          <td>1781 - 1827</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Basic residues</td>\n",
       "          <td>Compositional bias</td>\n",
       "          <td>1842 - 1914</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Basic residues</td>\n",
       "          <td>Compositional bias</td>\n",
       "          <td>1922 - 1945</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>2-1</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1926 - 1931</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>7 X 7 AA repeats of P-S-R-R-S-R-[TS]</td>\n",
       "          <td>Region</td>\n",
       "          <td>1926 - 1994</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>3-1</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1935 - 1952</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>2 X 19 AA repeats of P-S-R-R-R-R-S-R-S-V-V-R-R-R-S-F-S-I-S</td>\n",
       "          <td>Region</td>\n",
       "          <td>1935 - 2013</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>2-2</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1954 - 1959</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Basic residues</td>\n",
       "          <td>Compositional bias</td>\n",
       "          <td>1959 - 2006</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>2-3</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1961 - 1966</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>2-4</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1968 - 1973</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>2-5</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1975 - 1980</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>2-6</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1982 - 1987</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>2-7</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1989 - 1994</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>3-2</td>\n",
       "          <td>Repeat</td>\n",
       "          <td>1996 - 2013</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>3 X tandem repeats of [ST]-P-[VLI]-R-[RL]-[RK]-[RF]-S-R</td>\n",
       "          <td>Region</td>\n",
       "          <td>2014 - 2039</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Basic residues</td>\n",
       "          <td>Compositional bias</td>\n",
       "          <td>2020 - 2043</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>Disordered</td>\n",
       "          <td>Region</td>\n",
       "          <td>2201 - 2220</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>G-patch</td>\n",
       "          <td>Domain</td>\n",
       "          <td>2306 - 2351</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>0</td>\n",
       "          <td>DRBM</td>\n",
       "          <td>Domain</td>\n",
       "          <td>2372 - 2426</td>\n",
       "          <td></td>\n",
       "        </tr>\n",
       "        \n",
       "    </tbody>\n",
       "</table>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x16bdcdbd0>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import ProteinVariantViewer\n",
    "pvv = ProteinVariantViewer(protein_metadata=protein_meta, tx_id=mane_tx_id)\n",
    "pvv.process(cohort=cohort)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Variant alleles</h1>\n",
       "<div class=\"card\">\n",
       "  <table>\n",
       "    <caption>\n",
       "      A total of 35 unique alleles were identified in the cohort.\n",
       "    </caption>\n",
       "\n",
       "    <tbody>\n",
       "      <tr>\n",
       "        <th>Count</th>\n",
       "        <th>Variant key</th>\n",
       "        <th>HGVS</th>\n",
       "        <th>Overlapping Exons</th>\n",
       "        <th>Effects</th>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>13</td>\n",
       "        <td>21_33554981_33554985_CAGTT_C</td>\n",
       "        <td>c.5753_5756del (p.Val1918GlufsTer87)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td>21_33551111_33551113_CAG_C</td>\n",
       "        <td>c.1881_1882del (p.Val629AlafsTer56)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td>21_33552565_33552565_C_T</td>\n",
       "        <td>c.3334C>T (p.Arg1112Ter)</td>\n",
       "        <td>3</td>\n",
       "        <td>stop gained</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td>SO:1000029_HGNC:11183_SON</td>\n",
       "        <td>SV involving SON (p.?)</td>\n",
       "        <td></td>\n",
       "        <td>chromosomal deletion</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td>21_33553078_33553083_TGTATG_T</td>\n",
       "        <td>c.3852_3856del (p.Met1284IlefsTer2)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td>21_33555240_33555241_TG_T</td>\n",
       "        <td>c.6010del (p.Val2004TrpfsTer2)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33549687_33549688_TG_T</td>\n",
       "        <td>c.457del (p.Asp153IlefsTer4)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33550674_33550675_GC_G</td>\n",
       "        <td>c.1444del (p.Leu482CysfsTer4)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33555317_33555318_GC_G</td>\n",
       "        <td>c.6087del (p.Ser2029ArgfsTer22)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33549625_33549625_C_T</td>\n",
       "        <td>c.394C>T (p.Gln132Ter)</td>\n",
       "        <td>3</td>\n",
       "        <td>stop gained</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33553381_33553402_CTGGAGCCTTCGGTTGTGACTG_C</td>\n",
       "        <td>c.4152_4172del (p.Glu1385_Val1391del)</td>\n",
       "        <td>3</td>\n",
       "        <td>inframe deletion</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33552827_33552827_T_TGC</td>\n",
       "        <td>c.3597_3598dup (p.Pro1200ArgfsTer17)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33553905_33553906_TG_T</td>\n",
       "        <td>c.4678del (p.Glu1560LysfsTer63)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33549575_33549579_AAAAC_A</td>\n",
       "        <td>c.348_351del (p.Asn116LysfsTer32)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33552939_33552940_TC_T</td>\n",
       "        <td>c.3711del (p.Ser1238GlnfsTer3)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33550967_33550967_C_G</td>\n",
       "        <td>c.1736C>G (p.Thr579Ser)</td>\n",
       "        <td>3</td>\n",
       "        <td>missense</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33554460_33554461_AC_A</td>\n",
       "        <td>c.5230del (p.Arg1744ValfsTer29)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33551595_33551596_TA_T</td>\n",
       "        <td>c.2365del (p.Ser789AlafsTer8)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33554759_33554759_C_A</td>\n",
       "        <td>c.5528C>A (p.Ser1843Tyr)</td>\n",
       "        <td>3</td>\n",
       "        <td>missense</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33554776_33554778_AAG_A</td>\n",
       "        <td>c.5549_5550del (p.Arg1850IlefsTer3)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33553247_33553248_TG_T</td>\n",
       "        <td>c.4018del (p.Ala1340GlnfsTer26)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33549899_33549899_C_T</td>\n",
       "        <td>c.668C>T (p.Ser223Leu)</td>\n",
       "        <td>3</td>\n",
       "        <td>missense</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33557226_33557227_AC_A</td>\n",
       "        <td>c.6233del (p.Pro2078HisfsTer4)</td>\n",
       "        <td>4</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33553778_33553778_A_AG</td>\n",
       "        <td>c.4549dup (p.Glu1517GlyfsTer6)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33549607_33549608_GA_G</td>\n",
       "        <td>c.384del (p.Lys128AsnfsTer21)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33553586_33553588_TCA_T</td>\n",
       "        <td>c.4358_4359del (p.Thr1453SerfsTer11)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33552303_33552303_G_GA</td>\n",
       "        <td>c.3073dup (p.Met1025AsnfsTer6)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33554228_33554243_TTGATTTACCATCTAA_T</td>\n",
       "        <td>c.4999_5013del (p.Asp1667_Asn1671del)</td>\n",
       "        <td>3</td>\n",
       "        <td>inframe deletion</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33554144_33554149_AACATG_A</td>\n",
       "        <td>c.4919_4923del (p.Asp1640GlyfsTer7)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33555232_33555232_T_TCC</td>\n",
       "        <td>c.6002_6003insCC (p.Arg2002GlnfsTer5)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33549517_33549517_C_T</td>\n",
       "        <td>c.286C>T (p.Gln96Ter)</td>\n",
       "        <td>3</td>\n",
       "        <td>stop gained</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33549494_33549495_GA_G</td>\n",
       "        <td>c.268del (p.Ser90ValfsTer59)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33552434_33552434_C_G</td>\n",
       "        <td>c.3203C>G (p.Ser1068Ter)</td>\n",
       "        <td>3</td>\n",
       "        <td>stop gained</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33553870_33553871_CA_C</td>\n",
       "        <td>c.4640del (p.His1547LeufsTer76)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>21_33553282_33553283_GC_G</td>\n",
       "        <td>c.4055del (p.Pro1352GlnfsTer14)</td>\n",
       "        <td>3</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x16bdcdf90>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import CohortVariantViewer\n",
    "viewer = CohortVariantViewer(tx_id=mane_tx_id)\n",
    "viewer.process(cohort=cohort)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genotype-Phenotype Correlation (GPC) Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.clf import prepare_classifiers_for_terms_of_interest\n",
    "from gpsea.analysis.pcats import configure_hpo_term_analysis\n",
    "\n",
    "analysis = configure_hpo_term_analysis(hpo)\n",
    "pheno_clfs = prepare_classifiers_for_terms_of_interest(\n",
    "    cohort=cohort,\n",
    "    hpo=hpo,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Phenotype testing report</h1>\n",
       "<div class=\"cards\">\n",
       "  <div class=\"card\">\n",
       "    <ul>\n",
       "      <li>Phenotype MTC filter: <em>Independent filtering HPO filter</em></li>\n",
       "      <li>Multiple testing correction: <em>fdr_bh</em></li>\n",
       "    </ul>\n",
       "  </div>\n",
       "  <div class=\"card\">\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr>\n",
       "          <th>Code</th>\n",
       "          <th>Reason</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF01</td>\n",
       "          <td>Skipping term with maximum frequency that was less than threshold 0.4</td>\n",
       "          <td>25</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF03</td>\n",
       "          <td>Skipping term because of a child term with the same individual counts</td>\n",
       "          <td>11</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF08</td>\n",
       "          <td>Skipping general term</td>\n",
       "          <td>52</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF09</td>\n",
       "          <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>\n",
       "          <td>52</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "      <caption>\n",
       "        Performed statistical tests for 26 out of the total of 166 HPO terms.\n",
       "      </caption>\n",
       "    </table>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x16bd5ad70>"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.analysis.clf import monoallelic_classifier\n",
    "from gpsea.analysis.predicate import variant_key\n",
    "from gpsea.view import MtcStatsViewer\n",
    "\n",
    "c5753_5756del = variant_key('21_33554981_33554985_CAGTT_C') # c.5753_5756del\tp.Val1918GlufsTer87\n",
    "c5753_5756del_clf = monoallelic_classifier(\n",
    "    a_predicate=c5753_5756del,\n",
    "    b_predicate=~c5753_5756del,\n",
    "    a_label=\"c.5753_5756del\",\n",
    "    b_label=\"Other\"\n",
    ")\n",
    "c5753_5756del_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_clf=c5753_5756del_clf,\n",
    "    pheno_clfs=pheno_clfs,\n",
    ")\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(c5753_5756del_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Allele group</th>\n",
       "      <th>c.5753_5756del</th>\n",
       "      <th>Other</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Strabismus [HP:0000486]</th>\n",
       "      <td>2/11 (18%)</td>\n",
       "      <td>19/36 (53%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.080929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ventriculomegaly [HP:0002119]</th>\n",
       "      <td>4/11 (36%)</td>\n",
       "      <td>20/33 (61%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.185126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Low-set ears [HP:0000369]</th>\n",
       "      <td>5/12 (42%)</td>\n",
       "      <td>17/31 (55%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.509788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Seizure [HP:0001250]</th>\n",
       "      <td>5/13 (38%)</td>\n",
       "      <td>19/38 (50%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.533920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypotonia [HP:0001252]</th>\n",
       "      <td>7/13 (54%)</td>\n",
       "      <td>24/38 (63%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.743193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Intellectual disability [HP:0001249]</th>\n",
       "      <td>12/12 (100%)</td>\n",
       "      <td>37/38 (97%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neurodevelopmental abnormality [HP:0012759]</th>\n",
       "      <td>13/13 (100%)</td>\n",
       "      <td>38/38 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal conjugate eye movement [HP:0000549]</th>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>19/19 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal cerebral morphology [HP:0002060]</th>\n",
       "      <td>3/3 (100%)</td>\n",
       "      <td>19/19 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal cerebral ventricle morphology [HP:0002118]</th>\n",
       "      <td>4/4 (100%)</td>\n",
       "      <td>20/20 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Decreased body weight [HP:0004325]</th>\n",
       "      <td>4/4 (100%)</td>\n",
       "      <td>21/21 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Visual impairment [HP:0000505]</th>\n",
       "      <td>5/5 (100%)</td>\n",
       "      <td>16/16 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Downslanted palpebral fissures [HP:0000494]</th>\n",
       "      <td>6/11 (55%)</td>\n",
       "      <td>18/36 (50%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Slanting of the palpebral fissure [HP:0200006]</th>\n",
       "      <td>6/6 (100%)</td>\n",
       "      <td>18/18 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal brain morphology [HP:0012443]</th>\n",
       "      <td>6/6 (100%)</td>\n",
       "      <td>28/28 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Morphological central nervous system abnormality [HP:0002011]</th>\n",
       "      <td>6/6 (100%)</td>\n",
       "      <td>28/28 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Intellectual disability, severe [HP:0010864]</th>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>15/16 (94%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Short stature [HP:0004322]</th>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>20/20 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal muscle tone [HP:0003808]</th>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>24/24 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Feeding difficulties [HP:0011968]</th>\n",
       "      <td>8/12 (67%)</td>\n",
       "      <td>22/36 (61%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal location of ears [HP:0000357]</th>\n",
       "      <td>8/8 (100%)</td>\n",
       "      <td>19/19 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abdominal symptom [HP:0011458]</th>\n",
       "      <td>8/8 (100%)</td>\n",
       "      <td>22/22 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal eyelid morphology [HP:0000492]</th>\n",
       "      <td>8/8 (100%)</td>\n",
       "      <td>25/25 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal ocular adnexa morphology [HP:0030669]</th>\n",
       "      <td>8/8 (100%)</td>\n",
       "      <td>26/26 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Delayed speech and language development [HP:0000750]</th>\n",
       "      <td>9/9 (100%)</td>\n",
       "      <td>18/18 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal communication [HP:0034434]</th>\n",
       "      <td>9/9 (100%)</td>\n",
       "      <td>18/18 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Allele group                                       c.5753_5756del  \\\n",
       "Strabismus [HP:0000486]                                2/11 (18%)   \n",
       "Ventriculomegaly [HP:0002119]                          4/11 (36%)   \n",
       "Low-set ears [HP:0000369]                              5/12 (42%)   \n",
       "Seizure [HP:0001250]                                   5/13 (38%)   \n",
       "Hypotonia [HP:0001252]                                 7/13 (54%)   \n",
       "Intellectual disability [HP:0001249]                 12/12 (100%)   \n",
       "Neurodevelopmental abnormality [HP:0012759]          13/13 (100%)   \n",
       "Abnormal conjugate eye movement [HP:0000549]           2/2 (100%)   \n",
       "Abnormal cerebral morphology [HP:0002060]              3/3 (100%)   \n",
       "Abnormal cerebral ventricle morphology [HP:0002...     4/4 (100%)   \n",
       "Decreased body weight [HP:0004325]                     4/4 (100%)   \n",
       "Visual impairment [HP:0000505]                         5/5 (100%)   \n",
       "Downslanted palpebral fissures [HP:0000494]            6/11 (55%)   \n",
       "Slanting of the palpebral fissure [HP:0200006]         6/6 (100%)   \n",
       "Abnormal brain morphology [HP:0012443]                 6/6 (100%)   \n",
       "Morphological central nervous system abnormalit...     6/6 (100%)   \n",
       "Intellectual disability, severe [HP:0010864]           7/7 (100%)   \n",
       "Short stature [HP:0004322]                             7/7 (100%)   \n",
       "Abnormal muscle tone [HP:0003808]                      7/7 (100%)   \n",
       "Feeding difficulties [HP:0011968]                      8/12 (67%)   \n",
       "Abnormal location of ears [HP:0000357]                 8/8 (100%)   \n",
       "Abdominal symptom [HP:0011458]                         8/8 (100%)   \n",
       "Abnormal eyelid morphology [HP:0000492]                8/8 (100%)   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]         8/8 (100%)   \n",
       "Delayed speech and language development [HP:000...     9/9 (100%)   \n",
       "Abnormal communication [HP:0034434]                    9/9 (100%)   \n",
       "\n",
       "Allele group                                               Other  \\\n",
       "Strabismus [HP:0000486]                              19/36 (53%)   \n",
       "Ventriculomegaly [HP:0002119]                        20/33 (61%)   \n",
       "Low-set ears [HP:0000369]                            17/31 (55%)   \n",
       "Seizure [HP:0001250]                                 19/38 (50%)   \n",
       "Hypotonia [HP:0001252]                               24/38 (63%)   \n",
       "Intellectual disability [HP:0001249]                 37/38 (97%)   \n",
       "Neurodevelopmental abnormality [HP:0012759]         38/38 (100%)   \n",
       "Abnormal conjugate eye movement [HP:0000549]        19/19 (100%)   \n",
       "Abnormal cerebral morphology [HP:0002060]           19/19 (100%)   \n",
       "Abnormal cerebral ventricle morphology [HP:0002...  20/20 (100%)   \n",
       "Decreased body weight [HP:0004325]                  21/21 (100%)   \n",
       "Visual impairment [HP:0000505]                      16/16 (100%)   \n",
       "Downslanted palpebral fissures [HP:0000494]          18/36 (50%)   \n",
       "Slanting of the palpebral fissure [HP:0200006]      18/18 (100%)   \n",
       "Abnormal brain morphology [HP:0012443]              28/28 (100%)   \n",
       "Morphological central nervous system abnormalit...  28/28 (100%)   \n",
       "Intellectual disability, severe [HP:0010864]         15/16 (94%)   \n",
       "Short stature [HP:0004322]                          20/20 (100%)   \n",
       "Abnormal muscle tone [HP:0003808]                   24/24 (100%)   \n",
       "Feeding difficulties [HP:0011968]                    22/36 (61%)   \n",
       "Abnormal location of ears [HP:0000357]              19/19 (100%)   \n",
       "Abdominal symptom [HP:0011458]                      22/22 (100%)   \n",
       "Abnormal eyelid morphology [HP:0000492]             25/25 (100%)   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]      26/26 (100%)   \n",
       "Delayed speech and language development [HP:000...  18/18 (100%)   \n",
       "Abnormal communication [HP:0034434]                 18/18 (100%)   \n",
       "\n",
       "Allele group                                        Corrected p values  \\\n",
       "Strabismus [HP:0000486]                                            1.0   \n",
       "Ventriculomegaly [HP:0002119]                                      1.0   \n",
       "Low-set ears [HP:0000369]                                          1.0   \n",
       "Seizure [HP:0001250]                                               1.0   \n",
       "Hypotonia [HP:0001252]                                             1.0   \n",
       "Intellectual disability [HP:0001249]                               1.0   \n",
       "Neurodevelopmental abnormality [HP:0012759]                        1.0   \n",
       "Abnormal conjugate eye movement [HP:0000549]                       1.0   \n",
       "Abnormal cerebral morphology [HP:0002060]                          1.0   \n",
       "Abnormal cerebral ventricle morphology [HP:0002...                 1.0   \n",
       "Decreased body weight [HP:0004325]                                 1.0   \n",
       "Visual impairment [HP:0000505]                                     1.0   \n",
       "Downslanted palpebral fissures [HP:0000494]                        1.0   \n",
       "Slanting of the palpebral fissure [HP:0200006]                     1.0   \n",
       "Abnormal brain morphology [HP:0012443]                             1.0   \n",
       "Morphological central nervous system abnormalit...                 1.0   \n",
       "Intellectual disability, severe [HP:0010864]                       1.0   \n",
       "Short stature [HP:0004322]                                         1.0   \n",
       "Abnormal muscle tone [HP:0003808]                                  1.0   \n",
       "Feeding difficulties [HP:0011968]                                  1.0   \n",
       "Abnormal location of ears [HP:0000357]                             1.0   \n",
       "Abdominal symptom [HP:0011458]                                     1.0   \n",
       "Abnormal eyelid morphology [HP:0000492]                            1.0   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]                     1.0   \n",
       "Delayed speech and language development [HP:000...                 1.0   \n",
       "Abnormal communication [HP:0034434]                                1.0   \n",
       "\n",
       "Allele group                                        p values  \n",
       "Strabismus [HP:0000486]                             0.080929  \n",
       "Ventriculomegaly [HP:0002119]                       0.185126  \n",
       "Low-set ears [HP:0000369]                           0.509788  \n",
       "Seizure [HP:0001250]                                0.533920  \n",
       "Hypotonia [HP:0001252]                              0.743193  \n",
       "Intellectual disability [HP:0001249]                1.000000  \n",
       "Neurodevelopmental abnormality [HP:0012759]         1.000000  \n",
       "Abnormal conjugate eye movement [HP:0000549]        1.000000  \n",
       "Abnormal cerebral morphology [HP:0002060]           1.000000  \n",
       "Abnormal cerebral ventricle morphology [HP:0002...  1.000000  \n",
       "Decreased body weight [HP:0004325]                  1.000000  \n",
       "Visual impairment [HP:0000505]                      1.000000  \n",
       "Downslanted palpebral fissures [HP:0000494]         1.000000  \n",
       "Slanting of the palpebral fissure [HP:0200006]      1.000000  \n",
       "Abnormal brain morphology [HP:0012443]              1.000000  \n",
       "Morphological central nervous system abnormalit...  1.000000  \n",
       "Intellectual disability, severe [HP:0010864]        1.000000  \n",
       "Short stature [HP:0004322]                          1.000000  \n",
       "Abnormal muscle tone [HP:0003808]                   1.000000  \n",
       "Feeding difficulties [HP:0011968]                   1.000000  \n",
       "Abnormal location of ears [HP:0000357]              1.000000  \n",
       "Abdominal symptom [HP:0011458]                      1.000000  \n",
       "Abnormal eyelid morphology [HP:0000492]             1.000000  \n",
       "Abnormal ocular adnexa morphology [HP:0030669]      1.000000  \n",
       "Delayed speech and language development [HP:000...  1.000000  \n",
       "Abnormal communication [HP:0034434]                 1.000000  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "summarize_hpo_analysis(hpo, c5753_5756del_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Allele group</th>\n",
       "      <th>missense</th>\n",
       "      <th>Other</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Intellectual disability [HP:0001249]</th>\n",
       "      <td>1/2 (50%)</td>\n",
       "      <td>48/48 (100%)</td>\n",
       "      <td>0.695652</td>\n",
       "      <td>0.040000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ventriculomegaly [HP:0002119]</th>\n",
       "      <td>0/3 (0%)</td>\n",
       "      <td>24/41 (59%)</td>\n",
       "      <td>0.695652</td>\n",
       "      <td>0.086077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Intellectual disability, severe [HP:0010864]</th>\n",
       "      <td>1/2 (50%)</td>\n",
       "      <td>21/21 (100%)</td>\n",
       "      <td>0.695652</td>\n",
       "      <td>0.086957</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Epicanthus [HP:0000286]</th>\n",
       "      <td>2/3 (67%)</td>\n",
       "      <td>9/44 (20%)</td>\n",
       "      <td>0.793710</td>\n",
       "      <td>0.132285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Seizure [HP:0001250]</th>\n",
       "      <td>0/3 (0%)</td>\n",
       "      <td>24/48 (50%)</td>\n",
       "      <td>0.969473</td>\n",
       "      <td>0.237647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Strabismus [HP:0000486]</th>\n",
       "      <td>0/3 (0%)</td>\n",
       "      <td>21/44 (48%)</td>\n",
       "      <td>0.969473</td>\n",
       "      <td>0.242368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Feeding difficulties [HP:0011968]</th>\n",
       "      <td>1/3 (33%)</td>\n",
       "      <td>29/45 (64%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.547294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Low-set ears [HP:0000369]</th>\n",
       "      <td>1/3 (33%)</td>\n",
       "      <td>21/40 (52%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.606920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Downslanted palpebral fissures [HP:0000494]</th>\n",
       "      <td>1/3 (33%)</td>\n",
       "      <td>23/44 (52%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.608511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Slanting of the palpebral fissure [HP:0200006]</th>\n",
       "      <td>1/1 (100%)</td>\n",
       "      <td>23/23 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal location of ears [HP:0000357]</th>\n",
       "      <td>1/1 (100%)</td>\n",
       "      <td>26/26 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Short stature [HP:0004322]</th>\n",
       "      <td>1/1 (100%)</td>\n",
       "      <td>26/26 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abdominal symptom [HP:0011458]</th>\n",
       "      <td>1/1 (100%)</td>\n",
       "      <td>29/29 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal cerebral morphology [HP:0002060]</th>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>20/20 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Decreased body weight [HP:0004325]</th>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>23/23 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Delayed speech and language development [HP:0000750]</th>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>25/25 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal communication [HP:0034434]</th>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>25/25 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal muscle tone [HP:0003808]</th>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>29/29 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal eyelid morphology [HP:0000492]</th>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>31/31 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal brain morphology [HP:0012443]</th>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>32/32 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Morphological central nervous system abnormality [HP:0002011]</th>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>32/32 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal ocular adnexa morphology [HP:0030669]</th>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>32/32 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neurodevelopmental abnormality [HP:0012759]</th>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>49/49 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypotonia [HP:0001252]</th>\n",
       "      <td>2/3 (67%)</td>\n",
       "      <td>29/48 (60%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Allele group                                          missense         Other  \\\n",
       "Intellectual disability [HP:0001249]                 1/2 (50%)  48/48 (100%)   \n",
       "Ventriculomegaly [HP:0002119]                         0/3 (0%)   24/41 (59%)   \n",
       "Intellectual disability, severe [HP:0010864]         1/2 (50%)  21/21 (100%)   \n",
       "Epicanthus [HP:0000286]                              2/3 (67%)    9/44 (20%)   \n",
       "Seizure [HP:0001250]                                  0/3 (0%)   24/48 (50%)   \n",
       "Strabismus [HP:0000486]                               0/3 (0%)   21/44 (48%)   \n",
       "Feeding difficulties [HP:0011968]                    1/3 (33%)   29/45 (64%)   \n",
       "Low-set ears [HP:0000369]                            1/3 (33%)   21/40 (52%)   \n",
       "Downslanted palpebral fissures [HP:0000494]          1/3 (33%)   23/44 (52%)   \n",
       "Slanting of the palpebral fissure [HP:0200006]      1/1 (100%)  23/23 (100%)   \n",
       "Abnormal location of ears [HP:0000357]              1/1 (100%)  26/26 (100%)   \n",
       "Short stature [HP:0004322]                          1/1 (100%)  26/26 (100%)   \n",
       "Abdominal symptom [HP:0011458]                      1/1 (100%)  29/29 (100%)   \n",
       "Abnormal cerebral morphology [HP:0002060]           2/2 (100%)  20/20 (100%)   \n",
       "Decreased body weight [HP:0004325]                  2/2 (100%)  23/23 (100%)   \n",
       "Delayed speech and language development [HP:000...  2/2 (100%)  25/25 (100%)   \n",
       "Abnormal communication [HP:0034434]                 2/2 (100%)  25/25 (100%)   \n",
       "Abnormal muscle tone [HP:0003808]                   2/2 (100%)  29/29 (100%)   \n",
       "Abnormal eyelid morphology [HP:0000492]             2/2 (100%)  31/31 (100%)   \n",
       "Abnormal brain morphology [HP:0012443]              2/2 (100%)  32/32 (100%)   \n",
       "Morphological central nervous system abnormalit...  2/2 (100%)  32/32 (100%)   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]      2/2 (100%)  32/32 (100%)   \n",
       "Neurodevelopmental abnormality [HP:0012759]         2/2 (100%)  49/49 (100%)   \n",
       "Hypotonia [HP:0001252]                               2/3 (67%)   29/48 (60%)   \n",
       "\n",
       "Allele group                                        Corrected p values  \\\n",
       "Intellectual disability [HP:0001249]                          0.695652   \n",
       "Ventriculomegaly [HP:0002119]                                 0.695652   \n",
       "Intellectual disability, severe [HP:0010864]                  0.695652   \n",
       "Epicanthus [HP:0000286]                                       0.793710   \n",
       "Seizure [HP:0001250]                                          0.969473   \n",
       "Strabismus [HP:0000486]                                       0.969473   \n",
       "Feeding difficulties [HP:0011968]                             1.000000   \n",
       "Low-set ears [HP:0000369]                                     1.000000   \n",
       "Downslanted palpebral fissures [HP:0000494]                   1.000000   \n",
       "Slanting of the palpebral fissure [HP:0200006]                1.000000   \n",
       "Abnormal location of ears [HP:0000357]                        1.000000   \n",
       "Short stature [HP:0004322]                                    1.000000   \n",
       "Abdominal symptom [HP:0011458]                                1.000000   \n",
       "Abnormal cerebral morphology [HP:0002060]                     1.000000   \n",
       "Decreased body weight [HP:0004325]                            1.000000   \n",
       "Delayed speech and language development [HP:000...            1.000000   \n",
       "Abnormal communication [HP:0034434]                           1.000000   \n",
       "Abnormal muscle tone [HP:0003808]                             1.000000   \n",
       "Abnormal eyelid morphology [HP:0000492]                       1.000000   \n",
       "Abnormal brain morphology [HP:0012443]                        1.000000   \n",
       "Morphological central nervous system abnormalit...            1.000000   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]                1.000000   \n",
       "Neurodevelopmental abnormality [HP:0012759]                   1.000000   \n",
       "Hypotonia [HP:0001252]                                        1.000000   \n",
       "\n",
       "Allele group                                        p values  \n",
       "Intellectual disability [HP:0001249]                0.040000  \n",
       "Ventriculomegaly [HP:0002119]                       0.086077  \n",
       "Intellectual disability, severe [HP:0010864]        0.086957  \n",
       "Epicanthus [HP:0000286]                             0.132285  \n",
       "Seizure [HP:0001250]                                0.237647  \n",
       "Strabismus [HP:0000486]                             0.242368  \n",
       "Feeding difficulties [HP:0011968]                   0.547294  \n",
       "Low-set ears [HP:0000369]                           0.606920  \n",
       "Downslanted palpebral fissures [HP:0000494]         0.608511  \n",
       "Slanting of the palpebral fissure [HP:0200006]      1.000000  \n",
       "Abnormal location of ears [HP:0000357]              1.000000  \n",
       "Short stature [HP:0004322]                          1.000000  \n",
       "Abdominal symptom [HP:0011458]                      1.000000  \n",
       "Abnormal cerebral morphology [HP:0002060]           1.000000  \n",
       "Decreased body weight [HP:0004325]                  1.000000  \n",
       "Delayed speech and language development [HP:000...  1.000000  \n",
       "Abnormal communication [HP:0034434]                 1.000000  \n",
       "Abnormal muscle tone [HP:0003808]                   1.000000  \n",
       "Abnormal eyelid morphology [HP:0000492]             1.000000  \n",
       "Abnormal brain morphology [HP:0012443]              1.000000  \n",
       "Morphological central nervous system abnormalit...  1.000000  \n",
       "Abnormal ocular adnexa morphology [HP:0030669]      1.000000  \n",
       "Neurodevelopmental abnormality [HP:0012759]         1.000000  \n",
       "Hypotonia [HP:0001252]                              1.000000  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.model import VariantEffect\n",
    "from gpsea.analysis.predicate import variant_effect\n",
    "\n",
    "missense = variant_effect(effect=VariantEffect.MISSENSE_VARIANT, tx_id=mane_tx_id)\n",
    "missense_clf = monoallelic_classifier(\n",
    "    a_predicate=missense,\n",
    "    b_predicate=~missense,\n",
    "    a_label=\"missense\",\n",
    "    b_label=\"Other\"\n",
    ")\n",
    "missense_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_clf=missense_clf,\n",
    "    pheno_clfs=pheno_clfs,\n",
    ")\n",
    "summarize_hpo_analysis(hpo, missense_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Sex</th>\n",
       "      <th>FEMALE</th>\n",
       "      <th>MALE</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Ventriculomegaly [HP:0002119]</th>\n",
       "      <td>8/20 (40%)</td>\n",
       "      <td>16/24 (67%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.128152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Generalized joint hypermobility [HP:0002761]</th>\n",
       "      <td>6/25 (24%)</td>\n",
       "      <td>11/26 (42%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.236706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Low-set ears [HP:0000369]</th>\n",
       "      <td>9/21 (43%)</td>\n",
       "      <td>13/22 (59%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.366030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Feeding difficulties [HP:0011968]</th>\n",
       "      <td>17/24 (71%)</td>\n",
       "      <td>13/24 (54%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.371469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypotonia [HP:0001252]</th>\n",
       "      <td>17/25 (68%)</td>\n",
       "      <td>14/26 (54%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.392862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Intellectual disability, severe [HP:0010864]</th>\n",
       "      <td>12/12 (100%)</td>\n",
       "      <td>10/11 (91%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.478261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Strabismus [HP:0000486]</th>\n",
       "      <td>11/22 (50%)</td>\n",
       "      <td>10/25 (40%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.564277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Seizure [HP:0001250]</th>\n",
       "      <td>11/25 (44%)</td>\n",
       "      <td>13/26 (50%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.781251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal conjugate eye movement [HP:0000549]</th>\n",
       "      <td>11/11 (100%)</td>\n",
       "      <td>10/10 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Slanting of the palpebral fissure [HP:0200006]</th>\n",
       "      <td>11/11 (100%)</td>\n",
       "      <td>13/13 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Decreased body weight [HP:0004325]</th>\n",
       "      <td>11/11 (100%)</td>\n",
       "      <td>14/14 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal location of ears [HP:0000357]</th>\n",
       "      <td>11/11 (100%)</td>\n",
       "      <td>16/16 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Downslanted palpebral fissures [HP:0000494]</th>\n",
       "      <td>11/22 (50%)</td>\n",
       "      <td>13/25 (52%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal cerebral morphology [HP:0002060]</th>\n",
       "      <td>12/12 (100%)</td>\n",
       "      <td>10/10 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Delayed speech and language development [HP:0000750]</th>\n",
       "      <td>13/13 (100%)</td>\n",
       "      <td>14/14 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal communication [HP:0034434]</th>\n",
       "      <td>13/13 (100%)</td>\n",
       "      <td>14/14 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal eyelid morphology [HP:0000492]</th>\n",
       "      <td>14/14 (100%)</td>\n",
       "      <td>19/19 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Short stature [HP:0004322]</th>\n",
       "      <td>15/15 (100%)</td>\n",
       "      <td>12/12 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal ocular adnexa morphology [HP:0030669]</th>\n",
       "      <td>15/15 (100%)</td>\n",
       "      <td>19/19 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal brain morphology [HP:0012443]</th>\n",
       "      <td>16/16 (100%)</td>\n",
       "      <td>18/18 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Morphological central nervous system abnormality [HP:0002011]</th>\n",
       "      <td>16/16 (100%)</td>\n",
       "      <td>18/18 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abdominal symptom [HP:0011458]</th>\n",
       "      <td>17/17 (100%)</td>\n",
       "      <td>13/13 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal muscle tone [HP:0003808]</th>\n",
       "      <td>17/17 (100%)</td>\n",
       "      <td>14/14 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Intellectual disability [HP:0001249]</th>\n",
       "      <td>24/24 (100%)</td>\n",
       "      <td>25/26 (96%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neurodevelopmental abnormality [HP:0012759]</th>\n",
       "      <td>25/25 (100%)</td>\n",
       "      <td>26/26 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Visual impairment [HP:0000505]</th>\n",
       "      <td>8/8 (100%)</td>\n",
       "      <td>13/13 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal cerebral ventricle morphology [HP:0002118]</th>\n",
       "      <td>8/8 (100%)</td>\n",
       "      <td>16/16 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Sex                                                       FEMALE  \\\n",
       "Ventriculomegaly [HP:0002119]                         8/20 (40%)   \n",
       "Generalized joint hypermobility [HP:0002761]          6/25 (24%)   \n",
       "Low-set ears [HP:0000369]                             9/21 (43%)   \n",
       "Feeding difficulties [HP:0011968]                    17/24 (71%)   \n",
       "Hypotonia [HP:0001252]                               17/25 (68%)   \n",
       "Intellectual disability, severe [HP:0010864]        12/12 (100%)   \n",
       "Strabismus [HP:0000486]                              11/22 (50%)   \n",
       "Seizure [HP:0001250]                                 11/25 (44%)   \n",
       "Abnormal conjugate eye movement [HP:0000549]        11/11 (100%)   \n",
       "Slanting of the palpebral fissure [HP:0200006]      11/11 (100%)   \n",
       "Decreased body weight [HP:0004325]                  11/11 (100%)   \n",
       "Abnormal location of ears [HP:0000357]              11/11 (100%)   \n",
       "Downslanted palpebral fissures [HP:0000494]          11/22 (50%)   \n",
       "Abnormal cerebral morphology [HP:0002060]           12/12 (100%)   \n",
       "Delayed speech and language development [HP:000...  13/13 (100%)   \n",
       "Abnormal communication [HP:0034434]                 13/13 (100%)   \n",
       "Abnormal eyelid morphology [HP:0000492]             14/14 (100%)   \n",
       "Short stature [HP:0004322]                          15/15 (100%)   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]      15/15 (100%)   \n",
       "Abnormal brain morphology [HP:0012443]              16/16 (100%)   \n",
       "Morphological central nervous system abnormalit...  16/16 (100%)   \n",
       "Abdominal symptom [HP:0011458]                      17/17 (100%)   \n",
       "Abnormal muscle tone [HP:0003808]                   17/17 (100%)   \n",
       "Intellectual disability [HP:0001249]                24/24 (100%)   \n",
       "Neurodevelopmental abnormality [HP:0012759]         25/25 (100%)   \n",
       "Visual impairment [HP:0000505]                        8/8 (100%)   \n",
       "Abnormal cerebral ventricle morphology [HP:0002...    8/8 (100%)   \n",
       "\n",
       "Sex                                                         MALE  \\\n",
       "Ventriculomegaly [HP:0002119]                        16/24 (67%)   \n",
       "Generalized joint hypermobility [HP:0002761]         11/26 (42%)   \n",
       "Low-set ears [HP:0000369]                            13/22 (59%)   \n",
       "Feeding difficulties [HP:0011968]                    13/24 (54%)   \n",
       "Hypotonia [HP:0001252]                               14/26 (54%)   \n",
       "Intellectual disability, severe [HP:0010864]         10/11 (91%)   \n",
       "Strabismus [HP:0000486]                              10/25 (40%)   \n",
       "Seizure [HP:0001250]                                 13/26 (50%)   \n",
       "Abnormal conjugate eye movement [HP:0000549]        10/10 (100%)   \n",
       "Slanting of the palpebral fissure [HP:0200006]      13/13 (100%)   \n",
       "Decreased body weight [HP:0004325]                  14/14 (100%)   \n",
       "Abnormal location of ears [HP:0000357]              16/16 (100%)   \n",
       "Downslanted palpebral fissures [HP:0000494]          13/25 (52%)   \n",
       "Abnormal cerebral morphology [HP:0002060]           10/10 (100%)   \n",
       "Delayed speech and language development [HP:000...  14/14 (100%)   \n",
       "Abnormal communication [HP:0034434]                 14/14 (100%)   \n",
       "Abnormal eyelid morphology [HP:0000492]             19/19 (100%)   \n",
       "Short stature [HP:0004322]                          12/12 (100%)   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]      19/19 (100%)   \n",
       "Abnormal brain morphology [HP:0012443]              18/18 (100%)   \n",
       "Morphological central nervous system abnormalit...  18/18 (100%)   \n",
       "Abdominal symptom [HP:0011458]                      13/13 (100%)   \n",
       "Abnormal muscle tone [HP:0003808]                   14/14 (100%)   \n",
       "Intellectual disability [HP:0001249]                 25/26 (96%)   \n",
       "Neurodevelopmental abnormality [HP:0012759]         26/26 (100%)   \n",
       "Visual impairment [HP:0000505]                      13/13 (100%)   \n",
       "Abnormal cerebral ventricle morphology [HP:0002...  16/16 (100%)   \n",
       "\n",
       "Sex                                                 Corrected p values  \\\n",
       "Ventriculomegaly [HP:0002119]                                      1.0   \n",
       "Generalized joint hypermobility [HP:0002761]                       1.0   \n",
       "Low-set ears [HP:0000369]                                          1.0   \n",
       "Feeding difficulties [HP:0011968]                                  1.0   \n",
       "Hypotonia [HP:0001252]                                             1.0   \n",
       "Intellectual disability, severe [HP:0010864]                       1.0   \n",
       "Strabismus [HP:0000486]                                            1.0   \n",
       "Seizure [HP:0001250]                                               1.0   \n",
       "Abnormal conjugate eye movement [HP:0000549]                       1.0   \n",
       "Slanting of the palpebral fissure [HP:0200006]                     1.0   \n",
       "Decreased body weight [HP:0004325]                                 1.0   \n",
       "Abnormal location of ears [HP:0000357]                             1.0   \n",
       "Downslanted palpebral fissures [HP:0000494]                        1.0   \n",
       "Abnormal cerebral morphology [HP:0002060]                          1.0   \n",
       "Delayed speech and language development [HP:000...                 1.0   \n",
       "Abnormal communication [HP:0034434]                                1.0   \n",
       "Abnormal eyelid morphology [HP:0000492]                            1.0   \n",
       "Short stature [HP:0004322]                                         1.0   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]                     1.0   \n",
       "Abnormal brain morphology [HP:0012443]                             1.0   \n",
       "Morphological central nervous system abnormalit...                 1.0   \n",
       "Abdominal symptom [HP:0011458]                                     1.0   \n",
       "Abnormal muscle tone [HP:0003808]                                  1.0   \n",
       "Intellectual disability [HP:0001249]                               1.0   \n",
       "Neurodevelopmental abnormality [HP:0012759]                        1.0   \n",
       "Visual impairment [HP:0000505]                                     1.0   \n",
       "Abnormal cerebral ventricle morphology [HP:0002...                 1.0   \n",
       "\n",
       "Sex                                                 p values  \n",
       "Ventriculomegaly [HP:0002119]                       0.128152  \n",
       "Generalized joint hypermobility [HP:0002761]        0.236706  \n",
       "Low-set ears [HP:0000369]                           0.366030  \n",
       "Feeding difficulties [HP:0011968]                   0.371469  \n",
       "Hypotonia [HP:0001252]                              0.392862  \n",
       "Intellectual disability, severe [HP:0010864]        0.478261  \n",
       "Strabismus [HP:0000486]                             0.564277  \n",
       "Seizure [HP:0001250]                                0.781251  \n",
       "Abnormal conjugate eye movement [HP:0000549]        1.000000  \n",
       "Slanting of the palpebral fissure [HP:0200006]      1.000000  \n",
       "Decreased body weight [HP:0004325]                  1.000000  \n",
       "Abnormal location of ears [HP:0000357]              1.000000  \n",
       "Downslanted palpebral fissures [HP:0000494]         1.000000  \n",
       "Abnormal cerebral morphology [HP:0002060]           1.000000  \n",
       "Delayed speech and language development [HP:000...  1.000000  \n",
       "Abnormal communication [HP:0034434]                 1.000000  \n",
       "Abnormal eyelid morphology [HP:0000492]             1.000000  \n",
       "Short stature [HP:0004322]                          1.000000  \n",
       "Abnormal ocular adnexa morphology [HP:0030669]      1.000000  \n",
       "Abnormal brain morphology [HP:0012443]              1.000000  \n",
       "Morphological central nervous system abnormalit...  1.000000  \n",
       "Abdominal symptom [HP:0011458]                      1.000000  \n",
       "Abnormal muscle tone [HP:0003808]                   1.000000  \n",
       "Intellectual disability [HP:0001249]                1.000000  \n",
       "Neurodevelopmental abnormality [HP:0012759]         1.000000  \n",
       "Visual impairment [HP:0000505]                      1.000000  \n",
       "Abnormal cerebral ventricle morphology [HP:0002...  1.000000  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.analysis.clf import sex_classifier\n",
    "\n",
    "mf_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_clf=sex_classifier(),\n",
    "    pheno_clfs=pheno_clfs,\n",
    ")\n",
    "summarize_hpo_analysis(hpo, mf_result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpseacs.report import GpseaAnalysisReport, GPAnalysisResultSummary\n",
    "\n",
    "fet_results = (\n",
    "    GPAnalysisResultSummary.from_multi(\n",
    "        result=missense_result,\n",
    "    ),\n",
    "    GPAnalysisResultSummary.from_multi(\n",
    "        result=c5753_5756del_result,\n",
    "    ),\n",
    "    GPAnalysisResultSummary.from_multi(\n",
    "        result=mf_result,\n",
    "    ),\n",
    ")\n",
    "\n",
    "caption = \"\"\"Dingemans et al (2020) suggested a different pathomechanism for missense variants, but only our cohort only contains 3 individuals\n",
    " with missense variants, so there is no statistical power \\\\cite{PMID_34521999}.\"\"\"\n",
    "\n",
    "report = GpseaAnalysisReport(name=gene_symbol, \n",
    "                             cohort=cohort, \n",
    "                             fet_results=fet_results,\n",
    "                             gene_symbol=gene_symbol,\n",
    "                             mane_tx_id=mane_tx_id,\n",
    "                             mane_protein_id=mane_protein_id,\n",
    "                             caption=caption)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "      min-width: 100px;\n",
       "      border: 1px solid #ddd;\n",
       "      padding: 8px;\n",
       "        \n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    p {\n",
       "      width: 75%;\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>SON Summary</h1>\n",
       "  <p>The cohort comprised 52 individuals (26 females, 26 males). A total of 47 HPO terms were used to annotate the cohort. Disease diagnosis: ZTTK SYNDROME (OMIM:617140). Dingemans et al (2020) suggested a different pathomechanism for missense variants, but only our cohort only contains 3 individuals\n",
       " with missense variants, so there is no statistical power \\cite{PMID_34521999}. A total of 52 unique variant alleles were found in SON (transcript: NM_138927.4, protein id: NP_620305.2). HPO version: 2025-01-16. GPSEA version: 0.9.4.</p>\n",
       "  \n",
       "  \n",
       "    \n",
       "      <table>\n",
       "        <caption style=\"color: black;\">\n",
       "            Fisher Exact Test performed to compare HPO annotation frequency with respect to missense and Other.\n",
       "        </caption>\n",
       "            <tbody>\n",
       "                <tr class=\"strng\">\n",
       "                    <th>Genotype (A)</th>\n",
       "                    <th>Genotype (B)</th>\n",
       "                  <th>Tests performed</th>\n",
       "                  <th>Significant tests</th>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td class=\"lft\">missense</td>\n",
       "                    <td class=\"lft\">Other</td>\n",
       "                    <td class=\"lft\">24</td>\n",
       "                    <td class=\"lft\">0</td>\n",
       "                </tr>\n",
       "              </tbody>\n",
       "        </table>\n",
       "      \n",
       "\n",
       "  \n",
       "    \n",
       "      <table>\n",
       "        <caption style=\"color: black;\">\n",
       "            Fisher Exact Test performed to compare HPO annotation frequency with respect to c.5753_5756del and Other.\n",
       "        </caption>\n",
       "            <tbody>\n",
       "                <tr class=\"strng\">\n",
       "                    <th>Genotype (A)</th>\n",
       "                    <th>Genotype (B)</th>\n",
       "                  <th>Tests performed</th>\n",
       "                  <th>Significant tests</th>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td class=\"lft\">c.5753_5756del</td>\n",
       "                    <td class=\"lft\">Other</td>\n",
       "                    <td class=\"lft\">26</td>\n",
       "                    <td class=\"lft\">0</td>\n",
       "                </tr>\n",
       "              </tbody>\n",
       "        </table>\n",
       "      \n",
       "\n",
       "  \n",
       "    \n",
       "      <table>\n",
       "        <caption style=\"color: black;\">\n",
       "            Fisher Exact Test performed to compare HPO annotation frequency with respect to FEMALE and MALE.\n",
       "        </caption>\n",
       "            <tbody>\n",
       "                <tr class=\"strng\">\n",
       "                    <th>Genotype (A)</th>\n",
       "                    <th>Genotype (B)</th>\n",
       "                  <th>Tests performed</th>\n",
       "                  <th>Significant tests</th>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td class=\"lft\">FEMALE</td>\n",
       "                    <td class=\"lft\">MALE</td>\n",
       "                    <td class=\"lft\">27</td>\n",
       "                    <td class=\"lft\">0</td>\n",
       "                </tr>\n",
       "              </tbody>\n",
       "        </table>\n",
       "      \n",
       "\n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "\n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpseacs.report._api.HtmlGpseaNotebookSummarizer at 0x16bf452b0>"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpseacs.report import GpseaNotebookSummarizer\n",
    "summarizer = GpseaNotebookSummarizer(hpo=hpo, gpsea_version=gpsea.__version__)\n",
    "summarizer.summarize_report(report=report)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summarizer.process_latex(report=report, protein_fig=son_fig)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
