{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SUOX: Sulfite Oxidase Deficiency\n",
    "\n",
    "[Sulfite oxidase deficiency (ISOD)](https://omim.org/entry/272300) is caused by homozygous or compound heterozygous mutation in the sulfite oxidase gene (SUOX).\n",
    "\n",
    "TODO details\n",
    "\n",
    "Data from [Li JT, et al. Mutation analysis of SUOX in isolated sulfite oxidase deficiency with ectopia lentis as the presenting feature: insights into genotype-phenotype correlation](https://pubmed.ncbi.nlm.nih.gov/36303223/)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2024-08-13\n",
      "Using gpsea version 0.7.1\n"
     ]
    }
   ],
   "source": [
    "import hpotk\n",
    "import gpsea\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo(release='v2024-08-13')\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using gpsea version {gpsea.__version__}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### SUOX\n",
    "We user the [Matched Annotation from NCBI and EMBL-EBI (MANE)](https://www.ncbi.nlm.nih.gov/refseq/MANE/) transcript and the corresponding protein identifier for SUOX."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_symbol = 'NKX6-2'\n",
    "mane_tx_id = 'NM_001032386.2' \n",
    "mane_protein_id = 'NP_001027558.1' # sulfite oxidase, mitochondrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 33 phenopackets\n",
      "Individuals Processed: 100%|██████████| 33/33 [00:00<00:00, 514.85individuals/s]\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.21\") as ps:\n",
    "    phenopackets = tuple(ps.iter_cohort_phenopackets(gene_symbol))\n",
    "print(f'Loaded {len(phenopackets)} phenopackets')\n",
    "\n",
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort, validation = load_phenopackets(  \n",
    "    phenopackets=phenopackets,\n",
    "    cohort_creator=cohort_creator,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "  <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td,\n",
       "    tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px;\n",
       "      /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>GPSEA cohort analysis</h1>\n",
       "  <p>\n",
       "    Successfully loaded 33 individuals.\n",
       "\n",
       "    \n",
       "    18 were recorded as male,\n",
       "    15 as female,\n",
       "    and 0 as unknown sex.\n",
       "\n",
       "    \n",
       "    \n",
       "    No information about individuals' vital status was reported.\n",
       "    \n",
       "\n",
       "    \n",
       "    The cohort included 39 HPO terms.\n",
       "\n",
       "    \n",
       "    33 individuals had disease onset information\n",
       "    and 0 had information about the age of last encounter.\n",
       "\n",
       "\n",
       "    \n",
       "    \n",
       "  </p>\n",
       "\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "\n",
       "  \n",
       "  <table>\n",
       "    <caption style=\"color: black;\">\n",
       "      <h3>Top 10 HPO terms</h3>\n",
       "      A total of 39 HPO terms were used to annotated the cohort.\n",
       "      \n",
       "    </caption>\n",
       "    <tbody>\n",
       "      <tr class=\"strng\">\n",
       "        <th class=\"lft\">HPO Term</th>\n",
       "        <th>ID</th>\n",
       "        <th>Seen in <em>n</em> individuals</th>\n",
       "      </tr>\n",
       "\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Spasticity</td>\n",
       "        <td>HP:0001257</td>\n",
       "        <td>32</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Nystagmus</td>\n",
       "        <td>HP:0000639</td>\n",
       "        <td>29</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Motor delay</td>\n",
       "        <td>HP:0001270</td>\n",
       "        <td>24</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Global developmental delay</td>\n",
       "        <td>HP:0001263</td>\n",
       "        <td>24</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Hyperreflexia</td>\n",
       "        <td>HP:0001347</td>\n",
       "        <td>20</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Hypotonia</td>\n",
       "        <td>HP:0001252</td>\n",
       "        <td>19</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">CNS hypomyelination</td>\n",
       "        <td>HP:0003429</td>\n",
       "        <td>18</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Loss of ambulation</td>\n",
       "        <td>HP:0002505</td>\n",
       "        <td>13</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Ataxia</td>\n",
       "        <td>HP:0001251</td>\n",
       "        <td>9</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Abnormal pyramidal sign</td>\n",
       "        <td>HP:0007256</td>\n",
       "        <td>7</td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "\n",
       "  \n",
       "\n",
       "  \n",
       "  <table>\n",
       "    <caption style=\"color: black;\">\n",
       "      <h3>Top 10 variants</h3>\n",
       "      Variants are shown according to NM_001032386.2.\n",
       "      A total of 13 unique variants\n",
       "      were identified in the cohort.\n",
       "    </caption>\n",
       "    <tbody>\n",
       "      <tr class=\"strng\">\n",
       "        <th>Seen in <em>n</em> individuals</th>\n",
       "        <th class=\"lft\">Variant key</th>\n",
       "        <th>Variant Name</th>\n",
       "        <th>Protein Effect</th>\n",
       "        <th>Variant Class</th>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>9</td>\n",
       "        <td class=\"lft\">10_132785372_132785372_G_C</td>\n",
       "        <td>10_132785372_132785372_G_C</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>8</td>\n",
       "        <td class=\"lft\">10_132785752_132785753_CG_C</td>\n",
       "        <td>10_132785752_132785753_CG_C</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>6</td>\n",
       "        <td class=\"lft\">10_132785828_132785828_T_A</td>\n",
       "        <td>10_132785828_132785828_T_A</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">10_132785142_132785142_C_T</td>\n",
       "        <td>10_132785142_132785142_C_T</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">10_132785161_132785161_G_A</td>\n",
       "        <td>10_132785161_132785161_G_A</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">10_132785151_132785151_C_T</td>\n",
       "        <td>10_132785151_132785151_C_T</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">10_132785294_132785294_C_A</td>\n",
       "        <td>10_132785294_132785294_C_A</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td class=\"lft\">10_132785288_132785288_G_A</td>\n",
       "        <td>10_132785288_132785288_G_A</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td class=\"lft\">10_132785158_132785158_T_C</td>\n",
       "        <td>10_132785158_132785158_T_C</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td class=\"lft\">10_132785648_132785648_G_T</td>\n",
       "        <td>10_132785648_132785648_G_T</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "  \n",
       "  \n",
       "  <table>\n",
       "    <caption style=\"color: black;\">\n",
       "      <h3>Diseases</h3>\n",
       "    </caption>\n",
       "    <tbody>\n",
       "      <tr class=\"strng\">\n",
       "        <th class=\"lft\">Name</th>\n",
       "        <th>ID</th>\n",
       "        <th><em>N</em> diagnosed individuals</th>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Spastic ataxia 8, autosomal recessive, with hypomyelinating leukodystrophy</td>\n",
       "        <td>OMIM:617560</td>\n",
       "        <td>33</td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "  \n",
       "  \n",
       "  \n",
       "  <table>\n",
       "    <caption style=\"color: black;\">\n",
       "      <h3>Variant categories for NM_001032386.2</h3>\n",
       "    </caption>\n",
       "    <tbody>\n",
       "      <tr class=\"strng\">\n",
       "        <th class=\"lft\">Variant effect</th>\n",
       "        <th>Annotation count</th>\n",
       "        <th>Percent</th>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "  \n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpsea.view._report.HtmlGpseaReport at 0x116f00f80>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import CohortViewer\n",
    "\n",
    "viewer = CohortViewer(hpo)\n",
    "viewer.process(cohort=cohort, transcript_id=mane_tx_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Unrecognized protein feature type: \"Binding site\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[5], line 13\u001b[0m\n\u001b[1;32m      9\u001b[0m tx_coordinates \u001b[38;5;241m=\u001b[39m txc_service\u001b[38;5;241m.\u001b[39mfetch(mane_tx_id)\n\u001b[1;32m     10\u001b[0m protein_meta \u001b[38;5;241m=\u001b[39m pms\u001b[38;5;241m.\u001b[39mannotate(mane_protein_id)\n\u001b[0;32m---> 13\u001b[0m protein_meta2 \u001b[38;5;241m=\u001b[39m \u001b[43mProteinMetadata\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfrom_uniprot_json\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     14\u001b[0m \u001b[43m        \u001b[49m\u001b[43mprotein_id\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmane_protein_id\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     15\u001b[0m \u001b[43m        \u001b[49m\u001b[43mlabel\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mSUOX\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     16\u001b[0m \u001b[43m        \u001b[49m\u001b[43muniprot_json\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mP51687.json\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     17\u001b[0m \u001b[43m        \u001b[49m\u001b[43mprotein_length\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m550\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     18\u001b[0m \u001b[43m)\u001b[49m\n\u001b[1;32m     21\u001b[0m fig, ax \u001b[38;5;241m=\u001b[39m plt\u001b[38;5;241m.\u001b[39msubplots(figsize\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m15\u001b[39m, \u001b[38;5;241m8\u001b[39m))\n\u001b[1;32m     22\u001b[0m visualizer \u001b[38;5;241m=\u001b[39m ProteinVisualizer()\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.12/site-packages/gpsea/model/_protein.py:375\u001b[0m, in \u001b[0;36mProteinMetadata.from_uniprot_json\u001b[0;34m(protein_id, label, uniprot_json, protein_length)\u001b[0m\n\u001b[1;32m    373\u001b[0m region_start \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mint\u001b[39m(locus[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mstart\u001b[39m\u001b[38;5;124m\"\u001b[39m][\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mvalue\u001b[39m\u001b[38;5;124m\"\u001b[39m]) \u001b[38;5;241m-\u001b[39m \u001b[38;5;241m1\u001b[39m  \u001b[38;5;66;03m# convert to 0-based coordinates\u001b[39;00m\n\u001b[1;32m    374\u001b[0m region_end \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mint\u001b[39m(locus[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mend\u001b[39m\u001b[38;5;124m\"\u001b[39m][\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mvalue\u001b[39m\u001b[38;5;124m\"\u001b[39m])\n\u001b[0;32m--> 375\u001b[0m feature_type \u001b[38;5;241m=\u001b[39m \u001b[43mFeatureType\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfrom_string\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfeature\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtype\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    376\u001b[0m finfo \u001b[38;5;241m=\u001b[39m FeatureInfo(\n\u001b[1;32m    377\u001b[0m     name\u001b[38;5;241m=\u001b[39mregion_name, region\u001b[38;5;241m=\u001b[39mRegion(start\u001b[38;5;241m=\u001b[39mregion_start, end\u001b[38;5;241m=\u001b[39mregion_end)\n\u001b[1;32m    378\u001b[0m )\n\u001b[1;32m    379\u001b[0m pfeature \u001b[38;5;241m=\u001b[39m ProteinFeature\u001b[38;5;241m.\u001b[39mcreate(info\u001b[38;5;241m=\u001b[39mfinfo, feature_type\u001b[38;5;241m=\u001b[39mfeature_type)\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.12/site-packages/gpsea/model/_protein.py:144\u001b[0m, in \u001b[0;36mFeatureType.from_string\u001b[0;34m(category)\u001b[0m\n\u001b[1;32m    142\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m FeatureType\u001b[38;5;241m.\u001b[39mZINC_FINGER\n\u001b[1;32m    143\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 144\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mUnrecognized protein feature type: \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mcategory\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "\u001b[0;31mValueError\u001b[0m: Unrecognized protein feature type: \"Binding site\""
     ]
    }
   ],
   "source": [
    "from gpsea.model.genome import GRCh38\n",
    "from gpsea.model import ProteinMetadata\n",
    "from gpsea.preprocessing import configure_default_protein_metadata_service, VVMultiCoordinateService\n",
    "from gpsea.view import ProteinVisualizer\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "txc_service = VVMultiCoordinateService(genome_build=GRCh38)\n",
    "pms = configure_default_protein_metadata_service()\n",
    "tx_coordinates = txc_service.fetch(mane_tx_id)\n",
    "protein_meta = pms.annotate(mane_protein_id)\n",
    "\n",
    "\n",
    "protein_meta2 = ProteinMetadata.from_uniprot_json(\n",
    "        protein_id=mane_protein_id,\n",
    "        label=\"SUOX\",\n",
    "        uniprot_json=\"P51687.json\",\n",
    "        protein_length=550,\n",
    ")\n",
    "\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(15, 8))\n",
    "visualizer = ProteinVisualizer()\n",
    "visualizer.draw_protein_diagram(\n",
    "    tx_coordinates,\n",
    "    protein_meta,\n",
    "    cohort,\n",
    "    ax=ax,\n",
    ")\n",
    "fig.tight_layout()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_coordinates = txc_service.fetch(mane_tx_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "  <style>table {\n",
       "    border-collapse: collapse;\n",
       "    margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "  }\n",
       "\n",
       "\n",
       "  .table .column-1 {\n",
       "      text-align: left;\n",
       "  }\n",
       "  th {\n",
       "    background-color: LightSkyBlue;\n",
       "    border: 1px solid #dddddd;\n",
       "    text-align: left;\n",
       "    padding: 5px;\n",
       "  font-weight: bold;\n",
       "  font-size: 120%;\n",
       "}\n",
       "\n",
       "\n",
       "  tr {\n",
       "    border: 1px solid #dddddd;\n",
       "  }\n",
       "\n",
       "  td {\n",
       "    padding: 5px;\n",
       "    font-weight: bold;\n",
       "  }\n",
       "\n",
       "  tr:nth-child(even) {\n",
       "    background-color: #f2f2f2;\n",
       "  }\n",
       "\n",
       "  .table td,tr {\n",
       "    text-align: left;\n",
       "}\n",
       "\n",
       ".table td:first-child, tr:first-child {\n",
       "    text-align: left;\n",
       "}\n",
       "\n",
       "caption {\n",
       "  caption-side: top;\n",
       "  text-align: left;\n",
       "  padding-bottom: 10px;\n",
       "  font-weight: bold;\n",
       "}\n",
       "\n",
       "</style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>Sulfite oxidase, mitochondrial:  NP_001027558.1</h1>\n",
       "    <table>\n",
       "        <caption style=\"color:black;\">\n",
       "           Sulfite oxidase, mitochondrial (NP_001027558.1) has 4 annotated protein features.\n",
       "            0 were found to be located within these features, and\n",
       "            0 were located elsewhere. The total length of the protein is\n",
       "            545 amino acid residues.\n",
       "        </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th>Name</th>\n",
       "                <th>Type</th>\n",
       "                <th>Coordinates</th>\n",
       "                <th>Count</th>\n",
       "                <th>Variants</th>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td>Cytochrome b5 heme-binding</td>\n",
       "                <td>DOMAIN</td>\n",
       "                <td>82 - 161</td>\n",
       "                <td>0</td>\n",
       "                <td></td>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td>Hinge</td>\n",
       "                <td>REGION</td>\n",
       "                <td>165 - 174</td>\n",
       "                <td>0</td>\n",
       "                <td></td>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td>Moco domain</td>\n",
       "                <td>REGION</td>\n",
       "                <td>175 - 401</td>\n",
       "                <td>0</td>\n",
       "                <td></td>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td>Homodimerization</td>\n",
       "                <td>REGION</td>\n",
       "                <td>402 - 538</td>\n",
       "                <td>0</td>\n",
       "                <td></td>\n",
       "            </tr>\n",
       "            \n",
       "        </tbody>\n",
       "    </table>\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpsea.view._report.HtmlGpseaReport at 0x3237c71a0>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import ProteinVariantViewer\n",
    "viewer = ProteinVariantViewer(protein_meta, mane_tx_id)\n",
    "html_prot = viewer.process(cohort)\n",
    "html_prot"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
