\documentclass[12pt]{article}

\usepackage{times}
\usepackage{graphicx}
\usepackage{longtable}
\usepackage{array}
\usepackage{booktabs}
\usepackage{subcaption}  % For subfigures (panels A, B, C)% Adds support for breaklines
\usepackage{float}

\usepackage[font=small, justification=raggedright,labelfont=bf]{caption}
\usepackage[margin=1in]{geometry} % Set all margins to 1 inch


\renewcommand{\thefigure}{S\arabic{figure}}
\renewcommand{\thetable}{S\arabic{table}}
\captionsetup[figure]{labelfont={bf}, name={Figure}}




\begin{document}
\thispagestyle{empty}

\begin{table}
\centering
\renewcommand{\arraystretch}{1.2} % Optional: Increase row spacing
\begin{tabular}{l>{\raggedright\arraybackslash}p{6cm}>{\raggedright\arraybackslash}p{6cm}}
\toprule
\textbf{Name} & \textbf{Match criterion} & \textbf{Example}\\
\midrule
variant key & Specific variant & 1\_8364773\_8364773\_A\_AC (NM\_001042681.2:c.1512dup) \\
variant effect & Leads to the target effect & START\_LOST \\
variant class & Variant has the target type & SNV, DEL, DUP, INS, INV \\
gene & Variant affects gene & Affects RERE \\
transcript & Variant affects transcript & Overlaps NM\_12345.01 \\
exon & Variant overlaps exon & Located in exon 3 \\
region & Variant overlaps region on protein & Located between amino acid residues 1 to 77 \\
is large imprecise sv & Variant is structural variant without exact breakpoint coordinates & -- \\
is structural variant & The variant affects at least n base pairs (n=50 by default) or is a large imprecise SV or a translocation & -- \\
structural type & The target ontology class & Chromosomal deletion (SO:1000029) \\
is structural deletion & Variant is SV with deletion of at least a certain number of base pairs (default 50bp) & SV of at least 500 bp \\
ref length & Reference sequence affected by variant has specified bp length & Ref\_length less than 10 bp \\
change length & Change of length between the REF/ALT has specified bp length & Change\_length is exactly 10 bp \\
protein feature type & Variant affects protein feature type & DOMAIN \\
protein feature & Variant affects protein feature & Located in Bipartite nuclear localization signal \\
allof & Logical AND & missense AND exon 7 \\
anyof & Logical OR & START\_LOST OR STOP\_GAINED \\
\bottomrule
\end{tabular}
\caption{\textbf{Variant predicates.} Version 9.4 of GPSEA offers the variant predicates shown in this table. Each predicate evaluates a variant and returns True or False.}
\label{tab:varpredicates}
\end{table}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\input{hpo_fet}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\newpage

\begin{table}
\centering
\begin{scriptsize}
\begin{tabular}{l>{\raggedright}p{2.5cm}llllrr}
\toprule
\textbf{Cohort} & \textbf{HPO} & \textbf{disease A} & \textbf{} & \textbf{disease B} & \textbf{} & \textbf{p-val} & \textbf{adj. p}\\
\midrule
ATP13A2 & Parkinsonism [HP:0001300] &KRS & 28/28 (100\%) &SPG78 & 3/11 (27\%) & $2.7 \times 10^{-06}$ & $7.2 \times 10^{-05}$\\
ATP13A2 & Bradykinesia [HP:0002067] & KRS & 30/32 (94\%) & SPG78 & 4/10 (40\%) & $9.2 \times 10^{-04}$ & 0.012\\
Kabuki & Feeding difficulties [HP:0011968] & KABUK1 & 8/25 (32\%) & KABUK2 & 55/63 (87\%) & $7.4 \times 10^{-07}$ & $2.1 \times 10^{-05}$\\
Kabuki & Motor delay [HP:0001270] & KABUK1 & 4/10 (40\%) & KABUK2 & 58/61 (95\%) & $1.0 \times 10^{-04}$ & $1.0 \times 10^{-03}$\\
LDS 1/3 & Aortic aneurysm [HP:0004942] &LDS1 & 11/11 (100\%) & LDS3 & 26/48 (54\%) & 0.004 & 0.024\\
LDS 1/3 & Hypertelorism [HP:0000316] & LDS1 & 15/19 (79\%) & LDS3 & 13/35 (37\%) & 0.004 & 0.024\\
LDS 1/3 & Osteoarthritis [HP:0002758] &LDS1 & 0/11 (0\%) & LDS3 & 26/38 (68\%) & $4.6 \times 10^{-05}$ & $9.7 \times 10^{-04}$\\
LDS 1/3 & Scoliosis [HP:0002650] & LDS1 & 18/21 (86\%) & LDS3 & 20/43 (47\%) & 0.003 & 0.024\\
LDS 3/6 & Thoracic aortic aneurysm [HP:0012727] & LDS3 & 0/22 (0\%) & LDS6 & 10/15 (67\%) & $8.6 \times 10^{-06}$ & $1.8 \times 10^{-04}$\\
RPGRIP1 & Nystagmus [HP:0000639] & LCA6 & 64/66 (97\%) &CORD13& 11/16 (69\%) & 0.003 & 0.020\\
RPGRIP1 & Very low visual acuity [HP:0032122] & LCA6& 35/39 (90\%) & CORD13 & 4/16 (25\%) & $5.2 \times 10^{-06}$ & $7.8 \times 10^{-05}$\\
Robinow & Short stature [HP:0004322] & RRS1 & 29/29 (100\%) & DRS2 & 3/11 (27\%) & $2.2 \times 10^{-06}$ & $1.7 \times 10^{-04}$\\
Robinow & Orofacial cleft [HP:0000202] & RRS1 & 5/22 (23\%) & DRS2 & 5/5 (100\%) & 0.003 & 0.049\\
Robinow & Hearing impairment [HP:0000365] &RRS1& 3/22 (14\%) & DRS2 & 7/7 (100\%) & $7.7 \times 10^{-05}$ & 0.003\\
Robinow & Mesomelia [HP:0003027] & RRS1 & 31/31 (100\%) & DRS2 & 10/15 (67\%) & 0.002 & 0.043\\
Robinow & Cleft palate [HP:0000175] & RRS1 & 0/17 (0\%) & DRS2 & 5/8 (62\%) & $1.0 \times 10^{-03}$ & 0.028\\
\bottomrule
\end{tabular}
\end{scriptsize}
\caption{\textbf{Fischer exact test for association between disease diagnosis and phenotypic features.}
\textbf{OMIM:606693}: Kufor-Rakeb syndrome (KRS);
 \textbf{OMIM:617225}: Spastic paraplegia 78, autosomal recessive (SPG78);
 \textbf{OMIM:147920}: Kabuki syndrome 1 (KABUK1);
 \textbf{OMIM:300867}: Kabuki syndrome 2 (KABUK2);
 \textbf{OMIM:609192}: Loeys-Dietz syndrome 1 (LDS1);
 \textbf{OMIM:613795}:  Loeys-Dietz syndrome 3 (LDS3);
 \textbf{OMIM:619656}: Loeys-Dietz syndrome 6 (LDS6);
 \textbf{OMIM:613826}: Leber congenital amaurosis 6 (LCA6);
 \textbf{OMIM:608194}:  Cone-rod dystrophy 13 (CORD13);
 \textbf{OMIM:268310}: Robinow syndrome, autosomal recessive 1 (RRS1);
 \textbf{OMIM:616331}: Robinow syndrome, autosomal dominant 2 (DRS2).
}
\label{tab:to_do}
\end{table}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\newpage

\begin{table}
\centering
\begin{tabular}{l>{\raggedright}p{4cm}lp{1.5cm}lp{1.5cm}rr}
\toprule
\textbf{cohort} & \textbf{HPO} & \textbf{A} & \textbf{} & \textbf{B} & \textbf{} & \textbf{p-val} & \textbf{adj. p-val}\\
\midrule
KDM6A & Intellectual disability, severe [HP:0010864] & female & 7/25 (28\%) & male & 14/18 (78\%) & 0.002 & 0.008\\
\bottomrule
\end{tabular}
\caption{Fischer exact test for association between  between phenotypic features and sex (male, female).}
\label{tab:mf_hpo}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\newpage


\begin{table}
\centering
\begin{tabular}{lp{2cm}p{2cm}>{\raggedright}p{4.5cm}lr}
\toprule
\textbf{cohort} & \textbf{genotype (A)} & \textbf{genotype (B)} & \textbf{Outcome Variable} & \textbf{p-val} & \textbf{xrefs}\\
\midrule
ACADM & K329Q: 1/1 & K329Q: 1/0 or 0/0 & MCAD Activity\% [LOINC:74892-1] & $6.1 \times 10^{-10}$ & \cite{PMID_33580884}\\
ACADM & Y67H: 1/1 or 0/1  & Y67H: 0/0 & MCAD Activity\% [LOINC:74892-1] & $2.0 \times 10^{-05}$ & \cite{PMID_33580884}\\
CYP21A2 & missense: 0/1 or 0/0 & missense: 1/1 & 17-Hydroxyprogesterone [LOINC:1668-3] & $7.9 \times 10^{-06}$ & -\\
\bottomrule
\end{tabular}
\caption{Student t-tests performed using GPSEA. 1/1: biallelic with reference to the indicate variant; 0/1: heterozygous with reference to the indicated variant; 0/0: Neither allele has the indicated variant.
Citations in the xrefs column show previous publications that have presented similar findings.}
\label{tab:t_test}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\newpage


\begin{table}
\centering
\begin{tabular}{lp{2cm}p{2cm}>{\raggedright}p{4.5cm}lr}
\toprule
\textbf{cohort} & \textbf{genotype (A)} & \textbf{genotype (B)} & \textbf{HPO Term} & \textbf{p-val} & \textbf{xrefs}\\
\midrule
AIRE & R257*: 1/1 or 0/1 & R257*:0/0 & Chronic mucocutaneous candidiasis & 0.019 & -\\
CLDN16 & missense 1/1 or 0/1 & missense: 0/0 & Stage 5 chronic kidney disease & 0.034 & -\\
SCO2 & Glu140Lys: 1/1 & Glu140Lys: 0/1 or 0/0 & Hypertrophic cardiomyopathy & 0.219 & -\\
UMOD & 278\_289delins & other & Stage 5 chronic kidney disease & 0.835 & -\\
UMOD & EGF & other & Stage 5 chronic kidney disease & 0.284 & -\\
UMOD & Cys248Trp & Gln316Pro & Stage 5 chronic kidney disease & $4.1 \times 10^{-04}$ & -\\
\bottomrule
\end{tabular}
\caption{Log rank tests performed using GPSEA to assess association between a genotype and the age of
    onset of a phenotypic feature represented by an HPO term. 1/1: biallelic with reference to the indicate variant; 
    0/1: heterozygous with reference to the indicated variant; 0/0: Neither allele has the indicated variant.
    Citations in the xrefs column show previous publications that have presented similar findings.}
\label{tab:to_do}
\end{table}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\newpage



\begin{table}
\centering
\begin{tabular}{lp{3.5cm}p{3.5cm}>{\raggedright}p{2.5cm}lr}
\toprule
\textbf{cohort} & \textbf{genotype (A)} & \textbf{genotype (B)} & \textbf{Disease onset} & \textbf{p-val} & \textbf{xrefs}\\
\midrule
ATP6V0C & missense & other & OMIM:620465  & 0.193 & -\\
CLDN16 & missense: 1/1 or 0/1  & missense: 0/0 & OMIM:248250  & 0.333 & -\\
CNTNAP2 & 1 & 2 & OMIM:610042  & $6.2 \times 10^{-06}$ & -\\
COL3A1 & missense & other &  OMIM:130050  & 0.317 & -\\
FBXL4 &missense: 0/1 or 0/0 &missense: 1/1 & OMIM:615471  & 0.031 & -\\
HMGCS2 & missense: 1/1 or 0/1&  missense: 0/0 &  OMIM:605911  & 0.038 & -\\
MPV17 & missense: 1/1 or 0/1 &  missense: 0/0 &  OMIM:256810  & 0.002 & -\\
SCO2 & Glu140Lys: 1/1  & Glu140Lys: 0/1 or 0/0 &  OMIM:604377  & 0.415 & -\\
SETD2 & Missense & Other &  OMIM:616831  & $8.5 \times 10^{-05}$ & -\\
SUOX & hdim: 1/1 or 0/1 & hdim: 0/0&  OMIM:272300  & 0.853 & -\\
SUOX & missense: 1/1 or 0/1 &  missense: 0/0 &  OMIM:272300  & $9.2 \times 10^{-06}$ & -\\
\bottomrule
\end{tabular}
\caption{Log rank tests performed using GPSEA to assess association between a genotype and the age of
    onset of a disease. 1/1: biallelic with reference to the indicate variant; 
    0/1: heterozygous with reference to the indicated variant; 0/0: Neither allele has the indicated variant.
    Citations in the xrefs column show previous publications that have presented similar findings. \textbf{hdim}: homodimerization;
   \textbf{ OMIM:620465}: Epilepsy, early-onset, 3, with or without developmental delay; 
   \textbf{OMIM:248250}: Hypomagnesemia 3, renal;
   \textbf{OMIM:610042}: Pitt-Hopkins like syndrome 1;
   \textbf{OMIM:130050}: Ehlers-Danlos syndrome, vascular type;
   \textbf{OMIM:615471}: Mitochondrial DNA depletion syndrome 13 (encephalomyopathic type);
   \textbf{OMIM:605911}:  HMG-CoA synthase-2 deficiency;
   \textbf{OMIM:256810}: Mitochondrial DNA depletion syndrome 6 (hepatocerebral type);
   \textbf{OMIM:604377}: Mitochondrial complex IV deficiency, nuclear type 2;
   \textbf{OMIM:616831}: Luscan-Lumish syndrome;
  \textbf{OMIM:272300}: Sulfite oxidase deficiency.
   }
\label{tab:disease_onset}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\newpage


\begin{table}
\centering
\begin{tabular}{lp{3.5cm}p{3.5cm}>{\raggedright}p{3cm}lr}
\toprule
\textbf{cohort} & \textbf{genotype (A)} & \textbf{genotype (B)} & \textbf{Disease} & \textbf{p-val} & \textbf{xrefs}\\
\midrule
FBXL4 & missense: 0/1 or 0/0 & missense: 1/1 & OMIM:615471 & 0.080 & -\\
MPV17 & Pro98Leu: 1/1 & Pro98Leu: 0/1 or 0/0 & OMIM:256810 & 0.010 & -\\
NBAS & missense: 1/1 & missense: 0/1 or 0/0 & OMIM:614800 & 0.125 & -\\
\bottomrule
\end{tabular}
\caption{Log rank tests performed using GPSEA to assess association between a genotype and the age of
    death of individuals with a disease. 1/1: biallelic with reference to the indicate variant; 
    0/1: heterozygous with reference to the indicated variant; 0/0: Neither allele has the indicated variant.
    Citations in the xrefs column show previous publications that have presented similar findings.
    \textbf{OMIM:615471}: Mitochondrial DNA depletion syndrome 13 (encephalomyopathic type);
	\textbf{OMIM:256810}: Mitochondrial DNA depletion syndrome 6 (hepatocerebral type);
	\textbf{OMIM:614800}: Short stature, optic nerve atrophy, and Pelger-Huet anomaly.
    }
\label{tab:mortality}
\end{table}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\newpage


\begin{table}
\centering
\begin{tabular}{lp{3.5cm}p{3.5cm}>{\raggedright}p{3cm}lr}
\toprule
\textbf{cohort} & \textbf{genotype (A)} & \textbf{genotype (B)} & \textbf{Scorer} & \textbf{p-val} & \textbf{xrefs}\\
\midrule
ANKRD11 & SV & other & ANKRD11 phenotypical score(*) & $2.6 \times 10^{-04}$ & \cite{PMID_36446582}\\
ANKRD11 & FEMALE & MALE & ANKRD11 phenotypical score(*) & 0.007 & -\\
CHD8 & missense & other & DeVries score & $9.0 \times 10^{-04}$ & -\\
CHD8 & FEMALE & MALE & DeVries score & 0.006 & -\\
CTCF & missense & other & DeVries score & 0.009 & -\\
LMNA & Upstream of NLS & other & Cardiac phenotype score(*) & $1.8 \times 10^{-16}$ & -\\
LMNA & NLS Upstream & other & Cardiac phenotype score(*)& $1.8 \times 10^{-16}$ & -\\
POGZ & Missense & Other &  Nagy et al. Phenotypic  score(*)  & 0.429 & \cite{PMID_35052493}\\
POGZ & Missense & Other &  DeVries score & 0.429 & -\\
RERE & LoF & Atrophin & Jordan et al.  phenotype score(*)  & $1.0 \times 10^{-03}$ & \cite{PMID_29330883}\\
\bottomrule
\end{tabular}
\caption{Mann-Whitney U tests performed using GPSEA to assess association between a genotype and the age of
    death of individuals with a disease. 1/1: biallelic with reference to the indicate variant; 
    0/1: heterozygous with reference to the indicated variant; 0/0: Neither allele has the indicated variant.
    Citations in the xrefs column show previous publications that have presented similar findings. (*)= CountingPhenotypeScorer that assign a phenotype score that is equivalent to the count of present phenotypes that are either an exact match to the query terms or their descendants. See the notebooks for details. DeVries score: an adaption of the DeVries score \cite{PMID_34521999} using HPO.}
\label{tab:to_do}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\newpage

\end{document}