{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ae73ffda",
   "metadata": {},
   "source": [
    "**Generate manusript tables**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "755e11ae",
   "metadata": {},
   "source": [
    "# Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "91f189df",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pathlib\n",
    "import re\n",
    "import textwrap\n",
    "\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "555bdc3a",
   "metadata": {},
   "outputs": [],
   "source": [
    "version_ps = \"0.1.25\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5c9a9dcd",
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_cutoff = .05"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af5d18a2",
   "metadata": {},
   "source": [
    "# Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ebba79e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "analysis = 'v9_6'\n",
    "\n",
    "fpath_project_dir = pathlib.Path(os.path.join(os.getcwd(), os.pardir))\n",
    "fpath_supplement = fpath_project_dir.resolve().joinpath(\"supplement\")\n",
    "fpath_dashboard_dir = fpath_supplement.joinpath(\"scripts\").joinpath(analysis)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "335f458f",
   "metadata": {},
   "source": [
    "## Cohort dashboard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b396aba2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individuals</th>\n",
       "      <th>females</th>\n",
       "      <th>males</th>\n",
       "      <th>n_unknown_sex</th>\n",
       "      <th>total_hpo</th>\n",
       "      <th>total_measurements</th>\n",
       "      <th>hpo_version</th>\n",
       "      <th>gpsea_version</th>\n",
       "      <th>n_total_individual_count</th>\n",
       "      <th>n_alive</th>\n",
       "      <th>n_deceased</th>\n",
       "      <th>n_unknown_vital</th>\n",
       "      <th>n_with_age_of_last_encounter</th>\n",
       "      <th>n_with_onset</th>\n",
       "      <th>n_diseases</th>\n",
       "      <th>disease_string</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cohort</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABCB7</th>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>52</td>\n",
       "      <td>0</td>\n",
       "      <td>2025-01-16</td>\n",
       "      <td>0.9.6</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>Anemia, sideroblastic, and spinocerebellar ataxia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACADM</th>\n",
       "      <td>115</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>115</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>2025-01-16</td>\n",
       "      <td>0.9.6</td>\n",
       "      <td>115</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>115</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Acyl-CoA dehydrogenase, medium chain, deficien...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        individuals  females  males  n_unknown_sex  total_hpo  \\\n",
       "cohort                                                          \n",
       "ABCB7            18        0     18              0         52   \n",
       "ACADM           115        0      0            115          0   \n",
       "\n",
       "        total_measurements hpo_version gpsea_version  \\\n",
       "cohort                                                 \n",
       "ABCB7                    0  2025-01-16         0.9.6   \n",
       "ACADM                   31  2025-01-16         0.9.6   \n",
       "\n",
       "        n_total_individual_count  n_alive  n_deceased  n_unknown_vital  \\\n",
       "cohort                                                                   \n",
       "ABCB7                         18        0           0               18   \n",
       "ACADM                        115        0           0              115   \n",
       "\n",
       "        n_with_age_of_last_encounter  n_with_onset  n_diseases  \\\n",
       "cohort                                                           \n",
       "ABCB7                              0             8           1   \n",
       "ACADM                              0             0           1   \n",
       "\n",
       "                                           disease_string  \n",
       "cohort                                                     \n",
       "ABCB7   Anemia, sideroblastic, and spinocerebellar ataxia  \n",
       "ACADM   Acyl-CoA dehydrogenase, medium chain, deficien...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fpath_df = fpath_dashboard_dir.joinpath(\"cohort_dashboard.txt\")\n",
    "\n",
    "dashboard_cohort = pd.read_csv(fpath_df, sep=\"\\t\", index_col=\"#cohort\")\n",
    "dashboard_cohort.index.name = \"cohort\"\n",
    "\n",
    "dashboard_cohort.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3902808a",
   "metadata": {},
   "source": [
    "## Fisher exact test dashboard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "955c604b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>total_hpo_testable</th>\n",
       "      <th>total_hpo_tested</th>\n",
       "      <th>a_genotype</th>\n",
       "      <th>b_genotype</th>\n",
       "      <th>nsig</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cohort</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABCB7</th>\n",
       "      <td>149</td>\n",
       "      <td>10</td>\n",
       "      <td>p.Gly682Ser</td>\n",
       "      <td>Other variant</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCB7</th>\n",
       "      <td>149</td>\n",
       "      <td>12</td>\n",
       "      <td>ABC transmembrane type-1</td>\n",
       "      <td>Other region</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        total_hpo_testable  total_hpo_tested                a_genotype  \\\n",
       "cohort                                                                   \n",
       "ABCB7                  149                10               p.Gly682Ser   \n",
       "ABCB7                  149                12  ABC transmembrane type-1   \n",
       "\n",
       "           b_genotype  nsig  \n",
       "cohort                       \n",
       "ABCB7   Other variant     0  \n",
       "ABCB7    Other region     0  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fpath_df = fpath_dashboard_dir.joinpath(\"fisher_exact_test_dashboard.txt\")\n",
    "\n",
    "dashboard_fet = pd.read_csv(fpath_df, sep=\"\\t\", index_col=\"#cohort_name\")\n",
    "dashboard_fet.index.name = \"cohort\"\n",
    "\n",
    "dashboard_fet.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0cfd2f75",
   "metadata": {},
   "source": [
    "## Sig Fisher exact test dashboard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e8c8c7c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>a_genotype</th>\n",
       "      <th>b_genotype</th>\n",
       "      <th>nsig</th>\n",
       "      <th>n_tests_performed</th>\n",
       "      <th>hpo_item</th>\n",
       "      <th>with_geno_a</th>\n",
       "      <th>with_geno_b</th>\n",
       "      <th>pval</th>\n",
       "      <th>adj_pval</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cohort</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ATP13A2</th>\n",
       "      <td>OMIM:606693</td>\n",
       "      <td>OMIM:617225</td>\n",
       "      <td>2</td>\n",
       "      <td>29</td>\n",
       "      <td>Bradykinesia [HP:0002067]</td>\n",
       "      <td>30/32 (94%)</td>\n",
       "      <td>4/10 (40%)</td>\n",
       "      <td>0.000915</td>\n",
       "      <td>0.013000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATP13A2</th>\n",
       "      <td>OMIM:606693</td>\n",
       "      <td>OMIM:617225</td>\n",
       "      <td>2</td>\n",
       "      <td>29</td>\n",
       "      <td>Parkinsonism [HP:0001300]</td>\n",
       "      <td>28/28 (100%)</td>\n",
       "      <td>3/11 (27%)</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.000078</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          a_genotype   b_genotype  nsig  n_tests_performed  \\\n",
       "cohort                                                       \n",
       "ATP13A2  OMIM:606693  OMIM:617225     2                 29   \n",
       "ATP13A2  OMIM:606693  OMIM:617225     2                 29   \n",
       "\n",
       "                          hpo_item   with_geno_a with_geno_b      pval  \\\n",
       "cohort                                                                   \n",
       "ATP13A2  Bradykinesia [HP:0002067]   30/32 (94%)  4/10 (40%)  0.000915   \n",
       "ATP13A2  Parkinsonism [HP:0001300]  28/28 (100%)  3/11 (27%)  0.000003   \n",
       "\n",
       "         adj_pval  \n",
       "cohort             \n",
       "ATP13A2  0.013000  \n",
       "ATP13A2  0.000078  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fpath_df = fpath_dashboard_dir.joinpath(\"sig_fisher_exact_test_dashboard.txt\")\n",
    "\n",
    "dashboard_sig_fet = pd.read_csv(fpath_df, sep=\"\\t\", index_col=\"#cohort_name\")\n",
    "dashboard_sig_fet.index.name = \"cohort\"\n",
    "\n",
    "dashboard_sig_fet.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "62010a40",
   "metadata": {},
   "source": [
    "## Measurement dashboard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "bd66debc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>a_genotype</th>\n",
       "      <th>b_genotype</th>\n",
       "      <th>name</th>\n",
       "      <th>test_name</th>\n",
       "      <th>description</th>\n",
       "      <th>variable_name</th>\n",
       "      <th>pval</th>\n",
       "      <th>interpretation</th>\n",
       "      <th>xrefs</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cohort</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACADM</th>\n",
       "      <td>K329Q/K329Q</td>\n",
       "      <td>K329Q/other OR other/other</td>\n",
       "      <td>Measurement</td>\n",
       "      <td>t-test</td>\n",
       "      <td>Value of MCAD Activity% [LOINC:74892-1]</td>\n",
       "      <td>LOINC:74892-1</td>\n",
       "      <td>6.120000e-10</td>\n",
       "      <td>Mean MCAD activity for K329/K329: 0.52\\%, and ...</td>\n",
       "      <td>PMID:33580884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACADM</th>\n",
       "      <td>Y67H/Y67H OR Y67H/other</td>\n",
       "      <td>other/other</td>\n",
       "      <td>Measurement</td>\n",
       "      <td>t-test</td>\n",
       "      <td>Value of MCAD Activity% [LOINC:74892-1]</td>\n",
       "      <td>LOINC:74892-1</td>\n",
       "      <td>2.010000e-05</td>\n",
       "      <td>Mean MCAD activity for Y67H/Y67H: 18.60\\%, and...</td>\n",
       "      <td>PMID:33580884</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     a_genotype                  b_genotype         name  \\\n",
       "cohort                                                                     \n",
       "ACADM               K329Q/K329Q  K329Q/other OR other/other  Measurement   \n",
       "ACADM   Y67H/Y67H OR Y67H/other                 other/other  Measurement   \n",
       "\n",
       "       test_name                              description  variable_name  \\\n",
       "cohort                                                                     \n",
       "ACADM     t-test  Value of MCAD Activity% [LOINC:74892-1]  LOINC:74892-1   \n",
       "ACADM     t-test  Value of MCAD Activity% [LOINC:74892-1]  LOINC:74892-1   \n",
       "\n",
       "                pval                                     interpretation  \\\n",
       "cohort                                                                    \n",
       "ACADM   6.120000e-10  Mean MCAD activity for K329/K329: 0.52\\%, and ...   \n",
       "ACADM   2.010000e-05  Mean MCAD activity for Y67H/Y67H: 18.60\\%, and...   \n",
       "\n",
       "                xrefs  \n",
       "cohort                 \n",
       "ACADM   PMID:33580884  \n",
       "ACADM   PMID:33580884  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fpath_df = fpath_dashboard_dir.joinpath(\"measurement_dashboard.txt\")\n",
    "\n",
    "dashboard_measurement = pd.read_csv(fpath_df, sep=\"\\t\", index_col=\"#cohort\")\n",
    "dashboard_measurement.index.name = \"cohort\"\n",
    "\n",
    "dashboard_measurement.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edafa466",
   "metadata": {},
   "source": [
    "# Compute statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "867e9b79",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Simplify the access\n",
    "cohort_individuals = dashboard_cohort['individuals']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "98de3c81",
   "metadata": {},
   "outputs": [],
   "source": [
    "cohorts = dashboard_cohort.index.unique()\n",
    "\n",
    "# All fet, sig fet, and measurement cohorts are in the cohorts dashboard.\n",
    "assert len(dashboard_fet.index.difference(cohorts)) == 0\n",
    "assert len(dashboard_sig_fet.index.difference(cohorts)) == 0\n",
    "assert len(dashboard_measurement.index.difference(cohorts)) == 0"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81402b90",
   "metadata": {},
   "source": [
    "## N cohorts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d6054fb5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "85"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Number of analyzed cohorts\n",
    "n_cohorts = len(cohorts)\n",
    "n_cohorts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b173ed3e",
   "metadata": {},
   "source": [
    "## N affected individuals"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e29f3d7",
   "metadata": {},
   "source": [
    "Robinow syndrome cohorts consist of several Phenopacket Store cohorts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c3731281",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(16, 7, 7, 2)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import ppktstore\n",
    "from phenopackets.schema.v2.core.individual_pb2 import MALE, FEMALE, UNKNOWN_SEX\n",
    "\n",
    "# The gene responsible for ROBINOW SYNDROME, AUTOSOMAL DOMINANT 2; DRS2 \n",
    "rs_ad_gene = \"DVL1\"\n",
    "\n",
    "registry = ppktstore.registry.configure_phenopacket_registry()\n",
    "with registry.open_phenopacket_store(version_ps) as ps:\n",
    "    cohort = ps.cohort_for_name(\"DVL1\")\n",
    "    n_rs_ad = len(cohort.phenopackets)\n",
    "    n_rs_ad_males = len(list(filter(lambda pp: pp.phenopacket.subject.sex == MALE, cohort.phenopackets)))\n",
    "    n_rs_ad_females = len(list(filter(lambda pp: pp.phenopacket.subject.sex == FEMALE, cohort.phenopackets)))\n",
    "    n_rs_ad_unknown_sex = len(list(filter(lambda pp: pp.phenopacket.subject.sex == UNKNOWN_SEX, cohort.phenopackets)))\n",
    "\n",
    "# Count of individuals with `ROBINOW SYNDROME, AUTOSOMAL DOMINANT 2; DRS2`\n",
    "# which are only members of `Robinow syndrome` cohort.\n",
    "n_rs_ad, n_rs_ad_males, n_rs_ad_females, n_rs_ad_unknown_sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "cf781866",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ABCB7', 'ACADM', 'ACBD6', 'AIRE', 'ANKRD11', 'ASPM', 'ATP13A2',\n",
       "       'ATP6V0C', 'BRD4', 'CHD8', 'CLDN16', 'CNTNAP2', 'COL3A1', 'COQ4',\n",
       "       'CTCF', 'CYP21A2', 'EHMT1', 'EZH1', 'FBN1', 'FBXL4', 'FBXO11', 'FGD1',\n",
       "       'FZD5', 'GLI3', 'HMGCS2', 'IKZF1', 'ITPR1', 'KCNH5', 'KDM6A', 'KDM6B',\n",
       "       'KMT2D', 'LMNA', 'LZTR1', 'MAPK8IP3', 'MPV17', 'NBAS', 'NF1', 'NIPBL',\n",
       "       'NKX6-2', 'PIGA', 'POGZ', 'PPP2R1A', 'PTPN11', 'RERE', 'RNU4-2', 'ROR2',\n",
       "       'RPGRIP1', 'SAMD9L', 'SATB2', 'SCN2A', 'SCO2', 'SEC61A1', 'SETD2',\n",
       "       'SF3B4', 'SLC32A1', 'SLC45A2', 'SLC4A1', 'SMAD2', 'SMAD3', 'SMARCB1',\n",
       "       'SMARCC2', 'SON', 'SPTAN1', 'STXBP1', 'SUOX', 'TBCK', 'TBX1', 'TBX5',\n",
       "       'TGFB2', 'TGFB3', 'TGFBR1', 'TGFBR2', 'TRAF7', 'U2AF2', 'UMOD', 'WWOX',\n",
       "       'ZFX', 'ZMYM3', 'ZNF462', 'ZSWIM6'],\n",
       "      dtype='object', name='cohort')"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# These GPSEA-CS cohorts consist of multiple Phenopacket Store cohorts.\n",
    "\n",
    "non_multigene_cohort = dashboard_cohort.index[~(\n",
    "    dashboard_cohort.index.str.startswith(\"LDS\") |\\\n",
    "    dashboard_cohort.index.str.startswith(\"Robinow\") |\\\n",
    "    dashboard_cohort.index.str.startswith(\"Kabuki\")\n",
    ")]\n",
    "non_multigene_cohort"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "420b3339",
   "metadata": {},
   "source": [
    "### Number of analyzed individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "38ffe7ee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "np.int64(6179)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_individuals = dashboard_cohort.loc[non_multigene_cohort, 'individuals'].sum()\n",
    "\n",
    "n_individuals += n_rs_ad\n",
    "n_individuals"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e822fde",
   "metadata": {},
   "source": [
    "### Number of analyzed individuals with sex information available"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "742056b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "np.int64(5102)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_males_females = dashboard_cohort.loc[\n",
    "    non_multigene_cohort, \n",
    "    ['males', 'females'],\n",
    "].sum()\n",
    "n_males_females['males'] += n_rs_ad_males\n",
    "n_males_females['females'] += n_rs_ad_males\n",
    "\n",
    "n_w_sex_info = n_males_females.sum()\n",
    "n_w_sex_info"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91103edd",
   "metadata": {},
   "source": [
    "## N genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "bfb29b10",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "81"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#n_lds_genes = 4  # 4 LDS genes: {TGFBR1, TGFBR2, SMAD2, SMAD3}\n",
    "#n_kabuki = 2 # but each gene has a separate analysis\n",
    "n_robinow_genes = 1 # 2 Robinow syndrome genes: {DVL1, ROR2} but ROR2 has a separate analysis\n",
    "\n",
    "n_genes = len(non_multigene_cohort) + n_robinow_genes\n",
    "n_genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b45e21f",
   "metadata": {},
   "source": [
    "## N diseases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "e793f09b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "122"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diseases_uq = dashboard_cohort['disease_string'].str.split(';').explode().unique()\n",
    "n_diseases = len(diseases_uq)\n",
    "n_diseases"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e91266f8",
   "metadata": {},
   "source": [
    "## Cohorts with a significant GPC"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "604509e5",
   "metadata": {},
   "source": [
    "### Significant FET"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "fae6b1ad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ATP13A2', 'EHMT1', 'FBN1', 'FBN1', 'FBN1', 'FBN1', 'FBXL4', 'FGD1',\n",
       "       'GLI3', 'GLI3', 'GLI3', 'IKZF1', 'ITPR1', 'ITPR1', 'ITPR1', 'KCNH5',\n",
       "       'KDM6A', 'KDM6A', 'Kabuki', 'LDS 1 and 3', 'LDS 3 and 6', 'LMNA',\n",
       "       'LMNA', 'LMNA', 'LMNA', 'MPV17', 'NBAS', 'NF1', 'NF1', 'NF1', 'NF1',\n",
       "       'NF1', 'PTPN11', 'RPGRIP1', 'RPGRIP1', 'Robinow', 'SAMD9L', 'SATB2',\n",
       "       'SCN2A', 'SCN2A', 'SCO2', 'SETD2', 'SETD2', 'SMAD3', 'SMARCB1',\n",
       "       'SMARCC2', 'SPTAN1', 'SPTAN1', 'SPTAN1', 'SUOX', 'TBCK', 'TBX1', 'TBX5',\n",
       "       'TBX5', 'TGFBR1', 'TGFBR1', 'UMOD', 'UMOD', 'WWOX', 'ZFX', 'ZMYM3'],\n",
       "      dtype='object', name='cohort')"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fet_cohorts_w_sig_gpc = dashboard_fet.loc[dashboard_fet['nsig'] > 0].index\n",
    "fet_cohorts_w_sig_gpc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db60a41e",
   "metadata": {},
   "source": [
    "### Significant measurement or survival"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "dc1b3ed3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ACADM', 'ACADM', 'AIRE', 'ANKRD11', 'ANKRD11', 'CHD8', 'CHD8',\n",
       "       'CLDN16', 'CNTNAP2', 'CTCF', 'CYP21A2', 'FBXL4', 'HMGCS2', 'LMNA',\n",
       "       'MPV17', 'MPV17', 'RERE', 'SETD2', 'SUOX', 'UMOD'],\n",
       "      dtype='object', name='cohort')"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mes_surv_cohorts_w_sig_gpc = dashboard_measurement.loc[dashboard_measurement['pval'] < sig_cutoff].index\n",
    "mes_surv_cohorts_w_sig_gpc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "21f88789",
   "metadata": {},
   "source": [
    "### Cohorts with a significant GPC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8e0b494b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "48"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cohorts_w_sig_gpc = fet_cohorts_w_sig_gpc.union(mes_surv_cohorts_w_sig_gpc).unique()\n",
    "n_cohorts_w_sig_gpc = len(cohorts_w_sig_gpc)\n",
    "n_cohorts_w_sig_gpc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7d8dcffd",
   "metadata": {},
   "source": [
    "## Significant GPC count"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "380a78c5",
   "metadata": {},
   "source": [
    "### Significant FET"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "8a503672",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "np.int64(233)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_sig_fet = dashboard_fet['nsig'].sum()\n",
    "\n",
    "n_sig_fet"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a517ff85",
   "metadata": {},
   "source": [
    "### Significant measurement or survival"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "0b828d4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_measurement_or_survival = dashboard_measurement.loc[dashboard_measurement['pval'] < sig_cutoff].index\n",
    "n_sig_measurement_or_survival = len(sig_measurement_or_survival)\n",
    "\n",
    "n_sig_measurement_or_survival"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a45919b",
   "metadata": {},
   "source": [
    "### Significant GPC count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "fbaa4e46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "np.int64(253)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_sig_gpcs = n_sig_fet + n_sig_measurement_or_survival\n",
    "n_sig_gpcs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44506b82",
   "metadata": {},
   "source": [
    "# Manuscript sections"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e54f91ca",
   "metadata": {},
   "source": [
    "## Abstract"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "d8349a0c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['We applied GPSEA to 85 cohorts with 6179 previously published',\n",
       " 'individuals with variants in one of 81 genes associated with 122',\n",
       " 'Mendelian diseases and identified 253 significant GPCs, with 48',\n",
       " 'cohorts having at least one statistically significant GPC.']"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "textwrap.wrap(\n",
    "f'''We applied GPSEA to {n_cohorts} cohorts with {n_individuals} previously published individuals\n",
    "with variants in one of {n_genes} genes associated with {n_diseases} Mendelian diseases\n",
    "and identified {n_sig_gpcs} significant GPCs,\n",
    "with {n_cohorts_w_sig_gpc} cohorts having at least one statistically significant GPC.\n",
    "''',\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71cd2cb0",
   "metadata": {},
   "source": [
    "## Introduction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "d8daa769",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['We applied the software to 85 cohorts, 48 (56%) of which had at least',\n",
       " 'one statistically significant GPC (there were a total of 253',\n",
       " 'statistically significant results)']"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "textwrap.wrap(\n",
    "f\"\"\"We applied the software to {n_cohorts} cohorts,\n",
    "{n_cohorts_w_sig_gpc} ({n_cohorts_w_sig_gpc/n_cohorts:.0%}) of which had at least one statistically significant GPC\n",
    "(there were a total of {n_sig_gpcs} statistically significant results)\n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af64f6e6",
   "metadata": {},
   "source": [
    "## Material and methods"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb4a1ec9",
   "metadata": {},
   "source": [
    "### Input data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01eac7e3",
   "metadata": {},
   "source": [
    "Phenopacket Store statistics are summarized in [PhenopacketStoreStats](https://github.com/monarch-initiative/phenopacket-store/blob/0.1.25/PhenopacketStoreStats.ipynb) notebook."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0999b6e7",
   "metadata": {},
   "source": [
    "### Cohorts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "8989a79f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['A total of 85 cohorts were chosen for GPSEA analysis from version',\n",
       " '0.1.25 of Phenopacket Store. The cohorts had a mean of 77.8',\n",
       " 'individuals (median 49.0, minimum 16, maximum 462). The cohorts',\n",
       " 'comprised information in 6179 individuals. Information on the sex of',\n",
       " 'participants was available for 82.6% of these individuals, with 53%',\n",
       " 'being male and 47% female.']"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "textwrap.wrap(\n",
    "f\"\"\"A total of {n_cohorts} cohorts were chosen for GPSEA analysis from version {version_ps} of Phenopacket Store.\n",
    "The cohorts had a mean of {cohort_individuals.mean()} individuals (median {cohort_individuals.median()}, minimum {cohort_individuals.min()}, maximum {cohort_individuals.max()}).\n",
    "The cohorts comprised information in {n_individuals} individuals.\n",
    "Information on the sex of participants was available for {n_w_sex_info/n_individuals:.1%} of these individuals,\n",
    "with {n_males_females['males'] / n_males_females.sum():.0%} being male\n",
    "and {n_males_females['females'] / n_males_females.sum():.0%} female.\n",
    "\"\"\", \n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c83d62a7",
   "metadata": {},
   "source": [
    "## Results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0283cf1",
   "metadata": {},
   "source": [
    "### Table 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "ee9e412e",
   "metadata": {},
   "outputs": [],
   "source": [
    "stat_procedures = [\n",
    "    \"Categorical analysis\",\n",
    "    \"t test\",\n",
    "    \"HPO onset\",\n",
    "    \"Disease onset\",\n",
    "    \"Mortality\",\n",
    "    \"Phenotype scores\",\n",
    "    \"Disease diagnosis\",\n",
    "    \"Sex differences\",\n",
    "]\n",
    "stat_proc2cohorts_tested = {k: set() for k in stat_procedures}\n",
    "stat_proc2tests_performed = {k: 0 for k in stat_procedures}\n",
    "stat_proc2sig_tests = {k: 0 for k in stat_procedures}\n",
    "\n",
    "tableS10_sig = pd.DataFrame(\n",
    "    data=0,\n",
    "    index=cohorts,\n",
    "    columns=stat_procedures,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cdd636d4",
   "metadata": {},
   "source": [
    "#### FET"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "567b32c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "male_female = (\"MALE\", \"FEMALE\")\n",
    "\n",
    "sex_diffs = \"Sex differences\"\n",
    "dg_diff = \"Disease diagnosis\"\n",
    "cat_analysis = \"Categorical analysis\"\n",
    "\n",
    "for cohort_name, row in dashboard_fet.iterrows():\n",
    "    if row['a_genotype'] in male_female and row['b_genotype'] in male_female:\n",
    "        # Testing for sex differences\n",
    "        stat_proc2cohorts_tested[sex_diffs].add(cohort_name)\n",
    "        stat_proc2tests_performed[sex_diffs] += row[\"total_hpo_tested\"]\n",
    "        stat_proc2sig_tests[sex_diffs] += row[\"nsig\"]\n",
    "\n",
    "        tableS10_sig.loc[cohort_name, sex_diffs] += row[\"nsig\"] # type: ignore\n",
    "    elif row['a_genotype'].startswith(\"OMIM:\") and row[\"b_genotype\"].startswith(\"OMIM:\"):\n",
    "        # Testing for differences in phenotypes between disease diagnoses\n",
    "        stat_proc2cohorts_tested[dg_diff].add(cohort_name)\n",
    "        stat_proc2tests_performed[dg_diff] += row[\"total_hpo_tested\"]\n",
    "        stat_proc2sig_tests[dg_diff] += row[\"nsig\"]\n",
    "\n",
    "        tableS10_sig.loc[cohort_name, dg_diff] += row[\"nsig\"] # type: ignore\n",
    "    else:\n",
    "        # Categorical analysis\n",
    "        stat_proc2cohorts_tested[cat_analysis].add(cohort_name)\n",
    "        stat_proc2tests_performed[cat_analysis] += row[\"total_hpo_tested\"]\n",
    "        stat_proc2sig_tests[cat_analysis] += row[\"nsig\"]\n",
    "\n",
    "        tableS10_sig.loc[cohort_name, cat_analysis] += row[\"nsig\"] # type: ignore"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4ce2f98d",
   "metadata": {},
   "source": [
    "#### Measurements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "4c932d97",
   "metadata": {},
   "outputs": [],
   "source": [
    "def test_name2_stat_procedure(name: str) -> str:\n",
    "    if name.startswith(\"Onset\"):\n",
    "        if \"OMIM\" in name:\n",
    "            return \"Disease onset\"\n",
    "        else:\n",
    "            return \"HPO onset\"\n",
    "    elif name == \"Age of death\":\n",
    "        return \"Mortality\"\n",
    "    elif name == \"t-test\":\n",
    "        return \"t test\"\n",
    "    elif name.endswith(\"Score\") or name.endswith(\"Count\"):\n",
    "        return \"Phenotype scores\"\n",
    "    else:\n",
    "        raise ValueError(f\"Unexpected test name {name}\")\n",
    "\n",
    "# Not covering the following since they are not stored in the measurement dashboard:\n",
    "# - Sex differences\n",
    "# - Disease diagnosis\n",
    "# - Categorical analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "f6433cad",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "for cohort_name, row in dashboard_measurement.iterrows():\n",
    "    procedure = test_name2_stat_procedure(row['test_name'])\n",
    "    stat_proc2cohorts_tested[procedure].add(cohort_name)\n",
    "    stat_proc2tests_performed[procedure] += 1\n",
    "    if row['pval'] < sig_cutoff:\n",
    "        stat_proc2sig_tests[procedure] += 1\n",
    "        tableS10_sig.loc[cohort_name, procedure] += 1 # type: ignore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "b7fc497b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cohorts tested</th>\n",
       "      <th>Tests performed</th>\n",
       "      <th>Significant tests</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Categorical analysis</th>\n",
       "      <td>78</td>\n",
       "      <td>6736</td>\n",
       "      <td>217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>t test</th>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPO onset</th>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Disease onset</th>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mortality</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Phenotype scores</th>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Disease diagnosis</th>\n",
       "      <td>8</td>\n",
       "      <td>266</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sex differences</th>\n",
       "      <td>44</td>\n",
       "      <td>1979</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Cohorts tested  Tests performed  Significant tests\n",
       "Categorical analysis              78             6736                217\n",
       "t test                             2                3                  3\n",
       "HPO onset                          4                6                  3\n",
       "Disease onset                     10               11                  6\n",
       "Mortality                          3                3                  1\n",
       "Phenotype scores                   6                9                  7\n",
       "Disease diagnosis                  8              266                 15\n",
       "Sex differences                   44             1979                  1"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table1 = pd.DataFrame(\n",
    "    {\n",
    "        'Cohorts tested': {k: len(cohorts) for k, cohorts in stat_proc2cohorts_tested.items()},\n",
    "        'Tests performed': stat_proc2tests_performed,\n",
    "        'Significant tests': stat_proc2sig_tests,\n",
    "    }\n",
    ")\n",
    "table1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "662a8e11",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(85, np.int64(9013), np.int64(253))"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_cohorts, table1['Tests performed'].sum(), table1['Significant tests'].sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da3b4be3",
   "metadata": {},
   "source": [
    "### Check Supplemental tables S3-9\n",
    "\n",
    "Ensure the counts of significant tests are consistent between the supplemental tables and the `Significant tests` column.\n",
    "The tables S3-9 list only the significant findings.\n",
    "\n",
    "The review is done manually, by looking at `supplement_tables.tex` and the `table1` data frame."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2988e0c2",
   "metadata": {},
   "source": [
    "### Supplemental table S10"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df6d1cf2",
   "metadata": {},
   "source": [
    "#### Significant GPC counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "609b8356",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check that the cohorts with at least one significant GPC match\n",
    "# between Table S10 and the results derived from dashboards.\n",
    "\n",
    "assert set(cohorts_w_sig_gpc) == set(tableS10_sig.index[tableS10_sig.sum(axis=1) > 0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "87ed9fe8",
   "metadata": {},
   "outputs": [],
   "source": [
    "assert tableS10_sig.loc[cohorts_w_sig_gpc].sum(axis=1).sum() == n_sig_gpcs\n",
    "assert len(tableS10_sig.loc[cohorts_w_sig_gpc]) == n_cohorts_w_sig_gpc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d65c307",
   "metadata": {},
   "source": [
    "#### Published GPC counts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c65ab864",
   "metadata": {},
   "source": [
    "The numbers were ascertained by manual literature review."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "144ac39f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#                                   |<--  Survival -->|\n",
    "#   Cohort  Sig  Pub  Catg    t tst HPO   Dis.  Mort. Pscr  DisDg SexDiff\n",
    "published_gpcs_latex = \"\"\"\n",
    "ACADM       &  2 &  2 &   -   & 2/2 &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "AIRE        &  1 &  1 &   -   &  -  & 1/1 &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "ANKRD11     &  2 &  1 &   -   &  -  &  -  &  -  &  -  & 1/2 &  -  &  -  \\\\\n",
    "ATP13A2     &  2 &  2 &   -   &  -  &  -  &  -  &  -  &  -  & 2/2 &  -  \\\\\n",
    "CHD8        &  2 &  1 &   -   &  -  &  -  &  -  &  -  & 1/2 &  -  &  -  \\\\\n",
    "CLDN16      &  1 &  1 &   -   &  -  & 1/1 &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "CNTNAP2     &  1 &  0 &   -   &  -  &  -  & 0/1 &  -  &  -  &  -  &  -  \\\\\n",
    "CTCF        &  1 &  0 &   -   &  -  &  -  &  -  &  -  & 0/1 &  -  &  -  \\\\\n",
    "CYP21A2     &  1 &  1 &   -   & 1/1 &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "EHMT1       &  1 &  1 &  1/1  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "FBN1        & 14 & 14 & 14/14 &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "FBXL4       &  2 &  0 &  0/1  &  -  &  -  & 0/1 &  -  &  -  &  -  &  -  \\\\\n",
    "FGD1        &  1 &  1 &  1/1  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "GLI3        & 14 & 14 & 14/14 &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "HMGCS2      &  1 &  0 &    -  &  -  &  -  & 0/1 &  -  &  -  &  -  &  -  \\\\\n",
    "IKZF1       &  2 &  2 &  2/2  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "ITPR1       & 21 &  0 &  0/21 &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "Kabuki      &  2 &  0 &    -  &  -  &  -  &  -  &  -  &  -  & 0/2 &  -  \\\\\n",
    "KCNH5       &  1 &  1 &  1/1  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "KDM6A       &  2 &  1 &  1/1  &  -  &  -  &  -  &  -  &  -  &  -  & 0/1 \\\\\n",
    "LDS 1 and 3 &  4 &  0 &    -  &  -  &  -  &  -  &  -  &  -  & 0/4 &  -  \\\\\n",
    "LDS 3 and 6 &  1 &  0 &    -  &  -  &  -  &  -  &  -  &  -  & 0/1 &  -  \\\\\n",
    "LMNA        & 36 & 36 & 35/35 &  -  &  -  &  -  &  -  & 1/1 &  -  &  -  \\\\\n",
    "MPV17       &  3 &  1 &  0/1  &  -  &  -  & 0/1 & 1/1 &  -  &  -  &  -  \\\\\n",
    "NBAS        &  1 &  0 &  0/1  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "NF1         & 37 & 37 & 37/37 &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "PTPN11      &  4 &  0 &  0/4  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "RERE        &  1 &  1 &   -   &  -  &  -  &  -  &  -  & 1/1 &  -  &  -  \\\\\n",
    "Robinow     &  4 &  0 &   -   &  -  &  -  &  -  &  -  &  -  & 0/4 &  -  \\\\\n",
    "RPGRIP1     &  3 &  1 &  1/1  &  -  &  -  &  -  &  -  &  -  & 0/2 &  -  \\\\\n",
    "SAMD9L      &  3 &  0 &  0/3  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "SATB2       &  1 &  0 &  0/1  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "SCN2A       &  9 &  9 &  9/9  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "SCO2        &  1 &  0 &  0/1  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "SETD2       &  9 &  8 &  8/8  &  -  &  -  & 0/1 &  -  &  -  &  -  &  -  \\\\\n",
    "SMAD3       &  1 &  0 &  0/1  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "SMARCB1     &  6 &  0 &  0/6  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "SMARCC2     &  1 &  1 &  1/1  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "SPTAN1      & 35 & 35 & 35/35 &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "SUOX        &  2 &  1 &  1/1  &  -  &  -  & 0/1 &  -  &  -  &  -  &  -  \\\\\n",
    "TBCK        &  2 &  0 &  0/2  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "TBX1        &  2 &  0 &  0/2  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "TBX5        &  3 &  3 &  3/3  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "TGFBR1      &  4 &  1 &  1/4  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "UMOD        &  3 &  3 &  2/2  &  -  & 1/1 &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "WWOX        &  1 &  1 &  1/1  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "ZFX         &  1 &  1 &  1/1  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "ZMYM3       &  1 &  0 &  0/1  &  -  &  -  &  -  &  -  &  -  &  -  &  -  \\\\\n",
    "\"\"\".splitlines()[1:] # throw away the first blank line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "d239a9f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "fraction_re = re.compile(r'(?P<published>\\d+)/(?P<total>\\d+)')\n",
    "\n",
    "def parse_value(val: str) -> int:\n",
    "    stripped = val.strip()\n",
    "    matcher = fraction_re.match(stripped)\n",
    "    if matcher:\n",
    "        return int(matcher.group(\"published\"))\n",
    "    elif stripped == '-':\n",
    "        return 0\n",
    "    else:\n",
    "        raise ValueError(f\"Unexpected value {val}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "98b96905",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Categorical analysis</th>\n",
       "      <th>t test</th>\n",
       "      <th>HPO onset</th>\n",
       "      <th>Disease onset</th>\n",
       "      <th>Mortality</th>\n",
       "      <th>Phenotype scores</th>\n",
       "      <th>Disease diagnosis</th>\n",
       "      <th>Sex differences</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cohort</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABCB7</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACADM</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACBD6</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AIRE</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANKRD11</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASPM</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATP13A2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATP6V0C</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRD4</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHD8</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Categorical analysis  t test  HPO onset  Disease onset  Mortality  \\\n",
       "cohort                                                                       \n",
       "ABCB7                       0       0          0              0          0   \n",
       "ACADM                       0       2          0              0          0   \n",
       "ACBD6                       0       0          0              0          0   \n",
       "AIRE                        0       0          1              0          0   \n",
       "ANKRD11                     0       0          0              0          0   \n",
       "ASPM                        0       0          0              0          0   \n",
       "ATP13A2                     0       0          0              0          0   \n",
       "ATP6V0C                     0       0          0              0          0   \n",
       "BRD4                        0       0          0              0          0   \n",
       "CHD8                        0       0          0              0          0   \n",
       "\n",
       "         Phenotype scores  Disease diagnosis  Sex differences  \n",
       "cohort                                                         \n",
       "ABCB7                   0                  0                0  \n",
       "ACADM                   0                  0                0  \n",
       "ACBD6                   0                  0                0  \n",
       "AIRE                    0                  0                0  \n",
       "ANKRD11                 1                  0                0  \n",
       "ASPM                    0                  0                0  \n",
       "ATP13A2                 0                  2                0  \n",
       "ATP6V0C                 0                  0                0  \n",
       "BRD4                    0                  0                0  \n",
       "CHD8                    1                  0                0  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tableS10_published = pd.DataFrame(\n",
    "    data=0,\n",
    "    index=cohorts,\n",
    "    columns=stat_procedures,\n",
    ")\n",
    "\n",
    "# 0              1    2     3     4       5    6    7      8     9     10\n",
    "# Cohort         Sig  Pub   Catg  t tst   HPO  Dis. Mort.  Pscr  DisDg SexDiff\n",
    "\n",
    "for row in published_gpcs_latex:\n",
    "    cols = row.replace(\"\\\\\", \"\").split('&')\n",
    "    assert len(cols) == 11, f\"{len(cols)} should be 11\"\n",
    "    cohort_name = cols[0].strip()\n",
    "\n",
    "    tableS10_published.loc[cohort_name, cat_analysis] += parse_value(cols[3]) # type: ignore\n",
    "    tableS10_published.loc[cohort_name, 't test'] += parse_value(cols[4]) # type: ignore\n",
    "    tableS10_published.loc[cohort_name, 'HPO onset'] += parse_value(cols[5]) # type: ignore\n",
    "    tableS10_published.loc[cohort_name, 'Disease onset'] += parse_value(cols[6]) # type: ignore\n",
    "    tableS10_published.loc[cohort_name, 'Mortality'] += parse_value(cols[7]) # type: ignore\n",
    "    tableS10_published.loc[cohort_name, 'Phenotype scores'] += parse_value(cols[8]) # type: ignore\n",
    "    tableS10_published.loc[cohort_name, dg_diff] += parse_value(cols[9]) # type: ignore\n",
    "    tableS10_published.loc[cohort_name, sex_diffs] += parse_value(cols[10]) # type: ignore\n",
    "\n",
    "tableS10_published.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "d5e9fd4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sig</th>\n",
       "      <th>Pub</th>\n",
       "      <th>Categorical analysis</th>\n",
       "      <th>t test</th>\n",
       "      <th>HPO onset</th>\n",
       "      <th>Disease onset</th>\n",
       "      <th>Mortality</th>\n",
       "      <th>Phenotype scores</th>\n",
       "      <th>Disease diagnosis</th>\n",
       "      <th>Sex differences</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cohort</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACADM</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>-</td>\n",
       "      <td>2/2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AIRE</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1/1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Sig Pub Categorical analysis t test HPO onset Disease onset Mortality  \\\n",
       "cohort                                                                         \n",
       "ACADM    2   2                    -    2/2         -             -         -   \n",
       "AIRE     1   1                    -      -       1/1             -         -   \n",
       "\n",
       "       Phenotype scores Disease diagnosis Sex differences  \n",
       "cohort                                                     \n",
       "ACADM                 -                 -               -  \n",
       "AIRE                  -                 -               -  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tableS10 = pd.DataFrame(\n",
    "    \"\",\n",
    "    index=cohorts,\n",
    "    columns=[\"Sig\", \"Pub\"] + stat_procedures\n",
    ")\n",
    "\n",
    "for cohort_name in cohorts:\n",
    "    for procedure in stat_procedures:\n",
    "        nsig = tableS10_sig.loc[cohort_name, procedure]\n",
    "        if nsig > 0:\n",
    "            pub = tableS10_published.loc[cohort_name, procedure]\n",
    "            tableS10.loc[cohort_name, procedure] = f\"{pub}/{nsig}\"\n",
    "        else:\n",
    "            tableS10.loc[cohort_name, procedure] = \"-\"\n",
    "    tableS10.loc[cohort_name, \"Sig\"] = tableS10_sig.loc[cohort_name].sum()\n",
    "    tableS10.loc[cohort_name, \"Pub\"] = tableS10_published.loc[cohort_name].sum()\n",
    "\n",
    "tableS10.loc[cohorts_w_sig_gpc].head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "70bcdb5f",
   "metadata": {},
   "source": [
    "Generate LaTeX-like content for the Supplemental Table S10 body. The content is to be copy-pasted into `supplement_tables.tex`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "d1787774",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACADM & 2 & 2 & - & 2/2 & - & - & - & - & - & - \\\\\n",
      "AIRE & 1 & 1 & - & - & 1/1 & - & - & - & - & - \\\\\n",
      "ANKRD11 & 2 & 1 & - & - & - & - & - & 1/2 & - & - \\\\\n",
      "ATP13A2 & 2 & 2 & - & - & - & - & - & - & 2/2 & - \\\\\n",
      "CHD8 & 2 & 1 & - & - & - & - & - & 1/2 & - & - \\\\\n",
      "CLDN16 & 1 & 1 & - & - & 1/1 & - & - & - & - & - \\\\\n",
      "CNTNAP2 & 1 & 0 & - & - & - & 0/1 & - & - & - & - \\\\\n",
      "CTCF & 1 & 0 & - & - & - & - & - & 0/1 & - & - \\\\\n",
      "CYP21A2 & 1 & 1 & - & 1/1 & - & - & - & - & - & - \\\\\n",
      "EHMT1 & 1 & 1 & 1/1 & - & - & - & - & - & - & - \\\\\n",
      "FBN1 & 14 & 14 & 14/14 & - & - & - & - & - & - & - \\\\\n",
      "FBXL4 & 2 & 0 & 0/1 & - & - & 0/1 & - & - & - & - \\\\\n",
      "FGD1 & 1 & 1 & 1/1 & - & - & - & - & - & - & - \\\\\n",
      "GLI3 & 14 & 14 & 14/14 & - & - & - & - & - & - & - \\\\\n",
      "HMGCS2 & 1 & 0 & - & - & - & 0/1 & - & - & - & - \\\\\n",
      "IKZF1 & 2 & 2 & 2/2 & - & - & - & - & - & - & - \\\\\n",
      "ITPR1 & 21 & 0 & 0/21 & - & - & - & - & - & - & - \\\\\n",
      "KCNH5 & 1 & 1 & 1/1 & - & - & - & - & - & - & - \\\\\n",
      "KDM6A & 2 & 1 & 1/1 & - & - & - & - & - & - & 0/1 \\\\\n",
      "Kabuki & 2 & 0 & - & - & - & - & - & - & 0/2 & - \\\\\n",
      "LDS 1 and 3 & 4 & 0 & - & - & - & - & - & - & 0/4 & - \\\\\n",
      "LDS 3 and 6 & 1 & 0 & - & - & - & - & - & - & 0/1 & - \\\\\n",
      "LMNA & 36 & 36 & 35/35 & - & - & - & - & 1/1 & - & - \\\\\n",
      "MPV17 & 3 & 1 & 0/1 & - & - & 0/1 & 1/1 & - & - & - \\\\\n",
      "NBAS & 1 & 0 & 0/1 & - & - & - & - & - & - & - \\\\\n",
      "NF1 & 37 & 37 & 37/37 & - & - & - & - & - & - & - \\\\\n",
      "PTPN11 & 4 & 0 & 0/4 & - & - & - & - & - & - & - \\\\\n",
      "RERE & 1 & 1 & - & - & - & - & - & 1/1 & - & - \\\\\n",
      "RPGRIP1 & 3 & 1 & 1/1 & - & - & - & - & - & 0/2 & - \\\\\n",
      "Robinow & 4 & 0 & - & - & - & - & - & - & 0/4 & - \\\\\n",
      "SAMD9L & 3 & 0 & 0/3 & - & - & - & - & - & - & - \\\\\n",
      "SATB2 & 1 & 0 & 0/1 & - & - & - & - & - & - & - \\\\\n",
      "SCN2A & 9 & 9 & 9/9 & - & - & - & - & - & - & - \\\\\n",
      "SCO2 & 1 & 0 & 0/1 & - & - & - & - & - & - & - \\\\\n",
      "SETD2 & 9 & 8 & 8/8 & - & - & 0/1 & - & - & - & - \\\\\n",
      "SMAD3 & 1 & 0 & 0/1 & - & - & - & - & - & - & - \\\\\n",
      "SMARCB1 & 6 & 0 & 0/6 & - & - & - & - & - & - & - \\\\\n",
      "SMARCC2 & 1 & 1 & 1/1 & - & - & - & - & - & - & - \\\\\n",
      "SPTAN1 & 35 & 35 & 35/35 & - & - & - & - & - & - & - \\\\\n",
      "SUOX & 2 & 1 & 1/1 & - & - & 0/1 & - & - & - & - \\\\\n",
      "TBCK & 2 & 0 & 0/2 & - & - & - & - & - & - & - \\\\\n",
      "TBX1 & 2 & 0 & 0/2 & - & - & - & - & - & - & - \\\\\n",
      "TBX5 & 3 & 3 & 3/3 & - & - & - & - & - & - & - \\\\\n",
      "TGFBR1 & 4 & 1 & 1/4 & - & - & - & - & - & - & - \\\\\n",
      "UMOD & 3 & 3 & 2/2 & - & 1/1 & - & - & - & - & - \\\\\n",
      "WWOX & 1 & 1 & 1/1 & - & - & - & - & - & - & - \\\\\n",
      "ZFX & 1 & 1 & 1/1 & - & - & - & - & - & - & - \\\\\n",
      "ZMYM3 & 1 & 0 & 0/1 & - & - & - & - & - & - & - \\\\\n"
     ]
    }
   ],
   "source": [
    "latex_lines = []\n",
    "\n",
    "for cohort_name, row in tableS10.loc[cohorts_w_sig_gpc].iterrows():\n",
    "    latex_lines.append(\n",
    "        \" & \".join(\n",
    "            [cohort_name] + [f\"{val}\" for val in row]\n",
    "        ) + \" \\\\\\\\\"\n",
    "    )\n",
    "print(os.linesep.join(latex_lines))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5197fa4",
   "metadata": {},
   "source": [
    "### Characterizing genotype-phenotype correlations with GPSEA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "0107f920",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['We tested GPSEA on 85 cohorts, covering 81 genes and 122 diseases. We',\n",
       " 'first explain the algorithmic approaches to setting up GPC testing and',\n",
       " 'then present an overview of 253 significant correlations identified in',\n",
       " 'the cohorts.']"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "textwrap.wrap(\n",
    "f\"\"\"We tested GPSEA on {n_cohorts} cohorts, covering {n_genes} genes and {n_diseases} diseases.\n",
    "We first explain the algorithmic approaches to setting up GPC testing and then present\n",
    "an overview of {n_sig_gpcs} significant correlations identified in the cohorts.\n",
    "\"\"\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e5286b4b",
   "metadata": {},
   "source": [
    "### Independent Filtering for Human Phenotype Ontology (IF-HPO)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "bd1508cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['IF-HPO reduces the total number of tested terms by over ten-fold in',\n",
       " 'the cohorts analyzed here (before filtering: mean 304, median 276.5,',\n",
       " 'min 45, max 967; following independent filtering: mean 40, median',\n",
       " '28.0, min 1, max 225).']"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_hpo_pre_filter = dashboard_fet['total_hpo_testable']\n",
    "n_hpo_post_filter = dashboard_fet['total_hpo_tested']\n",
    "\n",
    "textwrap.wrap(\n",
    "f\"\"\"IF-HPO reduces the total number of tested terms by over ten-fold in the cohorts analyzed here\n",
    "(before filtering: mean {n_hpo_pre_filter.mean():.0f}, median {n_hpo_pre_filter.median()}, min {n_hpo_pre_filter.min()}, max {n_hpo_pre_filter.max()};\n",
    "following independent filtering: mean {n_hpo_post_filter.mean():.0f}, median {n_hpo_post_filter.median()}, min {n_hpo_post_filter.min()}, max {n_hpo_post_filter.max()}).\n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "885a3802",
   "metadata": {},
   "source": [
    "### GPCs are common in Mendelian disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "35d9bdbc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['We analyzed 85 cohorts with 6179 individuals (median 49.0 per cohort,',\n",
       " 'range: 16-462) with 122 Mendelian diseases.']"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "textwrap.wrap(\n",
    "f\"\"\"We analyzed {n_cohorts} cohorts with {n_individuals} individuals\n",
    "(median {cohort_individuals.median()} per cohort, range: {cohort_individuals.min()}-{cohort_individuals.max()})\n",
    "with {n_diseases} Mendelian diseases.\n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "c0c9dcb0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'A total of 253 significant correlations were identified.'"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f\"A total of {n_sig_gpcs} significant correlations were identified.\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "fa7870d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Significant results were identified for 48 cohorts.'"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f\"Significant results were identified for {n_cohorts_w_sig_gpc} cohorts.\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "e44e772f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'We identified previously published GPCs for 29 of these cohorts, many of which overlapped with our findings ...'"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cohorts_wo_pub_gpc = tableS10_published.index[tableS10_published.sum(axis=1) == 0].intersection(cohorts_w_sig_gpc)\n",
    "cohorts_w_pub_gpc  = tableS10_published.index[tableS10_published.sum(axis=1) > 0].intersection(cohorts_w_sig_gpc)\n",
    "n_cohorts_w_pub_gpc = len(cohorts_w_pub_gpc)\n",
    "\n",
    "f\"We identified previously published GPCs for {n_cohorts_w_pub_gpc} of these cohorts, many of which overlapped with our findings ...\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "c5a062e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "np.int64(71)"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonpub_sig_gpc = tableS10_sig - tableS10_published\n",
    "n_nonpub_sig_gpc = nonpub_sig_gpc.sum(axis=1).sum()\n",
    "n_nonpub_sig_gpc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "2bc6f7fe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['71 significant findings in the remaining 19 cohorts represent',\n",
       " 'candidate GPCs that should be validated by independent studies on',\n",
       " 'validation cohorts ...']"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_cohorts_w_nonpub_sig_gpc = n_cohorts_w_sig_gpc - n_cohorts_w_pub_gpc\n",
    "\n",
    "textwrap.wrap(\n",
    "f\"\"\"{n_nonpub_sig_gpc} significant findings in the remaining {n_cohorts_w_nonpub_sig_gpc}\n",
    "cohorts represent candidate GPCs that should be validated by independent studies on validation cohorts ...\"\"\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c6ce1615",
   "metadata": {},
   "source": [
    "## Distribution of phenotypic features with significant GPCs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c91d523",
   "metadata": {},
   "source": [
    "See `scripts/README.md` > Proportions."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "190072e8",
   "metadata": {},
   "source": [
    "*-* EOF *-*"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv (3.12.3)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
