{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ACADM\n",
    "[medium-chain acyl-CoA dehydrogenase deficiency (ACADMD)](https://omim.org/entry/201450) is caused by homozygous or compound heterozygous mutation in the medium-chain acyl-CoA dehydrogenase gene (ACADM).\n",
    "Here we ingest a dataset reported in [Tucci S, et al. (2021) Genotype and residual enzyme activity in medium-chain acyl-CoA dehydrogenase (MCAD) deficiency: Are predictions possible? J Inherit Metab Dis.](https://pubmed.ncbi.nlm.nih.gov/33580884/). Limited clinical information is provided, but we can collect data on genotypes and enzyme activity to show how GPSEA can be used to perform tests of association of genotypes with numerical measurements using a t test."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-12T15:52:32.211511Z",
     "start_time": "2024-10-12T15:52:31.771672Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pyphetools version 0.9.109\n"
     ]
    }
   ],
   "source": [
    "from pyphetools.creation import Measurements, VariantManager, IntergenicVariant, StructuralVariant, OntologyTerms, MetaData\n",
    "from pyphetools.pp import *\n",
    "import pandas as pd\n",
    "import os\n",
    "from google.protobuf.json_format import MessageToJson\n",
    "import pyphetools\n",
    "print(f\"pyphetools version {pyphetools.__version__}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-12T15:52:41.063434Z",
     "start_time": "2024-10-12T15:52:40.865904Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Individual</th>\n",
       "      <th>Allel1</th>\n",
       "      <th>Amino Acid Change</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Allel2</th>\n",
       "      <th>Amino Acid Change.1</th>\n",
       "      <th>Exon.1</th>\n",
       "      <th>MCAD Activity%</th>\n",
       "      <th>VLCAD Activity</th>\n",
       "      <th>AC C8 [µmol/L]b</th>\n",
       "      <th>AC C6 [µmol/L]b</th>\n",
       "      <th>AC C10 [µmol/L]b</th>\n",
       "      <th>AC others [µmol/L] or ratio</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11.0</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11.0</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11.0</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C8/C10 17.2; C8/C12 24.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11.0</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>11.86 (ref. &lt;0.269)</td>\n",
       "      <td>192163)</td>\n",
       "      <td>1.23 (ref. &lt;0.403)</td>\n",
       "      <td>C10:1 0.46 (ref.&lt;0.106); C6/C2 0.13 (ref.&lt;0.00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11.0</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11</td>\n",
       "      <td>0.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Individual    Allel1 Amino Acid Change  Exon    Allel2 Amino Acid Change.1  \\\n",
       "0           1  c.985A>G           p.K329E  11.0  c.985A>G             p.K329E   \n",
       "1           2  c.985A>G           p.K329E  11.0  c.985A>G             p.K329E   \n",
       "2           3  c.985A>G           p.K329E  11.0  c.985A>G             p.K329E   \n",
       "3           4  c.985A>G           p.K329E  11.0  c.985A>G             p.K329E   \n",
       "4           5  c.985A>G           p.K329E  11.0  c.985A>G             p.K329E   \n",
       "\n",
       "  Exon.1  MCAD Activity%  VLCAD Activity      AC C8 [µmol/L]b AC C6 [µmol/L]b  \\\n",
       "0     11             0.0            72.0                  NaN             NaN   \n",
       "1     11             0.0            71.0                  NaN             NaN   \n",
       "2     11             0.0            82.0                  NaN             NaN   \n",
       "3     11             0.0            88.0  11.86 (ref. <0.269)         192163)   \n",
       "4     11             0.0            79.0                  NaN             NaN   \n",
       "\n",
       "     AC C10 [µmol/L]b                        AC others [µmol/L] or ratio  \n",
       "0                 NaN                                                NaN  \n",
       "1                 NaN                                                NaN  \n",
       "2                 NaN                           C8/C10 17.2; C8/C12 24.1  \n",
       "3  1.23 (ref. <0.403)  C10:1 0.46 (ref.<0.106); C6/C2 0.13 (ref.<0.00...  \n",
       "4                 NaN                                                NaN  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(\"input/Tucci_ACADM_data.xlsx\")\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove rows without date\n",
    "Some of the rows either have no variant data or no MCAD activity data.\n",
    "We also filter out rows that do not describe biallelic variants, e.g., one of the columns is WT."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-12T15:52:44.987784Z",
     "start_time": "2024-10-12T15:52:44.980173Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtered from 460 to 115\n"
     ]
    }
   ],
   "source": [
    "n_rows_1 = len(df)\n",
    "df = df[(df['Allel1'] != 'ns') & (df['Allel2'] != 'ns') & (df['MCAD Activity%'].astype(str).str.strip().replace('nan', '') != '')]\n",
    "df = df[(df['Allel1'] != 'WT') & (df['Allel2'] != 'WT') ]\n",
    "df = df[(df['Allel1'].astype(str).str.strip().replace('nan', '') != '') & (df['Allel2'].astype(str).str.strip().replace('nan', '') != '') ]\n",
    "n_rows_2 = len(df)\n",
    "print(f\"Filtered from {n_rows_1} to {n_rows_2}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-12T15:52:46.436473Z",
     "start_time": "2024-10-12T15:52:46.429224Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Individual</th>\n",
       "      <th>Allel1</th>\n",
       "      <th>Amino Acid Change</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Allel2</th>\n",
       "      <th>Amino Acid Change.1</th>\n",
       "      <th>Exon.1</th>\n",
       "      <th>MCAD Activity%</th>\n",
       "      <th>VLCAD Activity</th>\n",
       "      <th>AC C8 [µmol/L]b</th>\n",
       "      <th>AC C6 [µmol/L]b</th>\n",
       "      <th>AC C10 [µmol/L]b</th>\n",
       "      <th>AC others [µmol/L] or ratio</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>198</td>\n",
       "      <td>c.127G&gt;A</td>\n",
       "      <td>p.E43K</td>\n",
       "      <td>3.0</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11</td>\n",
       "      <td>33.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.78 (ref.&lt;0.19)</td>\n",
       "      <td>0.48 (ref.&lt;0.19)</td>\n",
       "      <td>0.45 (ref.&lt;0.31)</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>205</td>\n",
       "      <td>c.653C&gt;A</td>\n",
       "      <td>p.A218D</td>\n",
       "      <td>8.0</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11</td>\n",
       "      <td>34.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>209</td>\n",
       "      <td>c.455A&gt;G</td>\n",
       "      <td>p.E152G</td>\n",
       "      <td>6.0</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11</td>\n",
       "      <td>35.0</td>\n",
       "      <td>81.0</td>\n",
       "      <td>0.98 (ref. 0-0.28)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C8/C16 0.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>215</td>\n",
       "      <td>c.127G&gt;A</td>\n",
       "      <td>p.E43K</td>\n",
       "      <td>3.0</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11</td>\n",
       "      <td>36.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>223</td>\n",
       "      <td>c.985A&gt;G</td>\n",
       "      <td>p.K329E</td>\n",
       "      <td>11.0</td>\n",
       "      <td>c.1091T&gt;C</td>\n",
       "      <td>p.I364T</td>\n",
       "      <td>11</td>\n",
       "      <td>38.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Individual    Allel1 Amino Acid Change  Exon     Allel2  \\\n",
       "197         198  c.127G>A            p.E43K   3.0   c.985A>G   \n",
       "204         205  c.653C>A           p.A218D   8.0   c.985A>G   \n",
       "208         209  c.455A>G           p.E152G   6.0   c.985A>G   \n",
       "214         215  c.127G>A            p.E43K   3.0   c.985A>G   \n",
       "222         223  c.985A>G           p.K329E  11.0  c.1091T>C   \n",
       "\n",
       "    Amino Acid Change.1 Exon.1  MCAD Activity%  VLCAD Activity  \\\n",
       "197             p.K329E     11            33.0             NaN   \n",
       "204             p.K329E     11            34.0            90.0   \n",
       "208             p.K329E     11            35.0            81.0   \n",
       "214             p.K329E     11            36.0            96.0   \n",
       "222             p.I364T     11            38.0            72.0   \n",
       "\n",
       "        AC C8 [µmol/L]b   AC C6 [µmol/L]b  AC C10 [µmol/L]b  \\\n",
       "197    0.78 (ref.<0.19)  0.48 (ref.<0.19)  0.45 (ref.<0.31)   \n",
       "204                 NaN               NaN               NaN   \n",
       "208  0.98 (ref. 0-0.28)               NaN               NaN   \n",
       "214                 NaN               NaN               NaN   \n",
       "222                 NaN               NaN               NaN   \n",
       "\n",
       "    AC others [µmol/L] or ratio  \n",
       "197                         NaN  \n",
       "204                         NaN  \n",
       "208                 C8/C16 0.28  \n",
       "214                         NaN  \n",
       "222                         NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.tail()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MCAD activity\n",
    "[LOINC:74892-1](https://loinc.org/74892-1) refers to Medium-chain Acyl CoA dehydrogenase [Enzymatic activity/mass] in Fibroblast\n",
    "The following function takes the activity level in the \"MCAD Activity%\" column and creates a corresponding Phenopacket Measurement.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-12T15:52:49.280154Z",
     "start_time": "2024-10-12T15:52:49.274225Z"
    }
   },
   "outputs": [],
   "source": [
    "def mcad_measurement(row):\n",
    "    value = row[\"MCAD Activity%\"]\n",
    "    loinc_code = \"LOINC:74892-1\"\n",
    "    loinc_label = \"Medium-chain Acyl CoA dehydrogenase [Enzymatic activity/mass] in Fibroblast\"\n",
    "    if value == \"na\":\n",
    "        return None\n",
    "    if isinstance(value,str) and value.endswith(\" \"):\n",
    "        raise ValueError(f\"Maformed MCAD: {value}\")\n",
    "    try:\n",
    "        concentration = int(value)\n",
    "        m = Measurements.percent(code=loinc_code,\n",
    "                                      label=loinc_label,\n",
    "                                      concentration=concentration)\n",
    "        return m\n",
    "    except ValueError:\n",
    "        print(f\"Could not parse \\\"{value}\\\"\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-12T15:52:58.723896Z",
     "start_time": "2024-10-12T15:52:58.715475Z"
    }
   },
   "outputs": [],
   "source": [
    "acadm_symbol = \"ACADM\"\n",
    "acadm_id = \"HGNC:89\"\n",
    "acadm_MANE_transcript = \"NM_000016.6\"\n",
    "vmanager = VariantManager(df=df, \n",
    "                          individual_column_name=\"Individual\", \n",
    "                          transcript=acadm_MANE_transcript, \n",
    "                          gene_id=acadm_id, \n",
    "                          gene_symbol=acadm_symbol, \n",
    "                          allele_1_column_name=\"Allel1\", \n",
    "                          allele_2_column_name=\"Allel2\")\n",
    "variant_d = vmanager.get_variant_d()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-12T15:57:45.965718Z",
     "start_time": "2024-10-12T15:57:45.962896Z"
    }
   },
   "outputs": [],
   "source": [
    "from pyphetools.pp.v202 import MetaData\n",
    "PMID = \"PMID:33580884\"\n",
    "title = \"Genotype and residual enzyme activity in medium-chain acyl-CoA dehydrogenase (MCAD) deficiency: Are predictions possible?\"\n",
    "\n",
    "created_by=\"ORCID:0000-0002-5648-2155\"\n",
    "metadata = MetaData.metadata_for_pmid(created_by=created_by, pmid=PMID, citation_title=title, include_loinc=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-12T16:03:00.586771Z",
     "start_time": "2024-10-12T16:03:00.580277Z"
    }
   },
   "outputs": [],
   "source": [
    "from pyphetools.pp.v202 import *\n",
    "\n",
    "\n",
    "def row_to_phenopacket(row:pd.Series):\n",
    "    individual_id = str(row[\"Individual\"])\n",
    "    phenopacket_id = \"PMID_33580884_individual_{}\".format(individual_id)  \n",
    "    i = Individual(id=individual_id)\n",
    "    \n",
    "    allele_1 = row[\"Allel1\"]\n",
    "    allele_2 = row[\"Allel2\"]\n",
    "    var_list = list()\n",
    "    if allele_1 == allele_2:\n",
    "        var = variant_d.get(allele_1)\n",
    "        var_obj = var.to_variant_interpretation_202()\n",
    "        var_obj.variation_descriptor.allelic_state = OntologyTerms.homozygous()\n",
    "        var_list.append(var_obj)\n",
    "    else:\n",
    "        var1 = variant_d.get(allele_1)\n",
    "        var_obj1 = var1.to_variant_interpretation_202()\n",
    "        var2 = variant_d.get(allele_2)\n",
    "        var_obj2 = var2.to_variant_interpretation_202()\n",
    "        var_obj1.variation_descriptor.allelic_state = OntologyTerms.heterozygous()\n",
    "        var_obj2.variation_descriptor.allelic_state = OntologyTerms.heterozygous()\n",
    "        var_list.append(var_obj1)\n",
    "        var_list.append(var_obj2)\n",
    "    ## create genomic interpretation\n",
    "    interpretation_list = list()\n",
    "    for var in var_list:\n",
    "        genomic_interpretation = GenomicInterpretation(subject_or_biosample_id=individual_id, \n",
    "                                                       interpretation_status=GenomicInterpretation.InterpretationStatus.CAUSATIVE,\n",
    "                                                       call=var)\n",
    "        interpretation_list.append(genomic_interpretation)\n",
    "    ## Disease is always OMIM:201910 for this cohort.\n",
    "    diseaseClass = OntologyClass(id=\"OMIM:201450\", label=\"Acyl-CoA dehydrogenase, medium chain, deficiency of\")\n",
    "\n",
    "    disease = Disease(term=diseaseClass)\n",
    "    diagnosis = Diagnosis(disease=diseaseClass, genomic_interpretations=interpretation_list)\n",
    "    interpretation = Interpretation(id=individual_id, progress_status=Interpretation.ProgressStatus.SOLVED, diagnosis=diagnosis)\n",
    "   \n",
    "    measurements = list()\n",
    "    m = mcad_measurement(row=row) \n",
    "    measurements.append(m)\n",
    "    ppkt = Phenopacket(id=phenopacket_id, \n",
    "                       subject=i, \n",
    "                       diseases=[disease],\n",
    "                       measurements=measurements, \n",
    "                       interpretations=[interpretation], \n",
    "                       meta_data=metadata)\n",
    "    return ppkt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-12T16:03:01.546007Z",
     "start_time": "2024-10-12T16:03:01.464919Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "0.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "1.0\n",
      "2.0\n",
      "2.0\n",
      "2.0\n",
      "2.0\n",
      "2.0\n",
      "3.0\n",
      "3.0\n",
      "3.0\n",
      "3.0\n",
      "4.0\n",
      "4.0\n",
      "4.0\n",
      "4.0\n",
      "4.0\n",
      "4.0\n",
      "5.0\n",
      "5.0\n",
      "7.0\n",
      "8.0\n",
      "9.0\n",
      "9.0\n",
      "10.0\n",
      "10.0\n",
      "11.0\n",
      "13.0\n",
      "14.0\n",
      "14.0\n",
      "14.0\n",
      "15.0\n",
      "15.0\n",
      "15.0\n",
      "16.0\n",
      "16.0\n",
      "16.0\n",
      "17.0\n",
      "17.0\n",
      "18.0\n",
      "18.0\n",
      "18.0\n",
      "18.0\n",
      "19.0\n",
      "19.0\n",
      "20.0\n",
      "20.0\n",
      "20.0\n",
      "20.0\n",
      "20.0\n",
      "20.0\n",
      "20.0\n",
      "21.0\n",
      "21.0\n",
      "21.0\n",
      "21.0\n",
      "22.0\n",
      "24.0\n",
      "24.0\n",
      "27.0\n",
      "28.0\n",
      "28.0\n",
      "28.0\n",
      "28.0\n",
      "29.0\n",
      "32.0\n",
      "33.0\n",
      "34.0\n",
      "35.0\n",
      "36.0\n",
      "38.0\n",
      "We output 115 GA4GH phenopackets to the directory phenopackets\n"
     ]
    }
   ],
   "source": [
    "phenopacket_list = list()\n",
    "for _, row in df.iterrows():\n",
    "    ppkt = row_to_phenopacket(row)\n",
    "    phenopacket_list.append(ppkt.to_message())\n",
    "outdir = \"phenopackets\"\n",
    "if not os.path.isdir(outdir):\n",
    "    os.makedirs(outdir)\n",
    "written = 0\n",
    "json_list = list()\n",
    "for ppkt in phenopacket_list:    \n",
    "    json_string = MessageToJson(ppkt)\n",
    "    fname = ppkt.id.replace(\" \", \"_\") + \".json\"\n",
    "    outpth = os.path.join(outdir, fname)\n",
    "    with open(outpth, \"wt\") as fh:\n",
    "        fh.write(json_string)\n",
    "        json_list.append(json_string)\n",
    "        written += 1\n",
    "print(f\"We output {written} GA4GH phenopackets to the directory {outdir}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
