{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Loeys-Dietz syndrome 1 and 3\n",
    "Loeys-Dietz syndrome (LDS) is an autosomal dominant aortic aneurysm syndrome characterized by multisystemic involvement. The most typical clinical triad includes hypertelorism, bifid uvula or cleft palate and aortic aneurysm with tortuosity. Affected individuals may expereince aortic dissection at smaller aortic diameter and arterial aneurysms throughout the arterial tree. The genetic cause is heterogeneous and includes mutations in genes encoding for components of the transforming growth factor beta (TGFβ) signalling pathway: TGFBR1, TGFBR2, SMAD2, SMAD3, TGFB2 and TGFB3 (See [Velchev JD, et al. (2021). Loeys-Dietz Syndrome. Adv Exp Med Biol](https://pubmed.ncbi.nlm.nih.gov/34807423/)).\n",
    "\n",
    "This notebook will explore whether there are significant differences in phenotypic features between LDS1 (TGFBR1) and LDS3 (SMAD3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2024-12-12\n",
      "Using genophenocorr version 0.9.1\n"
     ]
    }
   ],
   "source": [
    "import gpsea\n",
    "import hpotk\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo()\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using genophenocorr version {gpsea.__version__}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "41 LDS1 phenopackets\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "\n",
    "tgfbr1_symbol = 'TGFBR1'\n",
    "tgfbr1_mane_tx_id = 'NM_004612.4'\n",
    "tgfbr1_mane_protein_id = 'NP_004603.1' # TGF-beta receptor type-1 isoform 1 precursor\"\n",
    "lds1_disease_id = \"OMIM:609192\"\n",
    "\n",
    "\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.23\") as ps:\n",
    "    lds1_phenopackets = tuple(ps.iter_cohort_phenopackets(tgfbr1_symbol))\n",
    "tgfbr1_len = len(lds1_phenopackets)\n",
    "print(f\"{len(lds1_phenopackets)} LDS1 phenopackets\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS3\n",
    "[Loeys-Dietz syndrome-3 (LDS3)](https://omim.org/entry/613795) is caused by heterozygous mutation in the SMAD3 gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "49 LDS3 phenopackets\n"
     ]
    }
   ],
   "source": [
    "smad3_symbol = 'SMAD3'\n",
    "smad3_mane_tx_id = 'NM_005902.4'\n",
    "smad3_mane_protein_id = 'NP_005893.1' # mothers against decapentaplegic homolog 3\n",
    "lds3_disease_id = \"OMIM:613795\"\n",
    "\n",
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.23\") as ps:\n",
    "    lds3_phenopackets = tuple(ps.iter_cohort_phenopackets(smad3_symbol))\n",
    "\n",
    "print(f\"{len(lds3_phenopackets)} LDS3 phenopackets\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 100%|██████████| 90/90 [00:00<00:00, 541.94individuals/s]\n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "\n",
    "lds_phenopackets = list()\n",
    "lds_phenopackets.extend(lds1_phenopackets)\n",
    "lds_phenopackets.extend(lds3_phenopackets)\n",
    "\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort, validation = load_phenopackets(\n",
    "    phenopackets=lds_phenopackets, \n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "\n",
    "validation.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.pcats import configure_hpo_term_analysis\n",
    "from gpsea.analysis.clf import prepare_classifiers_for_terms_of_interest\n",
    "\n",
    "analysis = configure_hpo_term_analysis(hpo)\n",
    "\n",
    "pheno_clfs = prepare_classifiers_for_terms_of_interest(\n",
    "    cohort=cohort,\n",
    "    hpo=hpo,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Phenotype testing report</h1>\n",
       "<div class=\"cards\">\n",
       "  <div class=\"card\">\n",
       "    <ul>\n",
       "      <li>Phenotype MTC filter: <em>HPO MTC filter</em></li>\n",
       "      <li>Multiple testing correction: <em>fdr_bh</em></li>\n",
       "    </ul>\n",
       "  </div>\n",
       "  <div class=\"card\">\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr>\n",
       "          <th>Code</th>\n",
       "          <th>Reason</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF01</td>\n",
       "          <td>Skipping term with maximum frequency that was less than threshold 0.4</td>\n",
       "          <td>42</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF03</td>\n",
       "          <td>Skipping term because of a child term with the same individual counts</td>\n",
       "          <td>2</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF08</td>\n",
       "          <td>Skipping general term</td>\n",
       "          <td>63</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF09</td>\n",
       "          <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>\n",
       "          <td>150</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "      <caption>\n",
       "        Performed statistical tests for 21 out of the total of 278 HPO terms.\n",
       "      </caption>\n",
       "    </table>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x129249d10>"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.analysis.clf import diagnosis_classifier\n",
    "from gpsea.view import MtcStatsViewer\n",
    "\n",
    "lds_1_3_disease_clf = diagnosis_classifier (\n",
    "    diagnoses=(lds1_disease_id, lds3_disease_id),\n",
    "    labels=('LDS1', 'LDS3'),\n",
    ")\n",
    "\n",
    "lds1_3_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_clf=lds_1_3_disease_clf,\n",
    "    pheno_clfs=pheno_clfs,\n",
    ")\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(lds1_3_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Diagnosis</th>\n",
       "      <th colspan=\"2\" halign=\"left\">OMIM:609192</th>\n",
       "      <th colspan=\"2\" halign=\"left\">OMIM:613795</th>\n",
       "      <th colspan=\"2\" halign=\"left\"></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Osteoarthritis [HP:0002758]</th>\n",
       "      <td>0/11</td>\n",
       "      <td>0%</td>\n",
       "      <td>26/38</td>\n",
       "      <td>68%</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>0.000046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Scoliosis [HP:0002650]</th>\n",
       "      <td>18/21</td>\n",
       "      <td>86%</td>\n",
       "      <td>20/43</td>\n",
       "      <td>47%</td>\n",
       "      <td>0.023527</td>\n",
       "      <td>0.003015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aortic aneurysm [HP:0004942]</th>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>26/48</td>\n",
       "      <td>54%</td>\n",
       "      <td>0.023527</td>\n",
       "      <td>0.004327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypertelorism [HP:0000316]</th>\n",
       "      <td>15/19</td>\n",
       "      <td>79%</td>\n",
       "      <td>13/35</td>\n",
       "      <td>37%</td>\n",
       "      <td>0.023527</td>\n",
       "      <td>0.004481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Joint hypermobility [HP:0001382]</th>\n",
       "      <td>12/19</td>\n",
       "      <td>63%</td>\n",
       "      <td>12/36</td>\n",
       "      <td>33%</td>\n",
       "      <td>0.197692</td>\n",
       "      <td>0.047070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>High palate [HP:0000218]</th>\n",
       "      <td>12/16</td>\n",
       "      <td>75%</td>\n",
       "      <td>12/20</td>\n",
       "      <td>60%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.481498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inguinal hernia [HP:0000023]</th>\n",
       "      <td>6/14</td>\n",
       "      <td>43%</td>\n",
       "      <td>12/39</td>\n",
       "      <td>31%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.514672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Arterial tortuosity [HP:0005116]</th>\n",
       "      <td>9/17</td>\n",
       "      <td>53%</td>\n",
       "      <td>11/26</td>\n",
       "      <td>42%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.545009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Disproportionate tall stature [HP:0001519]</th>\n",
       "      <td>9/17</td>\n",
       "      <td>53%</td>\n",
       "      <td>8/19</td>\n",
       "      <td>42%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.738795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal oral cavity morphology [HP:0000163]</th>\n",
       "      <td>17/17</td>\n",
       "      <td>100%</td>\n",
       "      <td>24/24</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal palate morphology [HP:0000174]</th>\n",
       "      <td>17/17</td>\n",
       "      <td>100%</td>\n",
       "      <td>24/24</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal axial skeleton morphology [HP:0009121]</th>\n",
       "      <td>21/21</td>\n",
       "      <td>100%</td>\n",
       "      <td>33/33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal curvature of the vertebral column [HP:0010674]</th>\n",
       "      <td>18/18</td>\n",
       "      <td>100%</td>\n",
       "      <td>20/20</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormality of the vertebral column [HP:0000925]</th>\n",
       "      <td>19/19</td>\n",
       "      <td>100%</td>\n",
       "      <td>31/31</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Arachnodactyly [HP:0001166]</th>\n",
       "      <td>8/18</td>\n",
       "      <td>44%</td>\n",
       "      <td>7/18</td>\n",
       "      <td>39%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vascular dilatation [HP:0002617]</th>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>26/26</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal blood vessel morphology [HP:0033353]</th>\n",
       "      <td>15/15</td>\n",
       "      <td>100%</td>\n",
       "      <td>32/32</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal aortic morphology [HP:0001679]</th>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>26/26</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal morphology of the great vessels [HP:0030962]</th>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>26/26</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal systemic arterial morphology [HP:0011004]</th>\n",
       "      <td>14/14</td>\n",
       "      <td>100%</td>\n",
       "      <td>29/29</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Soft skin [HP:0000977]</th>\n",
       "      <td>3/5</td>\n",
       "      <td>60%</td>\n",
       "      <td>23/37</td>\n",
       "      <td>62%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Diagnosis                                          OMIM:609192          \\\n",
       "                                                         Count Percent   \n",
       "Osteoarthritis [HP:0002758]                               0/11      0%   \n",
       "Scoliosis [HP:0002650]                                   18/21     86%   \n",
       "Aortic aneurysm [HP:0004942]                             11/11    100%   \n",
       "Hypertelorism [HP:0000316]                               15/19     79%   \n",
       "Joint hypermobility [HP:0001382]                         12/19     63%   \n",
       "High palate [HP:0000218]                                 12/16     75%   \n",
       "Inguinal hernia [HP:0000023]                              6/14     43%   \n",
       "Arterial tortuosity [HP:0005116]                          9/17     53%   \n",
       "Disproportionate tall stature [HP:0001519]                9/17     53%   \n",
       "Abnormal oral cavity morphology [HP:0000163]             17/17    100%   \n",
       "Abnormal palate morphology [HP:0000174]                  17/17    100%   \n",
       "Abnormal axial skeleton morphology [HP:0009121]          21/21    100%   \n",
       "Abnormal curvature of the vertebral column [HP:...       18/18    100%   \n",
       "Abnormality of the vertebral column [HP:0000925]         19/19    100%   \n",
       "Arachnodactyly [HP:0001166]                               8/18     44%   \n",
       "Vascular dilatation [HP:0002617]                         11/11    100%   \n",
       "Abnormal blood vessel morphology [HP:0033353]            15/15    100%   \n",
       "Abnormal aortic morphology [HP:0001679]                  11/11    100%   \n",
       "Abnormal morphology of the great vessels [HP:00...       12/12    100%   \n",
       "Abnormal systemic arterial morphology [HP:0011004]       14/14    100%   \n",
       "Soft skin [HP:0000977]                                     3/5     60%   \n",
       "\n",
       "Diagnosis                                          OMIM:613795          \\\n",
       "                                                         Count Percent   \n",
       "Osteoarthritis [HP:0002758]                              26/38     68%   \n",
       "Scoliosis [HP:0002650]                                   20/43     47%   \n",
       "Aortic aneurysm [HP:0004942]                             26/48     54%   \n",
       "Hypertelorism [HP:0000316]                               13/35     37%   \n",
       "Joint hypermobility [HP:0001382]                         12/36     33%   \n",
       "High palate [HP:0000218]                                 12/20     60%   \n",
       "Inguinal hernia [HP:0000023]                             12/39     31%   \n",
       "Arterial tortuosity [HP:0005116]                         11/26     42%   \n",
       "Disproportionate tall stature [HP:0001519]                8/19     42%   \n",
       "Abnormal oral cavity morphology [HP:0000163]             24/24    100%   \n",
       "Abnormal palate morphology [HP:0000174]                  24/24    100%   \n",
       "Abnormal axial skeleton morphology [HP:0009121]          33/33    100%   \n",
       "Abnormal curvature of the vertebral column [HP:...       20/20    100%   \n",
       "Abnormality of the vertebral column [HP:0000925]         31/31    100%   \n",
       "Arachnodactyly [HP:0001166]                               7/18     39%   \n",
       "Vascular dilatation [HP:0002617]                         26/26    100%   \n",
       "Abnormal blood vessel morphology [HP:0033353]            32/32    100%   \n",
       "Abnormal aortic morphology [HP:0001679]                  26/26    100%   \n",
       "Abnormal morphology of the great vessels [HP:00...       26/26    100%   \n",
       "Abnormal systemic arterial morphology [HP:0011004]       29/29    100%   \n",
       "Soft skin [HP:0000977]                                   23/37     62%   \n",
       "\n",
       "Diagnosis                                                              \\\n",
       "                                                   Corrected p values   \n",
       "Osteoarthritis [HP:0002758]                                  0.000975   \n",
       "Scoliosis [HP:0002650]                                       0.023527   \n",
       "Aortic aneurysm [HP:0004942]                                 0.023527   \n",
       "Hypertelorism [HP:0000316]                                   0.023527   \n",
       "Joint hypermobility [HP:0001382]                             0.197692   \n",
       "High palate [HP:0000218]                                     1.000000   \n",
       "Inguinal hernia [HP:0000023]                                 1.000000   \n",
       "Arterial tortuosity [HP:0005116]                             1.000000   \n",
       "Disproportionate tall stature [HP:0001519]                   1.000000   \n",
       "Abnormal oral cavity morphology [HP:0000163]                 1.000000   \n",
       "Abnormal palate morphology [HP:0000174]                      1.000000   \n",
       "Abnormal axial skeleton morphology [HP:0009121]              1.000000   \n",
       "Abnormal curvature of the vertebral column [HP:...           1.000000   \n",
       "Abnormality of the vertebral column [HP:0000925]             1.000000   \n",
       "Arachnodactyly [HP:0001166]                                  1.000000   \n",
       "Vascular dilatation [HP:0002617]                             1.000000   \n",
       "Abnormal blood vessel morphology [HP:0033353]                1.000000   \n",
       "Abnormal aortic morphology [HP:0001679]                      1.000000   \n",
       "Abnormal morphology of the great vessels [HP:00...           1.000000   \n",
       "Abnormal systemic arterial morphology [HP:0011004]           1.000000   \n",
       "Soft skin [HP:0000977]                                       1.000000   \n",
       "\n",
       "Diagnosis                                                     \n",
       "                                                    p values  \n",
       "Osteoarthritis [HP:0002758]                         0.000046  \n",
       "Scoliosis [HP:0002650]                              0.003015  \n",
       "Aortic aneurysm [HP:0004942]                        0.004327  \n",
       "Hypertelorism [HP:0000316]                          0.004481  \n",
       "Joint hypermobility [HP:0001382]                    0.047070  \n",
       "High palate [HP:0000218]                            0.481498  \n",
       "Inguinal hernia [HP:0000023]                        0.514672  \n",
       "Arterial tortuosity [HP:0005116]                    0.545009  \n",
       "Disproportionate tall stature [HP:0001519]          0.738795  \n",
       "Abnormal oral cavity morphology [HP:0000163]        1.000000  \n",
       "Abnormal palate morphology [HP:0000174]             1.000000  \n",
       "Abnormal axial skeleton morphology [HP:0009121]     1.000000  \n",
       "Abnormal curvature of the vertebral column [HP:...  1.000000  \n",
       "Abnormality of the vertebral column [HP:0000925]    1.000000  \n",
       "Arachnodactyly [HP:0001166]                         1.000000  \n",
       "Vascular dilatation [HP:0002617]                    1.000000  \n",
       "Abnormal blood vessel morphology [HP:0033353]       1.000000  \n",
       "Abnormal aortic morphology [HP:0001679]             1.000000  \n",
       "Abnormal morphology of the great vessels [HP:00...  1.000000  \n",
       "Abnormal systemic arterial morphology [HP:0011004]  1.000000  \n",
       "Soft skin [HP:0000977]                              1.000000  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "summarize_hpo_analysis(hpo=hpo, result=lds1_3_result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpseacs.report import GpseaAnalysisReport, GPAnalysisResultSummary\n",
    "\n",
    "f_results = (\n",
    "  GPAnalysisResultSummary.from_multi( result=lds1_3_result,  ),\n",
    ")\n",
    "\n",
    "caption = \"\"\".\"\"\"\n",
    "report = GpseaAnalysisReport(name=\"LDS 1 and 3\", \n",
    "                             cohort=cohort, \n",
    "                             fet_results=f_results,\n",
    "                             gene_symbol=\"n/a\",\n",
    "                             mane_tx_id=\"n/a\",\n",
    "                             mane_protein_id=\"n/a\",\n",
    "                             caption=caption)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "      min-width: 100px;\n",
       "      border: 1px solid #ddd;\n",
       "      padding: 8px;\n",
       "        \n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    p {\n",
       "      width: 75%;\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>LDS 1 and 3 Summary</h1>\n",
       "  <p>The cohort comprised 90 individuals (29 females, 43 males, 18 with unknown sex). 2 of these individuals were reported to be deceased. A total of 89 HPO terms were used to annotate the cohort. Disease diagnoses: Loeys-Dietz syndrome 3 (OMIM:613795) (49 individuals), Loeys-Dietz syndrome 1 (OMIM:609192) (23 individuals), Multiple self-healing squamous epithelioma, susceptibility to (OMIM:132800) (18 individuals). . A total of 90 unique variant alleles were found in n/a (transcript: n/a, protein id: n/a). HPO version: 2024-12-12. GPSEA version: 0.9.1.</p>\n",
       "  \n",
       "  \n",
       "    \n",
       "    <table>\n",
       "      <caption style=\"color: black;\">\n",
       "          Fisher Exact Test performed to compare HPO annotation frequency with respect to OMIM:609192 and OMIM:613795. Total of\n",
       "          21 tests were performed.\n",
       "      </caption>\n",
       "          <tbody>\n",
       "              <tr class=\"strng\">\n",
       "                  <th>HPO Term</th>\n",
       "                  <th>OMIM:609192</th>\n",
       "                  <th>OMIM:613795</th>\n",
       "                  <th>p-val</th>\n",
       "                  <th>adj. p-val</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                  <td class=\"lft\">Scoliosis [HP:0002650]</td>\n",
       "                  <td class=\"lft\">18/21 (86%)</td>\n",
       "                  <td class=\"lft\">20/43 (47%)</td>\n",
       "                  <td class=\"lft\">0.003</td>\n",
       "                  <td class=\"lft\">0.024</td>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                  <td class=\"lft\">Hypertelorism [HP:0000316]</td>\n",
       "                  <td class=\"lft\">15/19 (79%)</td>\n",
       "                  <td class=\"lft\">13/35 (37%)</td>\n",
       "                  <td class=\"lft\">0.004</td>\n",
       "                  <td class=\"lft\">0.024</td>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                  <td class=\"lft\">Aortic aneurysm [HP:0004942]</td>\n",
       "                  <td class=\"lft\">11/11 (100%)</td>\n",
       "                  <td class=\"lft\">26/48 (54%)</td>\n",
       "                  <td class=\"lft\">0.004</td>\n",
       "                  <td class=\"lft\">0.024</td>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                  <td class=\"lft\">Osteoarthritis [HP:0002758]</td>\n",
       "                  <td class=\"lft\">0/11 (0%)</td>\n",
       "                  <td class=\"lft\">26/38 (68%)</td>\n",
       "                  <td class=\"lft\">4.64e-05</td>\n",
       "                  <td class=\"lft\">9.75e-04</td>\n",
       "              </tr>\n",
       "              \n",
       "            </tbody>\n",
       "      </table>\n",
       "      \n",
       "\n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "\n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpseacs.report._api.HtmlGpseaNotebookSummarizer at 0x119115a90>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpseacs.report import GpseaNotebookSummarizer\n",
    "summarizer = GpseaNotebookSummarizer(hpo=hpo, gpsea_version=gpsea.__version__)\n",
    "summarizer.summarize_report(report=report)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output to ../../supplement/tex/LDS_1_and_3_summary_draft.tex\n"
     ]
    }
   ],
   "source": [
    "summarizer.process_latex(report=report)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
