{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Loeys-Dietz syndrome 3 and 6\n",
    "Loeys-Dietz syndrome (LDS) is an autosomal dominant aortic aneurysm syndrome characterized by multisystemic involvement. The most typical clinical triad includes hypertelorism, bifid uvula or cleft palate and aortic aneurysm with tortuosity. Affected individuals may expereince aortic dissection at smaller aortic diameter and arterial aneurysms throughout the arterial tree. The genetic cause is heterogeneous and includes mutations in genes encoding for components of the transforming growth factor beta (TGFβ) signalling pathway: TGFBR1, TGFBR2, SMAD2, SMAD3, TGFB2 and TGFB3 (See [Velchev JD, et al. (2021). Loeys-Dietz Syndrome. Adv Exp Med Biol](https://pubmed.ncbi.nlm.nih.gov/34807423/)).\n",
    "\n",
    "This notebook will explore whether there are significant differences in phenotypic features between LDS3 (SMAD3) and LDS6 (SMAD2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2025-01-16\n",
      "Using genophenocorr version 0.9.6.dev0\n"
     ]
    }
   ],
   "source": [
    "import gpsea\n",
    "import hpotk\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo()\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using genophenocorr version {gpsea.__version__}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS3\n",
    "[Loeys-Dietz syndrome-3 (LDS3)](https://omim.org/entry/613795) is caused by heterozygous mutation in the SMAD3 gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "49 LDS3 phenopackets\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "\n",
    "smad3_symbol = 'SMAD3'\n",
    "smad3_mane_tx_id = 'NM_005902.4'\n",
    "smad3_mane_protein_id = 'NP_005893.1' # mothers against decapentaplegic homolog 3\n",
    "lds3_disease_id = \"OMIM:613795\"\n",
    "\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.24\") as ps:\n",
    "    lds3_phenopackets = tuple(ps.iter_cohort_phenopackets(smad3_symbol))\n",
    "\n",
    "print(f\"{len(lds3_phenopackets)} LDS3 phenopackets\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS6\n",
    "[Loeys-Dietz syndrome-6 (LDS6)](https://omim.org/entry/619656) is caused by heterozygous mutation in the SMAD2 gene "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 LDS6 phenopackets\n"
     ]
    }
   ],
   "source": [
    "smad2_symbol = 'SMAD2'\n",
    "smad2_mane_tx_id = 'NM_005901.6'\n",
    "smad2_mane_protein_id = 'NP_005892.1' # mothers against decapentaplegic homolog 2 isoform 1\n",
    "\n",
    "lds6_disease_id = \"OMIM:619656\"\n",
    "\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.24\") as ps:\n",
    "    lds6_phenopackets = tuple(ps.iter_cohort_phenopackets(smad2_symbol))\n",
    "\n",
    "print(f\"{len(lds6_phenopackets)} LDS6 phenopackets\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Got 72 LDS3 and LDS6 phenopackets\n",
      "Individuals Processed: 100%|██████████| 72/72 [00:00<00:00, 249.38 individuals/s]\n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "\n",
    "lds3_and_lds6_phenopackets = list()\n",
    "lds3_and_lds6_phenopackets.extend(lds3_phenopackets)\n",
    "lds3_and_lds6_phenopackets.extend(lds6_phenopackets)\n",
    "print(f\"Got {len(lds3_and_lds6_phenopackets)} LDS3 and LDS6 phenopackets\")\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort, validation = load_phenopackets(\n",
    "    phenopackets=lds3_and_lds6_phenopackets, \n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "\n",
    "validation.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.pcats import configure_hpo_term_analysis\n",
    "from gpsea.analysis.clf import prepare_classifiers_for_terms_of_interest\n",
    "\n",
    "analysis = configure_hpo_term_analysis(hpo)\n",
    "\n",
    "pheno_clfs = prepare_classifiers_for_terms_of_interest(\n",
    "    cohort=cohort,\n",
    "    hpo=hpo,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Phenotype testing report</h1>\n",
       "<div class=\"cards\">\n",
       "  <div class=\"card\">\n",
       "    <ul>\n",
       "      <li>Phenotype MTC filter: <em>Independent filtering HPO filter</em></li>\n",
       "      <li>Multiple testing correction: <em>fdr_bh</em></li>\n",
       "    </ul>\n",
       "  </div>\n",
       "  <div class=\"card\">\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr>\n",
       "          <th>Reason</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          \n",
       "            <td><a href=\"https://P2GX.github.io/gpsea/stable/user-guide/analyses/mtc.html#skip-terms-if-all-counts-are-identical-to-counts-for-a-child-term\">Skip terms if all counts are identical to counts for a child term</a></td>\n",
       "          \n",
       "          <td>2</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          \n",
       "            <td><a href=\"https://P2GX.github.io/gpsea/stable/user-guide/analyses/mtc.html#skipping-general-level-terms\">Skipping \"general\" level terms</a></td>\n",
       "          \n",
       "          <td>71</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          \n",
       "            <td><a href=\"https://P2GX.github.io/gpsea/stable/user-guide/analyses/mtc.html#skipping-terms-that-are-rare-on-the-cohort-level\">Skipping terms that are rare on the cohort level (in less than 40% of the cohort members)</a></td>\n",
       "          \n",
       "          <td>211</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "      <caption>\n",
       "        Performed statistical tests for 25 out of the total of 309 HPO terms.\n",
       "      </caption>\n",
       "    </table>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x7b38eaa6b4a0>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.analysis.clf import diagnosis_classifier\n",
    "from gpsea.view import MtcStatsViewer\n",
    "\n",
    "lds_3_6_disease_clf = diagnosis_classifier (\n",
    "    diagnoses=(lds3_disease_id, lds6_disease_id),\n",
    "    labels=('LDS3', 'LDS6'),\n",
    ")\n",
    "\n",
    "lds_3_6_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_clf=lds_3_6_disease_clf,\n",
    "    pheno_clfs=pheno_clfs,\n",
    ")\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(lds_3_6_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "OMIM:613795",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "OMIM:619656",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Corrected p values",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "p values",
         "rawType": "float64",
         "type": "float"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "a2504623-591f-4352-9485-b15f56a8204b",
       "rows": [
        [
         "Thoracic aortic aneurysm [HP:0012727]",
         "0/22 (0%)",
         "10/16 (62%)",
         "0.00042349419194003993",
         "1.6939767677601597e-05"
        ],
        [
         "Aortic aneurysm [HP:0004942]",
         "26/48 (54%)",
         "10/10 (100%)",
         "0.11359726577341209",
         "0.009087781261872967"
        ],
        [
         "Bifid uvula [HP:0000193]",
         "13/38 (34%)",
         "1/14 (7%)",
         "0.6508688638507408",
         "0.07810426366208889"
        ],
        [
         "Soft skin [HP:0000977]",
         "23/37 (62%)",
         "1/5 (20%)",
         "0.7315965673077438",
         "0.14636967653655714"
        ],
        [
         "Varicose veins [HP:0002619]",
         "14/22 (64%)",
         "4/12 (33%)",
         "0.7315965673077438",
         "0.15141461527300865"
        ],
        [
         "High palate [HP:0000218]",
         "12/20 (60%)",
         "5/15 (33%)",
         "0.7315965673077438",
         "0.17558317615385852"
        ],
        [
         "Abdominal aortic aneurysm [HP:0005112]",
         "0/22 (0%)",
         "1/11 (9%)",
         "1.0",
         "0.3333333333333333"
        ],
        [
         "Inguinal hernia [HP:0000023]",
         "12/39 (31%)",
         "5/11 (45%)",
         "1.0",
         "0.475053878140841"
        ],
        [
         "Umbilical hernia [HP:0001537]",
         "12/39 (31%)",
         "2/4 (50%)",
         "1.0",
         "0.5854792966534317"
        ],
        [
         "Hypertelorism [HP:0000316]",
         "13/35 (37%)",
         "4/14 (29%)",
         "1.0",
         "0.7429599937751151"
        ],
        [
         "Joint hypermobility [HP:0001382]",
         "12/36 (33%)",
         "2/6 (33%)",
         "1.0",
         "1.0"
        ],
        [
         "Spontaneous pneumothorax [HP:0002108]",
         "2/20 (10%)",
         "0/9 (0%)",
         "1.0",
         "1.0"
        ],
        [
         "Scoliosis [HP:0002650]",
         "20/43 (47%)",
         "7/14 (50%)",
         "1.0",
         "1.0"
        ],
        [
         "Abnormal palate morphology [HP:0000174]",
         "24/24 (100%)",
         "7/7 (100%)",
         "1.0",
         "1.0"
        ],
        [
         "Abnormal oral cavity morphology [HP:0000163]",
         "24/24 (100%)",
         "7/7 (100%)",
         "1.0",
         "1.0"
        ],
        [
         "Vascular dilatation [HP:0002617]",
         "26/26 (100%)",
         "11/11 (100%)",
         "1.0",
         "1.0"
        ],
        [
         "Abnormal aortic morphology [HP:0001679]",
         "26/26 (100%)",
         "12/12 (100%)",
         "1.0",
         "1.0"
        ],
        [
         "Abnormal morphology of the great vessels [HP:0030962]",
         "26/26 (100%)",
         "12/12 (100%)",
         "1.0",
         "1.0"
        ],
        [
         "Osteoarthritis [HP:0002758]",
         "26/38 (68%)",
         "2/2 (100%)",
         "1.0",
         "1.0"
        ],
        [
         "Abnormal systemic arterial morphology [HP:0011004]",
         "29/29 (100%)",
         "16/16 (100%)",
         "1.0",
         "1.0"
        ],
        [
         "Abnormality of the vertebral column [HP:0000925]",
         "31/31 (100%)",
         "10/10 (100%)",
         "1.0",
         "1.0"
        ],
        [
         "Abnormal blood vessel morphology [HP:0033353]",
         "32/32 (100%)",
         "17/17 (100%)",
         "1.0",
         "1.0"
        ],
        [
         "Abnormal axial skeleton morphology [HP:0009121]",
         "33/33 (100%)",
         "11/11 (100%)",
         "1.0",
         "1.0"
        ],
        [
         "Abnormal sternum morphology [HP:0000766]",
         "6/20 (30%)",
         "5/15 (33%)",
         "1.0",
         "1.0"
        ],
        [
         "Arachnodactyly [HP:0001166]",
         "7/18 (39%)",
         "6/14 (43%)",
         "1.0",
         "1.0"
        ]
       ],
       "shape": {
        "columns": 4,
        "rows": 25
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Diagnosis</th>\n",
       "      <th>OMIM:613795</th>\n",
       "      <th>OMIM:619656</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Thoracic aortic aneurysm [HP:0012727]</th>\n",
       "      <td>0/22 (0%)</td>\n",
       "      <td>10/16 (62%)</td>\n",
       "      <td>0.000423</td>\n",
       "      <td>0.000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aortic aneurysm [HP:0004942]</th>\n",
       "      <td>26/48 (54%)</td>\n",
       "      <td>10/10 (100%)</td>\n",
       "      <td>0.113597</td>\n",
       "      <td>0.009088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bifid uvula [HP:0000193]</th>\n",
       "      <td>13/38 (34%)</td>\n",
       "      <td>1/14 (7%)</td>\n",
       "      <td>0.650869</td>\n",
       "      <td>0.078104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Soft skin [HP:0000977]</th>\n",
       "      <td>23/37 (62%)</td>\n",
       "      <td>1/5 (20%)</td>\n",
       "      <td>0.731597</td>\n",
       "      <td>0.146370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Varicose veins [HP:0002619]</th>\n",
       "      <td>14/22 (64%)</td>\n",
       "      <td>4/12 (33%)</td>\n",
       "      <td>0.731597</td>\n",
       "      <td>0.151415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>High palate [HP:0000218]</th>\n",
       "      <td>12/20 (60%)</td>\n",
       "      <td>5/15 (33%)</td>\n",
       "      <td>0.731597</td>\n",
       "      <td>0.175583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abdominal aortic aneurysm [HP:0005112]</th>\n",
       "      <td>0/22 (0%)</td>\n",
       "      <td>1/11 (9%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inguinal hernia [HP:0000023]</th>\n",
       "      <td>12/39 (31%)</td>\n",
       "      <td>5/11 (45%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.475054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Umbilical hernia [HP:0001537]</th>\n",
       "      <td>12/39 (31%)</td>\n",
       "      <td>2/4 (50%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.585479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypertelorism [HP:0000316]</th>\n",
       "      <td>13/35 (37%)</td>\n",
       "      <td>4/14 (29%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.742960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Joint hypermobility [HP:0001382]</th>\n",
       "      <td>12/36 (33%)</td>\n",
       "      <td>2/6 (33%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Spontaneous pneumothorax [HP:0002108]</th>\n",
       "      <td>2/20 (10%)</td>\n",
       "      <td>0/9 (0%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Scoliosis [HP:0002650]</th>\n",
       "      <td>20/43 (47%)</td>\n",
       "      <td>7/14 (50%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal palate morphology [HP:0000174]</th>\n",
       "      <td>24/24 (100%)</td>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal oral cavity morphology [HP:0000163]</th>\n",
       "      <td>24/24 (100%)</td>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vascular dilatation [HP:0002617]</th>\n",
       "      <td>26/26 (100%)</td>\n",
       "      <td>11/11 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal aortic morphology [HP:0001679]</th>\n",
       "      <td>26/26 (100%)</td>\n",
       "      <td>12/12 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal morphology of the great vessels [HP:0030962]</th>\n",
       "      <td>26/26 (100%)</td>\n",
       "      <td>12/12 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Osteoarthritis [HP:0002758]</th>\n",
       "      <td>26/38 (68%)</td>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal systemic arterial morphology [HP:0011004]</th>\n",
       "      <td>29/29 (100%)</td>\n",
       "      <td>16/16 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormality of the vertebral column [HP:0000925]</th>\n",
       "      <td>31/31 (100%)</td>\n",
       "      <td>10/10 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal blood vessel morphology [HP:0033353]</th>\n",
       "      <td>32/32 (100%)</td>\n",
       "      <td>17/17 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal axial skeleton morphology [HP:0009121]</th>\n",
       "      <td>33/33 (100%)</td>\n",
       "      <td>11/11 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal sternum morphology [HP:0000766]</th>\n",
       "      <td>6/20 (30%)</td>\n",
       "      <td>5/15 (33%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Arachnodactyly [HP:0001166]</th>\n",
       "      <td>7/18 (39%)</td>\n",
       "      <td>6/14 (43%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Diagnosis                                            OMIM:613795  \\\n",
       "Thoracic aortic aneurysm [HP:0012727]                  0/22 (0%)   \n",
       "Aortic aneurysm [HP:0004942]                         26/48 (54%)   \n",
       "Bifid uvula [HP:0000193]                             13/38 (34%)   \n",
       "Soft skin [HP:0000977]                               23/37 (62%)   \n",
       "Varicose veins [HP:0002619]                          14/22 (64%)   \n",
       "High palate [HP:0000218]                             12/20 (60%)   \n",
       "Abdominal aortic aneurysm [HP:0005112]                 0/22 (0%)   \n",
       "Inguinal hernia [HP:0000023]                         12/39 (31%)   \n",
       "Umbilical hernia [HP:0001537]                        12/39 (31%)   \n",
       "Hypertelorism [HP:0000316]                           13/35 (37%)   \n",
       "Joint hypermobility [HP:0001382]                     12/36 (33%)   \n",
       "Spontaneous pneumothorax [HP:0002108]                 2/20 (10%)   \n",
       "Scoliosis [HP:0002650]                               20/43 (47%)   \n",
       "Abnormal palate morphology [HP:0000174]             24/24 (100%)   \n",
       "Abnormal oral cavity morphology [HP:0000163]        24/24 (100%)   \n",
       "Vascular dilatation [HP:0002617]                    26/26 (100%)   \n",
       "Abnormal aortic morphology [HP:0001679]             26/26 (100%)   \n",
       "Abnormal morphology of the great vessels [HP:00...  26/26 (100%)   \n",
       "Osteoarthritis [HP:0002758]                          26/38 (68%)   \n",
       "Abnormal systemic arterial morphology [HP:0011004]  29/29 (100%)   \n",
       "Abnormality of the vertebral column [HP:0000925]    31/31 (100%)   \n",
       "Abnormal blood vessel morphology [HP:0033353]       32/32 (100%)   \n",
       "Abnormal axial skeleton morphology [HP:0009121]     33/33 (100%)   \n",
       "Abnormal sternum morphology [HP:0000766]              6/20 (30%)   \n",
       "Arachnodactyly [HP:0001166]                           7/18 (39%)   \n",
       "\n",
       "Diagnosis                                            OMIM:619656  \\\n",
       "Thoracic aortic aneurysm [HP:0012727]                10/16 (62%)   \n",
       "Aortic aneurysm [HP:0004942]                        10/10 (100%)   \n",
       "Bifid uvula [HP:0000193]                               1/14 (7%)   \n",
       "Soft skin [HP:0000977]                                 1/5 (20%)   \n",
       "Varicose veins [HP:0002619]                           4/12 (33%)   \n",
       "High palate [HP:0000218]                              5/15 (33%)   \n",
       "Abdominal aortic aneurysm [HP:0005112]                 1/11 (9%)   \n",
       "Inguinal hernia [HP:0000023]                          5/11 (45%)   \n",
       "Umbilical hernia [HP:0001537]                          2/4 (50%)   \n",
       "Hypertelorism [HP:0000316]                            4/14 (29%)   \n",
       "Joint hypermobility [HP:0001382]                       2/6 (33%)   \n",
       "Spontaneous pneumothorax [HP:0002108]                   0/9 (0%)   \n",
       "Scoliosis [HP:0002650]                                7/14 (50%)   \n",
       "Abnormal palate morphology [HP:0000174]               7/7 (100%)   \n",
       "Abnormal oral cavity morphology [HP:0000163]          7/7 (100%)   \n",
       "Vascular dilatation [HP:0002617]                    11/11 (100%)   \n",
       "Abnormal aortic morphology [HP:0001679]             12/12 (100%)   \n",
       "Abnormal morphology of the great vessels [HP:00...  12/12 (100%)   \n",
       "Osteoarthritis [HP:0002758]                           2/2 (100%)   \n",
       "Abnormal systemic arterial morphology [HP:0011004]  16/16 (100%)   \n",
       "Abnormality of the vertebral column [HP:0000925]    10/10 (100%)   \n",
       "Abnormal blood vessel morphology [HP:0033353]       17/17 (100%)   \n",
       "Abnormal axial skeleton morphology [HP:0009121]     11/11 (100%)   \n",
       "Abnormal sternum morphology [HP:0000766]              5/15 (33%)   \n",
       "Arachnodactyly [HP:0001166]                           6/14 (43%)   \n",
       "\n",
       "Diagnosis                                           Corrected p values  \\\n",
       "Thoracic aortic aneurysm [HP:0012727]                         0.000423   \n",
       "Aortic aneurysm [HP:0004942]                                  0.113597   \n",
       "Bifid uvula [HP:0000193]                                      0.650869   \n",
       "Soft skin [HP:0000977]                                        0.731597   \n",
       "Varicose veins [HP:0002619]                                   0.731597   \n",
       "High palate [HP:0000218]                                      0.731597   \n",
       "Abdominal aortic aneurysm [HP:0005112]                        1.000000   \n",
       "Inguinal hernia [HP:0000023]                                  1.000000   \n",
       "Umbilical hernia [HP:0001537]                                 1.000000   \n",
       "Hypertelorism [HP:0000316]                                    1.000000   \n",
       "Joint hypermobility [HP:0001382]                              1.000000   \n",
       "Spontaneous pneumothorax [HP:0002108]                         1.000000   \n",
       "Scoliosis [HP:0002650]                                        1.000000   \n",
       "Abnormal palate morphology [HP:0000174]                       1.000000   \n",
       "Abnormal oral cavity morphology [HP:0000163]                  1.000000   \n",
       "Vascular dilatation [HP:0002617]                              1.000000   \n",
       "Abnormal aortic morphology [HP:0001679]                       1.000000   \n",
       "Abnormal morphology of the great vessels [HP:00...            1.000000   \n",
       "Osteoarthritis [HP:0002758]                                   1.000000   \n",
       "Abnormal systemic arterial morphology [HP:0011004]            1.000000   \n",
       "Abnormality of the vertebral column [HP:0000925]              1.000000   \n",
       "Abnormal blood vessel morphology [HP:0033353]                 1.000000   \n",
       "Abnormal axial skeleton morphology [HP:0009121]               1.000000   \n",
       "Abnormal sternum morphology [HP:0000766]                      1.000000   \n",
       "Arachnodactyly [HP:0001166]                                   1.000000   \n",
       "\n",
       "Diagnosis                                           p values  \n",
       "Thoracic aortic aneurysm [HP:0012727]               0.000017  \n",
       "Aortic aneurysm [HP:0004942]                        0.009088  \n",
       "Bifid uvula [HP:0000193]                            0.078104  \n",
       "Soft skin [HP:0000977]                              0.146370  \n",
       "Varicose veins [HP:0002619]                         0.151415  \n",
       "High palate [HP:0000218]                            0.175583  \n",
       "Abdominal aortic aneurysm [HP:0005112]              0.333333  \n",
       "Inguinal hernia [HP:0000023]                        0.475054  \n",
       "Umbilical hernia [HP:0001537]                       0.585479  \n",
       "Hypertelorism [HP:0000316]                          0.742960  \n",
       "Joint hypermobility [HP:0001382]                    1.000000  \n",
       "Spontaneous pneumothorax [HP:0002108]               1.000000  \n",
       "Scoliosis [HP:0002650]                              1.000000  \n",
       "Abnormal palate morphology [HP:0000174]             1.000000  \n",
       "Abnormal oral cavity morphology [HP:0000163]        1.000000  \n",
       "Vascular dilatation [HP:0002617]                    1.000000  \n",
       "Abnormal aortic morphology [HP:0001679]             1.000000  \n",
       "Abnormal morphology of the great vessels [HP:00...  1.000000  \n",
       "Osteoarthritis [HP:0002758]                         1.000000  \n",
       "Abnormal systemic arterial morphology [HP:0011004]  1.000000  \n",
       "Abnormality of the vertebral column [HP:0000925]    1.000000  \n",
       "Abnormal blood vessel morphology [HP:0033353]       1.000000  \n",
       "Abnormal axial skeleton morphology [HP:0009121]     1.000000  \n",
       "Abnormal sternum morphology [HP:0000766]            1.000000  \n",
       "Arachnodactyly [HP:0001166]                         1.000000  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "summarize_hpo_analysis(hpo=hpo, result=lds_3_6_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpseacs.report import GpseaAnalysisReport, GPAnalysisResultSummary\n",
    "\n",
    "f_results = (\n",
    "  GPAnalysisResultSummary.from_multi( result=lds_3_6_result,  ),\n",
    ")\n",
    "\n",
    "\n",
    "caption = \"\"\".\"\"\"\n",
    "report = GpseaAnalysisReport(name=\"LDS 3 and 6\", \n",
    "                             cohort=cohort, \n",
    "                             fet_results=f_results,\n",
    "                             gene_symbol=\"n/a\",\n",
    "                             mane_tx_id=\"n/a\",\n",
    "                             mane_protein_id=\"n/a\",\n",
    "                             caption=caption)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "      min-width: 100px;\n",
       "      border: 1px solid #ddd;\n",
       "      padding: 8px;\n",
       "        \n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    p {\n",
       "      width: 75%;\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>LDS 3 and 6 Summary</h1>\n",
       "  <p>The cohort comprised 72 individuals (35 females, 35 males, 2 with unknown sex). A total of 98 HPO terms were used to annotate the cohort. Disease diagnoses: Loeys-Dietz syndrome 3 (OMIM:613795) (49 individuals), Loeys-Dietz syndrome 6 (OMIM:619656) (18 individuals), Congenital heart defects, multiple types, 8, with or without heterotaxy (OMIM:619657) (5 individuals). . A total of 25 unique variant alleles were found in n/a (transcript: n/a, protein id: n/a). HPO version: 2025-01-16. GPSEA version: 0.9.6.dev0.</p>\n",
       "  \n",
       "  \n",
       "    \n",
       "    <table>\n",
       "      <caption style=\"color: black;\">\n",
       "          Fisher Exact Test performed to compare HPO annotation frequency with respect to OMIM:613795 and OMIM:619656. Total of\n",
       "          25 tests were performed.\n",
       "      </caption>\n",
       "          <tbody>\n",
       "              <tr class=\"strng\">\n",
       "                  <th>HPO Term</th>\n",
       "                  <th>OMIM:613795</th>\n",
       "                  <th>OMIM:619656</th>\n",
       "                  <th>p-val</th>\n",
       "                  <th>adj. p-val</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                  <td class=\"lft\">Thoracic aortic aneurysm [HP:0012727]</td>\n",
       "                  <td class=\"lft\">0/22 (0%)</td>\n",
       "                  <td class=\"lft\">10/16 (62%)</td>\n",
       "                  <td class=\"lft\">1.69e-05</td>\n",
       "                  <td class=\"lft\">4.23e-04</td>\n",
       "              </tr>\n",
       "              \n",
       "            </tbody>\n",
       "      </table>\n",
       "      \n",
       "\n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "\n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpseacs.report._api.HtmlGpseaNotebookSummarizer at 0x7b38ebc93740>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpseacs.report import GpseaNotebookSummarizer\n",
    "summarizer = GpseaNotebookSummarizer(hpo=hpo, gpsea_version=gpsea.__version__)\n",
    "summarizer.summarize_report(report=report)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output to ../../supplement/tex/LDS_3_and_6_summary_draft.tex\n"
     ]
    }
   ],
   "source": [
    "summarizer.process_latex(report=report)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
