{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SLC45A2\n",
    "[Albinism type IV (OCA4)](https://omim.org/entry/606574) is caused by homozygous or compound heterozygous mutation SLC45A2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2023-10-09\n",
      "Using genophenocorr version 0.6.1.dev0\n"
     ]
    }
   ],
   "source": [
    "import gpsea\n",
    "import hpotk\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo(release='v2023-10-09')\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using genophenocorr version {gpsea.__version__}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SLC45A2\n",
    "We user the [Matched Annotation from NCBI and EMBL-EBI (MANE)](https://www.ncbi.nlm.nih.gov/refseq/MANE/) transcript and the corresponding protein identifier for SLC45A2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_symbol = 'SLC45A2'\n",
    "mane_tx_id = 'NM_016180.5'\n",
    "mane_protein_id = \"NP_057264.4\" #membrane-associated transporter protein isoform"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 100%|██████████| 30/30 [00:00<00:00, 36.33individuals/s]\n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "\n",
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.20\") as ps:\n",
    "    phenopackets = tuple(ps.iter_cohort_phenopackets(gene_symbol))\n",
    "len(phenopackets)\n",
    "\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort, validation = load_phenopackets(\n",
    "    phenopackets=phenopackets, \n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "\n",
    "validation.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>GPSEA cohort analysis</h1>\n",
       "  <p>Successfully loaded 30 individuals.</p>\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Top 10 HPO Terms</h3>\n",
       "        A total of 21 HPO terms were used to annotated the cohort.\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">HPO Term</th>\n",
       "                <th>ID</th>\n",
       "                <th>Seen in <em>n</em> individuals</th>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Hypopigmentation of hair</td>\n",
       "                <td>HP:0005599</td>\n",
       "                <td>30</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">White eyelashes</td>\n",
       "                <td>HP:0002227</td>\n",
       "                <td>29</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">White eyebrow</td>\n",
       "                <td>HP:0002226</td>\n",
       "                <td>29</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Nystagmus</td>\n",
       "                <td>HP:0000639</td>\n",
       "                <td>28</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Iris hypopigmentation</td>\n",
       "                <td>HP:0007730</td>\n",
       "                <td>28</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Reduced visual acuity</td>\n",
       "                <td>HP:0007663</td>\n",
       "                <td>26</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Hypoplasia of the fovea</td>\n",
       "                <td>HP:0007750</td>\n",
       "                <td>26</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Iris transillumination defect</td>\n",
       "                <td>HP:0012805</td>\n",
       "                <td>24</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Melanocytic nevus</td>\n",
       "                <td>HP:0000995</td>\n",
       "                <td>12</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Nevus</td>\n",
       "                <td>HP:0003764</td>\n",
       "                <td>10</td>\n",
       "                </tr>\n",
       "            \n",
       "                </tbody>\n",
       "    </table>\n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Top 10 Variants</h3>\n",
       "        Variants are shown according to NM_016180.5. A total of 52 unique variants were identified in the cohort.\n",
       "    </caption>\n",
       "            <tbody>\n",
       "                <tr class=\"strng\">\n",
       "                    <th>Count</th>\n",
       "                    <th class=\"lft\">Variant key</th>\n",
       "                    <th>Variant Name</th>\n",
       "                    <th>Protein Variant</th>\n",
       "                    <th>Variant Class</th>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>12</td>\n",
       "                    <td class=\"lft\">5_33954406_33954407_TG_T</td>\n",
       "                    <td>c.986del</td>\n",
       "                    <td>p.Thr329LysfsTer69</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>4</td>\n",
       "                    <td class=\"lft\">5_33944770_33944770_C_T</td>\n",
       "                    <td>c.1471G>A</td>\n",
       "                    <td>p.Gly491Arg</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>3</td>\n",
       "                    <td class=\"lft\">5_33947363_33947365_CTT_C</td>\n",
       "                    <td>c.1166_1167del</td>\n",
       "                    <td>p.Lys389SerfsTer55</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">5_33944709_33944709_G_T</td>\n",
       "                    <td>c.1532C>A</td>\n",
       "                    <td>p.Ala511Glu</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">5_33947257_33947258_AG_A</td>\n",
       "                    <td>c.1273del</td>\n",
       "                    <td>p.Leu425TrpfsTer9</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">5_33944734_33944735_TC_T</td>\n",
       "                    <td>c.1506del</td>\n",
       "                    <td>p.Thr503ProfsTer6</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">5_33944723_33944723_G_A</td>\n",
       "                    <td>c.1518C>T</td>\n",
       "                    <td>p.Val506=</td>\n",
       "                    <td>SYNONYMOUS_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">5_33954363_33954365_GGC_G</td>\n",
       "                    <td>c.1028_1029del</td>\n",
       "                    <td>p.Gly343AlafsTer10</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">5_33951674_33951674_C_A</td>\n",
       "                    <td>c.1036G>T</td>\n",
       "                    <td>p.Val346Leu</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">5_33984325_33984326_CG_C</td>\n",
       "                    <td>c.258del</td>\n",
       "                    <td>p.Val87TrpfsTer26</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "            </tbody>\n",
       "        </table>\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Diseases</h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Disease Name</th>\n",
       "                <th >Disease ID</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">Albinism, oculocutaneous, type IV</td>\n",
       "                <td>OMIM:606574</td>\n",
       "                <td>30</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       "    \n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Variant categories for NM_016180.5</h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Variant effect</th>\n",
       "                <th>Annotation Count</th>\n",
       "                <th>Percent</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">FRAMESHIFT_VARIANT</td>\n",
       "                <td>26</td>\n",
       "               <td>48%</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">MISSENSE_VARIANT</td>\n",
       "                <td>21</td>\n",
       "               <td>39%</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">SYNONYMOUS_VARIANT</td>\n",
       "                <td>2</td>\n",
       "               <td>4%</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">SPLICE_REGION_VARIANT</td>\n",
       "                <td>1</td>\n",
       "               <td>2%</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">SPLICE_POLYPYRIMIDINE_TRACT_VARIANT</td>\n",
       "                <td>1</td>\n",
       "               <td>2%</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">INTRON_VARIANT</td>\n",
       "                <td>1</td>\n",
       "               <td>2%</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">TRANSCRIPT_ABLATION</td>\n",
       "                <td>1</td>\n",
       "               <td>2%</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">STOP_GAINED</td>\n",
       "                <td>1</td>\n",
       "               <td>2%</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       "  \n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpsea.view._report.HtmlGpseaReport at 0x13fbaaa50>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import CohortViewer\n",
    "cv = CohortViewer(hpo=hpo)\n",
    "cv.process(cohort=cohort, transcript_id=mane_tx_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Note we do not show the strcture because no domains/motifs are shown in UniProt (Q9UMX9)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>GPSEA cohort analysis: All variant alleles</h1>\n",
       "\n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Variant alleles</h3>\n",
       "        A total of 28 unique alleles were identified in the cohort.\n",
       "    </caption>\n",
       "\n",
       "\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Variant key</th>\n",
       "                <th>Variant (cDNA)</th>\n",
       "                <th>Variant (protein)</th>\n",
       "                <th>Overlapping Exons</th>\n",
       "                <th>Effects</th>\n",
       "                <th>Count</th>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33954406_33954407_TG_T</td>\n",
       "                <td>c.986del</td>\n",
       "                <td>p.Thr329LysfsTer69</td>\n",
       "                <td>4</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>12</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33944770_33944770_C_T</td>\n",
       "                <td>c.1471G>A</td>\n",
       "                <td>p.Gly491Arg</td>\n",
       "                <td>7</td>\n",
       "                <td>missense</td>\n",
       "                <td>4</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33947363_33947365_CTT_C</td>\n",
       "                <td>c.1166_1167del</td>\n",
       "                <td>p.Lys389SerfsTer55</td>\n",
       "                <td>6</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>3</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33947257_33947258_AG_A</td>\n",
       "                <td>c.1273del</td>\n",
       "                <td>p.Leu425TrpfsTer9</td>\n",
       "                <td>6</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33944709_33944709_G_T</td>\n",
       "                <td>c.1532C>A</td>\n",
       "                <td>p.Ala511Glu</td>\n",
       "                <td>7</td>\n",
       "                <td>missense</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33984325_33984326_CG_C</td>\n",
       "                <td>c.258del</td>\n",
       "                <td>p.Val87TrpfsTer26</td>\n",
       "                <td>1</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33944723_33944723_G_A</td>\n",
       "                <td>c.1518C>T</td>\n",
       "                <td>p.Val506=</td>\n",
       "                <td>7</td>\n",
       "                <td>synonymous</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33954443_33954443_T_C</td>\n",
       "                <td>c.950A>G</td>\n",
       "                <td>p.Tyr317Cys</td>\n",
       "                <td>4</td>\n",
       "                <td>missense</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33954363_33954365_GGC_G</td>\n",
       "                <td>c.1028_1029del</td>\n",
       "                <td>p.Gly343AlafsTer10</td>\n",
       "                <td>4</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33951674_33951674_C_A</td>\n",
       "                <td>c.1036G>T</td>\n",
       "                <td>p.Val346Leu</td>\n",
       "                <td>5</td>\n",
       "                <td>missense</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33944734_33944735_TC_T</td>\n",
       "                <td>c.1506del</td>\n",
       "                <td>p.Thr503ProfsTer6</td>\n",
       "                <td>7</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33963973_33963973_C_G</td>\n",
       "                <td>c.606G>C</td>\n",
       "                <td>p.Trp202Cys</td>\n",
       "                <td>3</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33951665_33951665_C_T</td>\n",
       "                <td>c.1045G>A</td>\n",
       "                <td>p.Gly349Arg</td>\n",
       "                <td>5</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33944775_33944775_A_G</td>\n",
       "                <td>c.1466T>C</td>\n",
       "                <td>p.Leu489Pro</td>\n",
       "                <td>7</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33944709_33944709_G_A</td>\n",
       "                <td>c.1532C>T</td>\n",
       "                <td>p.Ala511Val</td>\n",
       "                <td>7</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33982263_33982263_C_CCT</td>\n",
       "                <td>c.533_534dup</td>\n",
       "                <td>p.Gly179ArgfsTer23</td>\n",
       "                <td>2</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33951679_33951683_TGAAA_T</td>\n",
       "                <td>c.1033-6_1033-3del</td>\n",
       "                <td>None</td>\n",
       "                <td>None</td>\n",
       "                <td>splice region, splice polypyrimidine, intronic</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33963773_33963773_C_T</td>\n",
       "                <td>c.806G>A</td>\n",
       "                <td>p.Gly269Asp</td>\n",
       "                <td>3</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33954440_33954440_C_T</td>\n",
       "                <td>c.953G>A</td>\n",
       "                <td>p.Arg318His</td>\n",
       "                <td>4</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33984312_33984317_GCCGAT_G</td>\n",
       "                <td>c.267_271del</td>\n",
       "                <td>p.Ser90GlnfsTer42</td>\n",
       "                <td>1</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33954416_33954416_A_T</td>\n",
       "                <td>c.977T>A</td>\n",
       "                <td>p.Ile326Asn</td>\n",
       "                <td>4</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33951642_33951642_G_C</td>\n",
       "                <td>c.1068C>G</td>\n",
       "                <td>p.Asn356Lys</td>\n",
       "                <td>5</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33984454_33984454_C_T</td>\n",
       "                <td>c.130G>A</td>\n",
       "                <td>p.Gly44Arg</td>\n",
       "                <td>1</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33947276_33947276_G_T</td>\n",
       "                <td>c.1255C>A</td>\n",
       "                <td>p.Pro419Thr</td>\n",
       "                <td>6</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33984310_33984311_TG_T</td>\n",
       "                <td>c.273del</td>\n",
       "                <td>p.Ser92AlafsTer21</td>\n",
       "                <td>1</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33984405_33984405_A_C</td>\n",
       "                <td>c.179T>G</td>\n",
       "                <td>p.Leu60Arg</td>\n",
       "                <td>1</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">SO:1000029_HGNC:16472_SLC45A2</td>\n",
       "                <td>SV involving SLC45A2</td>\n",
       "                <td>p.?</td>\n",
       "                <td></td>\n",
       "                <td>chromosomal deletion</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">5_33984437_33984437_G_C</td>\n",
       "                <td>c.147C>G</td>\n",
       "                <td>p.Tyr49Ter</td>\n",
       "                <td>1</td>\n",
       "                <td>stop gained</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "                </tbody>\n",
       "    </table>\n",
       "<!--\n",
       "   <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Variant category counts for </h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Variant effect</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">frameshift</td>\n",
       "                <td>26</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">missense</td>\n",
       "                <td>21</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">synonymous</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">splice region</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">splice polypyrimidine</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">intronic</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">chromosomal deletion</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">stop gained</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       " -->\n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpsea.view._report.HtmlGpseaReport at 0x324846b40>"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import CohortVariantViewer\n",
    "\n",
    "viewer = CohortVariantViewer(tx_id=mane_tx_id)\n",
    "viewer.process(cohort)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.predicate.phenotype import prepare_predicates_for_terms_of_interest\n",
    "\n",
    "pheno_predicates = prepare_predicates_for_terms_of_interest(\n",
    "    cohort=cohort,\n",
    "    hpo=hpo,\n",
    ")\n",
    "len(pheno_predicates)\n",
    "from gpsea.analysis.mtc_filter import HpoMtcFilter\n",
    "mtc_filter = HpoMtcFilter.default_filter( hpo=hpo)\n",
    "mtc_correction = 'fdr_bh'\n",
    "mtc_alpha = 0.05\n",
    "from gpsea.analysis.pcats.stats import FisherExactTest\n",
    "\n",
    "count_statistic = FisherExactTest()\n",
    "from gpsea.analysis.pcats import HpoTermAnalysis\n",
    "\n",
    "analysis = HpoTermAnalysis(\n",
    "    count_statistic=count_statistic,\n",
    "    mtc_filter=mtc_filter,\n",
    "    mtc_correction=mtc_correction,\n",
    "     mtc_alpha=mtc_alpha,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Allele group: missense/missense OR missense/other, other/other\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "  <style>table {\n",
       "    border-collapse: collapse;\n",
       "    margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "  }\n",
       "\n",
       "\n",
       "  th {\n",
       "    background-color: #f2f2f2;\n",
       "    border: 1px solid #dddddd;\n",
       "    text-align: left;\n",
       "    padding: 2px;\n",
       "  }\n",
       "\n",
       "  tr {\n",
       "    border: 1px solid #dddddd;\n",
       "  }\n",
       "\n",
       "  td {\n",
       "    padding: 2px;\n",
       "    font-weight: bold;\n",
       "  }\n",
       "\n",
       "  tr:nth-child(even) {\n",
       "    background-color: #f2f2f2;\n",
       "  }\n",
       "\n",
       "  .table td,tr {\n",
       "    text-align: left;\n",
       "}\n",
       "\n",
       ".table td:first-child, tr:first-child {\n",
       "    text-align: left;\n",
       "}\n",
       "\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>Phenotype testing report</h1>\n",
       "  <p>Phenotype MTC filter: <em>HPO MTC filter</em></p>\n",
       "  <p>Multiple testing correction: <em>fdr_bh</em></p>\n",
       "  <p>Performed statistical tests for 24 out of the total of 63 HPO terms.</p>\n",
       "    <table>\n",
       "      <caption>Using <em>HPO MTC filter</em>, 39 term(s) were omitted from statistical analysis.</caption>\n",
       "        <tbody>\n",
       "          <tr>\n",
       "            <th>Code</th>\n",
       "            <th>Reason</th>\n",
       "            <th>Count</th>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF01</td>\n",
       "            <td>Skipping term with maximum frequency that was less than threshold 0.4</td>\n",
       "            <td>2</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF03</td>\n",
       "            <td>Skipping term because of a child term with the same individual counts</td>\n",
       "            <td>13</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF08</td>\n",
       "            <td>Skipping general term</td>\n",
       "            <td>19</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF09</td>\n",
       "            <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>\n",
       "            <td>5</td>\n",
       "          </tr>\n",
       "          \n",
       "        </tbody>\n",
       "    </table>\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpsea.view._report.HtmlGpseaReport at 0x3264e2720>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.model import VariantEffect\n",
    "from gpsea.analysis.predicate.genotype import VariantPredicates, biallelic_predicate\n",
    "\n",
    "missense = VariantPredicates.variant_effect(VariantEffect.MISSENSE_VARIANT, tx_id=mane_tx_id)\n",
    "\n",
    "\n",
    "gt_predicate = biallelic_predicate(a_predicate=missense, \n",
    "                                   b_predicate=~missense, \n",
    "                                   a_label=\"missense\",\n",
    "                                   b_label=\"other\",\n",
    "                                    partitions=((0,1),(2,))\n",
    "                                   )\n",
    "print(gt_predicate.display_question())\n",
    "result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_predicate=gt_predicate,\n",
    "    pheno_predicates=pheno_predicates,\n",
    ")\n",
    "from gpsea.view import MtcStatsViewer\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Allele group</th>\n",
       "      <th colspan=\"2\" halign=\"left\">missense/missense OR missense/other</th>\n",
       "      <th colspan=\"2\" halign=\"left\">other/other</th>\n",
       "      <th colspan=\"2\" halign=\"left\"></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Iris transillumination defect [HP:0012805]</th>\n",
       "      <td>13/17</td>\n",
       "      <td>76%</td>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.132357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Esotropia [HP:0000565]</th>\n",
       "      <td>4/16</td>\n",
       "      <td>25%</td>\n",
       "      <td>5/10</td>\n",
       "      <td>50%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.234117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Heterotropia [HP:0032012]</th>\n",
       "      <td>6/18</td>\n",
       "      <td>33%</td>\n",
       "      <td>7/12</td>\n",
       "      <td>58%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.264122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Strabismus [HP:0000486]</th>\n",
       "      <td>6/18</td>\n",
       "      <td>33%</td>\n",
       "      <td>7/12</td>\n",
       "      <td>58%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.264122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nystagmus [HP:0000639]</th>\n",
       "      <td>16/18</td>\n",
       "      <td>89%</td>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.503448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Iris hypopigmentation [HP:0007730]</th>\n",
       "      <td>16/18</td>\n",
       "      <td>89%</td>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.503448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Melanocytic nevus [HP:0000995]</th>\n",
       "      <td>9/14</td>\n",
       "      <td>64%</td>\n",
       "      <td>3/6</td>\n",
       "      <td>50%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.642415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nevus [HP:0003764]</th>\n",
       "      <td>13/18</td>\n",
       "      <td>72%</td>\n",
       "      <td>9/12</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Localized skin lesion [HP:0011355]</th>\n",
       "      <td>13/13</td>\n",
       "      <td>100%</td>\n",
       "      <td>9/9</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal iris pigmentation [HP:0008034]</th>\n",
       "      <td>17/17</td>\n",
       "      <td>100%</td>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>White eyelashes [HP:0002227]</th>\n",
       "      <td>17/17</td>\n",
       "      <td>100%</td>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormality of hair pigmentation [HP:0009887]</th>\n",
       "      <td>18/18</td>\n",
       "      <td>100%</td>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal ocular adnexa morphology [HP:0030669]</th>\n",
       "      <td>17/17</td>\n",
       "      <td>100%</td>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>White eyebrow [HP:0002226]</th>\n",
       "      <td>17/18</td>\n",
       "      <td>94%</td>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal eyebrow morphology [HP:0000534]</th>\n",
       "      <td>17/17</td>\n",
       "      <td>100%</td>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reduced visual acuity [HP:0007663]</th>\n",
       "      <td>16/16</td>\n",
       "      <td>100%</td>\n",
       "      <td>10/10</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal involuntary eye movements [HP:0012547]</th>\n",
       "      <td>16/16</td>\n",
       "      <td>100%</td>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypopigmentation of hair [HP:0005599]</th>\n",
       "      <td>18/18</td>\n",
       "      <td>100%</td>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypoplasia of the fovea [HP:0007750]</th>\n",
       "      <td>15/15</td>\n",
       "      <td>100%</td>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal fundus morphology [HP:0001098]</th>\n",
       "      <td>15/15</td>\n",
       "      <td>100%</td>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal retinal morphology [HP:0000479]</th>\n",
       "      <td>15/15</td>\n",
       "      <td>100%</td>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal macular morphology [HP:0001103]</th>\n",
       "      <td>15/15</td>\n",
       "      <td>100%</td>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal conjugate eye movement [HP:0000549]</th>\n",
       "      <td>6/6</td>\n",
       "      <td>100%</td>\n",
       "      <td>7/7</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormality of skin pigmentation [HP:0001000]</th>\n",
       "      <td>9/9</td>\n",
       "      <td>100%</td>\n",
       "      <td>3/3</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Allele group                                    missense/missense OR missense/other  \\\n",
       "                                                                              Count   \n",
       "Iris transillumination defect [HP:0012805]                                    13/17   \n",
       "Esotropia [HP:0000565]                                                         4/16   \n",
       "Heterotropia [HP:0032012]                                                      6/18   \n",
       "Strabismus [HP:0000486]                                                        6/18   \n",
       "Nystagmus [HP:0000639]                                                        16/18   \n",
       "Iris hypopigmentation [HP:0007730]                                            16/18   \n",
       "Melanocytic nevus [HP:0000995]                                                 9/14   \n",
       "Nevus [HP:0003764]                                                            13/18   \n",
       "Localized skin lesion [HP:0011355]                                            13/13   \n",
       "Abnormal iris pigmentation [HP:0008034]                                       17/17   \n",
       "White eyelashes [HP:0002227]                                                  17/17   \n",
       "Abnormality of hair pigmentation [HP:0009887]                                 18/18   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]                                17/17   \n",
       "White eyebrow [HP:0002226]                                                    17/18   \n",
       "Abnormal eyebrow morphology [HP:0000534]                                      17/17   \n",
       "Reduced visual acuity [HP:0007663]                                            16/16   \n",
       "Abnormal involuntary eye movements [HP:0012547]                               16/16   \n",
       "Hypopigmentation of hair [HP:0005599]                                         18/18   \n",
       "Hypoplasia of the fovea [HP:0007750]                                          15/15   \n",
       "Abnormal fundus morphology [HP:0001098]                                       15/15   \n",
       "Abnormal retinal morphology [HP:0000479]                                      15/15   \n",
       "Abnormal macular morphology [HP:0001103]                                      15/15   \n",
       "Abnormal conjugate eye movement [HP:0000549]                                    6/6   \n",
       "Abnormality of skin pigmentation [HP:0001000]                                   9/9   \n",
       "\n",
       "Allele group                                            other/other          \\\n",
       "                                                Percent       Count Percent   \n",
       "Iris transillumination defect [HP:0012805]          76%       11/11    100%   \n",
       "Esotropia [HP:0000565]                              25%        5/10     50%   \n",
       "Heterotropia [HP:0032012]                           33%        7/12     58%   \n",
       "Strabismus [HP:0000486]                             33%        7/12     58%   \n",
       "Nystagmus [HP:0000639]                              89%       12/12    100%   \n",
       "Iris hypopigmentation [HP:0007730]                  89%       12/12    100%   \n",
       "Melanocytic nevus [HP:0000995]                      64%         3/6     50%   \n",
       "Nevus [HP:0003764]                                  72%        9/12     75%   \n",
       "Localized skin lesion [HP:0011355]                 100%         9/9    100%   \n",
       "Abnormal iris pigmentation [HP:0008034]            100%       12/12    100%   \n",
       "White eyelashes [HP:0002227]                       100%       12/12    100%   \n",
       "Abnormality of hair pigmentation [HP:0009887]      100%       12/12    100%   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]     100%       12/12    100%   \n",
       "White eyebrow [HP:0002226]                          94%       12/12    100%   \n",
       "Abnormal eyebrow morphology [HP:0000534]           100%       12/12    100%   \n",
       "Reduced visual acuity [HP:0007663]                 100%       10/10    100%   \n",
       "Abnormal involuntary eye movements [HP:0012547]    100%       12/12    100%   \n",
       "Hypopigmentation of hair [HP:0005599]              100%       12/12    100%   \n",
       "Hypoplasia of the fovea [HP:0007750]               100%       11/11    100%   \n",
       "Abnormal fundus morphology [HP:0001098]            100%       11/11    100%   \n",
       "Abnormal retinal morphology [HP:0000479]           100%       11/11    100%   \n",
       "Abnormal macular morphology [HP:0001103]           100%       11/11    100%   \n",
       "Abnormal conjugate eye movement [HP:0000549]       100%         7/7    100%   \n",
       "Abnormality of skin pigmentation [HP:0001000]      100%         3/3    100%   \n",
       "\n",
       "Allele group                                                                  \n",
       "                                                Corrected p values  p values  \n",
       "Iris transillumination defect [HP:0012805]                     1.0  0.132357  \n",
       "Esotropia [HP:0000565]                                         1.0  0.234117  \n",
       "Heterotropia [HP:0032012]                                      1.0  0.264122  \n",
       "Strabismus [HP:0000486]                                        1.0  0.264122  \n",
       "Nystagmus [HP:0000639]                                         1.0  0.503448  \n",
       "Iris hypopigmentation [HP:0007730]                             1.0  0.503448  \n",
       "Melanocytic nevus [HP:0000995]                                 1.0  0.642415  \n",
       "Nevus [HP:0003764]                                             1.0  1.000000  \n",
       "Localized skin lesion [HP:0011355]                             1.0  1.000000  \n",
       "Abnormal iris pigmentation [HP:0008034]                        1.0  1.000000  \n",
       "White eyelashes [HP:0002227]                                   1.0  1.000000  \n",
       "Abnormality of hair pigmentation [HP:0009887]                  1.0  1.000000  \n",
       "Abnormal ocular adnexa morphology [HP:0030669]                 1.0  1.000000  \n",
       "White eyebrow [HP:0002226]                                     1.0  1.000000  \n",
       "Abnormal eyebrow morphology [HP:0000534]                       1.0  1.000000  \n",
       "Reduced visual acuity [HP:0007663]                             1.0  1.000000  \n",
       "Abnormal involuntary eye movements [HP:0012547]                1.0  1.000000  \n",
       "Hypopigmentation of hair [HP:0005599]                          1.0  1.000000  \n",
       "Hypoplasia of the fovea [HP:0007750]                           1.0  1.000000  \n",
       "Abnormal fundus morphology [HP:0001098]                        1.0  1.000000  \n",
       "Abnormal retinal morphology [HP:0000479]                       1.0  1.000000  \n",
       "Abnormal macular morphology [HP:0001103]                       1.0  1.000000  \n",
       "Abnormal conjugate eye movement [HP:0000549]                   1.0  1.000000  \n",
       "Abnormality of skin pigmentation [HP:0001000]                  1.0  1.000000  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "\n",
    "summarize_hpo_analysis(hpo=hpo, result=result)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
