{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ZSWIM6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2023-10-09\n",
      "Using gpsea version 0.9.4\n"
     ]
    }
   ],
   "source": [
    "import gpsea\n",
    "import hpotk\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo(release='v2023-10-09')\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using gpsea version {gpsea.__version__}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ZSWIM6\n",
    "We user the [Matched Annotation from NCBI and EMBL-EBI (MANE)](https://www.ncbi.nlm.nih.gov/refseq/MANE/) transcript and the corresponding protein identifier for ZSWIM6."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.preprocessing import configure_default_protein_metadata_service\n",
    "gene_symbol = 'ZSWIM6'\n",
    "mane_tx_id = 'NM_020928.2'\n",
    "mane_protein_id = 'NP_001685.1' # zinc finger SWIM domain-containing protein 6\n",
    "pms = configure_default_protein_metadata_service()\n",
    "protein_meta = pms.annotate(mane_protein_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 100%|██████████| 16/16 [00:00<00:00, 274.26 individuals/s]\n",
      "Validated under permissive policy\n",
      "Showing errors and warnings\n",
      "Phenopackets\n",
      "  patient #14\n",
      "    individual\n",
      "     warnings:\n",
      "     ·ontology_class of the time_at_last_encounter field cannot be parsed into age. Consider formatting the age as ISO8601 duration (e.g., \"P31Y2M\" for 31 years and 2 months)\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store('0.1.24') as ps:\n",
    "        phenopackets = tuple(ps.iter_cohort_phenopackets(gene_symbol))\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort, qc_results = load_phenopackets(  \n",
    "    phenopackets=phenopackets,\n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "qc_results.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "\n",
       "<h1>GPSEA cohort analysis</h1>\n",
       "<div id=\"cohort-summary\">\n",
       "  <p>\n",
       "    Successfully loaded 16 individuals.\n",
       "\n",
       "    \n",
       "    8 were recorded as male,\n",
       "    8 as female,\n",
       "    and 0 as unknown sex.\n",
       "\n",
       "    \n",
       "    \n",
       "    No information about individuals' vital status was reported.\n",
       "    \n",
       "\n",
       "    \n",
       "    16 individuals had disease onset information\n",
       "    and 0 had information about the age of last encounter.\n",
       "  </p>\n",
       "\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "</div>\n",
       "\n",
       "\n",
       "<div class=\"cards\">\n",
       "  \n",
       "  <div id=\"hpo\" class=\"card\">\n",
       "    <h2>HPO terms</h2>\n",
       "    \n",
       "    \n",
       "    <table>\n",
       "      The cohort included 85 distinct HPO terms.\n",
       "      <caption>\n",
       "        \n",
       "        Top 10 most common HPO terms.\n",
       "        \n",
       "        \n",
       "      </caption>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>HPO Term</th>\n",
       "        </tr>\n",
       "\n",
       "        \n",
       "        <tr>\n",
       "          <td>13</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0001263\">Global developmental delay</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>9</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000316\">Hypertelorism</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>8</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0011803\">Bifid nose</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>7</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0001249\">Intellectual disability</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>7</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0001252\">Hypotonia</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>6</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0002317\">Unsteady gait</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>6</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0001344\">Absent speech</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>6</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0006866\">Midline central nervous system lipomas</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>5</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0040082\">Happy demeanor</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>5</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000729\">Autistic behavior</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div id=\"measurements\" class=\"card\">\n",
       "    <h2>Measurements</h2>\n",
       "    \n",
       "    \n",
       "    No data regarding measurement assays were provided.\n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div id=\"diseases\" class=\"card\">\n",
       "    <h2>Diseases</h2>\n",
       "    \n",
       "    \n",
       "    The cohort members were diagnosed with 2 diseases.\n",
       "    <table>\n",
       "      \n",
       "\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>Disease</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>9</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/disease/OMIM:603671\">Acromelic frontonasal dysostosis</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>7</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/disease/OMIM:617865\">Neurodevelopmental disorder with movement abnormalities, abnormal gait, and autistic features</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  \n",
       "  <div class=\"card\">\n",
       "    <h2>Variants</h2>\n",
       "    <p>\n",
       "      A total of 2 unique variants\n",
       "      were identified in the cohort. Variants were annotated with respect to NM_020928.2.\n",
       "    </p>\n",
       "    <table>\n",
       "      <caption>\n",
       "        <h3>Top 10 variants</h3>\n",
       "      </caption>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>Variant key</th>\n",
       "          <th>HGVS</th>\n",
       "          <th>Variant Class</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>9</td>\n",
       "          <td>5_61544156_61544156_C_T</td>\n",
       "          <td>c.3487C>T (p.Arg1163Trp)</td>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>7</td>\n",
       "          <td>5_61541917_61541917_C_T</td>\n",
       "          <td>c.2737C>T (p.Arg913Ter)</td>\n",
       "          <td>STOP_GAINED</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div class=\"card\">\n",
       "    \n",
       "    <h2>Variant effects</h2>\n",
       "    <p>\n",
       "      The effects were predicted for NM_020928.2\n",
       "    </p>\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th>Variant effect</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "          <td>9 (56%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>STOP_GAINED</td>\n",
       "          <td>7 (44%)</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x13aa946e0>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import CohortViewer\n",
    "\n",
    "cv = CohortViewer(hpo)\n",
    "cv.process(cohort, transcript_id=mane_tx_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.view import ProteinVisualizer\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "zswim6_fig, ax = plt.subplots(figsize=(15, 8))\n",
    "visualizer = ProteinVisualizer()\n",
    "visualizer.draw_protein(\n",
    "    protein_metadata=protein_meta,\n",
    "    cohort=cohort,\n",
    "    ax=ax,\n",
    ")\n",
    "zswim6_fig.tight_layout()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genotype-Phenotype Correlation (GPC) Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.pcats import configure_hpo_term_analysis\n",
    "from gpsea.analysis.clf import prepare_classifiers_for_terms_of_interest\n",
    "\n",
    "analysis = configure_hpo_term_analysis(hpo)\n",
    "pheno_clfs = prepare_classifiers_for_terms_of_interest(\n",
    "    cohort=cohort,\n",
    "    hpo=hpo,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Phenotype testing report</h1>\n",
       "<div class=\"cards\">\n",
       "  <div class=\"card\">\n",
       "    <ul>\n",
       "      <li>Phenotype MTC filter: <em>Independent filtering HPO filter</em></li>\n",
       "      <li>Multiple testing correction: <em>fdr_bh</em></li>\n",
       "    </ul>\n",
       "  </div>\n",
       "  <div class=\"card\">\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr>\n",
       "          <th>Code</th>\n",
       "          <th>Reason</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF01</td>\n",
       "          <td>Skipping term with maximum frequency that was less than threshold 0.4</td>\n",
       "          <td>40</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF03</td>\n",
       "          <td>Skipping term because of a child term with the same individual counts</td>\n",
       "          <td>6</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF05</td>\n",
       "          <td>Skipping term because one genotype had zero observations</td>\n",
       "          <td>19</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF08</td>\n",
       "          <td>Skipping general term</td>\n",
       "          <td>59</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF09</td>\n",
       "          <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>\n",
       "          <td>191</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "      <caption>\n",
       "        Performed statistical tests for 12 out of the total of 327 HPO terms.\n",
       "      </caption>\n",
       "    </table>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x13aa4bc50>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import MtcStatsViewer\n",
    "from gpsea.analysis.clf import monoallelic_classifier\n",
    "from gpsea.analysis.predicate import variant_key\n",
    "\n",
    "r913ter = variant_key(\"5_61541917_61541917_C_T\") # c.2737C>T; p.Arg913Ter\n",
    "r913w = variant_key(\"5_61544156_61544156_C_T\")   # c.3487C>T; p.Arg1163Trp\n",
    "\n",
    "r913_clf = monoallelic_classifier(\n",
    "    a_predicate=r913ter, \n",
    "    b_predicate=r913w, \n",
    "    a_label=\"Arg913Ter\", \n",
    "    b_label= \"Arg1163Trp\")\n",
    "r913_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_clf=r913_clf,\n",
    "    pheno_clfs=pheno_clfs,\n",
    ")\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(r913_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Allele group</th>\n",
       "      <th>Arg913Ter</th>\n",
       "      <th>Arg1163Trp</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Abnormal nasal morphology [HP:0005105]</th>\n",
       "      <td>1/1 (100%)</td>\n",
       "      <td>9/9 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal eyelid morphology [HP:0000492]</th>\n",
       "      <td>3/3 (100%)</td>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aplasia/Hypoplasia involving the central nervous system [HP:0002977]</th>\n",
       "      <td>5/5 (100%)</td>\n",
       "      <td>2/2 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal cerebral morphology [HP:0002060]</th>\n",
       "      <td>5/5 (100%)</td>\n",
       "      <td>3/3 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal forebrain morphology [HP:0100547]</th>\n",
       "      <td>5/5 (100%)</td>\n",
       "      <td>3/3 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal brain morphology [HP:0012443]</th>\n",
       "      <td>5/5 (100%)</td>\n",
       "      <td>4/4 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal oral cavity morphology [HP:0000163]</th>\n",
       "      <td>5/5 (100%)</td>\n",
       "      <td>4/4 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Morphological central nervous system abnormality [HP:0002011]</th>\n",
       "      <td>5/5 (100%)</td>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal ocular adnexa morphology [HP:0030669]</th>\n",
       "      <td>6/6 (100%)</td>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neurodevelopmental abnormality [HP:0012759]</th>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>6/6 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neurodevelopmental delay [HP:0012758]</th>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>6/6 (100%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Global developmental delay [HP:0001263]</th>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>6/7 (86%)</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Allele group                                         Arg913Ter  Arg1163Trp  \\\n",
       "Abnormal nasal morphology [HP:0005105]              1/1 (100%)  9/9 (100%)   \n",
       "Abnormal eyelid morphology [HP:0000492]             3/3 (100%)  7/7 (100%)   \n",
       "Aplasia/Hypoplasia involving the central nervou...  5/5 (100%)  2/2 (100%)   \n",
       "Abnormal cerebral morphology [HP:0002060]           5/5 (100%)  3/3 (100%)   \n",
       "Abnormal forebrain morphology [HP:0100547]          5/5 (100%)  3/3 (100%)   \n",
       "Abnormal brain morphology [HP:0012443]              5/5 (100%)  4/4 (100%)   \n",
       "Abnormal oral cavity morphology [HP:0000163]        5/5 (100%)  4/4 (100%)   \n",
       "Morphological central nervous system abnormalit...  5/5 (100%)  7/7 (100%)   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]      6/6 (100%)  7/7 (100%)   \n",
       "Neurodevelopmental abnormality [HP:0012759]         7/7 (100%)  6/6 (100%)   \n",
       "Neurodevelopmental delay [HP:0012758]               7/7 (100%)  6/6 (100%)   \n",
       "Global developmental delay [HP:0001263]             7/7 (100%)   6/7 (86%)   \n",
       "\n",
       "Allele group                                        Corrected p values  \\\n",
       "Abnormal nasal morphology [HP:0005105]                             1.0   \n",
       "Abnormal eyelid morphology [HP:0000492]                            1.0   \n",
       "Aplasia/Hypoplasia involving the central nervou...                 1.0   \n",
       "Abnormal cerebral morphology [HP:0002060]                          1.0   \n",
       "Abnormal forebrain morphology [HP:0100547]                         1.0   \n",
       "Abnormal brain morphology [HP:0012443]                             1.0   \n",
       "Abnormal oral cavity morphology [HP:0000163]                       1.0   \n",
       "Morphological central nervous system abnormalit...                 1.0   \n",
       "Abnormal ocular adnexa morphology [HP:0030669]                     1.0   \n",
       "Neurodevelopmental abnormality [HP:0012759]                        1.0   \n",
       "Neurodevelopmental delay [HP:0012758]                              1.0   \n",
       "Global developmental delay [HP:0001263]                            1.0   \n",
       "\n",
       "Allele group                                        p values  \n",
       "Abnormal nasal morphology [HP:0005105]                   1.0  \n",
       "Abnormal eyelid morphology [HP:0000492]                  1.0  \n",
       "Aplasia/Hypoplasia involving the central nervou...       1.0  \n",
       "Abnormal cerebral morphology [HP:0002060]                1.0  \n",
       "Abnormal forebrain morphology [HP:0100547]               1.0  \n",
       "Abnormal brain morphology [HP:0012443]                   1.0  \n",
       "Abnormal oral cavity morphology [HP:0000163]             1.0  \n",
       "Morphological central nervous system abnormalit...       1.0  \n",
       "Abnormal ocular adnexa morphology [HP:0030669]           1.0  \n",
       "Neurodevelopmental abnormality [HP:0012759]              1.0  \n",
       "Neurodevelopmental delay [HP:0012758]                    1.0  \n",
       "Global developmental delay [HP:0001263]                  1.0  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "\n",
    "summarize_hpo_analysis(hpo=hpo, result=r913_result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpseacs.report import GpseaAnalysisReport, GPAnalysisResultSummary\n",
    "\n",
    "f_results = (\n",
    "    GPAnalysisResultSummary.from_multi(\n",
    "        result=r913_result,\n",
    "    ),\n",
    ")\n",
    "\n",
    "caption = \"No statistically significant results identified.\"\n",
    "report = GpseaAnalysisReport(name=gene_symbol, \n",
    "                             cohort=cohort, \n",
    "                             fet_results=f_results,\n",
    "                             gene_symbol=gene_symbol,\n",
    "                             mane_tx_id=mane_tx_id,\n",
    "                             mane_protein_id=mane_protein_id,\n",
    "                             caption=caption)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "      min-width: 100px;\n",
       "      border: 1px solid #ddd;\n",
       "      padding: 8px;\n",
       "        \n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    p {\n",
       "      width: 75%;\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>ZSWIM6 Summary</h1>\n",
       "  <p>The cohort comprised 16 individuals (8 females, 8 males). A total of 85 HPO terms were used to annotate the cohort. Disease diagnoses: Acromelic frontonasal dysostosis (OMIM:603671) (9 individuals), Neurodevelopmental disorder with movement abnormalities, abnormal gait, and autistic features (OMIM:617865) (7 individuals). No statistically significant results identified. A total of 16 unique variant alleles were found in ZSWIM6 (transcript: NM_020928.2, protein id: NP_001685.1). HPO version: 2023-10-09. GPSEA version: 0.9.4.</p>\n",
       "  \n",
       "  \n",
       "    \n",
       "      <table>\n",
       "        <caption style=\"color: black;\">\n",
       "            Fisher Exact Test performed to compare HPO annotation frequency with respect to Arg913Ter and Arg1163Trp.\n",
       "        </caption>\n",
       "            <tbody>\n",
       "                <tr class=\"strng\">\n",
       "                    <th>Genotype (A)</th>\n",
       "                    <th>Genotype (B)</th>\n",
       "                  <th>Tests performed</th>\n",
       "                  <th>Significant tests</th>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td class=\"lft\">Arg913Ter</td>\n",
       "                    <td class=\"lft\">Arg1163Trp</td>\n",
       "                    <td class=\"lft\">12</td>\n",
       "                    <td class=\"lft\">0</td>\n",
       "                </tr>\n",
       "              </tbody>\n",
       "        </table>\n",
       "      \n",
       "\n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "\n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpseacs.report._api.HtmlGpseaNotebookSummarizer at 0x13aa97b60>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpseacs.report import GpseaNotebookSummarizer\n",
    "summarizer = GpseaNotebookSummarizer(hpo=hpo, gpsea_version=gpsea.__version__)\n",
    "summarizer.summarize_report(report=report)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summarizer.process_latex(report=report, protein_fig=zswim6_fig)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
