{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ASPM\n",
    "[primary microcephaly-5 (MCPH5)](https://omim.org/entry/608716) is caused by homozygous or compound heterozygous mutation in the ASPM gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2024-08-13\n",
      "Using gpsea version 0.4.1.dev0\n"
     ]
    }
   ],
   "source": [
    "import gpsea\n",
    "from gpsea.analysis.predicate import PatientCategories\n",
    "from gpsea.preprocessing import load_phenopacket_folder\n",
    "from gpsea.preprocessing import configure_caching_cohort_creator\n",
    "from gpsea.model import FeatureType, VariantEffect\n",
    "from gpsea.view import CohortViewable\n",
    "from gpsea.preprocessing import UniprotProteinMetadataService\n",
    "from gpsea.model.genome import GRCh38\n",
    "from gpsea.preprocessing import VVMultiCoordinateService\n",
    "from gpsea.view import ProteinVisualizable, ProteinVisualizer, ProteinViewable\n",
    "import hpotk\n",
    "import gpsea\n",
    "\n",
    "from IPython.display import display, HTML\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo(release='v2024-08-13')\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using gpsea version {gpsea.__version__}\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The ASPM gene is the human ortholog of the Drosophila melanogaster 'abnormal spindle' gene (asp), which is essential for normal mitotic spindle function in embryonic neuroblasts. The mouse gene Aspm is expressed specifically in the primary sites of prenatal cerebral cortical neurogenesis ([OMIM:605481](https://omim.org/entry/605481))."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_symbol = 'ASPM'\n",
    "mane_tx_id = 'NM_018136.5'\n",
    "mane_protein_id = 'NP_060606.3' # abnormal spindle-like microcephaly-associated protein isoform 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load phenopackets\n",
    "\n",
    "The following code loads the 22 phenopackets available from release 0.1.19 of phenopacket-store."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 22 phenopackets\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "\n",
    "phenopacket_store_release = '0.1.19'  # Update, if necessary\n",
    "registry = configure_phenopacket_registry()\n",
    "\n",
    "with registry.open_phenopacket_store(release=phenopacket_store_release) as ps:\n",
    "    phenopackets = tuple(ps.iter_cohort_phenopackets(gene_symbol))\n",
    "\n",
    "print(f'Loaded {len(phenopackets)} phenopackets')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 100%|██████████| 22/22 [00:20<00:00,  1.07individuals/s]\n",
      "Loaded 22 individuals\n"
     ]
    }
   ],
   "source": [
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "\n",
    "cohort, qc = load_phenopackets(\n",
    "    phenopackets=phenopackets, \n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "print(f'Loaded {len(cohort)} individuals')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Validated under none policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "qc.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>GPSEA cohort analysis</h1>\n",
       "  <p>Successfully loaded 22 individuals.</p>\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Top 10 HPO Terms</h3>\n",
       "        A total of 23 HPO terms were used to annotated the cohort.\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">HPO Term</th>\n",
       "                <th>ID</th>\n",
       "                <th>Seen in <em>n</em> individuals</th>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Microcephaly</td>\n",
       "                <td>HP:0000252</td>\n",
       "                <td>22</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Intellectual disability</td>\n",
       "                <td>HP:0001249</td>\n",
       "                <td>18</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Short stature</td>\n",
       "                <td>HP:0004322</td>\n",
       "                <td>16</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Decreased body weight</td>\n",
       "                <td>HP:0004325</td>\n",
       "                <td>14</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Hyperactivity</td>\n",
       "                <td>HP:0000752</td>\n",
       "                <td>7</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Aggressive behavior</td>\n",
       "                <td>HP:0000718</td>\n",
       "                <td>3</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Self-injurious behavior</td>\n",
       "                <td>HP:0100716</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Motor delay</td>\n",
       "                <td>HP:0001270</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Opisthotonus</td>\n",
       "                <td>HP:0002179</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Simplified gyral pattern</td>\n",
       "                <td>HP:0009879</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "                </tbody>\n",
       "    </table>\n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Top 10 Variants</h3>\n",
       "        Variants are shown according to NM_018136.5. A total of 28 unique variants were identified in the cohort.\n",
       "    </caption>\n",
       "            <tbody>\n",
       "                <tr class=\"strng\">\n",
       "                    <th>Count</th>\n",
       "                    <th class=\"lft\">Variant key</th>\n",
       "                    <th>Variant Name</th>\n",
       "                    <th>Protein Variant</th>\n",
       "                    <th>Variant Class</th>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>5</td>\n",
       "                    <td class=\"lft\">1_197117875_197117875_G_A</td>\n",
       "                    <td>c.3979C>T</td>\n",
       "                    <td>p.Arg1327Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>4</td>\n",
       "                    <td class=\"lft\">1_197143563_197143564_TC_T</td>\n",
       "                    <td>c.688del</td>\n",
       "                    <td>p.Glu230AsnfsTer30</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>3</td>\n",
       "                    <td class=\"lft\">1_197093156_197093156_G_A</td>\n",
       "                    <td>c.9190C>T</td>\n",
       "                    <td>p.Arg3064Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">1_197124158_197124159_CT_C</td>\n",
       "                    <td>c.3341del</td>\n",
       "                    <td>p.Lys1114SerfsTer3</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">1_197090328_197090328_G_A</td>\n",
       "                    <td>c.9697C>T</td>\n",
       "                    <td>p.Arg3233Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">1_197090945_197090945_G_A</td>\n",
       "                    <td>c.9541C>T</td>\n",
       "                    <td>p.Arg3181Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">1_197124923_197124930_TATCCACA_T</td>\n",
       "                    <td>c.3108_3114del</td>\n",
       "                    <td>p.Val1037GlyfsTer13</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>1</td>\n",
       "                    <td class=\"lft\">1_197142463_197142463_G_A</td>\n",
       "                    <td>c.1789C>T</td>\n",
       "                    <td>p.Arg597Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>1</td>\n",
       "                    <td class=\"lft\">1_197086966_197086966_G_A</td>\n",
       "                    <td>c.10168C>T</td>\n",
       "                    <td>p.Arg3390Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>1</td>\n",
       "                    <td class=\"lft\">1_197101467_197101469_TTC_T</td>\n",
       "                    <td>c.7782_7783del</td>\n",
       "                    <td>p.Lys2595SerfsTer6</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "            </tbody>\n",
       "        </table>\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Diseases</h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Disease Name</th>\n",
       "                <th >Disease ID</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">Microcephaly 5, primary, autosomal recessive</td>\n",
       "                <td>OMIM:608716</td>\n",
       "                <td>22</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       "    \n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Variant categories for NM_018136.5</h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Variant effect</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">FRAMESHIFT_VARIANT</td>\n",
       "                <td>10</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">STOP_GAINED</td>\n",
       "                <td>16</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">INFRAME_DELETION</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">SPLICE_DONOR_VARIANT</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       "  \n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from gpsea.view import CohortViewable\n",
    "\n",
    "cv = CohortViewable(hpo)\n",
    "report = cv.process(cohort, transcript_id=mane_tx_id)\n",
    "display(HTML(report))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Could not find an entry for NP_060606.3 in Uniprot response",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[10], line 7\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mgpsea\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mview\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m ProteinVisualizable, ProteinViewable\n\u001b[1;32m      6\u001b[0m pms \u001b[38;5;241m=\u001b[39m configure_default_protein_metadata_service()\n\u001b[0;32m----> 7\u001b[0m protein_meta \u001b[38;5;241m=\u001b[39m \u001b[43mpms\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mannotate\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmane_protein_id\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      8\u001b[0m txc_service \u001b[38;5;241m=\u001b[39m VVMultiCoordinateService(genome_build\u001b[38;5;241m=\u001b[39mGRCh38)\n\u001b[1;32m      9\u001b[0m tx_coordinates \u001b[38;5;241m=\u001b[39m txc_service\u001b[38;5;241m.\u001b[39mfetch(mane_tx_id)\n",
      "File \u001b[0;32m~/GIT/gpsea/src/gpsea/preprocessing/_protein.py:94\u001b[0m, in \u001b[0;36mProtCachingMetadataService.annotate\u001b[0;34m(self, protein_id)\u001b[0m\n\u001b[1;32m     92\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m annotations\n\u001b[1;32m     93\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m---> 94\u001b[0m     ann \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_fallback\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mannotate\u001b[49m\u001b[43m(\u001b[49m\u001b[43mprotein_id\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     95\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_cache\u001b[38;5;241m.\u001b[39mstore_annotations(protein_id, ann)\n\u001b[1;32m     97\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m ann\n",
      "File \u001b[0;32m~/GIT/gpsea/src/gpsea/preprocessing/_uniprot.py:123\u001b[0m, in \u001b[0;36mUniprotProteinMetadataService.annotate\u001b[0;34m(self, protein_id)\u001b[0m\n\u001b[1;32m    120\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124monly works with a RefSeq database ID (e.g. NP_037407.4), but we got \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mprotein_id\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    122\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_fetch_uniprot_response(protein_id)\n\u001b[0;32m--> 123\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mUniprotProteinMetadataService\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mparse_uniprot_json\u001b[49m\u001b[43m(\u001b[49m\u001b[43mresponse\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mprotein_id\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/GIT/gpsea/src/gpsea/preprocessing/_uniprot.py:61\u001b[0m, in \u001b[0;36mUniprotProteinMetadataService.parse_uniprot_json\u001b[0;34m(payload, protein_id)\u001b[0m\n\u001b[1;32m     55\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28many\u001b[39m(uni[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mid\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m==\u001b[39m protein_id \u001b[38;5;28;01mfor\u001b[39;00m uni \u001b[38;5;129;01min\u001b[39;00m protein[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124muniProtKBCrossReferences\u001b[39m\u001b[38;5;124m'\u001b[39m]):\n\u001b[1;32m     56\u001b[0m             \u001b[38;5;28;01mreturn\u001b[39;00m UniprotProteinMetadataService\u001b[38;5;241m.\u001b[39m_extract_metadata(\n\u001b[1;32m     57\u001b[0m                 protein_id\u001b[38;5;241m=\u001b[39mprotein_id,\n\u001b[1;32m     58\u001b[0m                 data\u001b[38;5;241m=\u001b[39mprotein,\n\u001b[1;32m     59\u001b[0m             )\n\u001b[0;32m---> 61\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCould not find an entry for \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mprotein_id\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m in Uniprot response\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "\u001b[0;31mValueError\u001b[0m: Could not find an entry for NP_060606.3 in Uniprot response"
     ]
    }
   ],
   "source": [
    "from gpsea.model.genome import GRCh38\n",
    "from gpsea.preprocessing import configure_default_protein_metadata_service, VVMultiCoordinateService\n",
    "from gpsea.view import ProteinVisualizable, ProteinViewable\n",
    "\n",
    "\n",
    "pms = configure_default_protein_metadata_service()\n",
    "protein_meta = pms.annotate(mane_protein_id)\n",
    "txc_service = VVMultiCoordinateService(genome_build=GRCh38)\n",
    "tx_coordinates = txc_service.fetch(mane_tx_id)\n",
    "pvis = ProteinVisualizable(tx_coordinates=tx_coordinates, protein_meta=protein_meta, cohort=cohort)\n",
    "\n",
    "viewer = ProteinViewable()\n",
    "html_prot = viewer.process(cohort, pvis)\n",
    "display(HTML(html_prot))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>GPSEA cohort analysis: All variant alleles</h1>\n",
       "\n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Variant alleles</h3>\n",
       "        A total of 15 unique alleles were identified in the cohort.\n",
       "    </caption>\n",
       "\n",
       "\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Variant key</th>\n",
       "                <th>Variant (cDNA)</th>\n",
       "                <th>Variant (protein)</th>\n",
       "                <th>Effects</th>\n",
       "                <th>Count</th>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197117875_197117875_G_A</td>\n",
       "                <td>c.3979C>T</td>\n",
       "                <td>p.Arg1327Ter</td>\n",
       "                <td>stop gained</td>\n",
       "                <td>5</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197143563_197143564_TC_T</td>\n",
       "                <td>c.688del</td>\n",
       "                <td>p.Glu230AsnfsTer30</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>4</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197093156_197093156_G_A</td>\n",
       "                <td>c.9190C>T</td>\n",
       "                <td>p.Arg3064Ter</td>\n",
       "                <td>stop gained</td>\n",
       "                <td>3</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197124923_197124930_TATCCACA_T</td>\n",
       "                <td>c.3108_3114del</td>\n",
       "                <td>p.Val1037GlyfsTer13</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197124158_197124159_CT_C</td>\n",
       "                <td>c.3341del</td>\n",
       "                <td>p.Lys1114SerfsTer3</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197090328_197090328_G_A</td>\n",
       "                <td>c.9697C>T</td>\n",
       "                <td>p.Arg3233Ter</td>\n",
       "                <td>stop gained</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197090945_197090945_G_A</td>\n",
       "                <td>c.9541C>T</td>\n",
       "                <td>p.Arg3181Ter</td>\n",
       "                <td>stop gained</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197104639_197104639_G_A</td>\n",
       "                <td>c.4612C>T</td>\n",
       "                <td>p.Arg1538Ter</td>\n",
       "                <td>stop gained</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197142463_197142463_G_A</td>\n",
       "                <td>c.1789C>T</td>\n",
       "                <td>p.Arg597Ter</td>\n",
       "                <td>stop gained</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197139831_197139834_TTTG_T</td>\n",
       "                <td>c.1959_1961del</td>\n",
       "                <td>p.Asn653del</td>\n",
       "                <td>inframe deletion</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197101760_197101760_A_C</td>\n",
       "                <td>c.7491T>G</td>\n",
       "                <td>p.Tyr2497Ter</td>\n",
       "                <td>stop gained</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197086966_197086966_G_A</td>\n",
       "                <td>c.10168C>T</td>\n",
       "                <td>p.Arg3390Ter</td>\n",
       "                <td>stop gained</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197101467_197101469_TTC_T</td>\n",
       "                <td>c.7782_7783del</td>\n",
       "                <td>p.Lys2595SerfsTer6</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197128489_197128489_C_T</td>\n",
       "                <td>c.2936+1G>A</td>\n",
       "                <td>None</td>\n",
       "                <td>splice donor</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">1_197090279_197090283_AGTTT_A</td>\n",
       "                <td>c.9742_9745del</td>\n",
       "                <td>p.Lys3248SerfsTer13</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "                </tbody>\n",
       "    </table>\n",
       "<!--\n",
       "   <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Variant category counts for </h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Variant effect</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">stop gained</td>\n",
       "                <td>16</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">frameshift</td>\n",
       "                <td>10</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">inframe deletion</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">splice donor</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       " -->\n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from gpsea.view import CohortVariantViewer\n",
    "\n",
    "viewer = CohortVariantViewer(tx_id=mane_tx_id)\n",
    "report = viewer.process(cohort)\n",
    "display(HTML(report))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/hq/zttr8hjx127fgvh0bq7h4bc80000gn/T/ipykernel_12418/2202611031.py:5: DeprecationWarning: Use `gpsea.analysis.predicate.genotype.autosomal_recessive` instead\n",
      "  gt_predicate = ModeOfInheritancePredicate.autosomal_recessive(is_missense)\n"
     ]
    }
   ],
   "source": [
    "from gpsea.model import VariantEffect\n",
    "from gpsea.analysis.predicate.genotype import VariantPredicates, ModeOfInheritancePredicate\n",
    "\n",
    "is_missense = VariantPredicates.variant_effect(VariantEffect.FRAMESHIFT_VARIANT, tx_id=mane_tx_id)\n",
    "gt_predicate = ModeOfInheritancePredicate.autosomal_recessive(is_missense)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.model import FeatureType\n",
    "from gpsea.analysis.predicate.phenotype import prepare_predicates_for_terms_of_interest\n",
    "\n",
    "pheno_predicates = prepare_predicates_for_terms_of_interest(\n",
    "    cohort=cohort,\n",
    "    hpo=hpo,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.pcats.stats import FisherExactTest\n",
    "from gpsea.analysis.mtc_filter import HpoMtcFilter\n",
    "\n",
    "mtc_filter = HpoMtcFilter.default_filter(hpo=hpo, term_frequency_threshold=0.2)\n",
    "mtc_correction = 'fdr_bh'\n",
    "statistic = FisherExactTest()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.pcats import HpoTermAnalysis\n",
    "\n",
    "analysis = HpoTermAnalysis(\n",
    "    count_statistic=statistic,\n",
    "    mtc_filter=mtc_filter,\n",
    "    mtc_correction=mtc_correction,\n",
    "    mtc_alpha=0.05,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_predicate=gt_predicate,\n",
    "    pheno_predicates=pheno_predicates,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "  <style>table {\n",
       "    border-collapse: collapse;\n",
       "    margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "  }\n",
       "\n",
       "\n",
       "  th {\n",
       "    background-color: #f2f2f2;\n",
       "    border: 1px solid #dddddd;\n",
       "    text-align: left;\n",
       "    padding: 2px;\n",
       "  }\n",
       "\n",
       "  tr {\n",
       "    border: 1px solid #dddddd;\n",
       "  }\n",
       "\n",
       "  td {\n",
       "    padding: 2px;\n",
       "    font-weight: bold;\n",
       "  }\n",
       "\n",
       "  tr:nth-child(even) {\n",
       "    background-color: #f2f2f2;\n",
       "  }\n",
       "\n",
       "  .table td,tr {\n",
       "    text-align: left;\n",
       "}\n",
       "\n",
       ".table td:first-child, tr:first-child {\n",
       "    text-align: left;\n",
       "}\n",
       "\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>Phenotype testing report</h1>\n",
       "  <p>Phenotype MTC filter: <em>HPO MTC filter</em></p>\n",
       "  <p>Multiple testing correction: <em>fdr_bh</em></p>\n",
       "  <p>Performed statistical tests for 4 out of the total of 46 HPO terms.</p>\n",
       "    <table>\n",
       "      <caption>Using <em>HPO MTC filter</em>, 42 term(s) were omitted from statistical analysis.</caption>\n",
       "        <tbody>\n",
       "          <tr>\n",
       "            <th>Code</th>\n",
       "            <th>Reason</th>\n",
       "            <th>Count</th>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF01</td>\n",
       "            <td>Skipping term with maximum frequency that was less than threshold 0.2</td>\n",
       "            <td>1</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF02</td>\n",
       "            <td>Skipping term because no genotype has more than one observed HPO count</td>\n",
       "            <td>1</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF04</td>\n",
       "            <td>Skipping term because all genotypes have same HPO observed proportions</td>\n",
       "            <td>19</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF06</td>\n",
       "            <td>Skipping term with less than 6 observations (not powered for 2x3)</td>\n",
       "            <td>5</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF08</td>\n",
       "            <td>Skipping general term</td>\n",
       "            <td>16</td>\n",
       "          </tr>\n",
       "          \n",
       "        </tbody>\n",
       "    </table>\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from gpsea.view import MtcStatsViewer\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "report = viewer.process(result)\n",
    "display(HTML(report))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>What is the genotype group</th>\n",
       "      <th colspan=\"2\" halign=\"left\">HOM_REF</th>\n",
       "      <th colspan=\"2\" halign=\"left\">HET</th>\n",
       "      <th colspan=\"2\" halign=\"left\">BIALLELIC_ALT</th>\n",
       "      <th colspan=\"2\" halign=\"left\"></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Hyperactivity [HP:0000752]</th>\n",
       "      <td>5/15</td>\n",
       "      <td>33%</td>\n",
       "      <td>1/1</td>\n",
       "      <td>100%</td>\n",
       "      <td>1/4</td>\n",
       "      <td>25%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.509314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Short stature [HP:0004322]</th>\n",
       "      <td>10/15</td>\n",
       "      <td>67%</td>\n",
       "      <td>1/1</td>\n",
       "      <td>100%</td>\n",
       "      <td>5/6</td>\n",
       "      <td>83%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.725584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Decreased body weight [HP:0004325]</th>\n",
       "      <td>9/15</td>\n",
       "      <td>60%</td>\n",
       "      <td>1/1</td>\n",
       "      <td>100%</td>\n",
       "      <td>4/6</td>\n",
       "      <td>67%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aggressive behavior [HP:0000718]</th>\n",
       "      <td>2/15</td>\n",
       "      <td>13%</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0%</td>\n",
       "      <td>1/5</td>\n",
       "      <td>20%</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "What is the genotype group         HOM_REF           HET          \\\n",
       "                                     Count Percent Count Percent   \n",
       "Hyperactivity [HP:0000752]            5/15     33%   1/1    100%   \n",
       "Short stature [HP:0004322]           10/15     67%   1/1    100%   \n",
       "Decreased body weight [HP:0004325]    9/15     60%   1/1    100%   \n",
       "Aggressive behavior [HP:0000718]      2/15     13%   0/1      0%   \n",
       "\n",
       "What is the genotype group         BIALLELIC_ALT                             \\\n",
       "                                           Count Percent Corrected p values   \n",
       "Hyperactivity [HP:0000752]                   1/4     25%                1.0   \n",
       "Short stature [HP:0004322]                   5/6     83%                1.0   \n",
       "Decreased body weight [HP:0004325]           4/6     67%                1.0   \n",
       "Aggressive behavior [HP:0000718]             1/5     20%                1.0   \n",
       "\n",
       "What is the genotype group                    \n",
       "                                    p values  \n",
       "Hyperactivity [HP:0000752]          0.509314  \n",
       "Short stature [HP:0004322]          0.725584  \n",
       "Decreased body weight [HP:0004325]  1.000000  \n",
       "Aggressive behavior [HP:0000718]    1.000000  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "\n",
    "report = summarize_hpo_analysis(hpo=hpo, result=result)\n",
    "report"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
