{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Loeys-Dietz syndrome\n",
    "Loeys-Dietz syndrome (LDS) is an autosomal dominant aortic aneurysm syndrome characterized by multisystemic involvement. The most typical clinical triad includes hypertelorism, bifid uvula or cleft palate and aortic aneurysm with tortuosity. Affected individuals may expereince aortic dissection at smaller aortic diameter and arterial aneurysms throughout the arterial tree. The genetic cause is heterogeneous and includes mutations in genes encoding for components of the transforming growth factor beta (TGFβ) signalling pathway: TGFBR1, TGFBR2, SMAD2, SMAD3, TGFB2 and TGFB3 (See [Velchev JD, et al. (2021). Loeys-Dietz Syndrome. Adv Exp Med Biol](https://pubmed.ncbi.nlm.nih.gov/34807423/)).\n",
    "\n",
    "This notebook will explore whether there are significant differences in phenotypic features between different genetic forms of LDS. We will include TGFBR1 and TGFBR2 (which were the first forms of LDS to be identified) with "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2024-12-12\n",
      "Using genophenocorr version 0.9.1\n"
     ]
    }
   ],
   "source": [
    "import gpsea\n",
    "import hpotk\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo()\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using genophenocorr version {gpsea.__version__}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## LDS1\n",
    "\n",
    "Loeys Dietz syndrome 1 (LDS1) is caused by mutation in the TGFBR1 gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "41 LDS1 phenopackets\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "\n",
    "tgfbr1_symbol = 'TGFBR1'\n",
    "tgfbr1_mane_tx_id = 'NM_004612.4'\n",
    "tgfbr1_mane_protein_id = 'NP_004603.1' # TGF-beta receptor type-1 isoform 1 precursor\"\n",
    "lds1_disease_id = \"OMIM:609192\"\n",
    "\n",
    "\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.23\") as ps:\n",
    "    lds1_phenopackets = tuple(ps.iter_cohort_phenopackets(tgfbr1_symbol))\n",
    "tgfbr1_len = len(lds1_phenopackets)\n",
    "print(f\"{len(lds1_phenopackets)} LDS1 phenopackets\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS2\n",
    "[Loeys-Dietz syndrome 2 (OMIM:610168)](https://omim.org/entry/610168) is caused by heterozygous mutation in the TGFBR2 gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "53 LDS2 phenopackets\n"
     ]
    }
   ],
   "source": [
    "tgfbr2_symbol = 'TGFBR2'\n",
    "tgfbr2_mane_tx_id = 'NM_003242.6'\n",
    "tgfbr2_mane_protein_id = 'NP_003233.4' # TGF-beta receptor type-2 isoform B precursor\n",
    "lds2_disease_id = \"OMIM:610168\"\n",
    "\n",
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.23\") as ps:\n",
    "    lds2_phenopackets = tuple(ps.iter_cohort_phenopackets(tgfbr2_symbol))\n",
    "print(f\"{len(lds2_phenopackets)} LDS2 phenopackets\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS3\n",
    "[Loeys-Dietz syndrome-3 (LDS3)](https://omim.org/entry/613795) is caused by heterozygous mutation in the SMAD3 gene.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "49 LDS3 phenopackets\n"
     ]
    }
   ],
   "source": [
    "smad3_symbol = 'SMAD3'\n",
    "smad3_mane_tx_id = 'NM_005902.4'\n",
    "smad3_mane_protein_id = 'NP_005893.1' # mothers against decapentaplegic homolog 3\n",
    "lds3_disease_id = \"OMIM:613795\"\n",
    "\n",
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.23\") as ps:\n",
    "    lds3_phenopackets = tuple(ps.iter_cohort_phenopackets(smad3_symbol))\n",
    "\n",
    "print(f\"{len(lds3_phenopackets)} LDS3 phenopackets\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS4\n",
    "[Loeys-Dietz syndrome-4 (LDS4)](https://omim.org/entry/614816) is caused by heterozygous mutation in the TGFB2 gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "36 LDS4 phenopackets\n"
     ]
    }
   ],
   "source": [
    "tgfb2_symbol = 'TGFB2'\n",
    "tgfb2_mane_tx_id = 'NM_003238.6'\n",
    "tgfb2_mane_protein_id = 'NP_003229.1' # transforming growth factor beta-2 proprotein isoform 2 preproprotein\n",
    "lds4_disease_id = \"OMIM:614816\"\n",
    "\n",
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.20\") as ps:\n",
    "    lds4_phenopackets = tuple(ps.iter_cohort_phenopackets(tgfb2_symbol))\n",
    "\n",
    "print(f\"{len(lds4_phenopackets)} LDS4 phenopackets\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS5\n",
    "[Loeys-Dietz syndrome-5 (LDS5)](https://omim.org/entry/615582) is caused by heterozygous mutation in the TGFB3 gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "75 LDS5 phenopackets\n"
     ]
    }
   ],
   "source": [
    "tgfb3_symbol = 'TGFB3'\n",
    "tgfb3_mane_tx_id = 'NM_003239.5'\n",
    "tgfb3_mane_protein_id = 'NP_003230.1' # transforming growth factor beta-3 proprotein isoform 1 preproprotein\"\n",
    "lds5_disease_id = \"OMIM:615582\"\n",
    "\n",
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.23\") as ps:\n",
    "    lds5_phenopackets = tuple(ps.iter_cohort_phenopackets(tgfb3_symbol))\n",
    "\n",
    "print(f\"{len(lds5_phenopackets)} LDS5 phenopackets\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS6\n",
    "[Loeys-Dietz syndrome-6 (LDS6)](https://omim.org/entry/619656) is caused by heterozygous mutation in the SMAD2 gene "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 LDS6 phenopackets\n"
     ]
    }
   ],
   "source": [
    "smad2_symbol = 'SMAD2'\n",
    "smad2_mane_tx_id = 'NM_005901.6'\n",
    "smad2_mane_protein_id = 'NP_005892.1' # mothers against decapentaplegic homolog 2 isoform 1\n",
    "\n",
    "lds6_disease_id = \"OMIM:619656\"\n",
    "\n",
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.20\") as ps:\n",
    "    lds6_phenopackets = tuple(ps.iter_cohort_phenopackets(smad2_symbol))\n",
    "\n",
    "print(f\"{len(lds6_phenopackets)} LDS6 phenopackets\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create cohort\n",
    "We create a cohort with phenopackets representing individuals with all six forms of LDS.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 100%|██████████| 270/270 [00:50<00:00,  5.37individuals/s]\n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "\n",
    "lds_phenopackets = list()\n",
    "lds_phenopackets.extend(lds1_phenopackets)\n",
    "lds_phenopackets.extend(lds2_phenopackets)\n",
    "lds_phenopackets.extend(lds3_phenopackets)\n",
    "lds_phenopackets.extend(lds4_phenopackets)\n",
    "lds_phenopackets.extend(lds5_phenopackets)\n",
    "lds_phenopackets.extend(lds6_phenopackets)\n",
    "\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "lds_cohort, validation = load_phenopackets(\n",
    "    phenopackets=lds_phenopackets, \n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "\n",
    "validation.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.pcats import configure_hpo_term_analysis\n",
    "from gpsea.analysis.clf import prepare_classifiers_for_terms_of_interest\n",
    "\n",
    "analysis = configure_hpo_term_analysis(hpo)\n",
    "\n",
    "pheno_clfs = prepare_classifiers_for_terms_of_interest(\n",
    "    cohort=lds_cohort,\n",
    "    hpo=hpo,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Phenotype testing report</h1>\n",
       "<div class=\"cards\">\n",
       "  <div class=\"card\">\n",
       "    <ul>\n",
       "      <li>Phenotype MTC filter: <em>HPO MTC filter</em></li>\n",
       "      <li>Multiple testing correction: <em>fdr_bh</em></li>\n",
       "    </ul>\n",
       "  </div>\n",
       "  <div class=\"card\">\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr>\n",
       "          <th>Code</th>\n",
       "          <th>Reason</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF01</td>\n",
       "          <td>Skipping term with maximum frequency that was less than threshold 0.4</td>\n",
       "          <td>131</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF08</td>\n",
       "          <td>Skipping general term</td>\n",
       "          <td>76</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF09</td>\n",
       "          <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>\n",
       "          <td>249</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "      <caption>\n",
       "        Performed statistical tests for 0 out of the total of 456 HPO terms.\n",
       "      </caption>\n",
       "    </table>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x16b41d810>"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.analysis.clf import diagnosis_classifier\n",
    "from gpsea.view import MtcStatsViewer\n",
    "\n",
    "\n",
    "lds_1_2_disease_clf = diagnosis_classifier(\n",
    "    diagnoses=(lds1_disease_id, lds2_disease_id),\n",
    "    labels=('LDS1', 'LDS2'),\n",
    ")\n",
    "lds_1_2_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=lds_cohort,\n",
    "    gt_clf=lds_1_2_disease_clf,\n",
    "    pheno_clfs=pheno_clfs,\n",
    ")\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(lds_1_2_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Diagnosis</th>\n",
       "      <th colspan=\"2\" halign=\"left\">OMIM:609192</th>\n",
       "      <th colspan=\"2\" halign=\"left\">OMIM:610168</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [(OMIM:609192, Count), (OMIM:609192, Percent), (OMIM:610168, Count), (OMIM:610168, Percent), (, p values)]\n",
       "Index: []"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "\n",
    "summarize_hpo_analysis(hpo=hpo, result=lds_1_2_result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS1 vs LDS3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Got 90 LDS1 and LDS3 phenopackets\n",
      "Individuals Processed: 100%|██████████| 90/90 [00:22<00:00,  4.09individuals/s] \n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "lds1_and_lds3_phenopackets = list()\n",
    "lds1_and_lds3_phenopackets.extend(lds1_phenopackets)\n",
    "lds1_and_lds3_phenopackets.extend(lds3_phenopackets)\n",
    "print(f\"Got {len(lds1_and_lds3_phenopackets)} LDS1 and LDS3 phenopackets\")\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "lds1_lds3_cohort, validation = load_phenopackets(\n",
    "    phenopackets=lds1_and_lds3_phenopackets, \n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "\n",
    "validation.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Phenotype testing report</h1>\n",
       "<div class=\"cards\">\n",
       "  <div class=\"card\">\n",
       "    <ul>\n",
       "      <li>Phenotype MTC filter: <em>HPO MTC filter</em></li>\n",
       "      <li>Multiple testing correction: <em>fdr_bh</em></li>\n",
       "    </ul>\n",
       "  </div>\n",
       "  <div class=\"card\">\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr>\n",
       "          <th>Code</th>\n",
       "          <th>Reason</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF01</td>\n",
       "          <td>Skipping term with maximum frequency that was less than threshold 0.4</td>\n",
       "          <td>207</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF08</td>\n",
       "          <td>Skipping general term</td>\n",
       "          <td>76</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF09</td>\n",
       "          <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>\n",
       "          <td>173</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "      <caption>\n",
       "        Performed statistical tests for 0 out of the total of 456 HPO terms.\n",
       "      </caption>\n",
       "    </table>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x16b23a9c0>"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lds1_lds3_clfs = prepare_classifiers_for_terms_of_interest(\n",
    "    cohort=lds1_lds3_cohort,\n",
    "    hpo=hpo,\n",
    ")\n",
    "\n",
    "lds_1_3_disease_clf = diagnosis_classifier (\n",
    "    diagnoses=(lds1_disease_id, lds3_disease_id),\n",
    "    labels=('LDS1', 'LDS3'),\n",
    ")\n",
    "\n",
    "lds1_3_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=lds_cohort,\n",
    "    gt_clf=lds_1_3_disease_clf,\n",
    "    pheno_clfs=pheno_clfs,\n",
    ")\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(lds1_3_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Diagnosis</th>\n",
       "      <th colspan=\"2\" halign=\"left\">OMIM:609192</th>\n",
       "      <th colspan=\"2\" halign=\"left\">OMIM:613795</th>\n",
       "      <th colspan=\"2\" halign=\"left\"></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Osteoarthritis [HP:0002758]</th>\n",
       "      <td>0/11</td>\n",
       "      <td>0%</td>\n",
       "      <td>26/38</td>\n",
       "      <td>68%</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>0.000046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Scoliosis [HP:0002650]</th>\n",
       "      <td>18/21</td>\n",
       "      <td>86%</td>\n",
       "      <td>20/43</td>\n",
       "      <td>47%</td>\n",
       "      <td>0.023527</td>\n",
       "      <td>0.003015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aortic aneurysm [HP:0004942]</th>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>26/48</td>\n",
       "      <td>54%</td>\n",
       "      <td>0.023527</td>\n",
       "      <td>0.004327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypertelorism [HP:0000316]</th>\n",
       "      <td>15/19</td>\n",
       "      <td>79%</td>\n",
       "      <td>13/35</td>\n",
       "      <td>37%</td>\n",
       "      <td>0.023527</td>\n",
       "      <td>0.004481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Joint hypermobility [HP:0001382]</th>\n",
       "      <td>12/19</td>\n",
       "      <td>63%</td>\n",
       "      <td>12/36</td>\n",
       "      <td>33%</td>\n",
       "      <td>0.197692</td>\n",
       "      <td>0.047070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>High palate [HP:0000218]</th>\n",
       "      <td>12/16</td>\n",
       "      <td>75%</td>\n",
       "      <td>12/20</td>\n",
       "      <td>60%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.481498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inguinal hernia [HP:0000023]</th>\n",
       "      <td>6/14</td>\n",
       "      <td>43%</td>\n",
       "      <td>12/39</td>\n",
       "      <td>31%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.514672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Arterial tortuosity [HP:0005116]</th>\n",
       "      <td>9/17</td>\n",
       "      <td>53%</td>\n",
       "      <td>11/26</td>\n",
       "      <td>42%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.545009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Disproportionate tall stature [HP:0001519]</th>\n",
       "      <td>9/17</td>\n",
       "      <td>53%</td>\n",
       "      <td>8/19</td>\n",
       "      <td>42%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.738795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal oral cavity morphology [HP:0000163]</th>\n",
       "      <td>17/17</td>\n",
       "      <td>100%</td>\n",
       "      <td>24/24</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal palate morphology [HP:0000174]</th>\n",
       "      <td>17/17</td>\n",
       "      <td>100%</td>\n",
       "      <td>24/24</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal axial skeleton morphology [HP:0009121]</th>\n",
       "      <td>21/21</td>\n",
       "      <td>100%</td>\n",
       "      <td>33/33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal curvature of the vertebral column [HP:0010674]</th>\n",
       "      <td>18/18</td>\n",
       "      <td>100%</td>\n",
       "      <td>20/20</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormality of the vertebral column [HP:0000925]</th>\n",
       "      <td>19/19</td>\n",
       "      <td>100%</td>\n",
       "      <td>31/31</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Arachnodactyly [HP:0001166]</th>\n",
       "      <td>8/18</td>\n",
       "      <td>44%</td>\n",
       "      <td>7/18</td>\n",
       "      <td>39%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vascular dilatation [HP:0002617]</th>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>26/26</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal blood vessel morphology [HP:0033353]</th>\n",
       "      <td>15/15</td>\n",
       "      <td>100%</td>\n",
       "      <td>32/32</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal aortic morphology [HP:0001679]</th>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>26/26</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal morphology of the great vessels [HP:0030962]</th>\n",
       "      <td>12/12</td>\n",
       "      <td>100%</td>\n",
       "      <td>26/26</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal systemic arterial morphology [HP:0011004]</th>\n",
       "      <td>14/14</td>\n",
       "      <td>100%</td>\n",
       "      <td>29/29</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Soft skin [HP:0000977]</th>\n",
       "      <td>3/5</td>\n",
       "      <td>60%</td>\n",
       "      <td>23/37</td>\n",
       "      <td>62%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Diagnosis                                          OMIM:609192          \\\n",
       "                                                         Count Percent   \n",
       "Osteoarthritis [HP:0002758]                               0/11      0%   \n",
       "Scoliosis [HP:0002650]                                   18/21     86%   \n",
       "Aortic aneurysm [HP:0004942]                             11/11    100%   \n",
       "Hypertelorism [HP:0000316]                               15/19     79%   \n",
       "Joint hypermobility [HP:0001382]                         12/19     63%   \n",
       "High palate [HP:0000218]                                 12/16     75%   \n",
       "Inguinal hernia [HP:0000023]                              6/14     43%   \n",
       "Arterial tortuosity [HP:0005116]                          9/17     53%   \n",
       "Disproportionate tall stature [HP:0001519]                9/17     53%   \n",
       "Abnormal oral cavity morphology [HP:0000163]             17/17    100%   \n",
       "Abnormal palate morphology [HP:0000174]                  17/17    100%   \n",
       "Abnormal axial skeleton morphology [HP:0009121]          21/21    100%   \n",
       "Abnormal curvature of the vertebral column [HP:...       18/18    100%   \n",
       "Abnormality of the vertebral column [HP:0000925]         19/19    100%   \n",
       "Arachnodactyly [HP:0001166]                               8/18     44%   \n",
       "Vascular dilatation [HP:0002617]                         11/11    100%   \n",
       "Abnormal blood vessel morphology [HP:0033353]            15/15    100%   \n",
       "Abnormal aortic morphology [HP:0001679]                  11/11    100%   \n",
       "Abnormal morphology of the great vessels [HP:00...       12/12    100%   \n",
       "Abnormal systemic arterial morphology [HP:0011004]       14/14    100%   \n",
       "Soft skin [HP:0000977]                                     3/5     60%   \n",
       "\n",
       "Diagnosis                                          OMIM:613795          \\\n",
       "                                                         Count Percent   \n",
       "Osteoarthritis [HP:0002758]                              26/38     68%   \n",
       "Scoliosis [HP:0002650]                                   20/43     47%   \n",
       "Aortic aneurysm [HP:0004942]                             26/48     54%   \n",
       "Hypertelorism [HP:0000316]                               13/35     37%   \n",
       "Joint hypermobility [HP:0001382]                         12/36     33%   \n",
       "High palate [HP:0000218]                                 12/20     60%   \n",
       "Inguinal hernia [HP:0000023]                             12/39     31%   \n",
       "Arterial tortuosity [HP:0005116]                         11/26     42%   \n",
       "Disproportionate tall stature [HP:0001519]                8/19     42%   \n",
       "Abnormal oral cavity morphology [HP:0000163]             24/24    100%   \n",
       "Abnormal palate morphology [HP:0000174]                  24/24    100%   \n",
       "Abnormal axial skeleton morphology [HP:0009121]          33/33    100%   \n",
       "Abnormal curvature of the vertebral column [HP:...       20/20    100%   \n",
       "Abnormality of the vertebral column [HP:0000925]         31/31    100%   \n",
       "Arachnodactyly [HP:0001166]                               7/18     39%   \n",
       "Vascular dilatation [HP:0002617]                         26/26    100%   \n",
       "Abnormal blood vessel morphology [HP:0033353]            32/32    100%   \n",
       "Abnormal aortic morphology [HP:0001679]                  26/26    100%   \n",
       "Abnormal morphology of the great vessels [HP:00...       26/26    100%   \n",
       "Abnormal systemic arterial morphology [HP:0011004]       29/29    100%   \n",
       "Soft skin [HP:0000977]                                   23/37     62%   \n",
       "\n",
       "Diagnosis                                                              \\\n",
       "                                                   Corrected p values   \n",
       "Osteoarthritis [HP:0002758]                                  0.000975   \n",
       "Scoliosis [HP:0002650]                                       0.023527   \n",
       "Aortic aneurysm [HP:0004942]                                 0.023527   \n",
       "Hypertelorism [HP:0000316]                                   0.023527   \n",
       "Joint hypermobility [HP:0001382]                             0.197692   \n",
       "High palate [HP:0000218]                                     1.000000   \n",
       "Inguinal hernia [HP:0000023]                                 1.000000   \n",
       "Arterial tortuosity [HP:0005116]                             1.000000   \n",
       "Disproportionate tall stature [HP:0001519]                   1.000000   \n",
       "Abnormal oral cavity morphology [HP:0000163]                 1.000000   \n",
       "Abnormal palate morphology [HP:0000174]                      1.000000   \n",
       "Abnormal axial skeleton morphology [HP:0009121]              1.000000   \n",
       "Abnormal curvature of the vertebral column [HP:...           1.000000   \n",
       "Abnormality of the vertebral column [HP:0000925]             1.000000   \n",
       "Arachnodactyly [HP:0001166]                                  1.000000   \n",
       "Vascular dilatation [HP:0002617]                             1.000000   \n",
       "Abnormal blood vessel morphology [HP:0033353]                1.000000   \n",
       "Abnormal aortic morphology [HP:0001679]                      1.000000   \n",
       "Abnormal morphology of the great vessels [HP:00...           1.000000   \n",
       "Abnormal systemic arterial morphology [HP:0011004]           1.000000   \n",
       "Soft skin [HP:0000977]                                       1.000000   \n",
       "\n",
       "Diagnosis                                                     \n",
       "                                                    p values  \n",
       "Osteoarthritis [HP:0002758]                         0.000046  \n",
       "Scoliosis [HP:0002650]                              0.003015  \n",
       "Aortic aneurysm [HP:0004942]                        0.004327  \n",
       "Hypertelorism [HP:0000316]                          0.004481  \n",
       "Joint hypermobility [HP:0001382]                    0.047070  \n",
       "High palate [HP:0000218]                            0.481498  \n",
       "Inguinal hernia [HP:0000023]                        0.514672  \n",
       "Arterial tortuosity [HP:0005116]                    0.545009  \n",
       "Disproportionate tall stature [HP:0001519]          0.738795  \n",
       "Abnormal oral cavity morphology [HP:0000163]        1.000000  \n",
       "Abnormal palate morphology [HP:0000174]             1.000000  \n",
       "Abnormal axial skeleton morphology [HP:0009121]     1.000000  \n",
       "Abnormal curvature of the vertebral column [HP:...  1.000000  \n",
       "Abnormality of the vertebral column [HP:0000925]    1.000000  \n",
       "Arachnodactyly [HP:0001166]                         1.000000  \n",
       "Vascular dilatation [HP:0002617]                    1.000000  \n",
       "Abnormal blood vessel morphology [HP:0033353]       1.000000  \n",
       "Abnormal aortic morphology [HP:0001679]             1.000000  \n",
       "Abnormal morphology of the great vessels [HP:00...  1.000000  \n",
       "Abnormal systemic arterial morphology [HP:0011004]  1.000000  \n",
       "Soft skin [HP:0000977]                              1.000000  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summarize_hpo_analysis(hpo=hpo, result=lds1_3_result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS3 vs LDS6\n",
    "(SMAD3 and SMAD2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Got 65 LDS3 and LDS6 phenopackets\n",
      "Individuals Processed: 100%|██████████| 65/65 [00:12<00:00,  5.25individuals/s]\n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "lds3_and_lds6_phenopackets = list()\n",
    "lds3_and_lds6_phenopackets.extend(lds3_phenopackets)\n",
    "lds3_and_lds6_phenopackets.extend(lds6_phenopackets)\n",
    "print(f\"Got {len(lds3_and_lds6_phenopackets)} LDS3 and LDS6 phenopackets\")\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "lds3_lds6_cohort, validation = load_phenopackets(\n",
    "    phenopackets=lds3_and_lds6_phenopackets, \n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "\n",
    "validation.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Phenotype testing report</h1>\n",
       "<div class=\"cards\">\n",
       "  <div class=\"card\">\n",
       "    <ul>\n",
       "      <li>Phenotype MTC filter: <em>HPO MTC filter</em></li>\n",
       "      <li>Multiple testing correction: <em>fdr_bh</em></li>\n",
       "    </ul>\n",
       "  </div>\n",
       "  <div class=\"card\">\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr>\n",
       "          <th>Code</th>\n",
       "          <th>Reason</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF01</td>\n",
       "          <td>Skipping term with maximum frequency that was less than threshold 0.4</td>\n",
       "          <td>21</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF03</td>\n",
       "          <td>Skipping term because of a child term with the same individual counts</td>\n",
       "          <td>4</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF08</td>\n",
       "          <td>Skipping general term</td>\n",
       "          <td>49</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF09</td>\n",
       "          <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>\n",
       "          <td>71</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "      <caption>\n",
       "        Performed statistical tests for 21 out of the total of 166 HPO terms.\n",
       "      </caption>\n",
       "    </table>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x125373890>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lds3_lds6_pheno_clfs = prepare_classifiers_for_terms_of_interest(\n",
    "    cohort=lds3_lds6_cohort,\n",
    "    hpo=hpo,\n",
    ")\n",
    "\n",
    "lds_3_6_disease_clf = diagnosis_classifier(\n",
    "    diagnoses=(lds3_disease_id, lds6_disease_id),\n",
    "    labels=('LDS3', 'LDS6'),\n",
    ")\n",
    "lds3_6_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=lds3_lds6_cohort,\n",
    "    gt_clf=lds_3_6_disease_clf,\n",
    "    pheno_clfs=lds3_lds6_pheno_clfs,\n",
    ")\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(lds3_6_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Diagnosis</th>\n",
       "      <th colspan=\"2\" halign=\"left\">OMIM:613795</th>\n",
       "      <th colspan=\"2\" halign=\"left\">OMIM:619656</th>\n",
       "      <th colspan=\"2\" halign=\"left\"></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Thoracic aortic aneurysm [HP:0012727]</th>\n",
       "      <td>0/22</td>\n",
       "      <td>0%</td>\n",
       "      <td>10/15</td>\n",
       "      <td>67%</td>\n",
       "      <td>0.000181</td>\n",
       "      <td>0.000009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aortic aneurysm [HP:0004942]</th>\n",
       "      <td>26/48</td>\n",
       "      <td>54%</td>\n",
       "      <td>10/10</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.095422</td>\n",
       "      <td>0.009088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Soft skin [HP:0000977]</th>\n",
       "      <td>23/37</td>\n",
       "      <td>62%</td>\n",
       "      <td>0/4</td>\n",
       "      <td>0%</td>\n",
       "      <td>0.211514</td>\n",
       "      <td>0.030216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Varicose veins [HP:0002619]</th>\n",
       "      <td>14/22</td>\n",
       "      <td>64%</td>\n",
       "      <td>4/12</td>\n",
       "      <td>33%</td>\n",
       "      <td>0.737449</td>\n",
       "      <td>0.151415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>High palate [HP:0000218]</th>\n",
       "      <td>12/20</td>\n",
       "      <td>60%</td>\n",
       "      <td>5/15</td>\n",
       "      <td>33%</td>\n",
       "      <td>0.737449</td>\n",
       "      <td>0.175583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Arterial tortuosity [HP:0005116]</th>\n",
       "      <td>11/26</td>\n",
       "      <td>42%</td>\n",
       "      <td>1/1</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.444444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inguinal hernia [HP:0000023]</th>\n",
       "      <td>12/39</td>\n",
       "      <td>31%</td>\n",
       "      <td>5/11</td>\n",
       "      <td>45%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.475054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Umbilical hernia [HP:0001537]</th>\n",
       "      <td>12/39</td>\n",
       "      <td>31%</td>\n",
       "      <td>2/4</td>\n",
       "      <td>50%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.585479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Osteoarthritis [HP:0002758]</th>\n",
       "      <td>26/38</td>\n",
       "      <td>68%</td>\n",
       "      <td>2/2</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Arthritis [HP:0001369]</th>\n",
       "      <td>26/26</td>\n",
       "      <td>100%</td>\n",
       "      <td>2/2</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal joint morphology [HP:0001367]</th>\n",
       "      <td>26/26</td>\n",
       "      <td>100%</td>\n",
       "      <td>2/2</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormality of the vertebral column [HP:0000925]</th>\n",
       "      <td>31/31</td>\n",
       "      <td>100%</td>\n",
       "      <td>8/8</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal axial skeleton morphology [HP:0009121]</th>\n",
       "      <td>33/33</td>\n",
       "      <td>100%</td>\n",
       "      <td>9/9</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vascular dilatation [HP:0002617]</th>\n",
       "      <td>26/26</td>\n",
       "      <td>100%</td>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal blood vessel morphology [HP:0033353]</th>\n",
       "      <td>32/32</td>\n",
       "      <td>100%</td>\n",
       "      <td>15/15</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal aortic morphology [HP:0001679]</th>\n",
       "      <td>26/26</td>\n",
       "      <td>100%</td>\n",
       "      <td>10/10</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal systemic arterial morphology [HP:0011004]</th>\n",
       "      <td>29/29</td>\n",
       "      <td>100%</td>\n",
       "      <td>14/14</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Scoliosis [HP:0002650]</th>\n",
       "      <td>20/43</td>\n",
       "      <td>47%</td>\n",
       "      <td>5/12</td>\n",
       "      <td>42%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal palate morphology [HP:0000174]</th>\n",
       "      <td>24/24</td>\n",
       "      <td>100%</td>\n",
       "      <td>6/6</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal elasticity of skin [HP:0010647]</th>\n",
       "      <td>25/25</td>\n",
       "      <td>100%</td>\n",
       "      <td>1/1</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Arachnodactyly [HP:0001166]</th>\n",
       "      <td>7/18</td>\n",
       "      <td>39%</td>\n",
       "      <td>6/14</td>\n",
       "      <td>43%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Diagnosis                                          OMIM:613795          \\\n",
       "                                                         Count Percent   \n",
       "Thoracic aortic aneurysm [HP:0012727]                     0/22      0%   \n",
       "Aortic aneurysm [HP:0004942]                             26/48     54%   \n",
       "Soft skin [HP:0000977]                                   23/37     62%   \n",
       "Varicose veins [HP:0002619]                              14/22     64%   \n",
       "High palate [HP:0000218]                                 12/20     60%   \n",
       "Arterial tortuosity [HP:0005116]                         11/26     42%   \n",
       "Inguinal hernia [HP:0000023]                             12/39     31%   \n",
       "Umbilical hernia [HP:0001537]                            12/39     31%   \n",
       "Osteoarthritis [HP:0002758]                              26/38     68%   \n",
       "Arthritis [HP:0001369]                                   26/26    100%   \n",
       "Abnormal joint morphology [HP:0001367]                   26/26    100%   \n",
       "Abnormality of the vertebral column [HP:0000925]         31/31    100%   \n",
       "Abnormal axial skeleton morphology [HP:0009121]          33/33    100%   \n",
       "Vascular dilatation [HP:0002617]                         26/26    100%   \n",
       "Abnormal blood vessel morphology [HP:0033353]            32/32    100%   \n",
       "Abnormal aortic morphology [HP:0001679]                  26/26    100%   \n",
       "Abnormal systemic arterial morphology [HP:0011004]       29/29    100%   \n",
       "Scoliosis [HP:0002650]                                   20/43     47%   \n",
       "Abnormal palate morphology [HP:0000174]                  24/24    100%   \n",
       "Abnormal elasticity of skin [HP:0010647]                 25/25    100%   \n",
       "Arachnodactyly [HP:0001166]                               7/18     39%   \n",
       "\n",
       "Diagnosis                                          OMIM:619656          \\\n",
       "                                                         Count Percent   \n",
       "Thoracic aortic aneurysm [HP:0012727]                    10/15     67%   \n",
       "Aortic aneurysm [HP:0004942]                             10/10    100%   \n",
       "Soft skin [HP:0000977]                                     0/4      0%   \n",
       "Varicose veins [HP:0002619]                               4/12     33%   \n",
       "High palate [HP:0000218]                                  5/15     33%   \n",
       "Arterial tortuosity [HP:0005116]                           1/1    100%   \n",
       "Inguinal hernia [HP:0000023]                              5/11     45%   \n",
       "Umbilical hernia [HP:0001537]                              2/4     50%   \n",
       "Osteoarthritis [HP:0002758]                                2/2    100%   \n",
       "Arthritis [HP:0001369]                                     2/2    100%   \n",
       "Abnormal joint morphology [HP:0001367]                     2/2    100%   \n",
       "Abnormality of the vertebral column [HP:0000925]           8/8    100%   \n",
       "Abnormal axial skeleton morphology [HP:0009121]            9/9    100%   \n",
       "Vascular dilatation [HP:0002617]                         11/11    100%   \n",
       "Abnormal blood vessel morphology [HP:0033353]            15/15    100%   \n",
       "Abnormal aortic morphology [HP:0001679]                  10/10    100%   \n",
       "Abnormal systemic arterial morphology [HP:0011004]       14/14    100%   \n",
       "Scoliosis [HP:0002650]                                    5/12     42%   \n",
       "Abnormal palate morphology [HP:0000174]                    6/6    100%   \n",
       "Abnormal elasticity of skin [HP:0010647]                   1/1    100%   \n",
       "Arachnodactyly [HP:0001166]                               6/14     43%   \n",
       "\n",
       "Diagnosis                                                              \\\n",
       "                                                   Corrected p values   \n",
       "Thoracic aortic aneurysm [HP:0012727]                        0.000181   \n",
       "Aortic aneurysm [HP:0004942]                                 0.095422   \n",
       "Soft skin [HP:0000977]                                       0.211514   \n",
       "Varicose veins [HP:0002619]                                  0.737449   \n",
       "High palate [HP:0000218]                                     0.737449   \n",
       "Arterial tortuosity [HP:0005116]                             1.000000   \n",
       "Inguinal hernia [HP:0000023]                                 1.000000   \n",
       "Umbilical hernia [HP:0001537]                                1.000000   \n",
       "Osteoarthritis [HP:0002758]                                  1.000000   \n",
       "Arthritis [HP:0001369]                                       1.000000   \n",
       "Abnormal joint morphology [HP:0001367]                       1.000000   \n",
       "Abnormality of the vertebral column [HP:0000925]             1.000000   \n",
       "Abnormal axial skeleton morphology [HP:0009121]              1.000000   \n",
       "Vascular dilatation [HP:0002617]                             1.000000   \n",
       "Abnormal blood vessel morphology [HP:0033353]                1.000000   \n",
       "Abnormal aortic morphology [HP:0001679]                      1.000000   \n",
       "Abnormal systemic arterial morphology [HP:0011004]           1.000000   \n",
       "Scoliosis [HP:0002650]                                       1.000000   \n",
       "Abnormal palate morphology [HP:0000174]                      1.000000   \n",
       "Abnormal elasticity of skin [HP:0010647]                     1.000000   \n",
       "Arachnodactyly [HP:0001166]                                  1.000000   \n",
       "\n",
       "Diagnosis                                                     \n",
       "                                                    p values  \n",
       "Thoracic aortic aneurysm [HP:0012727]               0.000009  \n",
       "Aortic aneurysm [HP:0004942]                        0.009088  \n",
       "Soft skin [HP:0000977]                              0.030216  \n",
       "Varicose veins [HP:0002619]                         0.151415  \n",
       "High palate [HP:0000218]                            0.175583  \n",
       "Arterial tortuosity [HP:0005116]                    0.444444  \n",
       "Inguinal hernia [HP:0000023]                        0.475054  \n",
       "Umbilical hernia [HP:0001537]                       0.585479  \n",
       "Osteoarthritis [HP:0002758]                         1.000000  \n",
       "Arthritis [HP:0001369]                              1.000000  \n",
       "Abnormal joint morphology [HP:0001367]              1.000000  \n",
       "Abnormality of the vertebral column [HP:0000925]    1.000000  \n",
       "Abnormal axial skeleton morphology [HP:0009121]     1.000000  \n",
       "Vascular dilatation [HP:0002617]                    1.000000  \n",
       "Abnormal blood vessel morphology [HP:0033353]       1.000000  \n",
       "Abnormal aortic morphology [HP:0001679]             1.000000  \n",
       "Abnormal systemic arterial morphology [HP:0011004]  1.000000  \n",
       "Scoliosis [HP:0002650]                              1.000000  \n",
       "Abnormal palate morphology [HP:0000174]             1.000000  \n",
       "Abnormal elasticity of skin [HP:0010647]            1.000000  \n",
       "Arachnodactyly [HP:0001166]                         1.000000  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summarize_hpo_analysis(hpo=hpo, result=lds3_6_result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LDS2 vs LDS6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Got 57 LDS1 and LDS6 phenopackets\n",
      "Individuals Processed: 100%|██████████| 57/57 [00:00<00:00, 377.55individuals/s]\n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "lds1_and_lds6_phenopackets = list()\n",
    "lds1_and_lds6_phenopackets.extend(lds1_phenopackets)\n",
    "lds1_and_lds6_phenopackets.extend(lds6_phenopackets)\n",
    "print(f\"Got {len(lds1_and_lds6_phenopackets)} LDS1 and LDS6 phenopackets\")\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "lds1_lds6_cohort, validation = load_phenopackets(\n",
    "    phenopackets=lds1_and_lds6_phenopackets, \n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "\n",
    "validation.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Phenotype testing report</h1>\n",
       "<div class=\"cards\">\n",
       "  <div class=\"card\">\n",
       "    <ul>\n",
       "      <li>Phenotype MTC filter: <em>HPO MTC filter</em></li>\n",
       "      <li>Multiple testing correction: <em>fdr_bh</em></li>\n",
       "    </ul>\n",
       "  </div>\n",
       "  <div class=\"card\">\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr>\n",
       "          <th>Code</th>\n",
       "          <th>Reason</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF01</td>\n",
       "          <td>Skipping term with maximum frequency that was less than threshold 0.4</td>\n",
       "          <td>46</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF03</td>\n",
       "          <td>Skipping term because of a child term with the same individual counts</td>\n",
       "          <td>2</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF08</td>\n",
       "          <td>Skipping general term</td>\n",
       "          <td>63</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF09</td>\n",
       "          <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>\n",
       "          <td>157</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "      <caption>\n",
       "        Performed statistical tests for 16 out of the total of 284 HPO terms.\n",
       "      </caption>\n",
       "    </table>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x16b41d220>"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lds1_lds6_pheno_clfs = prepare_classifiers_for_terms_of_interest(\n",
    "    cohort=lds1_lds6_cohort,\n",
    "    hpo=hpo,\n",
    ")\n",
    "\n",
    "lds_1_6_disease_clf = diagnosis_classifier (\n",
    "    diagnoses=(lds1_disease_id, lds6_disease_id),\n",
    "    labels=('LDS1', 'LDS6'),\n",
    ")\n",
    "\n",
    "lds1_6_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=lds1_lds6_cohort,\n",
    "    gt_clf=lds_1_6_disease_clf,\n",
    "    pheno_clfs=lds1_lds6_pheno_clfs,\n",
    ")\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(lds1_6_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Diagnosis</th>\n",
       "      <th colspan=\"2\" halign=\"left\">OMIM:609192</th>\n",
       "      <th colspan=\"2\" halign=\"left\">OMIM:619656</th>\n",
       "      <th colspan=\"2\" halign=\"left\"></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Dolichocephaly [HP:0000268]</th>\n",
       "      <td>12/13</td>\n",
       "      <td>92%</td>\n",
       "      <td>3/11</td>\n",
       "      <td>27%</td>\n",
       "      <td>0.035671</td>\n",
       "      <td>0.002229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypertelorism [HP:0000316]</th>\n",
       "      <td>15/19</td>\n",
       "      <td>79%</td>\n",
       "      <td>4/14</td>\n",
       "      <td>29%</td>\n",
       "      <td>0.045333</td>\n",
       "      <td>0.005667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Scoliosis [HP:0002650]</th>\n",
       "      <td>18/21</td>\n",
       "      <td>86%</td>\n",
       "      <td>5/12</td>\n",
       "      <td>42%</td>\n",
       "      <td>0.087277</td>\n",
       "      <td>0.016364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>High palate [HP:0000218]</th>\n",
       "      <td>12/16</td>\n",
       "      <td>75%</td>\n",
       "      <td>5/15</td>\n",
       "      <td>33%</td>\n",
       "      <td>0.112478</td>\n",
       "      <td>0.031952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pes planus [HP:0001763]</th>\n",
       "      <td>11/14</td>\n",
       "      <td>79%</td>\n",
       "      <td>3/10</td>\n",
       "      <td>30%</td>\n",
       "      <td>0.112478</td>\n",
       "      <td>0.035149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Thoracic aortic aneurysm [HP:0012727]</th>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>10/15</td>\n",
       "      <td>67%</td>\n",
       "      <td>0.140468</td>\n",
       "      <td>0.052676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Joint hypermobility [HP:0001382]</th>\n",
       "      <td>12/19</td>\n",
       "      <td>63%</td>\n",
       "      <td>1/5</td>\n",
       "      <td>20%</td>\n",
       "      <td>0.324756</td>\n",
       "      <td>0.142081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal oral cavity morphology [HP:0000163]</th>\n",
       "      <td>17/17</td>\n",
       "      <td>100%</td>\n",
       "      <td>6/6</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal palate morphology [HP:0000174]</th>\n",
       "      <td>17/17</td>\n",
       "      <td>100%</td>\n",
       "      <td>6/6</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal axial skeleton morphology [HP:0009121]</th>\n",
       "      <td>21/21</td>\n",
       "      <td>100%</td>\n",
       "      <td>9/9</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal curvature of the vertebral column [HP:0010674]</th>\n",
       "      <td>18/18</td>\n",
       "      <td>100%</td>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormality of the vertebral column [HP:0000925]</th>\n",
       "      <td>19/19</td>\n",
       "      <td>100%</td>\n",
       "      <td>8/8</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Arachnodactyly [HP:0001166]</th>\n",
       "      <td>8/18</td>\n",
       "      <td>44%</td>\n",
       "      <td>6/14</td>\n",
       "      <td>43%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal blood vessel morphology [HP:0033353]</th>\n",
       "      <td>15/15</td>\n",
       "      <td>100%</td>\n",
       "      <td>15/15</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal systemic arterial morphology [HP:0011004]</th>\n",
       "      <td>14/14</td>\n",
       "      <td>100%</td>\n",
       "      <td>14/14</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inguinal hernia [HP:0000023]</th>\n",
       "      <td>6/14</td>\n",
       "      <td>43%</td>\n",
       "      <td>5/11</td>\n",
       "      <td>45%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Diagnosis                                          OMIM:609192          \\\n",
       "                                                         Count Percent   \n",
       "Dolichocephaly [HP:0000268]                              12/13     92%   \n",
       "Hypertelorism [HP:0000316]                               15/19     79%   \n",
       "Scoliosis [HP:0002650]                                   18/21     86%   \n",
       "High palate [HP:0000218]                                 12/16     75%   \n",
       "Pes planus [HP:0001763]                                  11/14     79%   \n",
       "Thoracic aortic aneurysm [HP:0012727]                    11/11    100%   \n",
       "Joint hypermobility [HP:0001382]                         12/19     63%   \n",
       "Abnormal oral cavity morphology [HP:0000163]             17/17    100%   \n",
       "Abnormal palate morphology [HP:0000174]                  17/17    100%   \n",
       "Abnormal axial skeleton morphology [HP:0009121]          21/21    100%   \n",
       "Abnormal curvature of the vertebral column [HP:...       18/18    100%   \n",
       "Abnormality of the vertebral column [HP:0000925]         19/19    100%   \n",
       "Arachnodactyly [HP:0001166]                               8/18     44%   \n",
       "Abnormal blood vessel morphology [HP:0033353]            15/15    100%   \n",
       "Abnormal systemic arterial morphology [HP:0011004]       14/14    100%   \n",
       "Inguinal hernia [HP:0000023]                              6/14     43%   \n",
       "\n",
       "Diagnosis                                          OMIM:619656          \\\n",
       "                                                         Count Percent   \n",
       "Dolichocephaly [HP:0000268]                               3/11     27%   \n",
       "Hypertelorism [HP:0000316]                                4/14     29%   \n",
       "Scoliosis [HP:0002650]                                    5/12     42%   \n",
       "High palate [HP:0000218]                                  5/15     33%   \n",
       "Pes planus [HP:0001763]                                   3/10     30%   \n",
       "Thoracic aortic aneurysm [HP:0012727]                    10/15     67%   \n",
       "Joint hypermobility [HP:0001382]                           1/5     20%   \n",
       "Abnormal oral cavity morphology [HP:0000163]               6/6    100%   \n",
       "Abnormal palate morphology [HP:0000174]                    6/6    100%   \n",
       "Abnormal axial skeleton morphology [HP:0009121]            9/9    100%   \n",
       "Abnormal curvature of the vertebral column [HP:...         5/5    100%   \n",
       "Abnormality of the vertebral column [HP:0000925]           8/8    100%   \n",
       "Arachnodactyly [HP:0001166]                               6/14     43%   \n",
       "Abnormal blood vessel morphology [HP:0033353]            15/15    100%   \n",
       "Abnormal systemic arterial morphology [HP:0011004]       14/14    100%   \n",
       "Inguinal hernia [HP:0000023]                              5/11     45%   \n",
       "\n",
       "Diagnosis                                                              \\\n",
       "                                                   Corrected p values   \n",
       "Dolichocephaly [HP:0000268]                                  0.035671   \n",
       "Hypertelorism [HP:0000316]                                   0.045333   \n",
       "Scoliosis [HP:0002650]                                       0.087277   \n",
       "High palate [HP:0000218]                                     0.112478   \n",
       "Pes planus [HP:0001763]                                      0.112478   \n",
       "Thoracic aortic aneurysm [HP:0012727]                        0.140468   \n",
       "Joint hypermobility [HP:0001382]                             0.324756   \n",
       "Abnormal oral cavity morphology [HP:0000163]                 1.000000   \n",
       "Abnormal palate morphology [HP:0000174]                      1.000000   \n",
       "Abnormal axial skeleton morphology [HP:0009121]              1.000000   \n",
       "Abnormal curvature of the vertebral column [HP:...           1.000000   \n",
       "Abnormality of the vertebral column [HP:0000925]             1.000000   \n",
       "Arachnodactyly [HP:0001166]                                  1.000000   \n",
       "Abnormal blood vessel morphology [HP:0033353]                1.000000   \n",
       "Abnormal systemic arterial morphology [HP:0011004]           1.000000   \n",
       "Inguinal hernia [HP:0000023]                                 1.000000   \n",
       "\n",
       "Diagnosis                                                     \n",
       "                                                    p values  \n",
       "Dolichocephaly [HP:0000268]                         0.002229  \n",
       "Hypertelorism [HP:0000316]                          0.005667  \n",
       "Scoliosis [HP:0002650]                              0.016364  \n",
       "High palate [HP:0000218]                            0.031952  \n",
       "Pes planus [HP:0001763]                             0.035149  \n",
       "Thoracic aortic aneurysm [HP:0012727]               0.052676  \n",
       "Joint hypermobility [HP:0001382]                    0.142081  \n",
       "Abnormal oral cavity morphology [HP:0000163]        1.000000  \n",
       "Abnormal palate morphology [HP:0000174]             1.000000  \n",
       "Abnormal axial skeleton morphology [HP:0009121]     1.000000  \n",
       "Abnormal curvature of the vertebral column [HP:...  1.000000  \n",
       "Abnormality of the vertebral column [HP:0000925]    1.000000  \n",
       "Arachnodactyly [HP:0001166]                         1.000000  \n",
       "Abnormal blood vessel morphology [HP:0033353]       1.000000  \n",
       "Abnormal systemic arterial morphology [HP:0011004]  1.000000  \n",
       "Inguinal hernia [HP:0000023]                        1.000000  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summarize_hpo_analysis(hpo=hpo, result=lds1_6_result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpseacs.report import GpseaAnalysisReport, GPAnalysisResultSummary\n",
    "\n",
    "f_results = (\n",
    "  GPAnalysisResultSummary.from_multi( result=lds_1_2_result,  ),\n",
    ")\n",
    "\n",
    "\n",
    "caption = \"\"\".\"\"\"\n",
    "report = GpseaAnalysisReport(name=\"Kabuki\", \n",
    "                             cohort=kabuki_1_and_2_cohort, \n",
    "                             fet_results=f_results,\n",
    "                             gene_symbol=\"n/a\",\n",
    "                             mane_tx_id=\"n/a\",\n",
    "                             mane_protein_id=\"n/a\",\n",
    "                             caption=caption)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
