{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# POLR1A\n",
    "\n",
    "POLR1A is the largest subunit of RNA polymerase I (Pol I), which catalyzes DNA-dependent synthesis of ribosomal RNA ([OMIM:616404](https://omim.org/entry/616404)).\n",
    "\n",
    "Pathogeniv POLR1A variants are associated with\n",
    "\n",
    "- [Acrofacial dysostosis, Cincinnati type, OMIM:616462](https://omim.org/entry/616462), and\n",
    "- [Leukodystrophy, hypomyelinating, 27, OMIM:620675](https://omim.org/entry/620675).\n",
    "\t\n",
    "\n",
    "\n",
    "[Acrofacial dysostosis, Cincinnati type](https://omim.org/entry/616462) is characterized by craniofacial anomalies reminiscent of Treacher Collins syndrome, as well as numerous additional phenotypes including neurodevelopmental abnormalities and structural cardiac defects, in combination with highly prevalent craniofacial anomalies and variable limb defects.\n",
    "[Hypomyelinating leukodystrophy-27](https://omim.org/entry/620675) is an autosomal recessive neurologic disorder characterized by global developmental delay with impaired motor and intellectual development apparent from infancy. \n",
    "\n",
    "> Genotype-phenotpye correlations related to specific variants or variant categories were not indentified in the published literature at the time of this writing (2024-09-19). However, the clinical manifestations of Acrofacial dysostosis, Cincinnati type, an autosomal dominant disease, and\n",
    "Hypomyelinating leukodystrophy-27, an autosomal recessive disease, are distinct. We were able to identify only four published cases of Hypomyelinating leukodystrophy-27, limiting statistical power, but we here test clinical manifestations of\n",
    "monoallic vs. biallelic genotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2024-08-13\n",
      "Using gpsea version 0.4.1.dev0\n"
     ]
    }
   ],
   "source": [
    "import gpsea\n",
    "import hpotk\n",
    "from IPython.display import display, HTML\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo()\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using gpsea version {gpsea.__version__}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found cohort with 22 phenopackets for POLR1A\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "registry = configure_phenopacket_registry()\n",
    "cohort_name = \"POLR1A\"\n",
    "with registry.open_phenopacket_store(release=\"0.1.19\") as ps:\n",
    "    phenopackets = list(ps.iter_cohort_phenopackets(cohort_name))\n",
    "print(f\"Found cohort with {len(phenopackets)} phenopackets for {cohort_name}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 100%|██████████| 22/22 [00:00<00:00, 423.20individuals/s]\n"
     ]
    }
   ],
   "source": [
    "POLR1A_MANE_transcript = 'NM_015425.6' # Homo sapiens RNA polymerase I subunit A (POLR1A), mRNA\n",
    "POLR1A_protein_id = 'NP_056240.2' #  DNA-directed RNA polymerase I subunit RPA1\n",
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort, validation = load_phenopackets(  \n",
    "    phenopackets=phenopackets,\n",
    "    cohort_creator=cohort_creator,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Validated under none policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "validation.summarize()  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>GPSEA cohort analysis</h1>\n",
       "  <p>Successfully loaded 22 individuals.</p>\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Top 10 HPO Terms</h3>\n",
       "        A total of 156 HPO terms were used to annotated the cohort.\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">HPO Term</th>\n",
       "                <th>ID</th>\n",
       "                <th>Seen in <em>n</em> individuals</th>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Global developmental delay</td>\n",
       "                <td>HP:0001263</td>\n",
       "                <td>12</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Hypotonia</td>\n",
       "                <td>HP:0001252</td>\n",
       "                <td>11</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Hypertelorism</td>\n",
       "                <td>HP:0000316</td>\n",
       "                <td>9</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Micrognathia</td>\n",
       "                <td>HP:0000347</td>\n",
       "                <td>7</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Seizure</td>\n",
       "                <td>HP:0001250</td>\n",
       "                <td>7</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Low-set ears</td>\n",
       "                <td>HP:0000369</td>\n",
       "                <td>6</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Abnormality of limbs</td>\n",
       "                <td>HP:0040064</td>\n",
       "                <td>6</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Cleft palate</td>\n",
       "                <td>HP:0000175</td>\n",
       "                <td>5</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Brain imaging abnormality</td>\n",
       "                <td>HP:0410263</td>\n",
       "                <td>5</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Microcephaly</td>\n",
       "                <td>HP:0000252</td>\n",
       "                <td>5</td>\n",
       "                </tr>\n",
       "            \n",
       "                </tbody>\n",
       "    </table>\n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Top 10 Variants</h3>\n",
       "        Variants are shown according to NM_015425.6. A total of 22 unique variants were identified in the cohort.\n",
       "    </caption>\n",
       "            <tbody>\n",
       "                <tr class=\"strng\">\n",
       "                    <th>Count</th>\n",
       "                    <th class=\"lft\">Variant key</th>\n",
       "                    <th>Variant Name</th>\n",
       "                    <th>Protein Variant</th>\n",
       "                    <th>Variant Class</th>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">2_86075199_86075199_C_T</td>\n",
       "                    <td>c.1442G>A</td>\n",
       "                    <td>p.Arg481Lys</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">2_86030290_86030290_C_A</td>\n",
       "                    <td>c.4685G>T</td>\n",
       "                    <td>p.Cys1562Phe</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">6_43519367_43519367_A_T</td>\n",
       "                    <td>6_43519367_43519367_A_T</td>\n",
       "                    <td>None</td>\n",
       "                    <td></td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">2_86045702_86045702_G_A</td>\n",
       "                    <td>c.2801C>T</td>\n",
       "                    <td>p.Ser934Leu</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">2_86078193_86078193_C_T</td>\n",
       "                    <td>c.1178G>A</td>\n",
       "                    <td>p.Arg393His</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">2_86038743_86038746_TCTC_T</td>\n",
       "                    <td>c.3988_3990del</td>\n",
       "                    <td>p.Glu1330del</td>\n",
       "                    <td>INFRAME_DELETION</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">2_86065407_86065407_G_T</td>\n",
       "                    <td>c.1925C>A</td>\n",
       "                    <td>p.Thr642Asn</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>1</td>\n",
       "                    <td class=\"lft\">2_86028600_86028600_C_T</td>\n",
       "                    <td>c.4891G>A</td>\n",
       "                    <td>p.Val1631Met</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>1</td>\n",
       "                    <td class=\"lft\">2_86100059_86100060_CA_C</td>\n",
       "                    <td>c.190del</td>\n",
       "                    <td>p.Cys64AlafsTer42</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>1</td>\n",
       "                    <td class=\"lft\">2_86048931_86048935_GATCA_G</td>\n",
       "                    <td>c.2583_2586del</td>\n",
       "                    <td>p.Asp862Ter</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "            </tbody>\n",
       "        </table>\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Diseases</h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Disease Name</th>\n",
       "                <th >Disease ID</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">Acrofacial dysostosis, Cincinnati type</td>\n",
       "                <td>OMIM:616462</td>\n",
       "                <td>18</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">Leukodystrophy, hypomyelinating, 27</td>\n",
       "                <td>OMIM:620675</td>\n",
       "                <td>4</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       "    \n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Variant categories for NM_015425.6</h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Variant effect</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">STOP_GAINED</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">MISSENSE_VARIANT</td>\n",
       "                <td>14</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">INFRAME_DELETION</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">FRAMESHIFT_VARIANT</td>\n",
       "                <td>3</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       "  \n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from gpsea.view import CohortViewable\n",
    "cv = CohortViewable(hpo=hpo)\n",
    "report = cv.process(cohort=cohort, transcript_id=POLR1A_MANE_transcript)\n",
    "display(HTML(report))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "A required `transcripts` field is missing in the response from Variant Validator API: \n{\n  \"error\": \"Unable to recognise gene symbol LOC90784\",\n  \"requested_symbol\": \"NM_015425.6\"\n}",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[8], line 5\u001b[0m\n\u001b[1;32m      3\u001b[0m txc_service \u001b[38;5;241m=\u001b[39m VVMultiCoordinateService(genome_build\u001b[38;5;241m=\u001b[39mGRCh38)\n\u001b[1;32m      4\u001b[0m pms \u001b[38;5;241m=\u001b[39m configure_default_protein_metadata_service()\n\u001b[0;32m----> 5\u001b[0m tx_coordinates \u001b[38;5;241m=\u001b[39m \u001b[43mtxc_service\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfetch\u001b[49m\u001b[43m(\u001b[49m\u001b[43mPOLR1A_MANE_transcript\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      6\u001b[0m \u001b[38;5;66;03m#protein_meta = pms.annotate(POLR1A_protein_id)\u001b[39;00m\n",
      "File \u001b[0;32m~/GIT/gpsea/src/gpsea/preprocessing/_vv.py:164\u001b[0m, in \u001b[0;36mVVMultiCoordinateService.fetch\u001b[0;34m(self, tx)\u001b[0m\n\u001b[1;32m    162\u001b[0m tx_id \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_parse_tx(tx)\n\u001b[1;32m    163\u001b[0m response_json \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mget_response(tx_id)\n\u001b[0;32m--> 164\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mparse_response\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtx_id\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mresponse_json\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/GIT/gpsea/src/gpsea/preprocessing/_vv.py:195\u001b[0m, in \u001b[0;36mVVMultiCoordinateService.parse_response\u001b[0;34m(self, tx_id, response)\u001b[0m\n\u001b[1;32m    193\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(error_string)\n\u001b[1;32m    194\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtranscripts\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m transcript_response:\n\u001b[0;32m--> 195\u001b[0m     \u001b[43mVVMultiCoordinateService\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_handle_missing_field\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    196\u001b[0m \u001b[43m        \u001b[49m\u001b[43mresponse\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mresponse\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m    197\u001b[0m \u001b[43m        \u001b[49m\u001b[43mfield\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mtranscripts\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    198\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    199\u001b[0m tx_data \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_find_tx_data(tx_id, transcript_response[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtranscripts\u001b[39m\u001b[38;5;124m'\u001b[39m])\n\u001b[1;32m    200\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgenomic_spans\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m tx_data:\n",
      "File \u001b[0;32m~/GIT/gpsea/src/gpsea/preprocessing/_vv.py:259\u001b[0m, in \u001b[0;36mVVMultiCoordinateService._handle_missing_field\u001b[0;34m(response, field)\u001b[0m\n\u001b[1;32m    257\u001b[0m json_formatted_str \u001b[38;5;241m=\u001b[39m json\u001b[38;5;241m.\u001b[39mdumps(response, indent\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m2\u001b[39m)\n\u001b[1;32m    258\u001b[0m error_string \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mA required `\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfield\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m` field is missing in the response from Variant Validator API: \u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;132;01m{\u001b[39;00mjson_formatted_str\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m--> 259\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(error_string)\n",
      "\u001b[0;31mValueError\u001b[0m: A required `transcripts` field is missing in the response from Variant Validator API: \n{\n  \"error\": \"Unable to recognise gene symbol LOC90784\",\n  \"requested_symbol\": \"NM_015425.6\"\n}"
     ]
    }
   ],
   "source": [
    "from gpsea.model.genome import GRCh38\n",
    "from gpsea.preprocessing import configure_default_protein_metadata_service, VVMultiCoordinateService\n",
    "txc_service = VVMultiCoordinateService(genome_build=GRCh38)\n",
    "pms = configure_default_protein_metadata_service()\n",
    "tx_coordinates = txc_service.fetch(POLR1A_MANE_transcript)\n",
    "#protein_meta = pms.annotate(POLR1A_protein_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>GPSEA cohort analysis: All variant alleles</h1>\n",
       "\n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Variant alleles</h3>\n",
       "        A total of 15 unique alleles were identified in the cohort.\n",
       "    </caption>\n",
       "\n",
       "\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Variant key</th>\n",
       "                <th>Variant (cDNA)</th>\n",
       "                <th>Variant (protein)</th>\n",
       "                <th>Effects</th>\n",
       "                <th>Count</th>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86078193_86078193_C_T</td>\n",
       "                <td>c.1178G>A</td>\n",
       "                <td>p.Arg393His</td>\n",
       "                <td>missense</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86075199_86075199_C_T</td>\n",
       "                <td>c.1442G>A</td>\n",
       "                <td>p.Arg481Lys</td>\n",
       "                <td>missense</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86065407_86065407_G_T</td>\n",
       "                <td>c.1925C>A</td>\n",
       "                <td>p.Thr642Asn</td>\n",
       "                <td>missense</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86045702_86045702_G_A</td>\n",
       "                <td>c.2801C>T</td>\n",
       "                <td>p.Ser934Leu</td>\n",
       "                <td>missense</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">6_43519367_43519367_A_T</td>\n",
       "                <td>6_43519367_43519367_A_T</td>\n",
       "                <td>None</td>\n",
       "                <td></td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86038743_86038746_TCTC_T</td>\n",
       "                <td>c.3988_3990del</td>\n",
       "                <td>p.Glu1330del</td>\n",
       "                <td>inframe deletion</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86030290_86030290_C_A</td>\n",
       "                <td>c.4685G>T</td>\n",
       "                <td>p.Cys1562Phe</td>\n",
       "                <td>missense</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86040411_86040411_C_T</td>\n",
       "                <td>c.3721G>A</td>\n",
       "                <td>p.Val1241Ile</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86100059_86100060_CA_C</td>\n",
       "                <td>c.190del</td>\n",
       "                <td>p.Cys64AlafsTer42</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86048931_86048935_GATCA_G</td>\n",
       "                <td>c.2583_2586del</td>\n",
       "                <td>p.Asp862Ter</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86028600_86028600_C_T</td>\n",
       "                <td>c.4891G>A</td>\n",
       "                <td>p.Val1631Met</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86028034_86028034_G_A</td>\n",
       "                <td>c.4913C>T</td>\n",
       "                <td>p.Pro1638Leu</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86039353_86039353_G_A</td>\n",
       "                <td>c.3850C>T</td>\n",
       "                <td>p.Gln1284Ter</td>\n",
       "                <td>stop gained</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86075153_86075153_C_A</td>\n",
       "                <td>c.1488G>T</td>\n",
       "                <td>p.Met496Ile</td>\n",
       "                <td>missense</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">2_86040482_86040483_TG_T</td>\n",
       "                <td>c.3649del</td>\n",
       "                <td>p.Gln1217ArgfsTer10</td>\n",
       "                <td>frameshift</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "                </tbody>\n",
       "    </table>\n",
       "<!--\n",
       "   <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Variant category counts for </h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Variant effect</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">missense</td>\n",
       "                <td>14</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">frameshift</td>\n",
       "                <td>3</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">inframe deletion</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">stop gained</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       " -->\n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from gpsea.view import CohortVariantViewer\n",
    "\n",
    "viewer = CohortVariantViewer(tx_id=POLR1A_MANE_transcript)\n",
    "report = viewer.process(cohort)\n",
    "display(HTML(report))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.model import VariantEffect\n",
    "from gpsea.analysis.predicate.genotype import VariantPredicates, ModeOfInheritancePredicate\n",
    "\n",
    "is_missense = VariantPredicates.variant_effect(VariantEffect.MISSENSE_VARIANT, tx_id=POLR1A_MANE_transcript)\n",
    "gt_predicate = ModeOfInheritancePredicate.autosomal_recessive(is_missense)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(Categorization(category=HOM_REF),\n",
       " Categorization(category=HET),\n",
       " Categorization(category=BIALLELIC_ALT))"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gt_predicate.get_categorizations()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'What is the genotype group: HET, BIALLELIC_ALT'"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.analysis.predicate.genotype import filtering_predicate\n",
    "\n",
    "gt_predicate = filtering_predicate(gt_predicate, gt_predicate.get_categorizations()[1:])\n",
    "gt_predicate.display_question()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.predicate.phenotype import prepare_predicates_for_terms_of_interest\n",
    "\n",
    "pheno_predicates = prepare_predicates_for_terms_of_interest(\n",
    "    cohort=cohort,\n",
    "    hpo=hpo,\n",
    ")\n",
    "from gpsea.analysis.pcats.stats import FisherExactTest\n",
    "from gpsea.analysis.mtc_filter import HpoMtcFilter\n",
    "\n",
    "mtc_filter = HpoMtcFilter.default_filter(hpo=hpo, term_frequency_threshold=0.2)\n",
    "mtc_correction = 'fdr_bh'\n",
    "statistic = FisherExactTest()\n",
    "from gpsea.analysis.pcats import HpoTermAnalysis\n",
    "\n",
    "analysis = HpoTermAnalysis(\n",
    "    count_statistic=statistic,\n",
    "    mtc_filter=mtc_filter,\n",
    "    mtc_correction=mtc_correction,\n",
    "    mtc_alpha=0.05,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_predicate=gt_predicate,\n",
    "    pheno_predicates=pheno_predicates,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "  <style>table {\n",
       "    border-collapse: collapse;\n",
       "    margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "  }\n",
       "\n",
       "\n",
       "  th {\n",
       "    background-color: #f2f2f2;\n",
       "    border: 1px solid #dddddd;\n",
       "    text-align: left;\n",
       "    padding: 2px;\n",
       "  }\n",
       "\n",
       "  tr {\n",
       "    border: 1px solid #dddddd;\n",
       "  }\n",
       "\n",
       "  td {\n",
       "    padding: 2px;\n",
       "    font-weight: bold;\n",
       "  }\n",
       "\n",
       "  tr:nth-child(even) {\n",
       "    background-color: #f2f2f2;\n",
       "  }\n",
       "\n",
       "  .table td,tr {\n",
       "    text-align: left;\n",
       "}\n",
       "\n",
       ".table td:first-child, tr:first-child {\n",
       "    text-align: left;\n",
       "}\n",
       "\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>Phenotype testing report</h1>\n",
       "  <p>Phenotype MTC filter: <em>HPO MTC filter</em></p>\n",
       "  <p>Multiple testing correction: <em>fdr_bh</em></p>\n",
       "  <p>Performed statistical tests for 9 out of the total of 254 HPO terms.</p>\n",
       "    <table>\n",
       "      <caption>Using <em>HPO MTC filter</em>, 245 term(s) were omitted from statistical analysis.</caption>\n",
       "        <tbody>\n",
       "          <tr>\n",
       "            <th>Code</th>\n",
       "            <th>Reason</th>\n",
       "            <th>Count</th>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF01</td>\n",
       "            <td>Skipping term with maximum frequency that was less than threshold 0.2</td>\n",
       "            <td>31</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF02</td>\n",
       "            <td>Skipping term because no genotype has more than one observed HPO count</td>\n",
       "            <td>2</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF03</td>\n",
       "            <td>Skipping term because of a child term with the same individual counts</td>\n",
       "            <td>1</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF04</td>\n",
       "            <td>Skipping term because all genotypes have same HPO observed proportions</td>\n",
       "            <td>8</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF05</td>\n",
       "            <td>Skipping term because one genotype had zero observations</td>\n",
       "            <td>13</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF06</td>\n",
       "            <td>Skipping term with less than 7 observations (not powered for 2x2)</td>\n",
       "            <td>129</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF08</td>\n",
       "            <td>Skipping general term</td>\n",
       "            <td>61</td>\n",
       "          </tr>\n",
       "          \n",
       "        </tbody>\n",
       "    </table>\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from gpsea.view import MtcStatsViewer\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "report = viewer.process(result)\n",
    "display(HTML(report))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>What is the genotype group</th>\n",
       "      <th colspan=\"2\" halign=\"left\">HET</th>\n",
       "      <th colspan=\"2\" halign=\"left\">BIALLELIC_ALT</th>\n",
       "      <th colspan=\"2\" halign=\"left\"></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Truncal ataxia [HP:0002078]</th>\n",
       "      <td>0/8</td>\n",
       "      <td>0%</td>\n",
       "      <td>3/3</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.027273</td>\n",
       "      <td>0.006061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ataxia [HP:0001251]</th>\n",
       "      <td>0/8</td>\n",
       "      <td>0%</td>\n",
       "      <td>3/3</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.027273</td>\n",
       "      <td>0.006061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Macrocephaly [HP:0000256]</th>\n",
       "      <td>0/8</td>\n",
       "      <td>0%</td>\n",
       "      <td>2/2</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.066667</td>\n",
       "      <td>0.022222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Developmental regression [HP:0002376]</th>\n",
       "      <td>0/8</td>\n",
       "      <td>0%</td>\n",
       "      <td>2/4</td>\n",
       "      <td>50%</td>\n",
       "      <td>0.163636</td>\n",
       "      <td>0.090909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Relative macrocephaly [HP:0004482]</th>\n",
       "      <td>0/8</td>\n",
       "      <td>0%</td>\n",
       "      <td>2/4</td>\n",
       "      <td>50%</td>\n",
       "      <td>0.163636</td>\n",
       "      <td>0.090909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ventriculomegaly [HP:0002119]</th>\n",
       "      <td>1/6</td>\n",
       "      <td>17%</td>\n",
       "      <td>3/4</td>\n",
       "      <td>75%</td>\n",
       "      <td>0.278721</td>\n",
       "      <td>0.190476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypotonia [HP:0001252]</th>\n",
       "      <td>7/9</td>\n",
       "      <td>78%</td>\n",
       "      <td>1/4</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.278721</td>\n",
       "      <td>0.216783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Seizure [HP:0001250]</th>\n",
       "      <td>4/4</td>\n",
       "      <td>100%</td>\n",
       "      <td>2/4</td>\n",
       "      <td>50%</td>\n",
       "      <td>0.482143</td>\n",
       "      <td>0.428571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Global developmental delay [HP:0001263]</th>\n",
       "      <td>6/8</td>\n",
       "      <td>75%</td>\n",
       "      <td>4/4</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.515152</td>\n",
       "      <td>0.515152</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "What is the genotype group                HET  ...          \n",
       "                                        Count  ...  p values\n",
       "Truncal ataxia [HP:0002078]               0/8  ...  0.006061\n",
       "Ataxia [HP:0001251]                       0/8  ...  0.006061\n",
       "Macrocephaly [HP:0000256]                 0/8  ...  0.022222\n",
       "Developmental regression [HP:0002376]     0/8  ...  0.090909\n",
       "Relative macrocephaly [HP:0004482]        0/8  ...  0.090909\n",
       "Ventriculomegaly [HP:0002119]             1/6  ...  0.190476\n",
       "Hypotonia [HP:0001252]                    7/9  ...  0.216783\n",
       "Seizure [HP:0001250]                      4/4  ...  0.428571\n",
       "Global developmental delay [HP:0001263]   6/8  ...  0.515152\n",
       "\n",
       "[9 rows x 6 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "\n",
    "report = summarize_hpo_analysis(hpo=hpo, result=result)\n",
    "report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
