{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# POLR1A\n",
    "\n",
    "POLR1A is the largest subunit of RNA polymerase I (Pol I), which catalyzes DNA-dependent synthesis of ribosomal RNA ([OMIM:616404](https://omim.org/entry/616404)).\n",
    "\n",
    "Pathogeniv POLR1A variants are associated with\n",
    "\n",
    "- [Acrofacial dysostosis, Cincinnati type, OMIM:616462](https://omim.org/entry/616462), and\n",
    "- [Leukodystrophy, hypomyelinating, 27, OMIM:620675](https://omim.org/entry/620675).\n",
    "\t\n",
    "\n",
    "\n",
    "[Acrofacial dysostosis, Cincinnati type](https://omim.org/entry/616462) is characterized by craniofacial anomalies reminiscent of Treacher Collins syndrome, as well as numerous additional phenotypes including neurodevelopmental abnormalities and structural cardiac defects, in combination with highly prevalent craniofacial anomalies and variable limb defects.\n",
    "[Hypomyelinating leukodystrophy-27](https://omim.org/entry/620675) is an autosomal recessive neurologic disorder characterized by global developmental delay with impaired motor and intellectual development apparent from infancy. \n",
    "\n",
    "> Genotype-phenotpye correlations related to specific variants or variant categories were not indentified in the published literature at the time of this writing (2024-09-19). However, the clinical manifestations of Acrofacial dysostosis, Cincinnati type, an autosomal dominant disease, and\n",
    "Hypomyelinating leukodystrophy-27, an autosomal recessive disease, are distinct. We were able to identify only four published cases of Hypomyelinating leukodystrophy-27, limiting statistical power, but we here test clinical manifestations of\n",
    "monoallic vs. biallelic genotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2024-08-13\n",
      "Using gpsea version 0.7.1\n"
     ]
    }
   ],
   "source": [
    "import gpsea\n",
    "import hpotk\n",
    "from IPython.display import display, HTML\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo()\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using gpsea version {gpsea.__version__}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# POGZ\n",
    "We user the [Matched Annotation from NCBI and EMBL-EBI (MANE)](https://www.ncbi.nlm.nih.gov/refseq/MANE/) transcript and the corresponding protein identifier for POGZ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_symbol = 'POLR1A'\n",
    "mane_tx_id =  'NM_015425.6'\n",
    "mane_protein_id = 'NP_056240.2' #  DNA-directed RNA polymerase I subunit RPA1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 22individuals [00:05,  4.36individuals/s]\n",
      "Loaded 22 individuals from Cohort(members=(Patient(labels:Individual 3[PMID_37075751_Individual_3], sex:Sex.FEMALE, age:Age(days=56.0, timeline=Timeline.GESTATIONAL), vital_status:VitalStatus(status=<Status.DECEASED: 2>, age_of_death=Age(days=56.0, timeline=Timeline.GESTATIONAL)), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86100058, end=86100060, strand=+), ref=CA, alt=C, change_length=-1), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.190del,is_preferred:True,variant_effects:(<VariantEffect.FRAMESHIFT_VARIANT: 'SO:0001589'>,),overlapping_exons:(2,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Cys64AlafsTer42,protein_effect_location:Region(start=63, end=64)),), genotypes=Genotypes(['Individual 3[PMID_37075751_Individual_3]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000023), DefaultTermId(idx=2, value=HP_0000238)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 11[PMID_37075751_Individual_11], sex:Sex.MALE, age:Age(days=4017.75, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86040410, end=86040411, strand=+), ref=C, alt=T, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.3721G>A,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(25,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Val1241Ile,protein_effect_location:Region(start=1240, end=1241)),), genotypes=Genotypes(['Individual 11[PMID_37075751_Individual_11]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000668), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000028), DefaultTermId(idx=2, value=HP_0000023), DefaultTermId(idx=2, value=HP_0000238)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 10[PMID_37075751_Individual_10], sex:Sex.FEMALE, age:Age(days=2769.8125, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86040481, end=86040483, strand=+), ref=TG, alt=T, change_length=-1), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.3649del,is_preferred:True,variant_effects:(<VariantEffect.FRAMESHIFT_VARIANT: 'SO:0001589'>,),overlapping_exons:(25,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Gln1217ArgfsTer10,protein_effect_location:Region(start=1216, end=1217)),), genotypes=Genotypes(['Individual 10[PMID_37075751_Individual_10]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000668), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000023), DefaultTermId(idx=2, value=HP_0000238)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Patient 1[PMID_36917474_Patient_1], sex:Sex.MALE, age:Age(days=6026.625, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.DECEASED: 2>, age_of_death=Age(days=6026.625, timeline=Timeline.POSTNATAL)), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86065406, end=86065407, strand=+), ref=G, alt=T, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.1925C>A,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(14,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Thr642Asn,protein_effect_location:Region(start=641, end=642)),), genotypes=Genotypes(['Patient 1[PMID_36917474_Patient_1]=1/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0001250), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0001332), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001347), DefaultTermId(idx=2, value=HP_0000737), DefaultTermId(idx=2, value=HP_0001276), DefaultTermId(idx=2, value=HP_0000486), DefaultTermId(idx=2, value=HP_0000518), DefaultTermId(idx=2, value=HP_0002415), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0001249), DefaultTermId(idx=2, value=HP_0007366), DefaultTermId(idx=2, value=HP_0003429), DefaultTermId(idx=2, value=HP_0001272), DefaultTermId(idx=2, value=HP_0002059), DefaultTermId(idx=2, value=HP_0002078), DefaultTermId(idx=2, value=HP_0033725), DefaultTermId(idx=2, value=HP_0000639), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0001344), DefaultTermId(idx=2, value=HP_0002599), DefaultTermId(idx=2, value=HP_0002066), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0000648), DefaultTermId(idx=2, value=HP_0000605), DefaultTermId(idx=2, value=HP_0001518), DefaultTermId(idx=2, value=HP_0001998), DefaultTermId(idx=2, value=HP_0004482)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:620675)], Patient(labels:Individual 2[PMID_37075751_Individual_2], sex:Sex.MALE, age:Age(days=1826.25, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86100073, end=86100074, strand=+), ref=T, alt=A, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.176A>T,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(2,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Asp59Val,protein_effect_location:Region(start=58, end=59)),), genotypes=Genotypes(['Individual 2[PMID_37075751_Individual_2]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000028), DefaultTermId(idx=2, value=HP_0000023)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Brother 1 (203)[PMID_28051070_Brother_1_203], sex:Sex.MALE, age:Age(days=4017.75, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86045701, end=86045702, strand=+), ref=G, alt=A, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.2801C>T,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(20,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Ser934Leu,protein_effect_location:Region(start=933, end=934)),), genotypes=Genotypes(['Brother 1 (203)[PMID_28051070_Brother_1_203]=1/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0002066), DefaultTermId(idx=2, value=HP_0002505), DefaultTermId(idx=2, value=HP_0001518), DefaultTermId(idx=2, value=HP_0003487), DefaultTermId(idx=2, value=HP_0002415), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0001249), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0002599), DefaultTermId(idx=2, value=HP_0031936), DefaultTermId(idx=2, value=HP_0001332), DefaultTermId(idx=2, value=HP_0001347), DefaultTermId(idx=2, value=HP_0001272), DefaultTermId(idx=2, value=HP_0002059), DefaultTermId(idx=2, value=HP_0002078), DefaultTermId(idx=2, value=HP_0033725), DefaultTermId(idx=2, value=HP_0001257), DefaultTermId(idx=2, value=HP_0001382), DefaultTermId(idx=2, value=HP_0001761), DefaultTermId(idx=2, value=HP_0000648), DefaultTermId(idx=2, value=HP_0000605), DefaultTermId(idx=2, value=HP_0004482), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0001250), DefaultTermId(idx=2, value=HP_0007366), DefaultTermId(idx=2, value=HP_0000737), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0000486), DefaultTermId(idx=2, value=HP_0000518), DefaultTermId(idx=2, value=HP_0000639), DefaultTermId(idx=2, value=HP_0001998)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:620675)], Patient(labels:Individual 17[PMID_37075751_Individual_17], sex:Sex.FEMALE, age:Age(days=2556.75, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86028599, end=86028600, strand=+), ref=C, alt=T, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.4891G>A,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(32,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Val1631Met,protein_effect_location:Region(start=1630, end=1631)),), genotypes=Genotypes(['Individual 17[PMID_37075751_Individual_17]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0001250), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000023), DefaultTermId(idx=2, value=HP_0000238)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 9[PMID_37075751_Individual_9], sex:Sex.MALE, age:Age(days=1461.0, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86048930, end=86048935, strand=+), ref=GATCA, alt=G, change_length=-4), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.2583_2586del,is_preferred:True,variant_effects:(<VariantEffect.FRAMESHIFT_VARIANT: 'SO:0001589'>,),overlapping_exons:(18,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Asp862Ter,protein_effect_location:Region(start=860, end=862)),), genotypes=Genotypes(['Individual 9[PMID_37075751_Individual_9]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000668), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000028), DefaultTermId(idx=2, value=HP_0000023), DefaultTermId(idx=2, value=HP_0000238)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 5[PMID_37075751_Individual_5], sex:Sex.FEMALE, age:None, vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86078192, end=86078193, strand=+), ref=C, alt=T, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.1178G>A,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(10,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Arg393His,protein_effect_location:Region(start=392, end=393)),), genotypes=Genotypes(['Individual 5[PMID_37075751_Individual_5]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000023)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 4[PMID_37075751_Individual_4], sex:Sex.MALE, age:Age(days=334.8125, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86078192, end=86078193, strand=+), ref=C, alt=T, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.1178G>A,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(10,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Arg393His,protein_effect_location:Region(start=392, end=393)),), genotypes=Genotypes(['Individual 4[PMID_37075751_Individual_4]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000028), DefaultTermId(idx=2, value=HP_0000023)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 8[PMID_37075751_Individual_8], sex:Sex.MALE, age:Age(days=730.5, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.DECEASED: 2>, age_of_death=Age(days=730.5, timeline=Timeline.POSTNATAL)), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86075152, end=86075153, strand=+), ref=C, alt=A, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.1488G>T,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(12,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Met496Ile,protein_effect_location:Region(start=495, end=496)),), genotypes=Genotypes(['Individual 8[PMID_37075751_Individual_8]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0001250), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000028), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000023), DefaultTermId(idx=2, value=HP_0000238)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 16[PMID_37075751_Individual_16], sex:Sex.MALE, age:Age(days=1095.75, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86030289, end=86030290, strand=+), ref=C, alt=A, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.4685G>T,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(31,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Cys1562Phe,protein_effect_location:Region(start=1561, end=1562)),), genotypes=Genotypes(['Individual 16[PMID_37075751_Individual_16]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0001250), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000028), DefaultTermId(idx=2, value=HP_0000023), DefaultTermId(idx=2, value=HP_0000238)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Brother 2 (204)[PMID_28051070_Brother_2_204], sex:Sex.MALE, age:Age(days=2374.125, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86045701, end=86045702, strand=+), ref=G, alt=A, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.2801C>T,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(20,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Ser934Leu,protein_effect_location:Region(start=933, end=934)),), genotypes=Genotypes(['Brother 2 (204)[PMID_28051070_Brother_2_204]=1/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0001250), DefaultTermId(idx=2, value=HP_0001344), DefaultTermId(idx=2, value=HP_0002421), DefaultTermId(idx=2, value=HP_0001518), DefaultTermId(idx=2, value=HP_0001998), DefaultTermId(idx=2, value=HP_0003487), DefaultTermId(idx=2, value=HP_0002415), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0001249), DefaultTermId(idx=2, value=HP_0002599), DefaultTermId(idx=2, value=HP_0031936), DefaultTermId(idx=2, value=HP_0001332), DefaultTermId(idx=2, value=HP_0001347), DefaultTermId(idx=2, value=HP_0001272), DefaultTermId(idx=2, value=HP_0002059), DefaultTermId(idx=2, value=HP_0002078), DefaultTermId(idx=2, value=HP_0033725), DefaultTermId(idx=2, value=HP_0001257), DefaultTermId(idx=2, value=HP_0001382), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0000648), DefaultTermId(idx=2, value=HP_0000605), DefaultTermId(idx=2, value=HP_0004482), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0007366), DefaultTermId(idx=2, value=HP_0000737), DefaultTermId(idx=2, value=HP_0002066), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0001761), DefaultTermId(idx=2, value=HP_0000486), DefaultTermId(idx=2, value=HP_0000518), DefaultTermId(idx=2, value=HP_0000639)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:620675)], Patient(labels:Individual 15[PMID_37075751_Individual_15], sex:Sex.MALE, age:Age(days=2191.5, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86030289, end=86030290, strand=+), ref=C, alt=A, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.4685G>T,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(31,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Cys1562Phe,protein_effect_location:Region(start=1561, end=1562)),), genotypes=Genotypes(['Individual 15[PMID_37075751_Individual_15]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000028), DefaultTermId(idx=2, value=HP_0000023), DefaultTermId(idx=2, value=HP_0000238)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 7[PMID_37075751_Individual_7], sex:Sex.FEMALE, age:Age(days=16071.0, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86075198, end=86075199, strand=+), ref=C, alt=T, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.1442G>A,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(12,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Arg481Lys,protein_effect_location:Region(start=480, end=481)),), genotypes=Genotypes(['Individual 7[PMID_37075751_Individual_7]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000668), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000023), DefaultTermId(idx=2, value=HP_0000238)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 6[PMID_37075751_Individual_6], sex:Sex.FEMALE, age:Age(days=3774.25, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86075198, end=86075199, strand=+), ref=C, alt=T, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.1442G>A,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(12,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Arg481Lys,protein_effect_location:Region(start=480, end=481)),), genotypes=Genotypes(['Individual 6[PMID_37075751_Individual_6]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000668), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000238), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000023)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 18[PMID_37075751_Individual_18], sex:Sex.FEMALE, age:Age(days=4017.75, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86028033, end=86028034, strand=+), ref=G, alt=A, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.4913C>T,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(33,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Pro1638Leu,protein_effect_location:Region(start=1637, end=1638)),), genotypes=Genotypes(['Individual 18[PMID_37075751_Individual_18]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000023)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 14[PMID_37075751_Individual_14], sex:Sex.MALE, age:Age(days=182.625, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86038742, end=86038746, strand=+), ref=TCTC, alt=T, change_length=-3), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.3988_3990del,is_preferred:True,variant_effects:(<VariantEffect.INFRAME_DELETION: 'SO:0001822'>,),overlapping_exons:(27,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Glu1330del,protein_effect_location:Region(start=1329, end=1330)),), genotypes=Genotypes(['Individual 14[PMID_37075751_Individual_14]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0001250), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000028), DefaultTermId(idx=2, value=HP_0000023), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000238)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 1[PMID_37075751_Individual_1], sex:Sex.MALE, age:Age(days=91.3125, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.DECEASED: 2>, age_of_death=Age(days=91.3125, timeline=Timeline.POSTNATAL)), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86100073, end=86100074, strand=+), ref=T, alt=A, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.176A>T,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(2,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Asp59Val,protein_effect_location:Region(start=58, end=59)),), genotypes=Genotypes(['Individual 1[PMID_37075751_Individual_1]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000028), DefaultTermId(idx=2, value=HP_0000023)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Individual 13[PMID_37075751_Individual_13], sex:Sex.MALE, age:Age(days=8766.0, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86038742, end=86038746, strand=+), ref=TCTC, alt=T, change_length=-3), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.3988_3990del,is_preferred:True,variant_effects:(<VariantEffect.INFRAME_DELETION: 'SO:0001822'>,),overlapping_exons:(27,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Glu1330del,protein_effect_location:Region(start=1329, end=1330)),), genotypes=Genotypes(['Individual 13[PMID_37075751_Individual_13]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0001250), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000668), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000028), DefaultTermId(idx=2, value=HP_0000023), DefaultTermId(idx=2, value=HP_0000238)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)], Patient(labels:Patient 2[PMID_36917474_Patient_2], sex:Sex.FEMALE, age:Age(days=1034.875, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86065406, end=86065407, strand=+), ref=G, alt=T, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.1925C>A,is_preferred:True,variant_effects:(<VariantEffect.MISSENSE_VARIANT: 'SO:0001583'>,),overlapping_exons:(14,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Thr642Asn,protein_effect_location:Region(start=641, end=642)),), genotypes=Genotypes(['Patient 2[PMID_36917474_Patient_2]=1/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0001332), DefaultTermId(idx=2, value=HP_0001347), DefaultTermId(idx=2, value=HP_0001257), DefaultTermId(idx=2, value=HP_0002415), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0003429), DefaultTermId(idx=2, value=HP_0001272), DefaultTermId(idx=2, value=HP_0033725), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0000639), DefaultTermId(idx=2, value=HP_0002119), DefaultTermId(idx=2, value=HP_0001250), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0001344), DefaultTermId(idx=2, value=HP_0007366), DefaultTermId(idx=2, value=HP_0002599), DefaultTermId(idx=2, value=HP_0000737), DefaultTermId(idx=2, value=HP_0002066), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0000486), DefaultTermId(idx=2, value=HP_0000518), DefaultTermId(idx=2, value=HP_0000648), DefaultTermId(idx=2, value=HP_0000605), DefaultTermId(idx=2, value=HP_0001518), DefaultTermId(idx=2, value=HP_0001998), DefaultTermId(idx=2, value=HP_0004482)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:620675)], Patient(labels:Individual 12[PMID_37075751_Individual_12], sex:Sex.MALE, age:Age(days=1552.3125, timeline=Timeline.POSTNATAL), vital_status:VitalStatus(status=<Status.ALIVE: 1>, age_of_death=None), variants:(Variant(variant_info=VariantInfo(variant_coordinates=VariantCoordinates(region=GenomicRegion(contig=2, start=86039352, end=86039353, strand=+), ref=G, alt=A, change_length=0), sv_info=None), tx_annotations=(TranscriptAnnotation(gene_id:POLR1A,transcript_id:NM_015425.6,hgvs_cdna:NM_015425.6:c.3850C>T,is_preferred:True,variant_effects:(<VariantEffect.STOP_GAINED: 'SO:0001587'>,),overlapping_exons:(26,),protein_id:NP_056240.2,hgvsp:NP_056240.2:p.Gln1284Ter,protein_effect_location:Region(start=1283, end=1284)),), genotypes=Genotypes(['Individual 12[PMID_37075751_Individual_12]=0/1'])),), phenotypes:[DefaultTermId(idx=2, value=HP_0004325), DefaultTermId(idx=2, value=HP_0001629), DefaultTermId(idx=2, value=HP_0000324), DefaultTermId(idx=2, value=HP_0000490), DefaultTermId(idx=2, value=HP_0000508), DefaultTermId(idx=2, value=HP_0008551), DefaultTermId(idx=2, value=HP_0000252), DefaultTermId(idx=2, value=HP_0001511), DefaultTermId(idx=2, value=HP_0000286), DefaultTermId(idx=2, value=HP_0000582), DefaultTermId(idx=2, value=HP_0000637), DefaultTermId(idx=2, value=HP_0001655), DefaultTermId(idx=2, value=HP_0011968), DefaultTermId(idx=2, value=HP_0000316), DefaultTermId(idx=2, value=HP_0000668), DefaultTermId(idx=2, value=HP_0000430), DefaultTermId(idx=2, value=HP_0000453), DefaultTermId(idx=2, value=HP_0012469), DefaultTermId(idx=2, value=HP_0010819), DefaultTermId(idx=2, value=HP_0001336), DefaultTermId(idx=2, value=HP_0200134), DefaultTermId(idx=2, value=HP_0001263), DefaultTermId(idx=2, value=HP_0002376), DefaultTermId(idx=2, value=HP_0000218), DefaultTermId(idx=2, value=HP_0000175), DefaultTermId(idx=2, value=HP_0000204), DefaultTermId(idx=2, value=HP_0000369), DefaultTermId(idx=2, value=HP_0000463), DefaultTermId(idx=2, value=HP_0000347), DefaultTermId(idx=2, value=HP_0009921), DefaultTermId(idx=2, value=HP_0002650), DefaultTermId(idx=2, value=HP_0001605), DefaultTermId(idx=2, value=HP_0004209), DefaultTermId(idx=2, value=HP_0004322), DefaultTermId(idx=2, value=HP_0000625), DefaultTermId(idx=2, value=HP_0001252), DefaultTermId(idx=2, value=HP_0005280), DefaultTermId(idx=2, value=HP_0011330), DefaultTermId(idx=2, value=HP_0000365), DefaultTermId(idx=2, value=HP_0000256), DefaultTermId(idx=2, value=HP_0000954), DefaultTermId(idx=2, value=HP_0000028), DefaultTermId(idx=2, value=HP_0000023)], measurements:[], diseases:[DefaultTermId(idx=4, value=OMIM:616462)]), excluded_count=0)\n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets, load_phenopacket_folder\n",
    "\n",
    "phenopacket_store_release = '0.1.21'  # Update, if necessary\n",
    "registry = configure_phenopacket_registry()\n",
    "\n",
    "with registry.open_phenopacket_store(release=phenopacket_store_release) as ps:\n",
    "    phenopackets = tuple(ps.iter_cohort_phenopackets(gene_symbol))\n",
    "\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort, qc = load_phenopacket_folder(\"../../../phenopacket-store/notebooks/POLR1A/phenopackets\", cohort_creator)\n",
    "\n",
    "#cohort, qc = load_phenopackets(\n",
    "#    phenopackets=phenopackets, \n",
    "#    cohort_creator=cohort_creator,\n",
    "#)\n",
    "print(f'Loaded {len(cohort)} individuals from {cohort}')\n",
    "qc.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "  <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td,\n",
       "    tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px;\n",
       "      /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>GPSEA cohort analysis</h1>\n",
       "  <p>\n",
       "    Successfully loaded 22 individuals.\n",
       "\n",
       "    \n",
       "    14 were recorded as male,\n",
       "    8 as female,\n",
       "    and 0 as unknown sex.\n",
       "\n",
       "    \n",
       "    \n",
       "    18 were reported to be alive at the time of last encounter,\n",
       "    4 were deceased,\n",
       "    and vital status was unreported for 0 individuals.\n",
       "    \n",
       "\n",
       "    \n",
       "    The cohort included 76 HPO terms.\n",
       "\n",
       "    \n",
       "    21 individuals had disease onset information\n",
       "    and 4 had information about the age of last encounter.\n",
       "\n",
       "\n",
       "    \n",
       "    \n",
       "  </p>\n",
       "\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "\n",
       "  \n",
       "  <table>\n",
       "    <caption style=\"color: black;\">\n",
       "      <h3>Top 10 HPO terms</h3>\n",
       "      A total of 76 HPO terms were used to annotated the cohort.\n",
       "      \n",
       "    </caption>\n",
       "    <tbody>\n",
       "      <tr class=\"strng\">\n",
       "        <th class=\"lft\">HPO Term</th>\n",
       "        <th>ID</th>\n",
       "        <th>Seen in <em>n</em> individuals</th>\n",
       "      </tr>\n",
       "\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Global developmental delay</td>\n",
       "        <td>HP:0001263</td>\n",
       "        <td>12</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Hypotonia</td>\n",
       "        <td>HP:0001252</td>\n",
       "        <td>11</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Hypertelorism</td>\n",
       "        <td>HP:0000316</td>\n",
       "        <td>9</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Short stature</td>\n",
       "        <td>HP:0004322</td>\n",
       "        <td>8</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Micrognathia</td>\n",
       "        <td>HP:0000347</td>\n",
       "        <td>7</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Seizure</td>\n",
       "        <td>HP:0001250</td>\n",
       "        <td>7</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Low-set ears</td>\n",
       "        <td>HP:0000369</td>\n",
       "        <td>6</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Microcephaly</td>\n",
       "        <td>HP:0000252</td>\n",
       "        <td>5</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Cleft palate</td>\n",
       "        <td>HP:0000175</td>\n",
       "        <td>5</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Ptosis</td>\n",
       "        <td>HP:0000508</td>\n",
       "        <td>5</td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "\n",
       "  \n",
       "\n",
       "  \n",
       "  <table>\n",
       "    <caption style=\"color: black;\">\n",
       "      <h3>Top 10 variants</h3>\n",
       "      Variants are shown according to MANE transcript ID.\n",
       "      A total of 15 unique variants\n",
       "      were identified in the cohort.\n",
       "    </caption>\n",
       "    <tbody>\n",
       "      <tr class=\"strng\">\n",
       "        <th>Seen in <em>n</em> individuals</th>\n",
       "        <th class=\"lft\">Variant key</th>\n",
       "        <th>Variant Name</th>\n",
       "        <th>Protein Effect</th>\n",
       "        <th>Variant Class</th>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">2_86065407_86065407_G_T</td>\n",
       "        <td>c.1925C>A</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">2_86100074_86100074_T_A</td>\n",
       "        <td>c.176A>T</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">2_86045702_86045702_G_A</td>\n",
       "        <td>c.2801C>T</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">2_86078193_86078193_C_T</td>\n",
       "        <td>c.1178G>A</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">2_86030290_86030290_C_A</td>\n",
       "        <td>c.4685G>T</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">2_86075199_86075199_C_T</td>\n",
       "        <td>c.1442G>A</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">2_86038743_86038746_TCTC_T</td>\n",
       "        <td>c.3988_3990del</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td class=\"lft\">2_86100059_86100060_CA_C</td>\n",
       "        <td>c.190del</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td class=\"lft\">2_86040411_86040411_C_T</td>\n",
       "        <td>c.3721G>A</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td class=\"lft\">2_86040482_86040483_TG_T</td>\n",
       "        <td>c.3649del</td>\n",
       "        <td>None</td>\n",
       "        <td></td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "  \n",
       "  \n",
       "  <table>\n",
       "    <caption style=\"color: black;\">\n",
       "      <h3>Diseases</h3>\n",
       "    </caption>\n",
       "    <tbody>\n",
       "      <tr class=\"strng\">\n",
       "        <th class=\"lft\">Name</th>\n",
       "        <th>ID</th>\n",
       "        <th><em>N</em> diagnosed individuals</th>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Acrofacial dysostosis, Cincinnati type</td>\n",
       "        <td>OMIM:616462</td>\n",
       "        <td>18</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Leukodystrophy, hypomyelinating, 27</td>\n",
       "        <td>OMIM:620675</td>\n",
       "        <td>4</td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "  \n",
       "  \n",
       "  \n",
       "  <p>Call this function with transcript to see table with variant effect counts.</p>\n",
       "  \n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpsea.view._report.HtmlGpseaReport at 0x17c54db50>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import CohortViewer\n",
    "cv = CohortViewer(hpo=hpo)\n",
    "cv.process(cohort=cohort)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "A required `transcripts` field is missing in the response from Variant Validator API: \n{\n  \"error\": \"Unable to recognise gene symbol LOC90784\",\n  \"requested_symbol\": \"NM_015425.6\"\n}",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[8], line 8\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mgpsea\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mview\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m ProteinVisualizer\n\u001b[1;32m      7\u001b[0m txc_service \u001b[38;5;241m=\u001b[39m VVMultiCoordinateService(genome_build\u001b[38;5;241m=\u001b[39mGRCh38)\n\u001b[0;32m----> 8\u001b[0m tx_coordinates \u001b[38;5;241m=\u001b[39m \u001b[43mtxc_service\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfetch\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmane_tx_id\u001b[49m\u001b[43m)\u001b[49m \n\u001b[1;32m      9\u001b[0m pms \u001b[38;5;241m=\u001b[39m configure_default_protein_metadata_service()\n\u001b[1;32m     10\u001b[0m \u001b[38;5;66;03m#protein_meta = pms.annotate(mane_protein_id)\u001b[39;00m\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.12/site-packages/gpsea/preprocessing/_vv.py:164\u001b[0m, in \u001b[0;36mVVMultiCoordinateService.fetch\u001b[0;34m(self, tx)\u001b[0m\n\u001b[1;32m    162\u001b[0m tx_id \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_parse_tx(tx)\n\u001b[1;32m    163\u001b[0m response_json \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mget_response(tx_id)\n\u001b[0;32m--> 164\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mparse_response\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtx_id\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mresponse_json\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.12/site-packages/gpsea/preprocessing/_vv.py:195\u001b[0m, in \u001b[0;36mVVMultiCoordinateService.parse_response\u001b[0;34m(self, tx_id, response)\u001b[0m\n\u001b[1;32m    193\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(error_string)\n\u001b[1;32m    194\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtranscripts\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m transcript_response:\n\u001b[0;32m--> 195\u001b[0m     \u001b[43mVVMultiCoordinateService\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_handle_missing_field\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    196\u001b[0m \u001b[43m        \u001b[49m\u001b[43mresponse\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mresponse\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m    197\u001b[0m \u001b[43m        \u001b[49m\u001b[43mfield\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mtranscripts\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    198\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    199\u001b[0m tx_data \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_find_tx_data(tx_id, transcript_response[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtranscripts\u001b[39m\u001b[38;5;124m'\u001b[39m])\n\u001b[1;32m    200\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgenomic_spans\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m tx_data:\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.12/site-packages/gpsea/preprocessing/_vv.py:259\u001b[0m, in \u001b[0;36mVVMultiCoordinateService._handle_missing_field\u001b[0;34m(response, field)\u001b[0m\n\u001b[1;32m    257\u001b[0m json_formatted_str \u001b[38;5;241m=\u001b[39m json\u001b[38;5;241m.\u001b[39mdumps(response, indent\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m2\u001b[39m)\n\u001b[1;32m    258\u001b[0m error_string \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mA required `\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfield\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m` field is missing in the response from Variant Validator API: \u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;132;01m{\u001b[39;00mjson_formatted_str\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m--> 259\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(error_string)\n",
      "\u001b[0;31mValueError\u001b[0m: A required `transcripts` field is missing in the response from Variant Validator API: \n{\n  \"error\": \"Unable to recognise gene symbol LOC90784\",\n  \"requested_symbol\": \"NM_015425.6\"\n}"
     ]
    }
   ],
   "source": [
    "from gpsea.model.genome import GRCh38\n",
    "from gpsea.model import ProteinMetadata\n",
    "from gpsea.preprocessing import VVMultiCoordinateService, configure_default_protein_metadata_service\n",
    "import matplotlib.pyplot as plt\n",
    "from gpsea.view import ProteinVisualizer\n",
    "\n",
    "txc_service = VVMultiCoordinateService(genome_build=GRCh38)\n",
    "tx_coordinates = txc_service.fetch(mane_tx_id) \n",
    "pms = configure_default_protein_metadata_service()\n",
    "#protein_meta = pms.annotate(mane_protein_id)\n",
    "\n",
    "protein_meta = ProteinMetadata.from_uniprot_json(protein_id=mane_protein_id, \n",
    "                                                 label=\"POLR1A\",\n",
    "                                                 uniprot_json=\"O95602.json\",\n",
    "                                                 protein_length=1720)\n",
    "\n",
    "polr1a_fig, ax = plt.subplots(figsize=(15, 8))\n",
    "visualizer = ProteinVisualizer()\n",
    "visualizer.draw_protein_diagram(\n",
    "    tx_coordinates,\n",
    "    protein_meta,\n",
    "    cohort,\n",
    "    ax=ax,\n",
    ")\n",
    "polr1a_fig.tight_layout()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'HtmlGpseaReport' object is not subscriptable",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[7], line 5\u001b[0m\n\u001b[1;32m      3\u001b[0m viewer \u001b[38;5;241m=\u001b[39m CohortVariantViewer(tx_id\u001b[38;5;241m=\u001b[39mmane_tx_id)\n\u001b[1;32m      4\u001b[0m report \u001b[38;5;241m=\u001b[39m viewer\u001b[38;5;241m.\u001b[39mprocess(cohort)\n\u001b[0;32m----> 5\u001b[0m display(\u001b[43mHTML\u001b[49m\u001b[43m(\u001b[49m\u001b[43mreport\u001b[49m\u001b[43m)\u001b[49m)\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.12/site-packages/IPython/core/display.py:434\u001b[0m, in \u001b[0;36mHTML.__init__\u001b[0;34m(self, data, url, filename, metadata)\u001b[0m\n\u001b[1;32m    431\u001b[0m     suffix \u001b[38;5;241m=\u001b[39m data[\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m10\u001b[39m:]\u001b[38;5;241m.\u001b[39mlower()\n\u001b[1;32m    432\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m prefix\u001b[38;5;241m.\u001b[39mstartswith(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m<iframe \u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;129;01mand\u001b[39;00m suffix\u001b[38;5;241m.\u001b[39mendswith(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m</iframe>\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m--> 434\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[43mwarn\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m:\n\u001b[1;32m    435\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConsider using IPython.display.IFrame instead\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    436\u001b[0m \u001b[38;5;28msuper\u001b[39m(HTML, \u001b[38;5;28mself\u001b[39m)\u001b[38;5;241m.\u001b[39m\u001b[38;5;21m__init__\u001b[39m(data\u001b[38;5;241m=\u001b[39mdata, url\u001b[38;5;241m=\u001b[39murl, filename\u001b[38;5;241m=\u001b[39mfilename, metadata\u001b[38;5;241m=\u001b[39mmetadata)\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.12/site-packages/IPython/core/display.py:430\u001b[0m, in \u001b[0;36mHTML.__init__.<locals>.warn\u001b[0;34m()\u001b[0m\n\u001b[1;32m    424\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[1;32m    426\u001b[0m \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[1;32m    427\u001b[0m \u001b[38;5;66;03m# Avoid calling lower() on the entire data, because it could be a\u001b[39;00m\n\u001b[1;32m    428\u001b[0m \u001b[38;5;66;03m# long string and we're only interested in its beginning and end.\u001b[39;00m\n\u001b[1;32m    429\u001b[0m \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[0;32m--> 430\u001b[0m prefix \u001b[38;5;241m=\u001b[39m \u001b[43mdata\u001b[49m\u001b[43m[\u001b[49m\u001b[43m:\u001b[49m\u001b[38;5;241;43m10\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241m.\u001b[39mlower()\n\u001b[1;32m    431\u001b[0m suffix \u001b[38;5;241m=\u001b[39m data[\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m10\u001b[39m:]\u001b[38;5;241m.\u001b[39mlower()\n\u001b[1;32m    432\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m prefix\u001b[38;5;241m.\u001b[39mstartswith(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m<iframe \u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;129;01mand\u001b[39;00m suffix\u001b[38;5;241m.\u001b[39mendswith(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m</iframe>\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[0;31mTypeError\u001b[0m: 'HtmlGpseaReport' object is not subscriptable"
     ]
    }
   ],
   "source": [
    "from gpsea.view import CohortVariantViewer\n",
    "\n",
    "viewer = CohortVariantViewer(tx_id=mane_tx_id)\n",
    "report = viewer.process(cohort)\n",
    "display(HTML(report))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.pcats import configure_hpo_term_analysis\n",
    "analysis = configure_hpo_term_analysis(hpo)\n",
    "\n",
    "from gpsea.analysis.predicate.phenotype import prepare_predicates_for_terms_of_interest\n",
    "pheno_predicates = prepare_predicates_for_terms_of_interest(\n",
    "    cohort=cohort,\n",
    "    hpo=hpo,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "  <style>table {\n",
       "    border-collapse: collapse;\n",
       "    margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "  }\n",
       "\n",
       "\n",
       "  th {\n",
       "    background-color: #f2f2f2;\n",
       "    border: 1px solid #dddddd;\n",
       "    text-align: left;\n",
       "    padding: 2px;\n",
       "  }\n",
       "\n",
       "  tr {\n",
       "    border: 1px solid #dddddd;\n",
       "  }\n",
       "\n",
       "  td {\n",
       "    padding: 2px;\n",
       "    font-weight: bold;\n",
       "  }\n",
       "\n",
       "  tr:nth-child(even) {\n",
       "    background-color: #f2f2f2;\n",
       "  }\n",
       "\n",
       "  .table td,tr {\n",
       "    text-align: left;\n",
       "}\n",
       "\n",
       ".table td:first-child, tr:first-child {\n",
       "    text-align: left;\n",
       "}\n",
       "\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>Phenotype testing report</h1>\n",
       "  <p>Phenotype MTC filter: <em>HPO MTC filter</em></p>\n",
       "  <p>Multiple testing correction: <em>fdr_bh</em></p>\n",
       "  <p>Performed statistical tests for 0 out of the total of 404 HPO terms.</p>\n",
       "    <table>\n",
       "      <caption>Using <em>HPO MTC filter</em>, 404 term(s) were omitted from statistical analysis.</caption>\n",
       "        <tbody>\n",
       "          <tr>\n",
       "            <th>Code</th>\n",
       "            <th>Reason</th>\n",
       "            <th>Count</th>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF01</td>\n",
       "            <td>Skipping term with maximum frequency that was less than threshold 0.4</td>\n",
       "            <td>236</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF08</td>\n",
       "            <td>Skipping general term</td>\n",
       "            <td>75</td>\n",
       "          </tr>\n",
       "          \n",
       "          <tr>\n",
       "            <td>HMF09</td>\n",
       "            <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>\n",
       "            <td>93</td>\n",
       "          </tr>\n",
       "          \n",
       "        </tbody>\n",
       "    </table>\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpsea.view._report.HtmlGpseaReport at 0x31e39a810>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.model import VariantEffect\n",
    "from gpsea.analysis.predicate.genotype import VariantPredicates, biallelic_predicate\n",
    "from gpsea.view import MtcStatsViewer\n",
    "\n",
    "is_missense = VariantPredicates.variant_effect(VariantEffect.MISSENSE_VARIANT, tx_id=mane_tx_id)\n",
    "missense_predicate = biallelic_predicate(\n",
    "    a_predicate=is_missense,\n",
    "    b_predicate=~is_missense,\n",
    "    a_label=\"missense\",\n",
    "    b_label=\"other\",\n",
    "    partitions=((0,1),(2,))\n",
    ")\n",
    "missense_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_predicate=missense_predicate,\n",
    "    pheno_predicates=pheno_predicates,\n",
    ")\n",
    "\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(missense_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Allele group</th>\n",
       "      <th colspan=\"2\" halign=\"left\">missense/missense OR missense/other</th>\n",
       "      <th colspan=\"2\" halign=\"left\">other/other</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [(missense/missense OR missense/other, Count), (missense/missense OR missense/other, Percent), (other/other, Count), (other/other, Percent), (, p values)]\n",
       "Index: []"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "\n",
    "summarize_hpo_analysis(hpo=hpo, result=missense_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>What is the genotype group</th>\n",
       "      <th colspan=\"2\" halign=\"left\">HET</th>\n",
       "      <th colspan=\"2\" halign=\"left\">BIALLELIC_ALT</th>\n",
       "      <th colspan=\"2\" halign=\"left\"></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Truncal ataxia [HP:0002078]</th>\n",
       "      <td>0/8</td>\n",
       "      <td>0%</td>\n",
       "      <td>3/3</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.027273</td>\n",
       "      <td>0.006061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ataxia [HP:0001251]</th>\n",
       "      <td>0/8</td>\n",
       "      <td>0%</td>\n",
       "      <td>3/3</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.027273</td>\n",
       "      <td>0.006061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Macrocephaly [HP:0000256]</th>\n",
       "      <td>0/8</td>\n",
       "      <td>0%</td>\n",
       "      <td>2/2</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.066667</td>\n",
       "      <td>0.022222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Developmental regression [HP:0002376]</th>\n",
       "      <td>0/8</td>\n",
       "      <td>0%</td>\n",
       "      <td>2/4</td>\n",
       "      <td>50%</td>\n",
       "      <td>0.163636</td>\n",
       "      <td>0.090909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Relative macrocephaly [HP:0004482]</th>\n",
       "      <td>0/8</td>\n",
       "      <td>0%</td>\n",
       "      <td>2/4</td>\n",
       "      <td>50%</td>\n",
       "      <td>0.163636</td>\n",
       "      <td>0.090909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ventriculomegaly [HP:0002119]</th>\n",
       "      <td>1/6</td>\n",
       "      <td>17%</td>\n",
       "      <td>3/4</td>\n",
       "      <td>75%</td>\n",
       "      <td>0.278721</td>\n",
       "      <td>0.190476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypotonia [HP:0001252]</th>\n",
       "      <td>7/9</td>\n",
       "      <td>78%</td>\n",
       "      <td>1/4</td>\n",
       "      <td>25%</td>\n",
       "      <td>0.278721</td>\n",
       "      <td>0.216783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Seizure [HP:0001250]</th>\n",
       "      <td>4/4</td>\n",
       "      <td>100%</td>\n",
       "      <td>2/4</td>\n",
       "      <td>50%</td>\n",
       "      <td>0.482143</td>\n",
       "      <td>0.428571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Global developmental delay [HP:0001263]</th>\n",
       "      <td>6/8</td>\n",
       "      <td>75%</td>\n",
       "      <td>4/4</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.515152</td>\n",
       "      <td>0.515152</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "What is the genotype group                HET  ...          \n",
       "                                        Count  ...  p values\n",
       "Truncal ataxia [HP:0002078]               0/8  ...  0.006061\n",
       "Ataxia [HP:0001251]                       0/8  ...  0.006061\n",
       "Macrocephaly [HP:0000256]                 0/8  ...  0.022222\n",
       "Developmental regression [HP:0002376]     0/8  ...  0.090909\n",
       "Relative macrocephaly [HP:0004482]        0/8  ...  0.090909\n",
       "Ventriculomegaly [HP:0002119]             1/6  ...  0.190476\n",
       "Hypotonia [HP:0001252]                    7/9  ...  0.216783\n",
       "Seizure [HP:0001250]                      4/4  ...  0.428571\n",
       "Global developmental delay [HP:0001263]   6/8  ...  0.515152\n",
       "\n",
       "[9 rows x 6 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "\n",
    "report = summarize_hpo_analysis(hpo=hpo, result=result)\n",
    "report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
