{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SLC32A1\n",
    "\n",
    "Heterozyogous variants in SLC32A1 are associated with\n",
    "two autosomal dominant conditions, [developmental and epileptic encephalopathy-114 (DEE114)](https://omim.org/entry/620774)\n",
    "and [generalized epilepsy with febrile seizures plus-12 (GEFSP12)](https://omim.org/entry/620755)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2024-08-13.\n",
      "Using gpsea version 0.7.1.\n"
     ]
    }
   ],
   "source": [
    "import hpotk\n",
    "import gpsea\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo(release='v2024-08-13')\n",
    "print(f'Loaded HPO v{hpo.version}.')\n",
    "print(f\"Using gpsea version {gpsea.__version__}.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### SLC32A1\n",
    "We user the [Matched Annotation from NCBI and EMBL-EBI (MANE)](https://www.ncbi.nlm.nih.gov/refseq/MANE/) transcript and the corresponding protein identifier for SLC32A1."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_symbol = 'SLC32A1'\n",
    "mane_tx_id = 'NM_080552.3'\n",
    "mane_protein_id = 'NP_542119.1' # vesicular inhibitory amino acid transporter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 38 phenopackets\n",
      "Individuals Processed: 100%|██████████| 38/38 [00:00<00:00, 679.44individuals/s]\n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store(\"0.1.21\") as ps:\n",
    "    phenopackets = tuple(ps.iter_cohort_phenopackets(gene_symbol))\n",
    "print(f'Loaded {len(phenopackets)} phenopackets')\n",
    "\n",
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort, validation = load_phenopackets(  \n",
    "    phenopackets=phenopackets,\n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "validation.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "  <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td,\n",
       "    tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px;\n",
       "      /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>GPSEA cohort analysis</h1>\n",
       "  <p>\n",
       "    Successfully loaded 38 individuals.\n",
       "\n",
       "    \n",
       "    19 were recorded as male,\n",
       "    19 as female,\n",
       "    and 0 as unknown sex.\n",
       "\n",
       "    \n",
       "    \n",
       "    No information about individuals' vital status was reported.\n",
       "    \n",
       "\n",
       "    \n",
       "    The cohort included 44 HPO terms.\n",
       "\n",
       "    \n",
       "    27 individuals had disease onset information\n",
       "    and 0 had information about the age of last encounter.\n",
       "\n",
       "\n",
       "    \n",
       "    \n",
       "  </p>\n",
       "\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "\n",
       "  \n",
       "  <table>\n",
       "    <caption style=\"color: black;\">\n",
       "      <h3>Top 10 HPO terms</h3>\n",
       "      A total of 44 HPO terms were used to annotated the cohort.\n",
       "      \n",
       "    </caption>\n",
       "    <tbody>\n",
       "      <tr class=\"strng\">\n",
       "        <th class=\"lft\">HPO Term</th>\n",
       "        <th>ID</th>\n",
       "        <th>Seen in <em>n</em> individuals</th>\n",
       "      </tr>\n",
       "\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Febrile seizure (within the age range of 3 months to 6 years)</td>\n",
       "        <td>HP:0002373</td>\n",
       "        <td>23</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Generalized-onset seizure</td>\n",
       "        <td>HP:0002197</td>\n",
       "        <td>11</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Focal-onset seizure</td>\n",
       "        <td>HP:0007359</td>\n",
       "        <td>7</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Hypotonia</td>\n",
       "        <td>HP:0001252</td>\n",
       "        <td>4</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Developmental regression</td>\n",
       "        <td>HP:0002376</td>\n",
       "        <td>4</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Motor delay</td>\n",
       "        <td>HP:0001270</td>\n",
       "        <td>4</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Delayed speech and language development</td>\n",
       "        <td>HP:0000750</td>\n",
       "        <td>4</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Intellectual disability</td>\n",
       "        <td>HP:0001249</td>\n",
       "        <td>4</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Bilateral tonic-clonic seizure with generalized onset</td>\n",
       "        <td>HP:0025190</td>\n",
       "        <td>3</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Focal impaired awareness seizure</td>\n",
       "        <td>HP:0002384</td>\n",
       "        <td>3</td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "\n",
       "  \n",
       "\n",
       "  \n",
       "  <table>\n",
       "    <caption style=\"color: black;\">\n",
       "      <h3>Top 10 variants</h3>\n",
       "      Variants are shown according to NM_080552.3.\n",
       "      A total of 12 unique variants\n",
       "      were identified in the cohort.\n",
       "    </caption>\n",
       "    <tbody>\n",
       "      <tr class=\"strng\">\n",
       "        <th>Seen in <em>n</em> individuals</th>\n",
       "        <th class=\"lft\">Variant key</th>\n",
       "        <th>Variant Name</th>\n",
       "        <th>Protein Effect</th>\n",
       "        <th>Variant Class</th>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>9</td>\n",
       "        <td class=\"lft\">20_38728050_38728050_T_C</td>\n",
       "        <td>c.989T>C</td>\n",
       "        <td>p.Met330Thr</td>\n",
       "        <td>MISSENSE_VARIANT</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>8</td>\n",
       "        <td class=\"lft\">20_38728464_38728464_T_C</td>\n",
       "        <td>c.1403T>C</td>\n",
       "        <td>p.Leu468Pro</td>\n",
       "        <td>MISSENSE_VARIANT</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>4</td>\n",
       "        <td class=\"lft\">20_38724851_38724851_G_T</td>\n",
       "        <td>c.127G>T</td>\n",
       "        <td>p.Gly43Cys</td>\n",
       "        <td>MISSENSE_VARIANT</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>4</td>\n",
       "        <td class=\"lft\">20_38728394_38728394_C_T</td>\n",
       "        <td>c.1333C>T</td>\n",
       "        <td>p.Leu445Phe</td>\n",
       "        <td>MISSENSE_VARIANT</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>3</td>\n",
       "        <td class=\"lft\">20_38728452_38728452_C_G</td>\n",
       "        <td>c.1391C>G</td>\n",
       "        <td>p.Thr464Arg</td>\n",
       "        <td>MISSENSE_VARIANT</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">20_38728454_38728454_G_A</td>\n",
       "        <td>c.1393G>A</td>\n",
       "        <td>p.Gly465Ser</td>\n",
       "        <td>MISSENSE_VARIANT</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">20_38727849_38727849_T_C</td>\n",
       "        <td>c.788T>C</td>\n",
       "        <td>p.Val263Ala</td>\n",
       "        <td>MISSENSE_VARIANT</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td class=\"lft\">20_38728443_38728443_G_A</td>\n",
       "        <td>c.1382G>A</td>\n",
       "        <td>p.Gly461Asp</td>\n",
       "        <td>MISSENSE_VARIANT</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td class=\"lft\">20_38728026_38728026_T_G</td>\n",
       "        <td>c.965T>G</td>\n",
       "        <td>p.Phe322Cys</td>\n",
       "        <td>MISSENSE_VARIANT</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td class=\"lft\">20_38727848_38727848_G_A</td>\n",
       "        <td>c.787G>A</td>\n",
       "        <td>p.Val263Met</td>\n",
       "        <td>MISSENSE_VARIANT</td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "  \n",
       "  \n",
       "  <table>\n",
       "    <caption style=\"color: black;\">\n",
       "      <h3>Diseases</h3>\n",
       "    </caption>\n",
       "    <tbody>\n",
       "      <tr class=\"strng\">\n",
       "        <th class=\"lft\">Name</th>\n",
       "        <th>ID</th>\n",
       "        <th><em>N</em> diagnosed individuals</th>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Generalized epilepsy with febrile seizures plus, type 12</td>\n",
       "        <td>OMIM:620755</td>\n",
       "        <td>34</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">Developmental and epileptic encephalopathy 114</td>\n",
       "        <td>OMIM:620774</td>\n",
       "        <td>4</td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "  \n",
       "  \n",
       "  \n",
       "  <table>\n",
       "    <caption style=\"color: black;\">\n",
       "      <h3>Variant categories for NM_080552.3</h3>\n",
       "    </caption>\n",
       "    <tbody>\n",
       "      <tr class=\"strng\">\n",
       "        <th class=\"lft\">Variant effect</th>\n",
       "        <th>Annotation count</th>\n",
       "        <th>Percent</th>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td class=\"lft\">MISSENSE_VARIANT</td>\n",
       "        <td>38</td>\n",
       "        <td>100%</td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "  \n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpsea.view._report.HtmlGpseaReport at 0x1314f8560>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import CohortViewer\n",
    "\n",
    "cv = CohortViewer(hpo)\n",
    "cv.process(cohort, transcript_id=mane_tx_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Unrecognized protein feature type: \"Topological domain\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[5], line 12\u001b[0m\n\u001b[1;32m      9\u001b[0m tx_coordinates \u001b[38;5;241m=\u001b[39m txc_service\u001b[38;5;241m.\u001b[39mfetch(mane_tx_id)\n\u001b[1;32m     10\u001b[0m protein_meta \u001b[38;5;241m=\u001b[39m pms\u001b[38;5;241m.\u001b[39mannotate(mane_protein_id)\n\u001b[0;32m---> 12\u001b[0m protein_meta \u001b[38;5;241m=\u001b[39m \u001b[43mProteinMetadata\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfrom_uniprot_json\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     13\u001b[0m \u001b[43m        \u001b[49m\u001b[43mprotein_id\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmane_protein_id\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     14\u001b[0m \u001b[43m        \u001b[49m\u001b[43mlabel\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mSLC31A1\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     15\u001b[0m \u001b[43m        \u001b[49m\u001b[43muniprot_json\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mQ9H598.json\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     16\u001b[0m \u001b[43m        \u001b[49m\u001b[43mprotein_length\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m550\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     17\u001b[0m \u001b[43m)\u001b[49m\n\u001b[1;32m     19\u001b[0m slc32a1_fig, ax \u001b[38;5;241m=\u001b[39m plt\u001b[38;5;241m.\u001b[39msubplots(figsize\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m15\u001b[39m, \u001b[38;5;241m8\u001b[39m))\n\u001b[1;32m     20\u001b[0m visualizer \u001b[38;5;241m=\u001b[39m ProteinVisualizer()\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.12/site-packages/gpsea/model/_protein.py:375\u001b[0m, in \u001b[0;36mProteinMetadata.from_uniprot_json\u001b[0;34m(protein_id, label, uniprot_json, protein_length)\u001b[0m\n\u001b[1;32m    373\u001b[0m region_start \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mint\u001b[39m(locus[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mstart\u001b[39m\u001b[38;5;124m\"\u001b[39m][\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mvalue\u001b[39m\u001b[38;5;124m\"\u001b[39m]) \u001b[38;5;241m-\u001b[39m \u001b[38;5;241m1\u001b[39m  \u001b[38;5;66;03m# convert to 0-based coordinates\u001b[39;00m\n\u001b[1;32m    374\u001b[0m region_end \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mint\u001b[39m(locus[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mend\u001b[39m\u001b[38;5;124m\"\u001b[39m][\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mvalue\u001b[39m\u001b[38;5;124m\"\u001b[39m])\n\u001b[0;32m--> 375\u001b[0m feature_type \u001b[38;5;241m=\u001b[39m \u001b[43mFeatureType\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfrom_string\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfeature\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mtype\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    376\u001b[0m finfo \u001b[38;5;241m=\u001b[39m FeatureInfo(\n\u001b[1;32m    377\u001b[0m     name\u001b[38;5;241m=\u001b[39mregion_name, region\u001b[38;5;241m=\u001b[39mRegion(start\u001b[38;5;241m=\u001b[39mregion_start, end\u001b[38;5;241m=\u001b[39mregion_end)\n\u001b[1;32m    378\u001b[0m )\n\u001b[1;32m    379\u001b[0m pfeature \u001b[38;5;241m=\u001b[39m ProteinFeature\u001b[38;5;241m.\u001b[39mcreate(info\u001b[38;5;241m=\u001b[39mfinfo, feature_type\u001b[38;5;241m=\u001b[39mfeature_type)\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.12/site-packages/gpsea/model/_protein.py:144\u001b[0m, in \u001b[0;36mFeatureType.from_string\u001b[0;34m(category)\u001b[0m\n\u001b[1;32m    142\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m FeatureType\u001b[38;5;241m.\u001b[39mZINC_FINGER\n\u001b[1;32m    143\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 144\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mUnrecognized protein feature type: \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mcategory\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "\u001b[0;31mValueError\u001b[0m: Unrecognized protein feature type: \"Topological domain\""
     ]
    }
   ],
   "source": [
    "from gpsea.model.genome import GRCh38\n",
    "from gpsea.model import ProteinMetadata\n",
    "from gpsea.preprocessing import configure_default_protein_metadata_service, VVMultiCoordinateService\n",
    "from gpsea.view import ProteinVisualizer\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "txc_service = VVMultiCoordinateService(genome_build=GRCh38)\n",
    "pms = configure_default_protein_metadata_service()\n",
    "tx_coordinates = txc_service.fetch(mane_tx_id)\n",
    "protein_meta = pms.annotate(mane_protein_id)\n",
    "\n",
    "protein_meta = ProteinMetadata.from_uniprot_json(\n",
    "        protein_id=mane_protein_id,\n",
    "        label=\"SLC31A1\",\n",
    "        uniprot_json=\"Q9H598.json\",\n",
    "        protein_length=550,\n",
    ")\n",
    "\n",
    "slc32a1_fig, ax = plt.subplots(figsize=(15, 8))\n",
    "visualizer = ProteinVisualizer()\n",
    "visualizer.draw_protein_diagram(\n",
    "    tx_coordinates,\n",
    "    protein_meta,\n",
    "    cohort,\n",
    "    ax=ax,\n",
    ")\n",
    "slc32a1_fig.tight_layout()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Strategy\n",
    "No genotype-phenotype correlations were identified in the published literature. All pathogenic variants are missense. Therefore, we will search for the most common variant and for N-terminal vs. C-terminal."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.pcats import configure_hpo_term_analysis\n",
    "analysis = configure_hpo_term_analysis(hpo)\n",
    "\n",
    "from gpsea.analysis.predicate.phenotype import prepare_predicates_for_terms_of_interest\n",
    "pheno_predicates = prepare_predicates_for_terms_of_interest(\n",
    "    cohort=cohort,\n",
    "    hpo=hpo,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.model.genome import Region\n",
    "from gpsea.view import MtcStatsViewer\n",
    "from gpsea.analysis.predicate.genotype import VariantPredicates, monoallelic_predicate\n",
    "\n",
    "n_term = VariantPredicates.region(region=Region(0, 360), tx_id=mane_tx_id)\n",
    "nterm_predicate = monoallelic_predicate(\n",
    "    a_predicate=n_term,\n",
    "    b_predicate=~n_term,\n",
    "    a_label=\"N term\",\n",
    "    b_label=\"other\"\n",
    ")\n",
    "\n",
    "nterm_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_predicate=nterm_predicate,\n",
    "    pheno_predicates=pheno_predicates,\n",
    ")\n",
    "\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(nterm_result)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "\n",
    "summarize_hpo_analysis(hpo=hpo, result=nterm_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.predicate.genotype import sex_predicate\n",
    "\n",
    "mf_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_predicate=sex_predicate(),\n",
    "    pheno_predicates=pheno_predicates,\n",
    ")\n",
    "\n",
    "summarize_hpo_analysis(hpo=hpo, result=mf_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.predicate.genotype import VariantPredicates, monoallelic_predicate\n",
    "met330thr = VariantPredicates.variant_key(\"20_38728050_38728050_T_C\") # c.989T>C\tp.Met330Thr\n",
    "met330thr_predicate = monoallelic_predicate(\n",
    "    a_predicate=met330thr,\n",
    "    b_predicate= ~met330thr,\n",
    "    a_label='p.Met330Thr',\n",
    "     b_label= 'Other variant'\n",
    ")\n",
    "met330thr_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_predicate=met330thr_predicate,\n",
    "    pheno_predicates=pheno_predicates,\n",
    ")\n",
    "summarize_hpo_analysis(hpo=hpo, result=met330thr_result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpseacs.report import GpseaAnalysisReport, GPAnalysisResultSummary\n",
    "\n",
    "fet_results = (\n",
    "    GPAnalysisResultSummary.from_multi(\n",
    "        result=nterm_result,\n",
    "    ),\n",
    "    GPAnalysisResultSummary.from_multi(\n",
    "        result=met330thr_result,\n",
    "    ),\n",
    "     GPAnalysisResultSummary.from_multi(\n",
    "        result=mf_result,\n",
    "    ),\n",
    ")\n",
    "\n",
    "caption = \"No significant correlation identified.\"\n",
    "report = GpseaAnalysisReport(name=gene_symbol, \n",
    "                             cohort=cohort, \n",
    "                             fet_results=fet_results,\n",
    "                             gene_symbol=gene_symbol,\n",
    "                             mane_tx_id=mane_tx_id,\n",
    "                             mane_protein_id=mane_protein_id,\n",
    "                             caption=caption)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpseacs.report import GpseaNotebookSummarizer\n",
    "summarizer = GpseaNotebookSummarizer(hpo=hpo, gpsea_version=gpsea.__version__)\n",
    "summarizer.summarize_report(report=report)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summarizer.process_latex(report=report, protein_fig=slc32a1_fig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
