{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TBX1 Cohort Analysis Report\n",
    "\n",
    "TBX1 or T box 1 is a transcription factor crucial for the prenatal establishment of the thymus [Lindsay et al. (2001)](https://pubmed.ncbi.nlm.nih.gov/11242049/). In humans, monoallelic loss-of-function mutations in TBX1 have been associated with the development of velo-cardio-facial, or DiGeorge, syndrome [Merscher et al. (2001)](https://pubmed.ncbi.nlm.nih.gov/11239417/). Interestingly, missense mutations leading to increased TBX1 activity also result in phenotypically similar syndrome [Zweier et al. (2007)](https://pubmed.ncbi.nlm.nih.gov/17273972/) - characterized by velopharyngeal insufficiency, hypoplastic thymus, immunodeficiency, hypoparathyroidism, craniofacial dysmorphia, hearing impairment and cardiac defect [Papangeli et al. (2013)](https://pubmed.ncbi.nlm.nih.gov/23799583/). All mutations in TBX1 seem highly variably penetrant, with some patients presenting without signs of immunodeficiency, while others present with life-threatening congenital athymia requiring corrective treatment with a thymic transplant [Kreins et al. (2021)](https://pubmed.ncbi.nlm.nih.gov/33815417/)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2024-08-13\n",
      "Using gpsea version 0.9.1\n"
     ]
    }
   ],
   "source": [
    "import hpotk\n",
    "import gpsea\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo(release='v2024-08-13')\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using gpsea version {gpsea.__version__}\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### TBX1\n",
    "We used the [Matched Annotation from NCBI and EMBL-EBI (MANE)](https://www.ncbi.nlm.nih.gov/refseq/MANE/) transcript and the corresponding protein identifier for TBX1."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_symbol = 'TBX1'\n",
    "mane_tx_id =  'NM_001379200.1' # the MANE select transcript.\n",
    "mane_protein_id ='NP_001366129.1'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 100%|██████████| 26/26 [00:00<00:00, 794.06individuals/s]\n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "\n",
    "phenopacket_store_release = '0.1.23' \n",
    "registry = configure_phenopacket_registry()\n",
    "\n",
    "with registry.open_phenopacket_store(release=phenopacket_store_release) as ps:\n",
    "    phenopackets = tuple(ps.iter_cohort_phenopackets(gene_symbol))\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "\n",
    "cohort, qc = load_phenopackets(\n",
    "    phenopackets=phenopackets, \n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "\n",
    "qc.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "\n",
       "<h1>GPSEA cohort analysis</h1>\n",
       "<div id=\"cohort-summary\">\n",
       "  <p>\n",
       "    Successfully loaded 26 individuals.\n",
       "\n",
       "    \n",
       "    15 were recorded as male,\n",
       "    10 as female,\n",
       "    and 1 as unknown sex.\n",
       "\n",
       "    \n",
       "    \n",
       "    No information about individuals' vital status was reported.\n",
       "    \n",
       "\n",
       "    \n",
       "    1 individual had disease onset information\n",
       "    and 0 had information about the age of last encounter.\n",
       "  </p>\n",
       "\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "</div>\n",
       "\n",
       "\n",
       "<div class=\"cards\">\n",
       "  \n",
       "  <div id=\"hpo\" class=\"card\">\n",
       "    <h2>HPO terms</h2>\n",
       "    \n",
       "    \n",
       "    <table>\n",
       "      The cohort included 25 distinct HPO terms.\n",
       "      <caption>\n",
       "        \n",
       "        Top 10 most common HPO terms.\n",
       "        \n",
       "        \n",
       "      </caption>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>HPO Term</th>\n",
       "        </tr>\n",
       "\n",
       "        \n",
       "        <tr>\n",
       "          <td>20</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0001999\">Abnormal facial shape</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>14</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000829\">Hypoparathyroidism</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>13</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000369\">Low-set ears</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>13</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000220\">Velopharyngeal insufficiency</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>12</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000316\">Hypertelorism</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>11</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000347\">Micrognathia</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>9</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000581\">Blepharophimosis</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>8</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0001263\">Global developmental delay</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>5</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000407\">Sensorineural hearing impairment</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>5</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000460\">Narrow nose</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div id=\"measurements\" class=\"card\">\n",
       "    <h2>Measurements</h2>\n",
       "    \n",
       "    \n",
       "    No data regarding measurement assays were provided.\n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div id=\"diseases\" class=\"card\">\n",
       "    <h2>Diseases</h2>\n",
       "    \n",
       "    \n",
       "    The cohort members were diagnosed with 1 disease.\n",
       "    <table>\n",
       "      \n",
       "\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>Disease</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>26</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/disease/OMIM:188400\">DiGeorge syndrome</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  \n",
       "  <div class=\"card\">\n",
       "    <h2>Variants</h2>\n",
       "    <p>\n",
       "      A total of 12 unique variants\n",
       "      were identified in the cohort. Variants were annotated with respect to NM_001379200.1.\n",
       "    </p>\n",
       "    <table>\n",
       "      <caption>\n",
       "        <h3>Top 10 variants</h3>\n",
       "      </caption>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>Variant key</th>\n",
       "          <th>HGVS</th>\n",
       "          <th>Variant Class</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>5</td>\n",
       "          <td>22_19766631_19766632_TA_T</td>\n",
       "          <td>c.1280del (p.Tyr427PhefsTer42)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>4</td>\n",
       "          <td>22_19766537_19766538_CG_T</td>\n",
       "          <td>c.1185_1186delinsT (p.Gly396AlafsTer73)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>3</td>\n",
       "          <td>22_19766003_19766003_G_C</td>\n",
       "          <td>c.1036+1G>C (None)</td>\n",
       "          <td>SPLICE_DONOR_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>3</td>\n",
       "          <td>22_19766601_19766602_TC_T</td>\n",
       "          <td>c.1250del (p.Ser417TrpfsTer52)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>3</td>\n",
       "          <td>22_19766671_19766694_GCCGGCCGGCGCCCTACCCGCTGC_G</td>\n",
       "          <td>c.1326_1348del (p.Pro444TrpfsTer174)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>22_19764224_19764224_C_G</td>\n",
       "          <td>c.609C>G (p.His203Gln)</td>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>1</td>\n",
       "          <td>22_19766651_19766659_CCACTATCT_C</td>\n",
       "          <td>c.1301_1308del (p.His434ArgfsTer189)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>1</td>\n",
       "          <td>22_19765959_19765959_G_GAACCCCGTGGC</td>\n",
       "          <td>c.994_1004dup (p.Ser336ThrfsTer47)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>1</td>\n",
       "          <td>22_19765921_19765921_G_A</td>\n",
       "          <td>c.955G>A (p.Gly319Ser)</td>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>1</td>\n",
       "          <td>22_19766765_19766765_C_--31bp--</td>\n",
       "          <td>c.1426_1455dup (p.Ala476_Ala485dup)</td>\n",
       "          <td>INFRAME_INSERTION</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div class=\"card\">\n",
       "    \n",
       "    <h2>Variant effects</h2>\n",
       "    <p>\n",
       "      The effects were predicted for NM_001379200.1\n",
       "    </p>\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th>Variant effect</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "          <td>17 (65%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "          <td>4 (15%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>SPLICE_DONOR_VARIANT</td>\n",
       "          <td>3 (12%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>INFRAME_INSERTION</td>\n",
       "          <td>1 (4%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>INFRAME_DELETION</td>\n",
       "          <td>1 (4%)</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x10ed09130>"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import CohortViewer\n",
    "\n",
    "viewer = CohortViewer(hpo)\n",
    "viewer.process(cohort=cohort, transcript_id=mane_tx_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The corresponding UnitProt entry (O43435) has a Disordered Region\t(23-72) and compositional bias at residues 37-72, and the length of the sequence is 398. \n",
    "# The API code that we use to generate the protein graphic is failing because no proteins found for ID NP_001366129.1 (the MANE select)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Variant alleles</h1>\n",
       "<div class=\"card\">\n",
       "  <table>\n",
       "    <caption>\n",
       "      A total of 12 unique alleles were identified in the cohort.\n",
       "    </caption>\n",
       "\n",
       "    <tbody>\n",
       "      <tr>\n",
       "        <th>Count</th>\n",
       "        <th>Variant key</th>\n",
       "        <th>HGVS</th>\n",
       "        <th>Overlapping Exons</th>\n",
       "        <th>Effects</th>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>5</td>\n",
       "        <td>22_19766631_19766632_TA_T</td>\n",
       "        <td>c.1280del (p.Tyr427PhefsTer42)</td>\n",
       "        <td>7</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>4</td>\n",
       "        <td>22_19766537_19766538_CG_T</td>\n",
       "        <td>c.1185_1186delinsT (p.Gly396AlafsTer73)</td>\n",
       "        <td>7</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>3</td>\n",
       "        <td>22_19766671_19766694_GCCGGCCGGCGCCCTACCCGCTGC_G</td>\n",
       "        <td>c.1326_1348del (p.Pro444TrpfsTer174)</td>\n",
       "        <td>7</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>3</td>\n",
       "        <td>22_19766003_19766003_G_C</td>\n",
       "        <td>c.1036+1G>C (-)</td>\n",
       "        <td>-</td>\n",
       "        <td>splice donor</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>3</td>\n",
       "        <td>22_19766601_19766602_TC_T</td>\n",
       "        <td>c.1250del (p.Ser417TrpfsTer52)</td>\n",
       "        <td>7</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>2</td>\n",
       "        <td>22_19764224_19764224_C_G</td>\n",
       "        <td>c.609C>G (p.His203Gln)</td>\n",
       "        <td>3</td>\n",
       "        <td>missense</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>22_19766765_19766765_C_--31bp--</td>\n",
       "        <td>c.1426_1455dup (p.Ala476_Ala485dup)</td>\n",
       "        <td>7</td>\n",
       "        <td>inframe insertion</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>22_19760998_19761055_--58bp--_C</td>\n",
       "        <td>c.173_229del (p.Arg58_Pro76del)</td>\n",
       "        <td>1</td>\n",
       "        <td>inframe deletion</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>22_19765921_19765921_G_A</td>\n",
       "        <td>c.955G>A (p.Gly319Ser)</td>\n",
       "        <td>6</td>\n",
       "        <td>missense</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>22_19765959_19765959_G_GAACCCCGTGGC</td>\n",
       "        <td>c.994_1004dup (p.Ser336ThrfsTer47)</td>\n",
       "        <td>6</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>22_19766651_19766659_CCACTATCT_C</td>\n",
       "        <td>c.1301_1308del (p.His434ArgfsTer189)</td>\n",
       "        <td>7</td>\n",
       "        <td>frameshift</td>\n",
       "      </tr>\n",
       "      \n",
       "      <tr>\n",
       "        <td>1</td>\n",
       "        <td>22_19763273_19763273_T_A</td>\n",
       "        <td>c.470T>A (p.Phe157Tyr)</td>\n",
       "        <td>2</td>\n",
       "        <td>missense</td>\n",
       "      </tr>\n",
       "      \n",
       "    </tbody>\n",
       "  </table>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x16a60f8a0>"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import CohortVariantViewer\n",
    "cvv = CohortVariantViewer(tx_id=mane_tx_id)\n",
    "cvv.process(cohort=cohort)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genotype-Phenotype Correlation (GPC) Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.pcats import configure_hpo_term_analysis\n",
    "from gpsea.analysis.clf import prepare_classifiers_for_terms_of_interest\n",
    "\n",
    "analysis = configure_hpo_term_analysis(hpo)\n",
    "\n",
    "pheno_clfs = prepare_classifiers_for_terms_of_interest(\n",
    "    cohort=cohort,\n",
    "    hpo=hpo,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Phenotype testing report</h1>\n",
       "<div class=\"cards\">\n",
       "  <div class=\"card\">\n",
       "    <ul>\n",
       "      <li>Phenotype MTC filter: <em>HPO MTC filter</em></li>\n",
       "      <li>Multiple testing correction: <em>fdr_bh</em></li>\n",
       "    </ul>\n",
       "  </div>\n",
       "  <div class=\"card\">\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr>\n",
       "          <th>Code</th>\n",
       "          <th>Reason</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF01</td>\n",
       "          <td>Skipping term with maximum frequency that was less than threshold 0.4</td>\n",
       "          <td>13</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF03</td>\n",
       "          <td>Skipping term because of a child term with the same individual counts</td>\n",
       "          <td>11</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF05</td>\n",
       "          <td>Skipping term because one genotype had zero observations</td>\n",
       "          <td>1</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF08</td>\n",
       "          <td>Skipping general term</td>\n",
       "          <td>42</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF09</td>\n",
       "          <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>\n",
       "          <td>41</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "      <caption>\n",
       "        Performed statistical tests for 12 out of the total of 120 HPO terms.\n",
       "      </caption>\n",
       "    </table>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x16b7ac050>"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.model import VariantEffect\n",
    "from gpsea.view import MtcStatsViewer\n",
    "from gpsea.analysis.clf import monoallelic_classifier\n",
    "from gpsea.analysis.predicate import variant_effect\n",
    "\n",
    "missense = variant_effect(VariantEffect.MISSENSE_VARIANT, mane_tx_id)\n",
    "\n",
    "missense_clf = monoallelic_classifier(\n",
    "    a_predicate=missense,\n",
    "    b_predicate=~missense,\n",
    "    a_label=\"Missense\",\n",
    "    b_label=\"Other\"\n",
    ")\n",
    "\n",
    "missense_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_clf=missense_clf,\n",
    "    pheno_clfs=pheno_clfs,\n",
    ")\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(missense_result)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Allele group</th>\n",
       "      <th colspan=\"2\" halign=\"left\">Missense</th>\n",
       "      <th colspan=\"2\" halign=\"left\">Other</th>\n",
       "      <th colspan=\"2\" halign=\"left\"></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Blepharophimosis [HP:0000581]</th>\n",
       "      <td>4/4</td>\n",
       "      <td>100%</td>\n",
       "      <td>5/14</td>\n",
       "      <td>36%</td>\n",
       "      <td>0.477124</td>\n",
       "      <td>0.082353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypoparathyroidism [HP:0000829]</th>\n",
       "      <td>1/4</td>\n",
       "      <td>25%</td>\n",
       "      <td>13/17</td>\n",
       "      <td>76%</td>\n",
       "      <td>0.477124</td>\n",
       "      <td>0.087719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Micrognathia [HP:0000347]</th>\n",
       "      <td>4/4</td>\n",
       "      <td>100%</td>\n",
       "      <td>7/14</td>\n",
       "      <td>50%</td>\n",
       "      <td>0.477124</td>\n",
       "      <td>0.119281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Low-set ears [HP:0000369]</th>\n",
       "      <td>4/4</td>\n",
       "      <td>100%</td>\n",
       "      <td>9/14</td>\n",
       "      <td>64%</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>0.277778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sensorineural hearing impairment [HP:0000407]</th>\n",
       "      <td>1/1</td>\n",
       "      <td>100%</td>\n",
       "      <td>4/13</td>\n",
       "      <td>31%</td>\n",
       "      <td>0.857143</td>\n",
       "      <td>0.357143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal facial shape [HP:0001999]</th>\n",
       "      <td>4/4</td>\n",
       "      <td>100%</td>\n",
       "      <td>16/21</td>\n",
       "      <td>76%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.549407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypertelorism [HP:0000316]</th>\n",
       "      <td>2/2</td>\n",
       "      <td>100%</td>\n",
       "      <td>10/13</td>\n",
       "      <td>77%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal location of ears [HP:0000357]</th>\n",
       "      <td>4/4</td>\n",
       "      <td>100%</td>\n",
       "      <td>9/9</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aplasia/Hypoplasia of the mandible [HP:0009118]</th>\n",
       "      <td>4/4</td>\n",
       "      <td>100%</td>\n",
       "      <td>7/7</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal axial skeleton morphology [HP:0009121]</th>\n",
       "      <td>4/4</td>\n",
       "      <td>100%</td>\n",
       "      <td>7/7</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Velopharyngeal insufficiency [HP:0000220]</th>\n",
       "      <td>2/4</td>\n",
       "      <td>50%</td>\n",
       "      <td>11/20</td>\n",
       "      <td>55%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal soft palate morphology [HP:0100736]</th>\n",
       "      <td>2/2</td>\n",
       "      <td>100%</td>\n",
       "      <td>11/11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Allele group                                    Missense          Other  \\\n",
       "                                                   Count Percent  Count   \n",
       "Blepharophimosis [HP:0000581]                        4/4    100%   5/14   \n",
       "Hypoparathyroidism [HP:0000829]                      1/4     25%  13/17   \n",
       "Micrognathia [HP:0000347]                            4/4    100%   7/14   \n",
       "Low-set ears [HP:0000369]                            4/4    100%   9/14   \n",
       "Sensorineural hearing impairment [HP:0000407]        1/1    100%   4/13   \n",
       "Abnormal facial shape [HP:0001999]                   4/4    100%  16/21   \n",
       "Hypertelorism [HP:0000316]                           2/2    100%  10/13   \n",
       "Abnormal location of ears [HP:0000357]               4/4    100%    9/9   \n",
       "Aplasia/Hypoplasia of the mandible [HP:0009118]      4/4    100%    7/7   \n",
       "Abnormal axial skeleton morphology [HP:0009121]      4/4    100%    7/7   \n",
       "Velopharyngeal insufficiency [HP:0000220]            2/4     50%  11/20   \n",
       "Abnormal soft palate morphology [HP:0100736]         2/2    100%  11/11   \n",
       "\n",
       "Allele group                                                                \\\n",
       "                                                Percent Corrected p values   \n",
       "Blepharophimosis [HP:0000581]                       36%           0.477124   \n",
       "Hypoparathyroidism [HP:0000829]                     76%           0.477124   \n",
       "Micrognathia [HP:0000347]                           50%           0.477124   \n",
       "Low-set ears [HP:0000369]                           64%           0.833333   \n",
       "Sensorineural hearing impairment [HP:0000407]       31%           0.857143   \n",
       "Abnormal facial shape [HP:0001999]                  76%           1.000000   \n",
       "Hypertelorism [HP:0000316]                          77%           1.000000   \n",
       "Abnormal location of ears [HP:0000357]             100%           1.000000   \n",
       "Aplasia/Hypoplasia of the mandible [HP:0009118]    100%           1.000000   \n",
       "Abnormal axial skeleton morphology [HP:0009121]    100%           1.000000   \n",
       "Velopharyngeal insufficiency [HP:0000220]           55%           1.000000   \n",
       "Abnormal soft palate morphology [HP:0100736]       100%           1.000000   \n",
       "\n",
       "Allele group                                               \n",
       "                                                 p values  \n",
       "Blepharophimosis [HP:0000581]                    0.082353  \n",
       "Hypoparathyroidism [HP:0000829]                  0.087719  \n",
       "Micrognathia [HP:0000347]                        0.119281  \n",
       "Low-set ears [HP:0000369]                        0.277778  \n",
       "Sensorineural hearing impairment [HP:0000407]    0.357143  \n",
       "Abnormal facial shape [HP:0001999]               0.549407  \n",
       "Hypertelorism [HP:0000316]                       1.000000  \n",
       "Abnormal location of ears [HP:0000357]           1.000000  \n",
       "Aplasia/Hypoplasia of the mandible [HP:0009118]  1.000000  \n",
       "Abnormal axial skeleton morphology [HP:0009121]  1.000000  \n",
       "Velopharyngeal insufficiency [HP:0000220]        1.000000  \n",
       "Abnormal soft palate morphology [HP:0100736]     1.000000  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "\n",
    "summarize_hpo_analysis(hpo=hpo, result=missense_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Allele group</th>\n",
       "      <th colspan=\"2\" halign=\"left\">Tyr418PhefsTer42</th>\n",
       "      <th colspan=\"2\" halign=\"left\">Other</th>\n",
       "      <th colspan=\"2\" halign=\"left\"></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Count</th>\n",
       "      <th>Percent</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Global developmental delay [HP:0001263]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>3/20</td>\n",
       "      <td>15%</td>\n",
       "      <td>0.014756</td>\n",
       "      <td>0.001054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Narrow nose [HP:0000460]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>0/6</td>\n",
       "      <td>0%</td>\n",
       "      <td>0.015152</td>\n",
       "      <td>0.002165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Blepharophimosis [HP:0000581]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>4/13</td>\n",
       "      <td>31%</td>\n",
       "      <td>0.137255</td>\n",
       "      <td>0.029412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Velopharyngeal insufficiency [HP:0000220]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>8/19</td>\n",
       "      <td>42%</td>\n",
       "      <td>0.144022</td>\n",
       "      <td>0.041149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Micrognathia [HP:0000347]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>6/13</td>\n",
       "      <td>46%</td>\n",
       "      <td>0.283660</td>\n",
       "      <td>0.101307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Low-set ears [HP:0000369]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>8/13</td>\n",
       "      <td>62%</td>\n",
       "      <td>0.580882</td>\n",
       "      <td>0.248950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypertelorism [HP:0000316]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>7/10</td>\n",
       "      <td>70%</td>\n",
       "      <td>0.902875</td>\n",
       "      <td>0.505495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal facial shape [HP:0001999]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>15/20</td>\n",
       "      <td>75%</td>\n",
       "      <td>0.902875</td>\n",
       "      <td>0.544043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sensorineural hearing impairment [HP:0000407]</th>\n",
       "      <td>1/5</td>\n",
       "      <td>20%</td>\n",
       "      <td>4/9</td>\n",
       "      <td>44%</td>\n",
       "      <td>0.902875</td>\n",
       "      <td>0.580420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal location of ears [HP:0000357]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>8/8</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aplasia/Hypoplasia of the mandible [HP:0009118]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>6/6</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal axial skeleton morphology [HP:0009121]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>6/6</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypoparathyroidism [HP:0000829]</th>\n",
       "      <td>3/5</td>\n",
       "      <td>60%</td>\n",
       "      <td>11/16</td>\n",
       "      <td>69%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal soft palate morphology [HP:0100736]</th>\n",
       "      <td>5/5</td>\n",
       "      <td>100%</td>\n",
       "      <td>8/8</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Allele group                                    Tyr418PhefsTer42          \\\n",
       "                                                           Count Percent   \n",
       "Global developmental delay [HP:0001263]                      5/5    100%   \n",
       "Narrow nose [HP:0000460]                                     5/5    100%   \n",
       "Blepharophimosis [HP:0000581]                                5/5    100%   \n",
       "Velopharyngeal insufficiency [HP:0000220]                    5/5    100%   \n",
       "Micrognathia [HP:0000347]                                    5/5    100%   \n",
       "Low-set ears [HP:0000369]                                    5/5    100%   \n",
       "Hypertelorism [HP:0000316]                                   5/5    100%   \n",
       "Abnormal facial shape [HP:0001999]                           5/5    100%   \n",
       "Sensorineural hearing impairment [HP:0000407]                1/5     20%   \n",
       "Abnormal location of ears [HP:0000357]                       5/5    100%   \n",
       "Aplasia/Hypoplasia of the mandible [HP:0009118]              5/5    100%   \n",
       "Abnormal axial skeleton morphology [HP:0009121]              5/5    100%   \n",
       "Hypoparathyroidism [HP:0000829]                              3/5     60%   \n",
       "Abnormal soft palate morphology [HP:0100736]                 5/5    100%   \n",
       "\n",
       "Allele group                                     Other          \\\n",
       "                                                 Count Percent   \n",
       "Global developmental delay [HP:0001263]           3/20     15%   \n",
       "Narrow nose [HP:0000460]                           0/6      0%   \n",
       "Blepharophimosis [HP:0000581]                     4/13     31%   \n",
       "Velopharyngeal insufficiency [HP:0000220]         8/19     42%   \n",
       "Micrognathia [HP:0000347]                         6/13     46%   \n",
       "Low-set ears [HP:0000369]                         8/13     62%   \n",
       "Hypertelorism [HP:0000316]                        7/10     70%   \n",
       "Abnormal facial shape [HP:0001999]               15/20     75%   \n",
       "Sensorineural hearing impairment [HP:0000407]      4/9     44%   \n",
       "Abnormal location of ears [HP:0000357]             8/8    100%   \n",
       "Aplasia/Hypoplasia of the mandible [HP:0009118]    6/6    100%   \n",
       "Abnormal axial skeleton morphology [HP:0009121]    6/6    100%   \n",
       "Hypoparathyroidism [HP:0000829]                  11/16     69%   \n",
       "Abnormal soft palate morphology [HP:0100736]       8/8    100%   \n",
       "\n",
       "Allele group                                                                  \n",
       "                                                Corrected p values  p values  \n",
       "Global developmental delay [HP:0001263]                   0.014756  0.001054  \n",
       "Narrow nose [HP:0000460]                                  0.015152  0.002165  \n",
       "Blepharophimosis [HP:0000581]                             0.137255  0.029412  \n",
       "Velopharyngeal insufficiency [HP:0000220]                 0.144022  0.041149  \n",
       "Micrognathia [HP:0000347]                                 0.283660  0.101307  \n",
       "Low-set ears [HP:0000369]                                 0.580882  0.248950  \n",
       "Hypertelorism [HP:0000316]                                0.902875  0.505495  \n",
       "Abnormal facial shape [HP:0001999]                        0.902875  0.544043  \n",
       "Sensorineural hearing impairment [HP:0000407]             0.902875  0.580420  \n",
       "Abnormal location of ears [HP:0000357]                    1.000000  1.000000  \n",
       "Aplasia/Hypoplasia of the mandible [HP:0009118]           1.000000  1.000000  \n",
       "Abnormal axial skeleton morphology [HP:0009121]           1.000000  1.000000  \n",
       "Hypoparathyroidism [HP:0000829]                           1.000000  1.000000  \n",
       "Abnormal soft palate morphology [HP:0100736]              1.000000  1.000000  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.analysis.predicate import variant_key\n",
    "\n",
    "Tyr418PhefsTer42 = variant_key(\"22_19766631_19766632_TA_T\") #  c.1253del; p.Tyr418PhefsTer42; most common variant\n",
    "\n",
    "Tyr418PhefsTer42_clf = monoallelic_classifier(\n",
    "    a_predicate=Tyr418PhefsTer42,\n",
    "    b_predicate=~Tyr418PhefsTer42,\n",
    "    a_label=\"Tyr418PhefsTer42\",\n",
    "    b_label=\"Other\"\n",
    ")\n",
    "\n",
    "Tyr418PhefsTer42_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_clf=Tyr418PhefsTer42_clf,\n",
    "    pheno_clfs=pheno_clfs,\n",
    ")\n",
    "\n",
    "summarize_hpo_analysis(hpo=hpo, result=Tyr418PhefsTer42_result)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpseacs.report import GpseaAnalysisReport, GPAnalysisResultSummary\n",
    "\n",
    "f_results = (\n",
    "  GPAnalysisResultSummary.from_multi( result=missense_result,  ),\n",
    "  GPAnalysisResultSummary.from_multi( result=Tyr418PhefsTer42_result,  ),\n",
    ")\n",
    "\n",
    "report = GpseaAnalysisReport(name=gene_symbol, \n",
    "                             cohort=cohort, \n",
    "                             fet_results=f_results,\n",
    "                             gene_symbol=gene_symbol,\n",
    "                             mane_tx_id=mane_tx_id,\n",
    "                             mane_protein_id=mane_protein_id,\n",
    "                             caption=\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "      min-width: 100px;\n",
       "      border: 1px solid #ddd;\n",
       "      padding: 8px;\n",
       "        \n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    p {\n",
       "      width: 75%;\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>TBX1 Summary</h1>\n",
       "  <p>The cohort comprised 26 individuals (10 females, 15 males, 1 with unknown sex). A total of 25 HPO terms were used to annotate the cohort. Disease diagnosis: DiGeorge syndrome (OMIM:188400).  A total of 26 unique variant alleles were found in TBX1 (transcript: NM_001379200.1, protein id: NP_001366129.1). HPO version: 2024-08-13. GPSEA version: 0.9.1.</p>\n",
       "  \n",
       "  \n",
       "    \n",
       "      <table>\n",
       "        <caption style=\"color: black;\">\n",
       "            Fisher Exact Test performed to compare HPO annotation frequency with respect to Missense and Other.\n",
       "        </caption>\n",
       "            <tbody>\n",
       "                <tr class=\"strng\">\n",
       "                    <th>Genotype (A)</th>\n",
       "                    <th>Genotype (B)</th>\n",
       "                  <th>Tests performed</th>\n",
       "                  <th>Significant tests</th>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td class=\"lft\">Missense</td>\n",
       "                    <td class=\"lft\">Other</td>\n",
       "                    <td class=\"lft\">12</td>\n",
       "                    <td class=\"lft\">0</td>\n",
       "                </tr>\n",
       "              </tbody>\n",
       "        </table>\n",
       "      \n",
       "\n",
       "  \n",
       "    \n",
       "    <table>\n",
       "      <caption style=\"color: black;\">\n",
       "          Fisher Exact Test performed to compare HPO annotation frequency with respect to Tyr418PhefsTer42 and Other. Total of\n",
       "          14 tests were performed.\n",
       "      </caption>\n",
       "          <tbody>\n",
       "              <tr class=\"strng\">\n",
       "                  <th>HPO Term</th>\n",
       "                  <th>Tyr418PhefsTer42</th>\n",
       "                  <th>Other</th>\n",
       "                  <th>p-val</th>\n",
       "                  <th>adj. p-val</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                  <td class=\"lft\">Global developmental delay [HP:0001263]</td>\n",
       "                  <td class=\"lft\">5/5 (100%)</td>\n",
       "                  <td class=\"lft\">3/20 (15%)</td>\n",
       "                  <td class=\"lft\">0.001</td>\n",
       "                  <td class=\"lft\">0.015</td>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                  <td class=\"lft\">Narrow nose [HP:0000460]</td>\n",
       "                  <td class=\"lft\">5/5 (100%)</td>\n",
       "                  <td class=\"lft\">0/6 (0%)</td>\n",
       "                  <td class=\"lft\">0.002</td>\n",
       "                  <td class=\"lft\">0.015</td>\n",
       "              </tr>\n",
       "              \n",
       "            </tbody>\n",
       "      </table>\n",
       "      \n",
       "\n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "\n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpseacs.report._api.HtmlGpseaNotebookSummarizer at 0x16b794ec0>"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpseacs.report import GpseaNotebookSummarizer\n",
    "summarizer = GpseaNotebookSummarizer(hpo=hpo, gpsea_version=gpsea.__version__)\n",
    "summarizer.summarize_report(report=report)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output to ../../supplement/tex/TBX1_summary_draft.tex\n"
     ]
    }
   ],
   "source": [
    "summarizer.process_latex(report=report)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
