{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# FZD5\n",
    "[microphthalmia/coloboma-11 (MCOPCB11)](https://omim.org/entry/620731) is caused by heterozygous mutation in the FZD5 gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-02-01T07:49:20.929383Z",
     "start_time": "2025-02-01T07:49:19.583193Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded HPO v2025-01-16\n",
      "Using gpsea version 0.9.4\n"
     ]
    }
   ],
   "source": [
    "import gpsea\n",
    "import hpotk\n",
    "\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo()\n",
    "print(f'Loaded HPO v{hpo.version}')\n",
    "print(f\"Using gpsea version {gpsea.__version__}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### FZD5\n",
    "We user the [Matched Annotation from NCBI and EMBL-EBI (MANE)](https://www.ncbi.nlm.nih.gov/refseq/MANE/) transcript and the corresponding protein identifier for FZD5."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-02-01T07:49:21.196116Z",
     "start_time": "2025-02-01T07:49:20.934225Z"
    }
   },
   "outputs": [],
   "source": [
    "gene_symbol = 'FZD5'\n",
    "mane_tx_id = 'NM_003468.4'\n",
    "mane_protein_id = 'NP_065979.1' # zinc finger SWIM domain-containing protein 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-02-01T07:49:24.143109Z",
     "start_time": "2025-02-01T07:49:23.995251Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 100%|██████████| 29/29 [00:00<00:00, 1755.98 individuals/s]\n",
      "Validated under permissive policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store('0.1.24') as ps:\n",
    "        phenopackets = tuple(ps.iter_cohort_phenopackets(gene_symbol))\n",
    "\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort, qc_results = load_phenopackets(  \n",
    "    phenopackets=phenopackets,\n",
    "    cohort_creator=cohort_creator,\n",
    ")\n",
    "\n",
    "qc_results.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-02-01T07:49:28.804914Z",
     "start_time": "2025-02-01T07:49:28.230161Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "\n",
       "<h1>GPSEA cohort analysis</h1>\n",
       "<div id=\"cohort-summary\">\n",
       "  <p>\n",
       "    Successfully loaded 29 individuals.\n",
       "\n",
       "    \n",
       "    10 were recorded as male,\n",
       "    11 as female,\n",
       "    and 8 as unknown sex.\n",
       "\n",
       "    \n",
       "    \n",
       "    29 were reported to be alive at the time of last encounter,\n",
       "    0 were deceased,\n",
       "    and vital status was unreported for 0 individuals.\n",
       "    \n",
       "\n",
       "    \n",
       "    29 individuals had disease onset information\n",
       "    and 0 had information about the age of last encounter.\n",
       "  </p>\n",
       "\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "</div>\n",
       "\n",
       "\n",
       "<div class=\"cards\">\n",
       "  \n",
       "  <div id=\"hpo\" class=\"card\">\n",
       "    <h2>HPO terms</h2>\n",
       "    \n",
       "    \n",
       "    <table>\n",
       "      The cohort included 16 distinct HPO terms.\n",
       "      <caption>\n",
       "        \n",
       "        Top 10 most common HPO terms.\n",
       "        \n",
       "        \n",
       "      </caption>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>HPO Term</th>\n",
       "        </tr>\n",
       "\n",
       "        \n",
       "        <tr>\n",
       "          <td>16</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000612\">Iris coloboma</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>9</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000639\">Nystagmus</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>7</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000567\">Chorioretinal coloboma</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>7</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000480\">Retinal coloboma</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>7</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0007766\">Optic disc hypoplasia</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>7</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0007663\">Reduced visual acuity</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>5</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000588\">Optic disc coloboma</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>5</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0011003\">High myopia</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>3</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0008499\">High hypermetropia</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/term/HP:0000568\">Microphthalmia</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div id=\"measurements\" class=\"card\">\n",
       "    <h2>Measurements</h2>\n",
       "    \n",
       "    \n",
       "    No data regarding measurement assays were provided.\n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div id=\"diseases\" class=\"card\">\n",
       "    <h2>Diseases</h2>\n",
       "    \n",
       "    \n",
       "    The cohort members were diagnosed with 1 disease.\n",
       "    <table>\n",
       "      \n",
       "\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>Disease</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>29</td>\n",
       "          <td><a href=\"https://hpo.jax.org/browse/disease/OMIM:620731\">Microphthalmia/coloboma 11</a>\n",
       "          </td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "  </div>\n",
       "\n",
       "  \n",
       "  \n",
       "  <div class=\"card\">\n",
       "    <h2>Variants</h2>\n",
       "    <p>\n",
       "      A total of 19 unique variants\n",
       "      were identified in the cohort. Variants were annotated with respect to NM_003468.4.\n",
       "    </p>\n",
       "    <table>\n",
       "      <caption>\n",
       "        <h3>Top 10 variants</h3>\n",
       "      </caption>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th><em>n</em></th>\n",
       "          <th>Variant key</th>\n",
       "          <th>HGVS</th>\n",
       "          <th>Variant Class</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>3</td>\n",
       "          <td>2_207767311_207767312_TC_T</td>\n",
       "          <td>c.1428del (p.Ser477AlafsTer130)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>3</td>\n",
       "          <td>2_207767333_207767333_C_CAGGT</td>\n",
       "          <td>c.1403_1406dup (p.Tyr470ProfsTer130)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>2_207767578_207767578_C_T</td>\n",
       "          <td>c.1162G>A (p.Gly388Ser)</td>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>2_207767508_207767508_T_C</td>\n",
       "          <td>c.1232A>G (p.Tyr411Cys)</td>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>2_207767230_207767230_T_A</td>\n",
       "          <td>c.1510A>T (p.Met504Leu)</td>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>2_207767493_207767559_--67bp--_G</td>\n",
       "          <td>c.1181_1246del (p.Asn394_Gly415del)</td>\n",
       "          <td>INFRAME_DELETION</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>2_207767946_207767946_C_A</td>\n",
       "          <td>c.794G>T (p.Arg265Leu)</td>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>2</td>\n",
       "          <td>2_207768352_207768352_G_T</td>\n",
       "          <td>c.388C>A (p.Arg130Ser)</td>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>1</td>\n",
       "          <td>2_207768200_207768200_G_GC</td>\n",
       "          <td>c.539dup (p.Glu181ArgfsTer88)</td>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>1</td>\n",
       "          <td>2_207767658_207767658_T_TTTC</td>\n",
       "          <td>c.1081_1082insGAA (p.His361delinsArgAsn)</td>\n",
       "          <td>PROTEIN_ALTERING_VARIANT</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "  </div>\n",
       "\n",
       "  \n",
       "  <div class=\"card\">\n",
       "    \n",
       "    <h2>Variant effects</h2>\n",
       "    <p>\n",
       "      The effects were predicted for NM_003468.4\n",
       "    </p>\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr class=\"strng\">\n",
       "          <th>Variant effect</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>MISSENSE_VARIANT</td>\n",
       "          <td>13 (45%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>FRAMESHIFT_VARIANT</td>\n",
       "          <td>10 (34%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>STOP_GAINED</td>\n",
       "          <td>3 (10%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>INFRAME_DELETION</td>\n",
       "          <td>2 (7%)</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>PROTEIN_ALTERING_VARIANT</td>\n",
       "          <td>1 (3%)</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "    </table>\n",
       "    \n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x1280bc6e0>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import CohortViewer\n",
    "\n",
    "cv = CohortViewer(hpo)\n",
    "cv.process(cohort, transcript_id=mane_tx_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-02-01T07:51:22.070383Z",
     "start_time": "2025-02-01T07:51:20.307897Z"
    }
   },
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "zero-size array to reduction operation maximum which has no identity",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[11], line 9\u001b[0m\n\u001b[1;32m      7\u001b[0m fzd5_fig, ax \u001b[38;5;241m=\u001b[39m plt\u001b[38;5;241m.\u001b[39msubplots(figsize\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m15\u001b[39m, \u001b[38;5;241m8\u001b[39m))\n\u001b[1;32m      8\u001b[0m visualizer \u001b[38;5;241m=\u001b[39m ProteinVisualizer()\n\u001b[0;32m----> 9\u001b[0m \u001b[43mvisualizer\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdraw_protein\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     10\u001b[0m \u001b[43m    \u001b[49m\u001b[43mprotein_metadata\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mprotein_meta\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     11\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcohort\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcohort\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     12\u001b[0m \u001b[43m    \u001b[49m\u001b[43max\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43max\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     13\u001b[0m \u001b[43m)\u001b[49m\n\u001b[1;32m     14\u001b[0m fzd5_fig\u001b[38;5;241m.\u001b[39mtight_layout()\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.13/site-packages/gpsea/view/_protein_visualizer.py:89\u001b[0m, in \u001b[0;36mProteinVisualizer.draw_protein\u001b[0;34m(self, cohort, protein_metadata, ax, **kwargs)\u001b[0m\n\u001b[1;32m     78\u001b[0m feature_handler \u001b[38;5;241m=\u001b[39m DrawableProteinFeatureHandler(\n\u001b[1;32m     79\u001b[0m     starts,\n\u001b[1;32m     80\u001b[0m     ends,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     83\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_available_colors,\n\u001b[1;32m     84\u001b[0m )\n\u001b[1;32m     85\u001b[0m max_overlapping_features \u001b[38;5;241m=\u001b[39m sweep_line(\n\u001b[1;32m     86\u001b[0m     [(f\u001b[38;5;241m.\u001b[39mmin_pos_abs, f\u001b[38;5;241m.\u001b[39mmax_pos_abs) \u001b[38;5;28;01mfor\u001b[39;00m f \u001b[38;5;129;01min\u001b[39;00m feature_handler\u001b[38;5;241m.\u001b[39mfeatures]  \u001b[38;5;66;03m# define intervals\u001b[39;00m\n\u001b[1;32m     87\u001b[0m )\n\u001b[0;32m---> 89\u001b[0m variant_handler \u001b[38;5;241m=\u001b[39m \u001b[43mDrawableProteinVariantHandler\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     90\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcohort\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcohort\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     91\u001b[0m \u001b[43m    \u001b[49m\u001b[43mprotein_meta\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mprotein_metadata\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     92\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     94\u001b[0m x_ticks \u001b[38;5;241m=\u001b[39m generate_ticks(apprx_n_ticks\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m6\u001b[39m, \u001b[38;5;28mmin\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m, \u001b[38;5;28mmax\u001b[39m\u001b[38;5;241m=\u001b[39mprotein_metadata\u001b[38;5;241m.\u001b[39mprotein_length)\n\u001b[1;32m     95\u001b[0m y_ticks \u001b[38;5;241m=\u001b[39m generate_ticks(apprx_n_ticks\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m5\u001b[39m, \u001b[38;5;28mmin\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0\u001b[39m, \u001b[38;5;28mmax\u001b[39m\u001b[38;5;241m=\u001b[39mvariant_handler\u001b[38;5;241m.\u001b[39mmax_marker_count)\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.13/site-packages/gpsea/view/_protein_visualizer.py:417\u001b[0m, in \u001b[0;36mDrawableProteinVariantHandler.__init__\u001b[0;34m(self, cohort, protein_meta, aggregation_method)\u001b[0m\n\u001b[1;32m    373\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mUnsupported aggregation method \u001b[39m\u001b[38;5;132;01m{\u001b[39;00maggregation_method\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m    375\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mvariant_effect2color \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m    376\u001b[0m     VariantEffect\u001b[38;5;241m.\u001b[39mTRANSCRIPT_ABLATION: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m#ff0000\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    377\u001b[0m     VariantEffect\u001b[38;5;241m.\u001b[39mSPLICE_ACCEPTOR_VARIANT: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m#00ff00\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    415\u001b[0m     VariantEffect\u001b[38;5;241m.\u001b[39mSEQUENCE_VARIANT: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m#33ff00\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    416\u001b[0m }\n\u001b[0;32m--> 417\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmax_marker_count \u001b[38;5;241m=\u001b[39m \u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_pvis\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmarker_counts\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    418\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39maggregation_method \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstandard\u001b[39m\u001b[38;5;124m'\u001b[39m:\n\u001b[1;32m    419\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mvariants \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_generate_variant_markers()\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.13/site-packages/numpy/_core/fromnumeric.py:3164\u001b[0m, in \u001b[0;36mmax\u001b[0;34m(a, axis, out, keepdims, initial, where)\u001b[0m\n\u001b[1;32m   3052\u001b[0m \u001b[38;5;129m@array_function_dispatch\u001b[39m(_max_dispatcher)\n\u001b[1;32m   3053\u001b[0m \u001b[38;5;129m@set_module\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mnumpy\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m   3054\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21mmax\u001b[39m(a, axis\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, out\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, keepdims\u001b[38;5;241m=\u001b[39mnp\u001b[38;5;241m.\u001b[39m_NoValue, initial\u001b[38;5;241m=\u001b[39mnp\u001b[38;5;241m.\u001b[39m_NoValue,\n\u001b[1;32m   3055\u001b[0m          where\u001b[38;5;241m=\u001b[39mnp\u001b[38;5;241m.\u001b[39m_NoValue):\n\u001b[1;32m   3056\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m   3057\u001b[0m \u001b[38;5;124;03m    Return the maximum of an array or maximum along an axis.\u001b[39;00m\n\u001b[1;32m   3058\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   3162\u001b[0m \u001b[38;5;124;03m    5\u001b[39;00m\n\u001b[1;32m   3163\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m-> 3164\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_wrapreduction\u001b[49m\u001b[43m(\u001b[49m\u001b[43ma\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmaximum\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mmax\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3165\u001b[0m \u001b[43m                          \u001b[49m\u001b[43mkeepdims\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mkeepdims\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43minitial\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minitial\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mwhere\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mwhere\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/GIT/gpsea-cs/venv/lib/python3.13/site-packages/numpy/_core/fromnumeric.py:86\u001b[0m, in \u001b[0;36m_wrapreduction\u001b[0;34m(obj, ufunc, method, axis, dtype, out, **kwargs)\u001b[0m\n\u001b[1;32m     83\u001b[0m         \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m     84\u001b[0m             \u001b[38;5;28;01mreturn\u001b[39;00m reduction(axis\u001b[38;5;241m=\u001b[39maxis, out\u001b[38;5;241m=\u001b[39mout, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mpasskwargs)\n\u001b[0;32m---> 86\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mufunc\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mreduce\u001b[49m\u001b[43m(\u001b[49m\u001b[43mobj\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mout\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mpasskwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[0;31mValueError\u001b[0m: zero-size array to reduction operation maximum which has no identity"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAABMkAAAKZCAYAAACiDnxZAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjAsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvlHJYcgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAJoRJREFUeJzt3XtsV/X9+PFXASmaCepQUMThvKFRQUERL1tMUBKNzj/MEI0Q5iVOJUrnBnihXqY4dYYlVIm3uH8YTKPGCIEpapyTjAiaaCYaRS0hcpuBMlRQ+Hxzzu/XjmpxFEtbeD0eySdyTs9pz8fkTcuz7/M+VZVKpRIAAAAAkFiXjr4AAAAAAOhoIhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOm1OpK99tprccEFF8QhhxwSVVVV8dxzz/3Pc1599dU4+eSTo7q6Oo488sh48sknd/Z6AQAAAKDjI9nGjRtj0KBBUVdXt0PHf/zxx3H++efH2WefHW+//XbceOONceWVV8b8+fN35noBAAAAoM1VVSqVyk6fXFUVzz77bFx00UXbPWbixIkxZ86cePfdd5v2XXLJJbFu3bqYN2/ezn5pAAAAAGgz3WIXW7hwYYwYMaLZvpEjR5YzyrZn06ZN5avR1q1b4/PPP48f//jHZZgDAAAAIKdKpRIbNmwolwLr0qXL7hPJVq5cGX369Gm2r9huaGiIL7/8Mvbee+/vnDN16tS44447dvWlAQAAALCbWr58eRx66KG7TyTbGZMnT46ampqm7fXr18dhhx1WvvmePXt26LUBAAAA0HGKiVf9+/ePfffdt00/7y6PZH379o1Vq1Y121dsF7GrpVlkheIpmMXr24pzRDIAAAAAqtp4Sa62u3FzO4YPHx4LFixotu/FF18s9wMAAABAZ9DqSPaf//wn3n777fJV+Pjjj8s/19fXN90qOWbMmKbjr7nmmli2bFn87ne/i6VLl8ZDDz0Uf/3rX2PChAlt+T4AAAAAoP0i2ZtvvhknnXRS+SoUa4cVf54yZUq5/dlnnzUFs8Lhhx8ec+bMKWePDRo0KP74xz/GY489Vj7hEgAAAAA6g6pK8dzM3WBBtl69epUL+FuTDAAAACCvhl3UiXb5mmQAAAAA0NmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAApCeSAQAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6OxXJ6urqYsCAAdGjR48YNmxYLFq06HuPnzZtWhxzzDGx9957R//+/WPChAnx1Vdf7ew1AwAAAEDHRrLZs2dHTU1N1NbWxpIlS2LQoEExcuTIWL16dYvHz5w5MyZNmlQe/95778Xjjz9efo6bb765La4fAAAAANo/kj344INx1VVXxbhx4+K4446LGTNmxD777BNPPPFEi8e/8cYbccYZZ8Sll15azj4799xzY/To0f9z9hkAAAAAdMpItnnz5li8eHGMGDHiv5+gS5dye+HChS2ec/rpp5fnNEaxZcuWxdy5c+O8887b7tfZtGlTNDQ0NHsBAAAAwK7SrTUHr127NrZs2RJ9+vRptr/YXrp0aYvnFDPIivPOPPPMqFQq8c0338Q111zzvbdbTp06Ne64447WXBoAAAAAdN6nW7766qtxzz33xEMPPVSuYfbMM8/EnDlz4q677truOZMnT47169c3vZYvX76rLxMAAACAxFo1k6x3797RtWvXWLVqVbP9xXbfvn1bPOe2226Lyy+/PK688spy+4QTToiNGzfG1VdfHbfcckt5u+a3VVdXly8AAAAA6HQzybp37x5DhgyJBQsWNO3bunVruT18+PAWz/niiy++E8KK0FYobr8EAAAAgN1qJlmhpqYmxo4dG0OHDo1TTz01pk2bVs4MK552WRgzZkz069evXFescMEFF5RPxDzppJNi2LBh8eGHH5azy4r9jbEMAAAAAHarSDZq1KhYs2ZNTJkyJVauXBmDBw+OefPmNS3mX19f32zm2K233hpVVVXlf1esWBEHHnhgGcjuvvvutn0nAAAAALCTqiq7wT2PDQ0N0atXr3IR/549e3b05QAAAACwh3WiXf50SwAAAADo7EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAIL2dimR1dXUxYMCA6NGjRwwbNiwWLVr0vcevW7currvuujj44IOjuro6jj766Jg7d+7OXjMAAAAAtKlurT1h9uzZUVNTEzNmzCgD2bRp02LkyJHx/vvvx0EHHfSd4zdv3hznnHNO+bGnn346+vXrF59++mnst99+bfUeAAAAAOAHqapUKpXWnFCEsVNOOSWmT59ebm/dujX69+8f48ePj0mTJn3n+CKm3X///bF06dLYa6+9duoiGxoaolevXrF+/fro2bPnTn0OAAAAAHZ/DbuoE7XqdstiVtjixYtjxIgR//0EXbqU2wsXLmzxnOeffz6GDx9e3m7Zp0+fOP744+Oee+6JLVu2/PCrBwAAAID2vt1y7dq1ZdwqYte2iu1iplhLli1bFi+//HJcdtll5TpkH374YVx77bXx9ddfR21tbYvnbNq0qXxtWwgBAAAAYLd9umVxO2axHtkjjzwSQ4YMiVGjRsUtt9xS3oa5PVOnTi2nzTW+its5AQAAAKBTRLLevXtH165dY9WqVc32F9t9+/Zt8ZziiZbF0yyL8xode+yxsXLlyvL2zZZMnjy5vK+08bV8+fLWXCYAAAAA7LpI1r1793I22IIFC5rNFCu2i3XHWnLGGWeUt1gWxzX64IMPynhWfL6WVFdXlwuvbfsCAAAAgE5zu2VNTU08+uij8ec//znee++9+PWvfx0bN26McePGlR8fM2ZMOROsUfHxzz//PG644YYyjs2ZM6dcuL9YyB8AAAAAdruF+wvFmmJr1qyJKVOmlLdMDh48OObNm9e0mH99fX35xMtGxXpi8+fPjwkTJsSJJ54Y/fr1K4PZxIkT2/adAAAAAMBOqqpUKpXo5IqnWxYL+Bfrk7n1EgAAACCvhl3UiXb50y0BAAAAoLMTyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0tupSFZXVxcDBgyIHj16xLBhw2LRokU7dN6sWbOiqqoqLrroop35sgAAAADQOSLZ7Nmzo6amJmpra2PJkiUxaNCgGDlyZKxevfp7z/vkk0/ipptuirPOOuuHXC8AAAAAdHwke/DBB+Oqq66KcePGxXHHHRczZsyIffbZJ5544ontnrNly5a47LLL4o477oif/vSnP/SaAQAAAKDjItnmzZtj8eLFMWLEiP9+gi5dyu2FCxdu97w777wzDjrooLjiiit26Ots2rQpGhoamr0AAAAAoFNEsrVr15azwvr06dNsf7G9cuXKFs95/fXX4/HHH49HH310h7/O1KlTo1evXk2v/v37t+YyAQAAAKDzPN1yw4YNcfnll5eBrHfv3jt83uTJk2P9+vVNr+XLl+/KywQAAAAguW6tObgIXV27do1Vq1Y1219s9+3b9zvHf/TRR+WC/RdccEHTvq1bt/6/L9ytW7z//vtxxBFHfOe86urq8gUAAAAAnW4mWffu3WPIkCGxYMGCZtGr2B4+fPh3jh84cGC888478fbbbze9Lrzwwjj77LPLP7uNEgAAAIDdbiZZoaamJsaOHRtDhw6NU089NaZNmxYbN24sn3ZZGDNmTPTr169cV6xHjx5x/PHHNzt/v/32K//77f0AAAAAsNtEslGjRsWaNWtiypQp5WL9gwcPjnnz5jUt5l9fX18+8RIAAAAAdhdVlUqlEp1cQ0ND+ZTLYhH/nj17dvTlAAAAALCHdSJTvgAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0tupSFZXVxcDBgyIHj16xLBhw2LRokXbPfbRRx+Ns846K/bff//yNWLEiO89HgAAAAA6fSSbPXt21NTURG1tbSxZsiQGDRoUI0eOjNWrV7d4/KuvvhqjR4+OV155JRYuXBj9+/ePc889N1asWNEW1w8AAAAAP1hVpVKptOaEYubYKaecEtOnTy+3t27dWoav8ePHx6RJk/7n+Vu2bClnlBXnjxkzZoe+ZkNDQ/Tq1SvWr18fPXv2bM3lAgAAALAHadhFnahVM8k2b94cixcvLm+ZbPoEXbqU28UssR3xxRdfxNdffx0HHHDAdo/ZtGlT+Ya3fQEAAADArtKqSLZ27dpyJlifPn2a7S+2V65cuUOfY+LEiXHIIYc0C23fNnXq1LIINr6KmWoAAAAAsEc83fLee++NWbNmxbPPPlsu+r89kydPLqfMNb6WL1/enpcJAAAAQDLdWnNw7969o2vXrrFq1apm+4vtvn37fu+5DzzwQBnJXnrppTjxxBO/99jq6uryBQAAAACdbiZZ9+7dY8iQIbFgwYKmfcXC/cX28OHDt3vefffdF3fddVfMmzcvhg4d+sOuGAAAAAA6ciZZoaamJsaOHVvGrlNPPTWmTZsWGzdujHHjxpUfL55Y2a9fv3JdscIf/vCHmDJlSsycOTMGDBjQtHbZj370o/IFAAAAALtdJBs1alSsWbOmDF9F8Bo8eHA5Q6xxMf/6+vryiZeNHn744fKpmBdffHGzz1NbWxu33357W7wHAAAAAPhBqiqVSiU6uYaGhvIpl8Ui/j179uzoywEAAABgD+tE7fp0SwAAAADojEQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAID2RDAAAAID0RDIAAAAA0hPJAAAAAEhPJAMAAAAgPZEMAAAAgPREMgAAAADSE8kAAAAASE8kAwAAACA9kQwAAACA9EQyAAAAANITyQAAAABITyQDAAAAIL2dimR1dXUxYMCA6NGjRwwbNiwWLVr0vcc/9dRTMXDgwPL4E044IebOnbuz1wsAAAAAHR/JZs+eHTU1NVFbWxtLliyJQYMGxciRI2P16tUtHv/GG2/E6NGj44orroi33norLrroovL17rvvtsX1AwAAAMAPVlWpVCqtOaGYOXbKKafE9OnTy+2tW7dG//79Y/z48TFp0qTvHD9q1KjYuHFjvPDCC037TjvttBg8eHDMmDFjh75mQ0ND9OrVK9avXx89e/ZszeUCAAAAsAdp2EWdqFtrDt68eXMsXrw4Jk+e3LSvS5cuMWLEiFi4cGGL5xT7i5ln2ypmnj333HPb/TqbNm0qX42KN934PwEAAACAvBr+fx9q5byvto1ka9eujS1btkSfPn2a7S+2ly5d2uI5K1eubPH4Yv/2TJ06Ne64447v7C9mrAEAAADAv//973JGWYdEsvZSzFTbdvbZunXr4ic/+UnU19e36ZsH2qbgFwF7+fLlboeGTsgYhc7L+ITOzRiFzqu44/Cwww6LAw44oE0/b6siWe/evaNr166xatWqZvuL7b59+7Z4TrG/NccXqqury9e3FYHMX07QORVj0/iEzssYhc7L+ITOzRiFzqtYAqxNP19rDu7evXsMGTIkFixY0LSvWLi/2B4+fHiL5xT7tz2+8OKLL273eAAAAABob62+3bK4DXLs2LExdOjQOPXUU2PatGnl0yvHjRtXfnzMmDHRr1+/cl2xwg033BA///nP449//GOcf/75MWvWrHjzzTfjkUceaft3AwAAAADtEclGjRoVa9asiSlTppSL7w8ePDjmzZvXtDh/sW7YttPdTj/99Jg5c2bceuutcfPNN8dRRx1VPtny+OOP3+GvWdx6WVtb2+ItmEDHMj6hczNGofMyPqFzM0Yh3/isqrT18zIBAAAAYDfTtiucAQAAAMBuSCQDAAAAID2RDAAAAID0RDIAAAAA0us0kayuri4GDBgQPXr0iGHDhsWiRYu+9/innnoqBg4cWB5/wgknxNy5c9vtWiGb1ozPRx99NM4666zYf//9y9eIESP+53gG2vd7aKNZs2ZFVVVVXHTRRbv8GiGr1o7PdevWxXXXXRcHH3xw+cSuo48+2s+50InG6LRp0+KYY46JvffeO/r37x8TJkyIr776qt2uF7J47bXX4oILLohDDjmk/Hn1ueee+5/nvPrqq3HyySeX3z+PPPLIePLJJ3fPSDZ79uyoqakpH9+5ZMmSGDRoUIwcOTJWr17d4vFvvPFGjB49Oq644op46623yh/ui9e7777b7tcOe7rWjs/iL6ZifL7yyiuxcOHC8oeHc889N1asWNHu1w4ZtHaMNvrkk0/ipptuKqM20DnG5+bNm+Occ84px+fTTz8d77//fvnLp379+rX7tUMGrR2jM2fOjEmTJpXHv/fee/H444+Xn+Pmm29u92uHPd3GjRvLMVmE7B3x8ccfx/nnnx9nn312vP3223HjjTfGlVdeGfPnz2/V162qVCqV6GBFsT/llFNi+vTp5fbWrVvLf1iPHz++/Evo20aNGlX+D3vhhRea9p122mkxePDgmDFjRrteO+zpWjs+v23Lli3ljLLi/DFjxrTDFUMuOzNGi3H5s5/9LH71q1/F3//+93Lmyo78dg7YteOz+Dn2/vvvj6VLl8Zee+3VAVcMubR2jF5//fVlHFuwYEHTvt/85jfxz3/+M15//fV2vXbIpKqqKp599tnvvfth4sSJMWfOnGaTpy655JLy59x58+btPjPJit+YLV68uLwlq1GXLl3K7WIWSkuK/dseXyiK//aOB9pvfH7bF198EV9//XUccMABu/BKIaedHaN33nlnHHTQQeWMbKDzjM/nn38+hg8fXt5u2adPnzj++OPjnnvuKcM20PFj9PTTTy/Pabwlc9myZeXt0Oedd167XTcQu7QTdYsOtnbt2vIbf/GDwLaK7eK3aC1ZuXJli8cX+4GOHZ8tFf3iPvJv/4UFdMwYLX7TXdweUkxDBzrX+Cz+wf3yyy/HZZddVv7D+8MPP4xrr722/GVTcXsX0LFj9NJLLy3PO/PMM6O4Ieubb76Ja665xu2W0AlsrxM1NDTEl19+Wa4juFvMJAP2XPfee2+5MHgxNbZYDBXoWBs2bIjLL7+8XOOod+/eHX05wLcUt3oVszwfeeSRGDJkSLnEyC233GI5EegkirV3i9mdDz30ULmG2TPPPFPe3nXXXXd19KUBbaTDZ5IVP6R37do1Vq1a1Wx/sd23b98Wzyn2t+Z4oP3GZ6MHHnigjGQvvfRSnHjiibv4SiGn1o7Rjz76qFwQvHhS0Lb/KC9069atXCT8iCOOaIcrhz3fznwPLZ5oWaxFVpzX6Nhjjy1/O17cGta9e/ddft2Qxc6M0dtuu638ZVOxGHjhhBNOKNfKvvrqq8ugXdyuCXSM7XWinj177vAsskKHj+Lim33xm7JtFz8sfmAvtos1GVpS7N/2+MKLL7643eOB9hufhfvuu6/8jVqxQOLQoUPb6Wohn9aO0YEDB8Y777xT3mrZ+LrwwgubngJULFYMdNz30DPOOKO8xbIxXhc++OCDMp4JZNDxY7RYa/fbIawxaneC5+FBasPbqhNVOoFZs2ZVqqurK08++WTlX//6V+Xqq6+u7LfffpWVK1eWH7/88ssrkyZNajr+H//4R6Vbt26VBx54oPLee+9VamtrK3vttVflnXfe6cB3AXum1o7Pe++9t9K9e/fK008/Xfnss8+aXhs2bOjAdwF7rtaO0W8bO3Zs5Re/+EU7XjHk0drxWV9fX9l3330r119/feX999+vvPDCC5WDDjqo8vvf/74D3wXsuVo7Rot/dxZj9C9/+Utl2bJllb/97W+VI444ovLLX/6yA98F7Jk2bNhQeeutt8pXka4efPDB8s+ffvpp+fFibBZjtFExJvfZZ5/Kb3/727IT1dXVVbp27VqZN29eq75uh99uWSjWW1izZk1MmTKlnE4+ePDgcgZK46Jr9fX1zYp98VSRmTNnxq233louknjUUUeVj64vngAEdOz4fPjhh8tbQi6++OJmn6dYcPj2229v9+uHPV1rxyjQecdnMZtz/vz5MWHChHKpgn79+sUNN9xQPgQH6PgxWvz7s6qqqvzvihUr4sADDyyXMLj77rs78F3AnunNN98s73ZoVFNTU/537Nix8eSTT8Znn31WjtFGhx9+eLlGYPE99E9/+lMceuih8dhjj5VPuGyNqqKUteH7AAAAAIDdjl8tAwAAAJCeSAYAAABAeiIZAAAAAOmJZAAAAACkJ5IBAAAAkJ5IBgAAAEB6IhkAAAAA6YlkAAAAAKQnkgEAAACQnkgGAAAAQHoiGQAAAADpiWQAAAAARHb/B3bDz3bg6PK3AAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 1500x800 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from gpsea.view import ProteinVisualizer\n",
    "import matplotlib.pyplot as plt\n",
    "from gpsea.preprocessing import configure_default_protein_metadata_service\n",
    "\n",
    "pms = configure_default_protein_metadata_service()\n",
    "protein_meta = pms.annotate(mane_protein_id)\n",
    "fzd5_fig, ax = plt.subplots(figsize=(15, 8))\n",
    "visualizer = ProteinVisualizer()\n",
    "visualizer.draw_protein(\n",
    "    protein_metadata=protein_meta,\n",
    "    cohort=cohort,\n",
    "    ax=ax,\n",
    ")\n",
    "fzd5_fig.tight_layout()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genotype-Phenotype Correlation (GPC) Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpsea.analysis.pcats import configure_hpo_term_analysis\n",
    "analysis = configure_hpo_term_analysis(hpo)\n",
    "\n",
    "from gpsea.analysis.clf import prepare_classifiers_for_terms_of_interest\n",
    "pheno_clfs = prepare_classifiers_for_terms_of_interest(\n",
    "    cohort=cohort,\n",
    "    hpo=hpo,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <style>\n",
       "        /** Light theme is the default */\n",
       "        :root {\n",
       "            --gpsea-color-text: black;\n",
       "            --gpsea-color-link-text: rgb(0, 118, 187);\n",
       "            --gpsea-color-link-hover: rgb(0, 99, 156);\n",
       "            --gpsea-color-primary-a0: #2f5170;\n",
       "            --gpsea-color-primary-a10: #4778a5;\n",
       "            --gpsea-color-primary-a20: #5995ce;\n",
       "            --gpsea-color-primary-a60: LightSkyBlue;\n",
       "            --gpsea-color-surface-a0: #fcfcfc;\n",
       "            --gpsea-color-surface-a10: #ffffff;\n",
       "            --gpsea-color-surface-a20: #b3b3b3;\n",
       "            --gpsea-color-surface-a30: #8d8d8d;\n",
       "            --gpsea-color-surface-a40: #646464;\n",
       "            --gpsea-color-surface-a50: #000000;\n",
       "        }\n",
       "\n",
       "        /** Dark theme */\n",
       "        /*\n",
       "        @media (prefers-color-scheme: dark) {\n",
       "            :root {\n",
       "                --gpsea-color-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-text: rgb(240, 240, 240);\n",
       "                --gpsea-color-link-hover: lightblue;\n",
       "                --gpsea-color-primary-a0: #64a3df;\n",
       "                --gpsea-color-primary-a10: #5286b8;\n",
       "                --gpsea-color-primary-a20: #375c7e;\n",
       "                --gpsea-color-primary-a60: #233b52;\n",
       "                --gpsea-color-surface-a0: #121212;\n",
       "                --gpsea-color-surface-a10: #282828;\n",
       "                --gpsea-color-surface-a20: #3f3f3f;\n",
       "                --gpsea-color-surface-a30: #575757;\n",
       "                --gpsea-color-surface-a40: #717171;\n",
       "                --gpsea-color-surface-a50: #8b8b8b;\n",
       "            }\n",
       "        }\n",
       "        */\n",
       "\n",
       "        #gpsea-body {\n",
       "            max-width: 1600px;\n",
       "            color: var(--gpsea-color-text);\n",
       "            background-color: var(--gpsea-color-surface-a0);\n",
       "            padding: 10px 0;\n",
       "        }\n",
       "\n",
       "        #gpsea-body {\n",
       "\n",
       "            h1,\n",
       "            h2,\n",
       "            h3,\n",
       "            h4,\n",
       "            h5 {\n",
       "                color: var(--gpsea-color-primary-a0);\n",
       "            }\n",
       "\n",
       "            a {\n",
       "                color: var(--gpsea-color-link-text);\n",
       "                text-decoration: none;\n",
       "            }\n",
       "\n",
       "            a:hover {\n",
       "                color: var(--gpsea-color-link-hover);\n",
       "            }\n",
       "\n",
       "            #cohort-summary {\n",
       "                max-width: 800px;\n",
       "            }\n",
       "\n",
       "            td {\n",
       "                padding: 5px 0;\n",
       "            }\n",
       "\n",
       "            .cards {\n",
       "                display: flex;\n",
       "                flex-flow: row wrap;\n",
       "                align-items: top;\n",
       "                gap: 1rem;\n",
       "            }\n",
       "\n",
       "            .card {\n",
       "                border-radius: 5px;\n",
       "                padding: 0px 10px;\n",
       "                background: var(--gpsea-color-surface-a10);\n",
       "                border: 1px solid var(--gpsea-color-surface-a30);\n",
       "                box-shadow: 0 4px 8px var(--gpsea-color-surface-a30);\n",
       "            }\n",
       "\n",
       "            table {\n",
       "                width: 100%;\n",
       "                border-collapse: collapse;\n",
       "                margin: 10px 0;\n",
       "                font-size: 0.9em;\n",
       "                font-family: sans-serif;\n",
       "                box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "                border: 1px solid var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            th {\n",
       "                text-align: left;\n",
       "                padding: 2px;\n",
       "                font-weight: bold;\n",
       "                font-size: 120%;\n",
       "                color: var(--gpsea-color-text);\n",
       "                background-color: var(--gpsea-color-primary-a60);\n",
       "            }\n",
       "\n",
       "            table td,\n",
       "            tr {\n",
       "                text-align: left;\n",
       "                padding: 0px 5px;\n",
       "            }\n",
       "\n",
       "            tr:nth-child(even) {\n",
       "                background-color: var(--gpsea-color-surface-a10);\n",
       "            }\n",
       "\n",
       "            tr:nth-child(odd) {\n",
       "                background-color: var(--gpsea-color-surface-a20);\n",
       "            }\n",
       "\n",
       "            caption {\n",
       "                caption-side: top;\n",
       "                text-align: left;\n",
       "                padding-bottom: 10px;\n",
       "                font-weight: bold;\n",
       "            }\n",
       "        }\n",
       "    </style>\n",
       "    <div id=\"gpsea-body\">\n",
       "        \n",
       "<h1>Phenotype testing report</h1>\n",
       "<div class=\"cards\">\n",
       "  <div class=\"card\">\n",
       "    <ul>\n",
       "      <li>Phenotype MTC filter: <em>Independent filtering HPO filter</em></li>\n",
       "      <li>Multiple testing correction: <em>fdr_bh</em></li>\n",
       "    </ul>\n",
       "  </div>\n",
       "  <div class=\"card\">\n",
       "    <table>\n",
       "      <tbody>\n",
       "        <tr>\n",
       "          <th>Code</th>\n",
       "          <th>Reason</th>\n",
       "          <th>Count</th>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF01</td>\n",
       "          <td>Skipping term with maximum frequency that was less than threshold 0.4</td>\n",
       "          <td>9</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF08</td>\n",
       "          <td>Skipping general term</td>\n",
       "          <td>15</td>\n",
       "        </tr>\n",
       "        \n",
       "        <tr>\n",
       "          <td>HMF09</td>\n",
       "          <td>Skipping term with maximum annotation frequency that was less than threshold 0.4</td>\n",
       "          <td>23</td>\n",
       "        </tr>\n",
       "        \n",
       "      </tbody>\n",
       "      <caption>\n",
       "        Performed statistical tests for 11 out of the total of 58 HPO terms.\n",
       "      </caption>\n",
       "    </table>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<gpsea.view._base.HtmlGpseaReport at 0x10e90e5d0>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.model import VariantEffect\n",
    "from gpsea.analysis.predicate import variant_effect\n",
    "from gpsea.analysis.clf import monoallelic_classifier\n",
    "from gpsea.view import MtcStatsViewer\n",
    "\n",
    "missense = variant_effect(VariantEffect.MISSENSE_VARIANT, tx_id=mane_tx_id)\n",
    "\n",
    "missense_clf = monoallelic_classifier(a_predicate=missense, \n",
    "                                           b_predicate=~missense, \n",
    "                                           a_label=\"missense\", \n",
    "                                           b_label=\"other\")\n",
    "missense_result = analysis.compare_genotype_vs_phenotypes(\n",
    "    cohort=cohort,\n",
    "    gt_clf=missense_clf,\n",
    "    pheno_clfs=pheno_clfs,\n",
    ")\n",
    "\n",
    "viewer = MtcStatsViewer()\n",
    "viewer.process(missense_result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Allele group</th>\n",
       "      <th>missense</th>\n",
       "      <th>other</th>\n",
       "      <th>Corrected p values</th>\n",
       "      <th>p values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Iris coloboma [HP:0000612]</th>\n",
       "      <td>4/13 (31%)</td>\n",
       "      <td>12/16 (75%)</td>\n",
       "      <td>0.297588</td>\n",
       "      <td>0.027053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Retinal coloboma [HP:0000480]</th>\n",
       "      <td>2/8 (25%)</td>\n",
       "      <td>5/6 (83%)</td>\n",
       "      <td>0.564103</td>\n",
       "      <td>0.102564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Optic disc coloboma [HP:0000588]</th>\n",
       "      <td>0/3 (0%)</td>\n",
       "      <td>5/9 (56%)</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.204545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Optic disc hypoplasia [HP:0007766]</th>\n",
       "      <td>5/8 (62%)</td>\n",
       "      <td>2/6 (33%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.592075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal fundus morphology [HP:0001098]</th>\n",
       "      <td>10/10 (100%)</td>\n",
       "      <td>15/15 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chorioretinal coloboma [HP:0000567]</th>\n",
       "      <td>2/3 (67%)</td>\n",
       "      <td>5/9 (56%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormality iris morphology [HP:0000525]</th>\n",
       "      <td>4/4 (100%)</td>\n",
       "      <td>12/12 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reduced visual acuity [HP:0007663]</th>\n",
       "      <td>4/8 (50%)</td>\n",
       "      <td>3/6 (50%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal retinal morphology [HP:0000479]</th>\n",
       "      <td>5/5 (100%)</td>\n",
       "      <td>10/10 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Coloboma [HP:0000589]</th>\n",
       "      <td>5/5 (100%)</td>\n",
       "      <td>14/14 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abnormal optic nerve morphology [HP:0000587]</th>\n",
       "      <td>5/5 (100%)</td>\n",
       "      <td>7/7 (100%)</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Allele group                                      missense         other  \\\n",
       "Iris coloboma [HP:0000612]                      4/13 (31%)   12/16 (75%)   \n",
       "Retinal coloboma [HP:0000480]                    2/8 (25%)     5/6 (83%)   \n",
       "Optic disc coloboma [HP:0000588]                  0/3 (0%)     5/9 (56%)   \n",
       "Optic disc hypoplasia [HP:0007766]               5/8 (62%)     2/6 (33%)   \n",
       "Abnormal fundus morphology [HP:0001098]       10/10 (100%)  15/15 (100%)   \n",
       "Chorioretinal coloboma [HP:0000567]              2/3 (67%)     5/9 (56%)   \n",
       "Abnormality iris morphology [HP:0000525]        4/4 (100%)  12/12 (100%)   \n",
       "Reduced visual acuity [HP:0007663]               4/8 (50%)     3/6 (50%)   \n",
       "Abnormal retinal morphology [HP:0000479]        5/5 (100%)  10/10 (100%)   \n",
       "Coloboma [HP:0000589]                           5/5 (100%)  14/14 (100%)   \n",
       "Abnormal optic nerve morphology [HP:0000587]    5/5 (100%)    7/7 (100%)   \n",
       "\n",
       "Allele group                                  Corrected p values  p values  \n",
       "Iris coloboma [HP:0000612]                              0.297588  0.027053  \n",
       "Retinal coloboma [HP:0000480]                           0.564103  0.102564  \n",
       "Optic disc coloboma [HP:0000588]                        0.750000  0.204545  \n",
       "Optic disc hypoplasia [HP:0007766]                      1.000000  0.592075  \n",
       "Abnormal fundus morphology [HP:0001098]                 1.000000  1.000000  \n",
       "Chorioretinal coloboma [HP:0000567]                     1.000000  1.000000  \n",
       "Abnormality iris morphology [HP:0000525]                1.000000  1.000000  \n",
       "Reduced visual acuity [HP:0007663]                      1.000000  1.000000  \n",
       "Abnormal retinal morphology [HP:0000479]                1.000000  1.000000  \n",
       "Coloboma [HP:0000589]                                   1.000000  1.000000  \n",
       "Abnormal optic nerve morphology [HP:0000587]            1.000000  1.000000  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpsea.view import summarize_hpo_analysis\n",
    "\n",
    "summarize_hpo_analysis(hpo=hpo, result=missense_result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gpseacs.report import GpseaAnalysisReport, GPAnalysisResultSummary\n",
    "\n",
    "fet_results = (\n",
    "    GPAnalysisResultSummary.from_multi(\n",
    "        result=missense_result,\n",
    "    ),\n",
    ")\n",
    "\n",
    "caption = \"No significant correlations identified.\"\n",
    "report = GpseaAnalysisReport(name=gene_symbol, \n",
    "                             cohort=cohort, \n",
    "                             fet_results=fet_results,\n",
    "                             gene_symbol=gene_symbol,\n",
    "                             mane_tx_id=mane_tx_id,\n",
    "                             mane_protein_id=mane_protein_id,\n",
    "                             caption=caption)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "      min-width: 100px;\n",
       "      border: 1px solid #ddd;\n",
       "      padding: 8px;\n",
       "        \n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    p {\n",
       "      width: 75%;\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>FZD5 Summary</h1>\n",
       "  <p>The cohort comprised 29 individuals (11 females, 10 males, 8 with unknown sex). A total of 16 HPO terms were used to annotate the cohort. Disease diagnosis: Microphthalmia/coloboma 11 (OMIM:620731). No significant correlations identified. A total of 29 unique variant alleles were found in FZD5 (transcript: NM_003468.4, protein id: NP_065979.1). HPO version: 2025-01-16. GPSEA version: 0.9.4.</p>\n",
       "  \n",
       "  \n",
       "    \n",
       "      <table>\n",
       "        <caption style=\"color: black;\">\n",
       "            Fisher Exact Test performed to compare HPO annotation frequency with respect to missense and other.\n",
       "        </caption>\n",
       "            <tbody>\n",
       "                <tr class=\"strng\">\n",
       "                    <th>Genotype (A)</th>\n",
       "                    <th>Genotype (B)</th>\n",
       "                  <th>Tests performed</th>\n",
       "                  <th>Significant tests</th>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td class=\"lft\">missense</td>\n",
       "                    <td class=\"lft\">other</td>\n",
       "                    <td class=\"lft\">11</td>\n",
       "                    <td class=\"lft\">0</td>\n",
       "                </tr>\n",
       "              </tbody>\n",
       "        </table>\n",
       "      \n",
       "\n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "  \n",
       "\n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<gpseacs.report._api.HtmlGpseaNotebookSummarizer at 0x1280bf620>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from gpseacs.report import GpseaNotebookSummarizer\n",
    "summarizer = GpseaNotebookSummarizer(hpo=hpo, gpsea_version=gpsea.__version__)\n",
    "summarizer.summarize_report(report=report)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summarizer.process_latex(report=report, protein_fig=fzd5_fig)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
