{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Kabuki 1 & Kabuki 2 Syndrome \n",
    "\n",
    "##### **Kabuki 1 Syndrome** - [OMIM:147920](https://omim.org/entry/147920), MONDO:0007843\n",
    "- LYSINE-SPECIFIC METHYLTRANSFERASE 2D; **KMT2D** ([OMIM:602113](https://omim.org/entry/602113))\n",
    "\n",
    "##### **Kabuki 2 Syndrome** - [OMIM:300867](https://www.omim.org/entry/300867), MONDO:0010465\n",
    "- LYSINE DEMETHYLASE 6A; **KDM6A** ([OMIM:300128](https://www.omim.org/entry/300128))\n",
    "\n",
    "#### **Genotype Phenotype Correlations**\n",
    "\n",
    "##### KMT2D & KDM6A\n",
    "\n",
    "- [Banka et al (2012)](https://pubmed.ncbi.nlm.nih.gov/22126750/): \n",
    "    - analyzed the KMT2D gene in a cohort of 116 patients with Kabuki syndrome, including 18 patients previously reported by Hannibal et al. (2011), and identified KMT2D variants in 74 (63.8%).\n",
    "    - KABUK1 patients were more likely to have feeding problems, kidney anomalies, early breast bud development, joint dislocations, and palatal malformations in comparison with KMT2D mutation-negative patients\n",
    "    - **Systematic Kabuki syndrome facial morphology study suggested that nearly all patients with typical Kabuki syndrome facial features have pathogenic KMT2D mutations, although the disorder can be phenotypically variable.**\n",
    "\n",
    "- [Miyake et al (2013)](https://pubmed.ncbi.nlm.nih.gov/23913813/)\n",
    "    - screened 81 patients with Kabuki syndrome for mutations in the MLL2 and KDM6A genes and identified MLL2 mutations in 50 (61.7%) and KDM6A mutations in 5 (6.2%)\n",
    "    - Patients with **KMT2D truncating mutations (70%) had facies that were more typical of those seen in the patients originally reported with Kabuki syndrome.** High-arched eyebrows, short fifth fingers, and infantile hypotonia were more commonly seen in patients with MLL2 mutations than in those with KDM6A mutations.\n",
    "    - **Only half of the patients with KMT2D mutations had short stature and postnatal growth retardation**, compared to all of the patients with KDM6A mutations.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_name_1 = 'KMT2D'\n",
    "mane_tx_id_1 = 'NM_003482.4'\n",
    "mane_px_id_1 = 'NP_003473.3' \n",
    "\n",
    "cohort_name_2 = 'KDM6A'\n",
    "mane_tx_id_2 = 'NM_001291415.2'\n",
    "mane_px_id_2 = 'NP_001278344.1' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import hpotk\n",
    "store = hpotk.configure_ontology_store()\n",
    "hpo = store.load_minimal_hpo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store('0.1.20') as ps:\n",
    "    phenopackets = tuple(ps.iter_cohort_phenopackets(cohort_name_1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 100%|██████████| 65/65 [00:00<00:00, 224.95individuals/s]\n"
     ]
    }
   ],
   "source": [
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort_1, validation = load_phenopackets(\n",
    "    phenopackets=phenopackets,\n",
    "    cohort_creator=cohort_creator,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ppktstore.registry import configure_phenopacket_registry\n",
    "phenopacket_registry = configure_phenopacket_registry()\n",
    "with phenopacket_registry.open_phenopacket_store('0.1.20') as ps:\n",
    "    phenopackets = tuple(ps.iter_cohort_phenopackets(cohort_name_2))\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Individuals Processed: 100%|██████████| 81/81 [00:12<00:00,  6.54individuals/s]\n"
     ]
    }
   ],
   "source": [
    "from gpsea.preprocessing import configure_caching_cohort_creator, load_phenopackets\n",
    "cohort_creator = configure_caching_cohort_creator(hpo)\n",
    "cohort_2, validation = load_phenopackets(\n",
    "    phenopackets=phenopackets,\n",
    "    cohort_creator=cohort_creator,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Validated under none policy\n",
      "No errors or warnings were found\n"
     ]
    }
   ],
   "source": [
    "validation.summarize() "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **Kabuki 1 Cohort** "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>GPSEA cohort analysis</h1>\n",
       "  <p>Successfully loaded 65 individuals.</p>\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Top 10 HPO Terms</h3>\n",
       "        A total of 238 HPO terms were used to annotated the cohort.\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">HPO Term</th>\n",
       "                <th>ID</th>\n",
       "                <th>Seen in <em>n</em> individuals</th>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Depressed nasal tip</td>\n",
       "                <td>HP:0000437</td>\n",
       "                <td>36</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Prominent fingertip pads</td>\n",
       "                <td>HP:0001212</td>\n",
       "                <td>30</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Global developmental delay</td>\n",
       "                <td>HP:0001263</td>\n",
       "                <td>30</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Long palpebral fissure</td>\n",
       "                <td>HP:0000637</td>\n",
       "                <td>25</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Macrotia</td>\n",
       "                <td>HP:0000400</td>\n",
       "                <td>24</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Highly arched eyebrow</td>\n",
       "                <td>HP:0002553</td>\n",
       "                <td>22</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Intellectual disability</td>\n",
       "                <td>HP:0001249</td>\n",
       "                <td>21</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Eversion of lateral third of lower eyelids</td>\n",
       "                <td>HP:0007655</td>\n",
       "                <td>18</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Cleft palate</td>\n",
       "                <td>HP:0000175</td>\n",
       "                <td>17</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Hearing impairment</td>\n",
       "                <td>HP:0000365</td>\n",
       "                <td>16</td>\n",
       "                </tr>\n",
       "            \n",
       "                </tbody>\n",
       "    </table>\n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Top 10 Variants</h3>\n",
       "        Variants are shown according to NM_003482.4. A total of 65 unique variants were identified in the cohort.\n",
       "    </caption>\n",
       "            <tbody>\n",
       "                <tr class=\"strng\">\n",
       "                    <th>Count</th>\n",
       "                    <th class=\"lft\">Variant key</th>\n",
       "                    <th>Variant Name</th>\n",
       "                    <th>Protein Variant</th>\n",
       "                    <th>Variant Class</th>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>4</td>\n",
       "                    <td class=\"lft\">12_49022655_49022655_C_T</td>\n",
       "                    <td>c.16273G>A</td>\n",
       "                    <td>p.Glu5425Lys</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>4</td>\n",
       "                    <td class=\"lft\">12_49034182_49034182_A_G</td>\n",
       "                    <td>c.10625T>C</td>\n",
       "                    <td>p.Leu3542Pro</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>3</td>\n",
       "                    <td class=\"lft\">12_49033960_49033960_C_T</td>\n",
       "                    <td>c.10745G>A</td>\n",
       "                    <td>p.Arg3582Gln</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>3</td>\n",
       "                    <td class=\"lft\">12_49034082_49034082_C_G</td>\n",
       "                    <td>c.10725G>C</td>\n",
       "                    <td>p.Gln3575His</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>3</td>\n",
       "                    <td class=\"lft\">12_49034225_49034225_G_C</td>\n",
       "                    <td>c.10582C>G</td>\n",
       "                    <td>p.Leu3528Val</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">12_49022634_49022634_G_A</td>\n",
       "                    <td>c.16294C>T</td>\n",
       "                    <td>p.Arg5432Trp</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>1</td>\n",
       "                    <td class=\"lft\">12_49032140_49032140_C_A</td>\n",
       "                    <td>c.12565G>T</td>\n",
       "                    <td>p.Gly4189Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>1</td>\n",
       "                    <td class=\"lft\">12_49033961_49033961_G_A</td>\n",
       "                    <td>c.10744C>T</td>\n",
       "                    <td>p.Arg3582Trp</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>1</td>\n",
       "                    <td class=\"lft\">12_49055320_49055320_T_C</td>\n",
       "                    <td>c.5A>G</td>\n",
       "                    <td>p.Asp2Gly</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>1</td>\n",
       "                    <td class=\"lft\">12_49039557_49039557_C_A</td>\n",
       "                    <td>c.8107G>T</td>\n",
       "                    <td>p.Glu2703Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "            </tbody>\n",
       "        </table>\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Diseases</h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Disease Name</th>\n",
       "                <th >Disease ID</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">Kabuki Syndrome 1</td>\n",
       "                <td>OMIM:147920</td>\n",
       "                <td>65</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       "    \n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Variant categories for NM_003482.4</h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Variant effect</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">MISSENSE_VARIANT</td>\n",
       "                <td>36</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">STOP_GAINED</td>\n",
       "                <td>15</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">FRAMESHIFT_VARIANT</td>\n",
       "                <td>14</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       "  \n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import HTML, display\n",
    "\n",
    "from gpsea.view import CohortViewable\n",
    "viewer = CohortViewable(hpo)\n",
    "report = viewer.process(cohort=cohort_1, transcript_id=mane_tx_id_1)\n",
    "display(HTML(report))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **Kabuki 2**Cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title>Cohort</title>\n",
       "    <style>\n",
       "    table {\n",
       "      border-collapse: collapse;\n",
       "      margin: 25px 0;\n",
       "      font-size: 0.9em;\n",
       "      font-family: sans-serif;\n",
       "      min-width: 400px;\n",
       "      box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);\n",
       "    }\n",
       "\n",
       "    .table .column-1 {\n",
       "      text-align: left;\n",
       "    }\n",
       "    th {\n",
       "      background-color: LightSkyBlue;\n",
       "      border: 1px solid #dddddd;\n",
       "      text-align: left;\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "      font-size: 120%;\n",
       "    }\n",
       "\n",
       "    tr {\n",
       "      border: 1px solid #dddddd;\n",
       "    }\n",
       "\n",
       "    td {\n",
       "      padding: 2px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "\n",
       "    tr:nth-child(even) {\n",
       "      background-color: #f2f2f2;\n",
       "    }\n",
       "\n",
       "    .table td, tr {\n",
       "      text-align: right;\n",
       "    }\n",
       "\n",
       "    .lft {\n",
       "      text-align: left;\n",
       "    }\n",
       "\n",
       "    div {\n",
       "      display: block;\n",
       "      width: 100%;\n",
       "      padding-right: 1%;\n",
       "      margin-bottom: 20px; /* Add margin to separate divs */\n",
       "    }\n",
       "\n",
       "    caption {\n",
       "      caption-side: top;\n",
       "      text-align: left;\n",
       "      padding-bottom: 10px;\n",
       "      font-weight: bold;\n",
       "    }\n",
       "  </style>\n",
       "</head>\n",
       "\n",
       "<body>\n",
       "  <h1>GPSEA cohort analysis</h1>\n",
       "  <p>Successfully loaded 81 individuals.</p>\n",
       "  \n",
       "  <p>No errors encountered.</p>\n",
       "  \n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Top 10 HPO Terms</h3>\n",
       "        A total of 127 HPO terms were used to annotated the cohort.\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">HPO Term</th>\n",
       "                <th>ID</th>\n",
       "                <th>Seen in <em>n</em> individuals</th>\n",
       "            </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Abnormal facial shape</td>\n",
       "                <td>HP:0001999</td>\n",
       "                <td>80</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Feeding difficulties</td>\n",
       "                <td>HP:0011968</td>\n",
       "                <td>55</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Delayed speech and language development</td>\n",
       "                <td>HP:0000750</td>\n",
       "                <td>54</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Hypotonia</td>\n",
       "                <td>HP:0001252</td>\n",
       "                <td>52</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Narrow palate</td>\n",
       "                <td>HP:0000189</td>\n",
       "                <td>38</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Prominent fingertip pads</td>\n",
       "                <td>HP:0001212</td>\n",
       "                <td>35</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Neonatal hypoglycemia</td>\n",
       "                <td>HP:0001998</td>\n",
       "                <td>31</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Delayed ability to walk</td>\n",
       "                <td>HP:0031936</td>\n",
       "                <td>31</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Motor delay</td>\n",
       "                <td>HP:0001270</td>\n",
       "                <td>27</td>\n",
       "                </tr>\n",
       "            \n",
       "            <tr>\n",
       "                <td class=\"lft\">Microcephaly</td>\n",
       "                <td>HP:0000252</td>\n",
       "                <td>24</td>\n",
       "                </tr>\n",
       "            \n",
       "                </tbody>\n",
       "    </table>\n",
       "\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Top 10 Variants</h3>\n",
       "        Variants are shown according to NM_001291415.2. A total of 81 unique variants were identified in the cohort.\n",
       "    </caption>\n",
       "            <tbody>\n",
       "                <tr class=\"strng\">\n",
       "                    <th>Count</th>\n",
       "                    <th class=\"lft\">Variant key</th>\n",
       "                    <th>Variant Name</th>\n",
       "                    <th>Protein Variant</th>\n",
       "                    <th>Variant Class</th>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>10</td>\n",
       "                    <td class=\"lft\">SO:1000029_HGNC:12637_KDM6A</td>\n",
       "                    <td>1000029_HGNC</td>\n",
       "                    <td>None</td>\n",
       "                    <td>TRANSCRIPT_ABLATION</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>4</td>\n",
       "                    <td class=\"lft\">X_45070166_45070170_CACAA_C</td>\n",
       "                    <td>c.2671_2674del</td>\n",
       "                    <td>p.Asn891ValfsTer27</td>\n",
       "                    <td>FRAMESHIFT_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>3</td>\n",
       "                    <td class=\"lft\">X_45020680_45020680_C_T</td>\n",
       "                    <td>c.514C>T</td>\n",
       "                    <td>p.Arg172Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>3</td>\n",
       "                    <td class=\"lft\">X_45090821_45090821_C_T</td>\n",
       "                    <td>c.3991C>T</td>\n",
       "                    <td>p.Arg1331Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>3</td>\n",
       "                    <td class=\"lft\">X_45059104_45059104_G_C</td>\n",
       "                    <td>c.974G>C</td>\n",
       "                    <td>p.Gly325Ala</td>\n",
       "                    <td>MISSENSE_VARIANT, SPLICE_REGION_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>3</td>\n",
       "                    <td class=\"lft\">X_44974688_44974688_C_G</td>\n",
       "                    <td>c.357C>G</td>\n",
       "                    <td>p.Tyr119Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>3</td>\n",
       "                    <td class=\"lft\">X_45010977_45010977_A_G</td>\n",
       "                    <td>c.401A>G</td>\n",
       "                    <td>p.Tyr134Cys</td>\n",
       "                    <td>MISSENSE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">X_45090861_45090865_TTAAG_T</td>\n",
       "                    <td>c.4034+3_4034+6del</td>\n",
       "                    <td>None</td>\n",
       "                    <td>SPLICE_DONOR_VARIANT, CODING_SEQUENCE_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">X_44974665_44974665_G_T</td>\n",
       "                    <td>c.335-1G>T</td>\n",
       "                    <td>None</td>\n",
       "                    <td>SPLICE_ACCEPTOR_VARIANT</td>\n",
       "                </tr>\n",
       "                \n",
       "                <tr>\n",
       "                    <td>2</td>\n",
       "                    <td class=\"lft\">X_45079316_45079316_C_T</td>\n",
       "                    <td>c.3265C>T</td>\n",
       "                    <td>p.Gln1089Ter</td>\n",
       "                    <td>STOP_GAINED</td>\n",
       "                </tr>\n",
       "                \n",
       "            </tbody>\n",
       "        </table>\n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Diseases</h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Disease Name</th>\n",
       "                <th >Disease ID</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">Kabuki syndrome 2</td>\n",
       "                <td>OMIM:300867</td>\n",
       "                <td>81</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       "    \n",
       "    <table>\n",
       "    <caption style=\"color: black;\">\n",
       "        <h3>Variant categories for NM_001291415.2</h3>\n",
       "    </caption>\n",
       "        <tbody>\n",
       "            <tr class=\"strng\">\n",
       "                <th class=\"lft\">Variant effect</th>\n",
       "                <th>Annotation Count</th>\n",
       "              </tr>\n",
       "              \n",
       "              <tr>\n",
       "                <td class=\"lft\">FRAMESHIFT_VARIANT</td>\n",
       "                <td>16</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">TRANSCRIPT_ABLATION</td>\n",
       "                <td>10</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">SPLICE_ACCEPTOR_VARIANT</td>\n",
       "                <td>6</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">MISSENSE_VARIANT</td>\n",
       "                <td>15</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">SPLICE_REGION_VARIANT</td>\n",
       "                <td>7</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">SPLICE_DONOR_5TH_BASE_VARIANT</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">INTRON_VARIANT</td>\n",
       "                <td>4</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">STOP_GAINED</td>\n",
       "                <td>21</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">SPLICE_DONOR_VARIANT</td>\n",
       "                <td>6</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">INFRAME_DELETION</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">CODING_SEQUENCE_VARIANT</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">PROTEIN_ALTERING_VARIANT</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">SPLICE_DONOR_REGION_VARIANT</td>\n",
       "                <td>2</td>\n",
       "                </tr>\n",
       "            \n",
       "              <tr>\n",
       "                <td class=\"lft\">SPLICE_POLYPYRIMIDINE_TRACT_VARIANT</td>\n",
       "                <td>1</td>\n",
       "                </tr>\n",
       "            \n",
       "           </tbody>\n",
       "    </table>\n",
       "  \n",
       "\n",
       "\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import HTML, display\n",
    "\n",
    "from gpsea.view import CohortViewable\n",
    "viewer = CohortViewable(hpo)\n",
    "report = viewer.process(cohort=cohort_2, transcript_id=mane_tx_id_2)\n",
    "display(HTML(report))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "gpcvenv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
